<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="apple-touch-icon.png">
<link rel="icon" sizes="any" href="favicon.ico">
<link rel="manifest" href="site.webmanifest">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data">
<meta name="description" content="This package performs various models of genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data.">
<meta property="og:description" content="This package performs various models of genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data.">
<meta property="og:image" content="/logo.svg">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/GXwasR_All_Functions.html">GXwasR Functions</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.svg" class="logo" alt=""><h1 id="gxwasr-">GXwasR <a class="anchor" aria-label="anchor" href="#gxwasr-"></a>
</h1>
</div>
<p>A tool for conducting sex-aware quality control, association analysis, and testing various models of sex-dependent genetic effects in complex traits.</p>
<!-- badges: start -->

<p>This package implements various statistical genetics models for Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA) analyses in a sex-combined or sex-stratified way considering X-Chromosome Inactivation (XCI) pattern. In addition to association analysis, the package also enables testing for sex differences in genetic effects, including the implementation of specific models and applying best practices for additional quality control (QC) of genetic data required for these tests. The package includes twenty-five different functions in six different categories (A-F) which enable a comprehensive pipeline for sex-aware genetic association analysis of common variants with unrelated individuals.</p>
<p><strong>(A)Pre-imputation QC:</strong> <code><a href="reference/QCsnp.html">QCsnp()</a></code>; <code><a href="reference/QCsample.html">QCsample()</a></code>; <code><a href="reference/AncestryCheck.html">AncestryCheck()</a></code>; <code><a href="reference/SexCheck.html">SexCheck()</a></code></p>
<p><strong>(B)Post-imputation QC:</strong> <code><a href="reference/QCsnp.html">QCsnp()</a></code>; <code>QCsample2()</code>; <code><a href="reference/Xhwe.html">Xhwe()</a></code>; <code><a href="reference/MAFdiffSexControl.html">MAFdiffSexControl()</a></code>; <code><a href="reference/FilterRegion.html">FilterRegion()</a></code></p>
<p><strong>(C)Sex-combined and sex-stratified GWAS with XWAS:</strong> <code><a href="reference/GXwasR-package.html">GXwasR()</a></code></p>
<p><strong>(D)Sex-differential test:</strong> <code><a href="reference/SexDiff.html">SexDiff()</a></code>; <code><a href="reference/SexDiffZscore.html">SexDiffZscore()</a></code>; <code><a href="reference/DiffZeroOne.html">DiffZeroOne()</a></code></p>
<p><strong>(E)High level analysis:</strong> <code><a href="reference/TestXGene.html">TestXGene()</a></code>; <code><a href="reference/MetaGWAS.html">MetaGWAS()</a></code>; <code><a href="reference/ComputePRS.html">ComputePRS()</a></code>; <code>ComputeCorrBT()</code>; <code><a href="reference/EstimateHerit.html">EstimateHerit()</a></code></p>
<p><strong>(F)Utility Functions:</strong> <code><a href="reference/FilterPlinkSample.html">FilterPlinkSample()</a></code>; <code><a href="reference/ComputeGeneticPC.html">ComputeGeneticPC()</a></code>; <code><a href="reference/ClumpLD.html">ClumpLD()</a></code>; <code><a href="reference/GetMFPlink.html">GetMFPlink()</a></code>; <code><a href="reference/plinkVCF.html">plinkVCF()</a></code>; <code><a href="reference/MergeRegion.html">MergeRegion()</a></code>; <code><a href="reference/FilterAllele.html">FilterAllele()</a></code>; <code><a href="reference/PlinkSummary.html">PlinkSummary()</a></code></p>
<div class="section level2">
<h2 id="package-installation">📦 Installation<a class="anchor" aria-label="anchor" href="#package-installation"></a>
</h2>
<p>You can install the development version of GXwasR from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")</span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"boseb/GXwasR"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="external-dependencies">External Dependencies<a class="anchor" aria-label="anchor" href="#external-dependencies"></a>
</h3>
<p>This package requires PLINK and GCTA, two widely used command-line tools for genetic data analysis:</p>
<ul>
<li>PLINK: A toolset for genome association and linkage analysis.</li>
<li>GCTA: Genome-wide Complex Trait Analysis, used for estimating genetic relationships and variance components.</li>
</ul>
<p>Please follow the instructions below to ensure both tools are installed and available to your system before use.</p>
<div class="section level4">
<h4 id="plink">PLINK<a class="anchor" aria-label="anchor" href="#plink"></a>
</h4>
<p>This package depends on the <a href="https://www.cog-genomics.org/plink/" class="external-link">PLINK</a> command-line tool (version 1.9). PLINK must be installed separately and made available on your system.</p>
<p>PLINK is not bundled with this package and must either:</p>
<ul>
<li>(<em>preferred</em>) be specified via the PLINK_PATH environment variable, or</li>
<li>be on your system PATH.</li>
</ul>
<div class="section level5">
<h5 id="wrench-plink-installation-instructions">🔧 PLINK Installation Instructions<a class="anchor" aria-label="anchor" href="#wrench-plink-installation-instructions"></a>
</h5>
<p>Binaries for all major platforms can be downloaded from:</p>
<ul>
<li><a href="https://www.cog-genomics.org/plink/1.9/" class="external-link">PLINK v1.9</a></li>
</ul>
<p>Detailed, platform-specific setup instructions can be found in the INSTALL file included with this package.</p>
</div>
<div class="section level5">
<h5 id="compass-configuring-the-plink-path">🧭 Configuring the PLINK Path<a class="anchor" aria-label="anchor" href="#compass-configuring-the-plink-path"></a>
</h5>
<p>This package will attempt to locate PLINK using:</p>
<ul>
<li>The <code>PLINK_PATH</code> environment variable, if set.</li>
<li>The system path, via <code>Sys.which("plink")</code>.</li>
</ul>
<p>If PLINK is not found, an error will be raised with guidance on how to resolve it.</p>
<p>You can manually set the path in your R session:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>PLINK_PATH <span class="op">=</span> <span class="st">"/path/to/plink"</span><span class="op">)</span></span></code></pre></div>
<p>For a persistent configuration, you can add this line to your .Renviron file:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>PLINK_PATH<span class="ot">=</span><span class="er">/</span>path<span class="sc">/</span>to<span class="sc">/</span>plink</span></code></pre></div>
<p>To verify that PLINK is discoverable:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plink_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"PLINK_PATH"</span>, unset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"plink"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">plink_path</span><span class="op">)</span> <span class="op">||</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nzchar</a></span><span class="op">(</span><span class="va">plink_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"PLINK binary not found. Please install PLINK and/or set the PLINK_PATH environment variable."</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="gcta">GCTA<a class="anchor" aria-label="anchor" href="#gcta"></a>
</h4>
<p>This package also utilizes the <a href="https://yanglab.westlake.edu.cn/software/gcta/#Overview" class="external-link">GCTA</a> command-line tool (Genome-wide Complex Trait Analysis). GCTA must be installed separately and made available on your system.</p>
<p>GCTA is not bundled with this package and must either:</p>
<ul>
<li>(<em>preferred</em>) be specified via the GCTA_PATH environment variable, or</li>
<li>be on your system PATH.</li>
</ul>
<div class="section level5">
<h5 id="wrench-gcta-installation-instructions">🔧 GCTA Installation Instructions<a class="anchor" aria-label="anchor" href="#wrench-gcta-installation-instructions"></a>
</h5>
<p>Binaries for all major platforms can be downloaded from the <a href="https://yanglab.westlake.edu.cn/software/gcta/#Overview" class="external-link">GCTA</a> website.</p>
<p>Detailed, platform-specific setup instructions can be found in the INSTALL file included with this package.</p>
</div>
<div class="section level5">
<h5 id="compass-configuring-the-gcta-path">🧭 Configuring the GCTA Path<a class="anchor" aria-label="anchor" href="#compass-configuring-the-gcta-path"></a>
</h5>
<p>This package will attempt to locate GCTA using:</p>
<ul>
<li>The GCTA_PATH environment variable, if set.</li>
<li>The system path, via <code>Sys.which("gcta64")</code>.</li>
</ul>
<p>If GCTA is not found, an error will be raised with guidance on how to resolve it.</p>
<p>You can manually set the path in your R session:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>GCTA_PATH <span class="op">=</span> <span class="st">"/path/to/gcta64"</span><span class="op">)</span></span></code></pre></div>
<p>For a persistent configuration, you can add this line to your .Renviron file:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>GCTA_PATH<span class="ot">=</span><span class="er">/</span>path<span class="sc">/</span>to<span class="sc">/</span>gcta64</span></code></pre></div>
<p>To verify that GCTA is discoverable:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gcta_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"GCTA_PATH"</span>, unset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.which.html" class="external-link">Sys.which</a></span><span class="op">(</span><span class="st">"gcta64"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">gcta_path</span><span class="op">)</span> <span class="op">||</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nzchar</a></span><span class="op">(</span><span class="va">gcta_path</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"GCTA binary not found. Please install GCTA and/or set the GCTA_PATH environment variable."</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="warning-macos-security-warning">⚠️ macOS Security Warning<a class="anchor" aria-label="anchor" href="#warning-macos-security-warning"></a>
</h4>
<p>macOS may block these applications from launching because they were downloaded from the internet and aren’t explicitly approved by Apple. If you see a warning like:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="st">"“(PLINK/GCTA)” can’t be opened because Apple cannot check it for malicious software."</span></span></code></pre></div>
<p>You can still run the app by following these steps:</p>
<ol style="list-style-type: decimal">
<li>Open System Settings (or System Preferences on older macOS versions).</li>
<li>Go to Privacy &amp; Security.</li>
<li>Scroll down to the Security section.</li>
<li>You should see a message about the blocked app — click “Open Anyway”.</li>
<li>Confirm when prompted.</li>
</ol>
<p>For more details, see Apple’s official guide: <a href="https://support.apple.com/en-us/102445" class="external-link uri">https://support.apple.com/en-us/102445</a></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="example-analysis">Example Analysis<a class="anchor" aria-label="anchor" href="#example-analysis"></a>
</h2>
<p>Run a genome-wide association study (GWAS) and X-chromosome-wide association study (XWAS):</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GXwasR: Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data</span></span>
<span><span class="co">#&gt; Version 0.99.0 () installed</span></span>
<span><span class="co">#&gt; Author: c(</span></span>
<span><span class="co">#&gt;     person(given = "Banabithi",</span></span>
<span><span class="co">#&gt;            family = "Bose",</span></span>
<span><span class="co">#&gt;            role = c("cre", "aut"),</span></span>
<span><span class="co">#&gt;            email = "banabithi.bose@gmail.com",</span></span>
<span><span class="co">#&gt;            comment = c(ORCID = "0000-0003-0842-8768"))</span></span>
<span><span class="co">#&gt;            )</span></span>
<span><span class="co">#&gt; Maintainer: Banabithi Bose &lt;banabithi.bose@gmail.com&gt;</span></span>
<span><span class="co">#&gt; Tutorial: https://github.com</span></span>
<span><span class="co">#&gt; Use citation("GXwasR") to know how to cite this work.</span></span>
<span><span class="va">ResultGXwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/GXwas.html">GXwas</a></span><span class="op">(</span></span>
<span>  DataDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>  finput <span class="op">=</span> <span class="st">"GXwasR_example"</span>, </span>
<span>  trait <span class="op">=</span> <span class="st">"binary"</span>, </span>
<span>  xmodel <span class="op">=</span> <span class="st">"FMstatrified"</span>, </span>
<span>  covarfile <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>  sex <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  xsex <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  combtest <span class="op">=</span> <span class="st">"fisher.method"</span>, </span>
<span>  MF.p.corr <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  snp_pval <span class="op">=</span> <span class="fl">1e-08</span>, </span>
<span>  plot.jpeg <span class="op">=</span> <span class="cn">FALSE</span>, </span>
<span>  suggestiveline <span class="op">=</span> <span class="fl">5</span>, </span>
<span>  genomewideline <span class="op">=</span> <span class="fl">7.3</span>,</span>
<span>  MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, </span>
<span>  ncores <span class="op">=</span> <span class="fl">0</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; • Running FMstatrified model</span></span>
<span><span class="co">#&gt; Using PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)</span></span>
<span><span class="co">#&gt; • Stratified test is running</span></span>
<span><span class="co">#&gt; • Stratified test is running</span></span>
<span><span class="co">#&gt; • If you want parallel computation, please provide non-zero value for argument ncores.</span></span>
<span><span class="co">#&gt; • If you want parallel computation, please provide non-zero value for argument ncores.</span></span>
<span><span class="co">#&gt; ℹ Plots are initiated.</span></span>
<span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp949z4L/Stratified_GWAS.png</span></span>
<span><span class="co">#&gt; This message is displayed once every 8 hours.</span></span></code></pre></div>
<p><img src="reference/figures/README-example-1.png" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="co">#&gt; ℹ Saving plot to</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp949z4L/Stratified_XWAS.png</span></span></code></pre></div>
<p><img src="reference/figures/README-example-2.png" width="100%"><img src="reference/figures/README-example-3.png" width="100%"><img src="reference/figures/README-example-4.png" width="100%"><img src="reference/figures/README-example-5.png" width="100%"><img src="reference/figures/README-example-6.png" width="100%"></p>
<div class="sourceCode" id="cb11"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co">#&gt; • Three dataframes such as, CombinedWAS, MaleWAS and FemaleWAS are produced</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a><span class="co">#&gt; in/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp949z4L</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>ResultGXwas</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co">#&gt; $CombinedWAS</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; Key: &lt;SNP&gt;</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a><span class="co">#&gt;               SNP   CHR        BP          P</span></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt;            &lt;char&gt; &lt;int&gt;     &lt;int&gt;      &lt;num&gt;</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a><span class="co">#&gt;     1: rs10000405     4  47716881 0.97684636</span></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a><span class="co">#&gt;     2: rs10000452     4  63234460 0.69697101</span></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co">#&gt;     3: rs10000465     4 120835814 0.53783097</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt;     4: rs10000605     4  13875675 0.72216716</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a><span class="co">#&gt;     5: rs10000675     4 121520624 0.82462199</span></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt;    ---                                      </span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="co">#&gt; 23089:  rs9997486     4  72333579 0.50805172</span></span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt; 23090:  rs9997787     4  37866912 0.34794418</span></span>
<span id="cb11-16"><a href="#cb11-16" tabindex="-1"></a><span class="co">#&gt; 23091:  rs9998003     4 187293933 0.06428729</span></span>
<span id="cb11-17"><a href="#cb11-17" tabindex="-1"></a><span class="co">#&gt; 23092:  rs9998694     4  59730463 0.08565972</span></span>
<span id="cb11-18"><a href="#cb11-18" tabindex="-1"></a><span class="co">#&gt; 23093:  rs9999463     4 166351966 0.42240473</span></span>
<span id="cb11-19"><a href="#cb11-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-20"><a href="#cb11-20" tabindex="-1"></a><span class="co">#&gt; $MaleWAS</span></span>
<span id="cb11-21"><a href="#cb11-21" tabindex="-1"></a><span class="co">#&gt;          CHR         SNP        BP     A1   </span><span class="al">TEST</span><span class="co"> NMISS     BETA     SE     L95</span></span>
<span id="cb11-22"><a href="#cb11-22" tabindex="-1"></a><span class="co">#&gt;        &lt;int&gt;      &lt;char&gt;     &lt;int&gt; &lt;char&gt; &lt;char&gt; &lt;int&gt;    &lt;num&gt;  &lt;num&gt;   &lt;num&gt;</span></span>
<span id="cb11-23"><a href="#cb11-23" tabindex="-1"></a><span class="co">#&gt;     1:     1 rs143773730     73841      T    ADD   125 -0.07890 0.2643 -0.5968</span></span>
<span id="cb11-24"><a href="#cb11-24" tabindex="-1"></a><span class="co">#&gt;     2:     1 rs147281566    775125      T    ADD   125 -0.39590 1.2380 -2.8230</span></span>
<span id="cb11-25"><a href="#cb11-25" tabindex="-1"></a><span class="co">#&gt;     3:     1  rs35854196    863863      A    ADD   125  1.05000 0.8858 -0.6864</span></span>
<span id="cb11-26"><a href="#cb11-26" tabindex="-1"></a><span class="co">#&gt;     4:     1  rs12041521   1109154      A    ADD   125 -0.44510 0.3387 -1.1090</span></span>
<span id="cb11-27"><a href="#cb11-27" tabindex="-1"></a><span class="co">#&gt;     5:     1 rs148527527   1127860      G    ADD   125  1.67700 0.6864  0.3317</span></span>
<span id="cb11-28"><a href="#cb11-28" tabindex="-1"></a><span class="co">#&gt;    ---                                                                        </span></span>
<span id="cb11-29"><a href="#cb11-29" tabindex="-1"></a><span class="co">#&gt; 24135:    23 rs113460214 154103785      G    ADD   125  2.00100 1.1110 -0.1771</span></span>
<span id="cb11-30"><a href="#cb11-30" tabindex="-1"></a><span class="co">#&gt; 24136:    23   rs5945265 154196387      A    ADD   125 -0.23380 0.5514 -1.3150</span></span>
<span id="cb11-31"><a href="#cb11-31" tabindex="-1"></a><span class="co">#&gt; 24137:    23 rs191928457 154429518      C    ADD   125  1.44900 1.1690 -0.8426</span></span>
<span id="cb11-32"><a href="#cb11-32" tabindex="-1"></a><span class="co">#&gt; 24138:    24   rs9785994  16222561      T    ADD    36 -0.47000 0.9811 -2.3930</span></span>
<span id="cb11-33"><a href="#cb11-33" tabindex="-1"></a><span class="co">#&gt; 24139:    24 rs201216761  22113254      A    ADD    38 -0.09353 0.8447 -1.7490</span></span>
<span id="cb11-34"><a href="#cb11-34" tabindex="-1"></a><span class="co">#&gt;           U95    STAT       P</span></span>
<span id="cb11-35"><a href="#cb11-35" tabindex="-1"></a><span class="co">#&gt;         &lt;num&gt;   &lt;num&gt;   &lt;num&gt;</span></span>
<span id="cb11-36"><a href="#cb11-36" tabindex="-1"></a><span class="co">#&gt;     1: 0.4390 -0.2986 0.76530</span></span>
<span id="cb11-37"><a href="#cb11-37" tabindex="-1"></a><span class="co">#&gt;     2: 2.0310 -0.3197 0.74920</span></span>
<span id="cb11-38"><a href="#cb11-38" tabindex="-1"></a><span class="co">#&gt;     3: 2.7860  1.1850 0.23600</span></span>
<span id="cb11-39"><a href="#cb11-39" tabindex="-1"></a><span class="co">#&gt;     4: 0.2188 -1.3140 0.18890</span></span>
<span id="cb11-40"><a href="#cb11-40" tabindex="-1"></a><span class="co">#&gt;     5: 3.0220  2.4430 0.01456</span></span>
<span id="cb11-41"><a href="#cb11-41" tabindex="-1"></a><span class="co">#&gt;    ---                       </span></span>
<span id="cb11-42"><a href="#cb11-42" tabindex="-1"></a><span class="co">#&gt; 24135: 4.1790  1.8010 0.07177</span></span>
<span id="cb11-43"><a href="#cb11-43" tabindex="-1"></a><span class="co">#&gt; 24136: 0.8469 -0.4241 0.67150</span></span>
<span id="cb11-44"><a href="#cb11-44" tabindex="-1"></a><span class="co">#&gt; 24137: 3.7410  1.2390 0.21520</span></span>
<span id="cb11-45"><a href="#cb11-45" tabindex="-1"></a><span class="co">#&gt; 24138: 1.4530 -0.4791 0.63190</span></span>
<span id="cb11-46"><a href="#cb11-46" tabindex="-1"></a><span class="co">#&gt; 24139: 1.5620 -0.1107 0.91180</span></span>
<span id="cb11-47"><a href="#cb11-47" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb11-48"><a href="#cb11-48" tabindex="-1"></a><span class="co">#&gt; $FemaleWAS</span></span>
<span id="cb11-49"><a href="#cb11-49" tabindex="-1"></a><span class="co">#&gt;          CHR         SNP        BP     A1   </span><span class="al">TEST</span><span class="co"> NMISS    BETA     SE     L95</span></span>
<span id="cb11-50"><a href="#cb11-50" tabindex="-1"></a><span class="co">#&gt;        &lt;int&gt;      &lt;char&gt;     &lt;int&gt; &lt;char&gt; &lt;char&gt; &lt;int&gt;   &lt;num&gt;  &lt;num&gt;   &lt;num&gt;</span></span>
<span id="cb11-51"><a href="#cb11-51" tabindex="-1"></a><span class="co">#&gt;     1:     1 rs143773730     73841      T    ADD   151  0.3459 0.2818 -0.2064</span></span>
<span id="cb11-52"><a href="#cb11-52" tabindex="-1"></a><span class="co">#&gt;     2:     1 rs147281566    775125      T    ADD   151  0.1568 0.9290 -1.6640</span></span>
<span id="cb11-53"><a href="#cb11-53" tabindex="-1"></a><span class="co">#&gt;     3:     1  rs35854196    863863      A    ADD   151 -0.1446 0.7282 -1.5720</span></span>
<span id="cb11-54"><a href="#cb11-54" tabindex="-1"></a><span class="co">#&gt;     4:     1 rs115490086    928969      T    ADD   151  0.5649 1.4240 -2.2270</span></span>
<span id="cb11-55"><a href="#cb11-55" tabindex="-1"></a><span class="co">#&gt;     5:     1  rs12041521   1109154      A    ADD   151 -0.6697 0.3314 -1.3190</span></span>
<span id="cb11-56"><a href="#cb11-56" tabindex="-1"></a><span class="co">#&gt;    ---                                                                       </span></span>
<span id="cb11-57"><a href="#cb11-57" tabindex="-1"></a><span class="co">#&gt; 24355:    23   rs2097214 153652198      G    ADD   151 -0.4137 0.5158 -1.4250</span></span>
<span id="cb11-58"><a href="#cb11-58" tabindex="-1"></a><span class="co">#&gt; 24356:    23 rs113519705 153715153      T    ADD   151  0.2942 0.3751 -0.4409</span></span>
<span id="cb11-59"><a href="#cb11-59" tabindex="-1"></a><span class="co">#&gt; 24357:    23 rs113460214 154103785      G    ADD   151 -0.7779 0.5943 -1.9430</span></span>
<span id="cb11-60"><a href="#cb11-60" tabindex="-1"></a><span class="co">#&gt; 24358:    23   rs5945265 154196387      A    ADD   151  0.5425 0.3360 -0.1160</span></span>
<span id="cb11-61"><a href="#cb11-61" tabindex="-1"></a><span class="co">#&gt; 24359:    23 rs191928457 154429518      C    ADD   151 -1.4460 1.0830 -3.5690</span></span>
<span id="cb11-62"><a href="#cb11-62" tabindex="-1"></a><span class="co">#&gt;             U95    STAT       P</span></span>
<span id="cb11-63"><a href="#cb11-63" tabindex="-1"></a><span class="co">#&gt;           &lt;num&gt;   &lt;num&gt;   &lt;num&gt;</span></span>
<span id="cb11-64"><a href="#cb11-64" tabindex="-1"></a><span class="co">#&gt;     1:  0.89820  1.2270 0.21970</span></span>
<span id="cb11-65"><a href="#cb11-65" tabindex="-1"></a><span class="co">#&gt;     2:  1.97800  0.1688 0.86590</span></span>
<span id="cb11-66"><a href="#cb11-66" tabindex="-1"></a><span class="co">#&gt;     3:  1.28300 -0.1985 0.84260</span></span>
<span id="cb11-67"><a href="#cb11-67" tabindex="-1"></a><span class="co">#&gt;     4:  3.35700  0.3966 0.69170</span></span>
<span id="cb11-68"><a href="#cb11-68" tabindex="-1"></a><span class="co">#&gt;     5: -0.02008 -2.0210 0.04332</span></span>
<span id="cb11-69"><a href="#cb11-69" tabindex="-1"></a><span class="co">#&gt;    ---                         </span></span>
<span id="cb11-70"><a href="#cb11-70" tabindex="-1"></a><span class="co">#&gt; 24355:  0.59740 -0.8019 0.42260</span></span>
<span id="cb11-71"><a href="#cb11-71" tabindex="-1"></a><span class="co">#&gt; 24356:  1.02900  0.7844 0.43280</span></span>
<span id="cb11-72"><a href="#cb11-72" tabindex="-1"></a><span class="co">#&gt; 24357:  0.38700 -1.3090 0.19060</span></span>
<span id="cb11-73"><a href="#cb11-73" tabindex="-1"></a><span class="co">#&gt; 24358:  1.20100  1.6150 0.10640</span></span>
<span id="cb11-74"><a href="#cb11-74" tabindex="-1"></a><span class="co">#&gt; 24359:  0.67610 -1.3360 0.18170</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the GXwasR project is released with a <a href="https://contributor-covenant.org/version/2/1/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing GXwasR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Banabithi Bose <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0003-0842-8768" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#stable" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-brightgreen.svg" alt="Lifecycle: stable"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
