---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# GXwasR <img src="man/figures/logo.png" align="right" height="120" alt="" />

A tool for conducting sex-aware quality control, association analysis, and testing various models of sex-dependent genetic effects in complex traits.

<!-- badges: start -->
[![Lifecycle: stable](https://img.shields.io/badge/lifecycle-stable-brightgreen.svg)](https://lifecycle.r-lib.org/articles/stages.html#stable)
<!-- badges: end -->

This package implements various statistical genetics models for Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA) analyses in a sex-combined or sex-stratified way considering X-Chromosome Inactivation (XCI) pattern. In addition to association analysis, the package also enables testing for sex differences in genetic effects, including the implementation of specific models and applying best practices for additional quality control (QC) of genetic data required for these tests. The package includes twenty-five different functions in six different categories (A-F) which enable a comprehensive pipeline for sex-aware genetic association analysis of common variants with unrelated individuals.

**(A)Pre-imputation QC:**  `QCsnp()`; `QCsample()`; `AncestryCheck()`; `SexCheck()`

**(B)Post-imputation QC:** `QCsnp()`; `QCsample2()`; `Xhwe()`; `MAFdiffSexControl()`; `FilterRegion()`

**(C)Sex-combined and sex-stratified GWAS with XWAS:** `GXwasR()`

**(D)Sex-differential test:** `SexDiff()`; `SexDiffZscore()`; `DiffZeroOne()`

**(E)High level analysis:** `TestXGene()`; `MetaGWAS()`; `ComputePRS()`; `ComputeCorrBT()`; `EstimateHerit()`

**(F)Utility Functions:** `FilterPlinkSample()`; `ComputeGeneticPC()`; `ClumpLD()`; `GetMFPlink()`; `plinkVCF()`; `MergeRegion()`; `FilterAllele()`; `PlinkSummary()`

## Installation

You can install the development version of GXwasR from [GitHub](https://github.com/) with:

``` r
# install.packages("pak")
pak::pak("boseb/GXwasR")
```

## Example

Run a genome-wide association study (GWAS) and X-chromosome-wide association study (XWAS):

```{r example}
library(GXwasR)
ResultGXwas <- GXwas(
  DataDir = system.file("extdata", package = "GXwasR"),
  ResultDir = tempdir(), 
  finput = "GXwasR_example", 
  trait = "binary", 
  xmodel = "FMstatrified", 
  covarfile = NULL, 
  sex = TRUE, 
  xsex = FALSE, 
  combtest = "fisher.method", 
  MF.p.corr = "none",
  snp_pval = 1e-08, 
  plot.jpeg = FALSE, 
  suggestiveline = 5, 
  genomewideline = 7.3,
  MF.mc.cores = 1, 
  ncores = 0
  )
ResultGXwas
```

## Code of Conduct

Please note that the GXwasR project is released with a [Contributor Code of Conduct](https://contributor-covenant.org/version/2/1/CODE_OF_CONDUCT.html). By contributing to this project, you agree to abide by its terms.
