---
title: "Introduction of GXwasR"
author: "By Banabithi Bose"
date: "1/09/2023"
output:      
    BiocStyle::html_document:
        toc: true
        toc_depth: 3  
        number_sections: true
        theme: united 
r_packages: rmarkdown
vignette: >
  %\VignetteIndexEntry{Introduction of GXwasR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}{inputenc}
---
```{r global_options, include = FALSE}
knitr::opts_chunk$set(message=FALSE,
tidy.opts=list(width.cutoff=60))
```

# <a id = "GXwasR Description"></a> GXwasR Description: 

This package implements various statistical genetics models for Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA) analyses in a sex-combined or sex-stratified way considering X-Chromosome Inactivation (XCI) pattern. In addition to association analysis, the package also enables testing for sex differences in genetic effects, including the implementation of specific models and applying best practices for additional quality control (QC) of genetic data required for these tests. The package includes twenty-five different functions in six different categories (A-F) which enable a comprehensive pipeline for sex-aware genetic association analysis of common variants with unrelated individuals.

**(A)Pre-imputation QC:**  QCsnp(); QCsample(); AncestryCheck(); SexCheck()

**(B)Post-imputation QC:** QCsnp(); QCsample2(); Xhwe(); MAFdiffSexControl(); FilterRegion()

**(C)Sex-combined and sex-stratified GWAS with XWAS:** GXwasR()

**(D)Sex-differential test:** SexDiff(); SexDiffZscore; DiffZeroOne()

**(E)High level analysis:** TestXGene(); MetaGWAS(); ComputePRS(); ComputeCorrBT(); EstimateHerit()

**(F)Utility Functions:** FilterPlinkSample(); ComputeGeneticPC(); ClumpLD(); GetMFPlink(); plinkVCF(); MergeRegion(); FilterAllele(); PlinkSummary()

In this vignette, we will show sex-aware QC followed by applying a GWAS with XWAS model. The example datasets for this tutorial are provided in “extdata”. 

# <b id = "Example Datasets"></b> Example Datasets:
The PLINK bed, bim, and fam files are the three mandatory files representing a genotype dataset to run this pipeline. To know about these file extensions, please check https://www.cog-genomics.org/plink/1.9/formats.

**(1)GXwasR_example:**

GXwasR_example.bed

GXwasR_example.bim

GXwasR_example.fam

These plink files contain genotypes for 276 individuals (males and females) simulated from 1000Genome from European decent with 26515 variants across twelve chromosomes (1-10,23,24). This dataset contains 125 males, 151 females, 108 cases and 168 controls. We will utilize this set of plink files as proxy of pre-imputated genotype data.

**(2)GXwasR_example_imputed:**

GXwasR_example_imputed.bed

GXwasR_example_imputed.bim

GXwasR_example_imputed.fam

These plink files will serve as a proxy for the imputed genotype dataset of GXwasR_example.

**(3)wrong_sex_samples:**
A plain text file containing the FID and IID of the four samples. This file will be used to extract samples from the input genotype dataset.

# <c id = "Pre-imputation QC steps:"> </c> Pre-imputation QC steps:
**(A)	QC at sample level:** 
1.Check the ancestry level of the input samples and find the outlier samples of Non-European descent.
2.Check the inconsistency between genetic sex and reported sex.
3.Filter out the samples based on missingness and heterozygosity rate.

**(B)	QC at SNP level for autosomes and sex chromosomes:** 
1.Filter out SNPs with genotyping efficiency/call rate < 0.98. 
2.Filter out SNPs with Hardy-Weinberg Equilibrium (cases < 1e-10; controls < 1e-6 for case-control data).
3.Filter out SNPs with case-control differential missingness > 0.02 for case-control data.
4.Filter SNPs with MAF filtering (Common variant MAF < 0.01; Rare variant MAF = 0). 
5.Remove monomorphic SNPs.  

```{r}
## Call some libraries
library(GXwasR)
library(printr)
library(rmarkdown)
## Load the example .Rda files
data("GXwasRData")
```
## <a id = "Pre-imputation QC pipeline:"> </a>Pre-imputation QC pipeline:
We will run the pre-imputation QC pipeline in this vignette utilizing the functions from GXwasR package. 

**Step 1:** Firstly, users need to ensure that their input plink files are bi-allelic. To ensure that, it is advised to run the FilterAllele() function with the input dataset.

### <a id = "Function1"> </a> Function 1: FilterAllele()
```{r}
help(FilterAllele, package = GXwasR)
```
#### <a id = "Running FilterAllele"> </a> Running FilterAllele
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "GXwasR_example"
foutput <- "Filter_Test"
x <- FilterAllele(DataDir, ResultDir, finput, foutput)
```
**Step 2:** We will check the ancestry label of the input samples and look for the outlier samples of Non-European descent.

### <a id = "Function2"> </a> Function 2: AncestryCheck
```{r}
help(AncestryCheck, package = GXwasR)
```
#### <a id = "Running AncestryCheck"> </a> Running AncestryCheck
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir <- tempdir()
finput <- "GXwasR_example"
reference <- "HapMapIII_NCBI36"
#load("~/b1137/BBose/GXwasR/data/highLD_hg19.Rda")
highLD_regions <- highLD_hg19 #"high-LD-regions-hg19-GRCh37.txt" # four columns, chr, start, end, index
#load("~/b1137/BBose/GXwasR/data/example_data_study_sample_ancestry.Rda")
study_pop <- example_data_study_sample_ancestry #PreimputeEX
studyLD_window_size = 50
studyLD_step_size = 5
studyLD_r2_threshold = 0.02
filterSNP = TRUE
studyLD = TRUE
referLD = TRUE
referLD_window_size = 50
referLD_step_size = 5
referLD_r2_threshold = 0.02
outlier = TRUE
outlier_threshold = 3
x <- AncestryCheck(DataDir = DataDir, ResultDir = ResultDir, finput = finput, reference = reference,highLD_regions = highLD_regions,
study_pop = study_pop, studyLD = studyLD, referLD = referLD, outlier = outlier, outlierOf = "CEU", outlier_threshold = outlier_threshold )
```
Since function AncestryCheck() didn’t produce any outlier samples, the users can proceed with the “GXwasR_example” dataset for the next steps.

**Step 3:** We will run SexCheck() function to check if there is any discrepancy in sex assignment between input dataset (i.e., “GXwasR_example”) and predicted sex. 

### <a id = "Function3"> </a> Function 3: SexCheck
```{r}
help(SexCheck, package = GXwasR)
```
#### <a id = "Running SexCheck"> </a> Running SexCheck
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "GXwasR_example"
LD = TRUE
LD_window_size <- 50
LD_step_size <- 5
LD_r2_threshold <- 0.02
fmax_F <- 0.2
mmin_F <- 0.8
# We will not impute sex
impute_sex <- FALSE 
compute_freq <- FALSE
SexCheckResult <-SexCheck(DataDir=DataDir,ResultDir = ResultDir, finput=finput,impute_sex=impute_sex,compute_freq =compute_freq,LD_window_size=LD_window_size,LD_step_size=LD_step_size,LD_r2_threshold=0.02,fmax_F=0.2,mmin_F=0.8)
```
```{r}
# Outputs
knitr::kable(SexCheckResult[1:10,], caption = 'First ten rows from the output of SexCheck() function.')
# Checking if there is any problematic sex assignment

knitr::kable(problematic_sex <- SexCheckResult[SexCheckResult$STATUS != "OK",], caption = 'Samples with mismatch sex label between reported and predicted sex.')
```
Four samples had problematic sex assignment. We will remove these samples from GXwasR_example dataset using FilterPlinkSample(). Users can retain these samples if they are confident about the reporting of the sex.

### <a id = "Function4"> </a> Function 4: FilterPlinkSample
```{r}
help(FilterPlinkSample, package = GXwasR)
```
#### <a id = "Running FilterPlinkSample"> </a> Running FilterPlinkSample
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "GXwasR_example"
foutput <- "PreimputeEX_QC1"
keep_remove_sample_file <- "wrong_sex_samples"
keep <- FALSE
FilterPlinkSample(DataDir = DataDir,ResultDir = ResultDir,finput = finput,foutput = foutput,keep_remove_sample_file = keep_remove_sample_file, keep = keep)
```
Now, we will copy the output file (i.e., "PreimputeEX_QC1") to DataDir to be used as input dataset for the next step of the pipeline.

```{r}
## Copying the plink files from ResultDir to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PreimputeEX_QC1")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 4:** We will filter out the samples based on missingness and heterozygosity rate.

### <a id = "Function5"> </a> Function 5: QCsample
```{r}
help(QCsample, package = GXwasR)
```
#### <a id = "Running QCsample"> </a> Running QCsample
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PreimputeEX_QC1"
foutput <- "PreimputeEX_QC2"
imiss = 0.01
het = 2
x = QCsample(DataDir = DataDir,ResultDir = ResultDir, finput = finput,foutput = "PreimputeEX_QC2", imiss = imiss,het = het)
## Samples filtered out for missingness and heterozygosity
x$HM
## Samples filtered out for missingness
x$Failed_Missingness
## Samples filtered out for heterozygosity
x$Failed_heterozygosity
```
```{r}
#Now, just for a test, let’s run the same function using “HapMapIII_NCBI36” dataset as input.
## Download the data in DataDir
Download_reference(refdata = "HapMapIII_NCBI36", wdir = DataDir)
y = QCsample(DataDir = DataDir,ResultDir = ResultDir, finput = "HapMapIII_NCBI36", foutput = "hapmap_test_qc", imiss = imiss,het = het)
## Remove the reference
 ftemp <- list.files(paste0(DataDir,"/"),pattern = "HapMapIII_NCBI36")
    invisible(do.call(file.remove,list(paste0(DataDir,"/",ftemp))))
```
Now, users need to move the QC-ed file (i.e., "PreimputeEX_QC2") to DataDir to be used as input dataset for the next step of the pipeline.
```{r}

## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
#invisible(file.remove(paste0(DataDir,"/",ftemp)))

## New QC-ed files
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PreimputeEX_QC2")
file.copy(paste0(ResultDir,"/",ftemp),DataDir)

```
**Step 5:** We will perform QC at SNP level for autosomes and sex chromosomes. We will filter out SNPs based on genotyping efficiency/call rate, Hardy-Weinberg Equilibrium (cases < 1e-10; controls < 1e-6 for case-control data), case-control differential missingness and, MAF. Also, we will remove monomorphic SNPs as well. 

### <a id = "Function6"> </a> Function 6: QCsnp
```{r}
help(QCsnp, package = GXwasR)
```
#### <a id = "Running QCsnp"> </a> Running QCsnp
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PreimputeEX_QC2"
foutput <- "PreimputeEX_QC3"
geno <- 0.02
maf <- 0.01
casecontrol <- TRUE
hweCase <- 0.000001
hweControl <- 0.000001
hweCase <- 1e-10
hweControl <- 1e-06
monomorphicSNPs <- TRUE
caldiffmiss <- TRUE
diffmissFilter <- TRUE
dmissX <- TRUE
dmissAutoY <- TRUE
ld_prunning <- FALSE
x <- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss, dmissX = dmissX, dmissAutoY = dmissAutoY, diffmissFilter = diffmissFilter)

```
```{r}
# Outputs
## There are 24 Monomorphic SNPs
x$MonomorSNPs
## There is no SNP with differential missingness between cases and controls
x$DiffMissSNPs
```

With “PreimputeEX_QC3” users could proceed to the imputation. Users could use Michigan Imputation server that contains X chromosome for this purpose. More details can be found here: https://imputationserver.readthedocs.io/en/latest/getting-started/

```{r}
# Before proceeding to that, we will remove the old QC-ed file from DataDir. 
invisible(file.remove(paste0(DataDir,"/",ftemp)))

```
# <d id = "Post-imputation QC steps:"> </d> Post-imputation QC steps:
**(A) Sex-combined QC:** 
1.Remove SNPs with IMPUTE2 info score < 0.6 and certainty < 0.8
This part should be done in the imputation server with the VCF files.

**(B) Sex-specific autosomal QC at the variant level:** 
1.Filter PAR, XTR, Ampliconic regions from the X-chromosome and combine with autosomes for autosomal QC.
2.Any SNPs in the controls with a sex difference in MAF should be carefully examined and flagged for further examination, since the differences may be due to technical confounding or sampling biases for the study cohorts
3.Prepare separate male and female subsets of the genotypes
4.Remove SNPs with MAF < 0.05 from each sex 

**(C) Sex-specific X-chromosomal QC at the variant level:**
1.Separate PAR, XTR, Ampliconic regions from the X-chromosome
2.Prepare separate male and female subsets of the genotypes
3.Remove SNPs with a missingness difference in cases vs controls, separately in each sex 
4.In each sex, filter out SNPs based on 1% MAF, 10% missingness per individual, and 5% missingness per genotype. We recommend setting the threshold filters to be the same in males and females for analyses of common variants.
5.Identify the set of SNPs found in both of the post-QC male and female datasets and create a combined dataset with only these SNPs
6.Testing for HWE across the X in females (cases and controls combined), and later removing these regions from analysis in all samples. If assuming equal allele frequency between males and females, HWE for chrX can be considered in females only as described in Khramatsova et. al., 2023.
7.Test for significantly different MAF (p < 0.05/#SNPs) between sexes in control samples only for binary traits. ChrX variants with MAF sex differences in controls may be biologically meaningful, one could consider flagging these variants rather than removing them (Khramatsova et. al., 2023.)

**(D)	Sex-specific QC at sample level: Filter samples if:**
1.Remove individual with missing genotype rate > 0.1
2.Remove individual with the absolute value of the heterozygosity F statistic  > 0.20 
3.Remove pair of individuals with IBD statistics pihat > 0.2.

## <a id = "Post-imputation QC pipeline:"> </a>Post-imputation QC pipeline:
We will use plink binary files from DataDir, prefixed as “GXwasR_example_imputed” for the input of the post-imputation QC functions and cosidered that the SNPs are passed IMPUTE2 info score < 0.6 and certainty < 0.8. 

Let's check the summary statistics of this toy imputed files.
Let's check the summary statistics of the output plink files (prefixed as “PreimputeEX_QC3”) using PlinkSummary() function.

### <a id = "Function7"> </a> Function 7: PlinkSummary
```{r}
help(PlinkSummary, package = GXwasR)
```
#### <a id = "Running PlinkSummary"> </a> Running PlinkSummary
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "GXwasR_example_imputed"
x <- PlinkSummary(DataDir, ResultDir, finput)
```
**Step 1:**Now, we will filter Pseudo-Autosomal Region (PAR), X-transposed region (XTR) and Ampliconic regions from the X-chromosome and will combine these regions with autosomes for autosomal QC. 

### <a id = "Function8"> </a> Function 8: FilterRegion
```{r}
help(FilterRegion, package = GXwasR)
```
#### <a id = "Running FilterRegion"> </a> Running FilterRegion
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "GXwasR_example_imputed"
foutput <- "PostimputeEX_QC1"
x <- FilterRegion(DataDir = DataDir,ResultDir = ResultDir,finput = finput,foutput = foutput,CHRX = TRUE,CHRY = FALSE, filterPAR = TRUE,filterXTR = TRUE,filterAmpliconic = TRUE,regionfile = FALSE,Hg = "38")
```
```{r}
# Output
# SNPs in PAR
PAR_SNPs <- x$PAR ## No PAR SNP
# SNPs in XTR
XTR_SNPs <- x$XTR ## 42 SNPs
knitr::kable(XTR_SNPs, caption = 'SNPs in XTR.')
# SNPs in Ampliconic region
Ampliconic_SNPs <- x$Ampliconic # 7 SNPs
knitr::kable(Ampliconic_SNPs, caption = 'SNPs in Ampliconic region.')
```
Now, users need to move the QC-ed files (i.e., "PostimputeEX_QC1" and “PostimputeEX_QC1_snps_extracted”) to DataDir to be used as input datasets for the next step of the pipeline.

```{r}
## Copying the new QC-ed files to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC1")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
Now, we will prepare two sets of plink binary files from “PostimputeEX_QC1” containing autosomes and X-chromosomes, respectively. For this purpose, we will again use FilterRegion().

```{r}
#Running the function to get plink files with autosomes and plink files with X chromosome
finput <- "PostimputeEX_QC1"
foutput <- "PostimputeEX_QC2"
filterCHR = 23
y <- FilterRegion(DataDir = DataDir,ResultDir = ResultDir,finput = finput,foutput = foutput,CHRX = TRUE,CHRY = FALSE, filterPAR = TRUE,filterXTR = TRUE,filterAmpliconic = TRUE,regionfile = FALSE,filterCHR = 23, Hg = "38",exclude = TRUE)
```
“PostimputeEX_QC2_snps_extracted” contains X-chromosome, while “PostimputeEX_QC2” contains autosomal SNPs.
```{r}
#Now, copying these files to DataDir.
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC2")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
We will now merge the autosomal files with PAR, XTR and Ampliconic regions using MergeRegion().

### <a id = "Function9"> </a> Function 9: MergeRegion
```{r}
help(MergeRegion, package = GXwasR)
```
#### <a id = "Running MergeRegion"> </a> Running MergeRegion
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput1 <- "PostimputeEX_QC2"
finput2 <- "PostimputeEX_QC1_snps_extracted"
foutput <- "PostimputeEX_QC3_autopxa"
y <- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput, use_common_snps = FALSE)
```
```{r}
#Now, copying these files to DataDir.
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC3")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC1")
invisible(file.remove(paste0(DataDir,"/",ftemp)))
```
**Step 2:** With “PostimputeEX_QC3_autopxa”, we will flag the SNPs in the controls with a sex difference in MAF, since the differences may be due to technical confounding or sampling biases for the study cohorts.

### <a id = "Function10"> </a> Function 10: MAFdiffSexControl
```{r}
help(MAFdiffSexControl, package = GXwasR)
```
#### <a id = "Running MAFdiffSexControl"> </a> Running MAFdiffSexControl
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC3_autopxa"
foutput <- "Test_output"
x <- MAFdiffSexControl(DataDir, ResultDir, finput,foutput, filterSNP = FALSE)
```
Since no SNP is having MAF difference in sexes in controls, we will proceed to the next steps of the pipeline.

**Step 3:**Prepare separate male and female subsets of the genotypes.

Using “PostimputeEX_QC3_autopxa” plink files. First, we will prepare sex-specific plink files.

### <a id = "Function11"> </a> Function 11: GetMFPlink
```{r}
help(GetMFPlink, package = GXwasR)
```
#### <a id = "Running GetMFPlink"> </a> Running GetMFPlink
```{r}
# Running
# Making female plink files
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC3_autopxa"
foutput <- "PostimputeEX_QC4_Female"
sex <- "females"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)

# Making male plink files
foutput <- "PostimputeEX_QC4_Male"
sex <- "males"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)
```

```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC3")
invisible(file.remove(paste0(DataDir,"/",ftemp)))

## Copying new plink files to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC4")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```

**Step 4:**Remove SNPs with MAF < 0.05 from each sex from the sex-specific autosomal plink files.

Now, using QCsnp(), we will remove SNPs with MAF < 0.05 from each sex. Except, argument “maf” all other arguments are set to NULL or FALSE so that the filtering could be done based on only MAF filter.

```{r}
# Running
# Applying MAF filter to female-specific plink files
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC4_Female"
foutput <- "PostimputeEX_QC5_Female"
geno <- NULL
maf <- 0.05
casecontrol <- FALSE
hweCase <- NULL
hweControl <- NULL
hweCase <- NULL
monomorphicSNPs <- FALSE
caldiffmiss <- FALSE
ld_prunning <- FALSE
x <- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss)

# Applying MAF filter to female-specific plink files
finput <- "PostimputeEX_QC4_Male"
foutput <- "PostimputeEX_QC5_Male"
x <- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss)
```
```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC4")
invisible(file.remove(paste0(DataDir,"/",ftemp)))

ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC5")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 5:** Combine sex-specific autosomal QC-ed files using common SNPs.
```{r}
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput1 <- "PostimputeEX_QC5_Female"
finput2 <- "PostimputeEX_QC5_Male"
foutput <- "PostimputeEX_Auto"
y <- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput, use_common_snps = TRUE)
```
```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC5")
invisible(file.remove(paste0(DataDir,"/",ftemp)))

ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_Auto")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 6:**Get sex-specific X-chromosomal plink files.
For this, we will use plink files prefixed as “PostimputeEX_QC2_snps_extracted”. As a reminder, these plink files contain X chromosomes without PAR, XTR and Ampliconic sites.
We will utilize GetMFPlink() function to get male and female plink binary files with X chromosomes. We will move these files from ResultDir to DataDir and will run the sex-specific QC steps.

```{r}
# Making female plink files with X chromosome
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC2_snps_extracted"
foutput <- "PostimputeEX_QC3_XFemale"
sex <- "females"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)

# Making male plink files X chromosome
foutput <- "PostimputeEX_QC3_XMale"
sex <- "males"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)

```
```{r}
## Copying sex-specific plink files
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC3_X")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```

**Step 7:** Apply variant level QC with sex-specific X-chromosomal plink files.
Now, we will remove SNPs with a missingness difference in cases vs controls, separately in each sex, and will filter out SNPs based on 1% MAF and 5% missingness per genotype. We recommend setting the threshold filters to be the same in males and females for analyses of common variants.
For this, we will use QCsnp() function.

```{r}
## Female-specific QC
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC3_XFemale"
foutput <- "PostimputeEX_QC4_XFemale"
geno <- 0.05 
maf <- 0.01 
casecontrol <- TRUE
hweCase <- NULL
hweControl <- NULL
hweCase <- NULL
monomorphicSNPs <- FALSE
caldiffmiss <- TRUE
diffmissFilter <- TRUE
dmissX <- TRUE # Since our input has X chromosome, setting this as TRUE wouldn’t change the result.
dmissAutoY <- FALSE # Since our input has X chromosome, setting this as TRUE wouldn’t change the result.
x <- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss, diffmissFilter = diffmissFilter, dmissX = dmissX, dmissAutoY = dmissAutoY)
```
```{r}
## For male-specific QC:
finput <- "PostimputeEX_QC3_XMale"
foutput <- "PostimputeEX_QC4_XMale"
x <- QCsnp(DataDir = DataDir, ResultDir = ResultDir, finput = finput,foutput = foutput, geno = geno, maf = maf,hweCase = hweCase, hweControl = hweControl, ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs, caldiffmiss = caldiffmiss, diffmissFilter = diffmissFilter, dmissX = dmissX, dmissAutoY = dmissAutoY)
```
```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC3")
invisible(file.remove(paste0(DataDir,"/",ftemp)))

## Copying sex-specific plink files
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC4_X")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 7:** Identify the set of SNPs found in both of the post-QC male and female datasets and create a combined dataset with only these SNPs.
```{r}
## Running the function
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput1 <- "PostimputeEX_QC4_XFemale"
finput2 <- "PostimputeEX_QC4_XMale"
foutput <- "PostimputeEX_QC5_Xchr"
y <- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput, use_common_snps = TRUE)
```
```{r}
## Copying merged plink files to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_QC5_Xchr")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
```{r}
## Removing other QC-ed files from DataDir
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC2")
invisible(file.remove(paste0(DataDir,"/",ftemp)))
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC4")
invisible(file.remove(paste0(DataDir,"/",ftemp)))
```
**Step 8:**Test for HWE across the X in females (cases and controls combined) and remove these regions from analysis for all samples. 
Users might just want to flag these SNPs instead of filtering.

### <a id = "Function12"> </a> Function 12: Xhwe
```{r}
help(Xhwe, package = GXwasR)
```
#### <a id = "Running Xhwe"> </a> Running Xhwe
```{r}
# Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeEX_QC5_Xchr"
foutput <- "PostimputeEX_Xchr"
x <- Xhwe(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput, filterSNP = TRUE)
```
```{r}
## No. of the SNPs failed the test.
length(x)# 20 SNPs in X chr in females failed HWE test.
## The failed SNPs
x
```
```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX_QC5_Xchr")
invisible(file.remove(paste0(DataDir,"/",ftemp)))

## Copying new QC-ed plink files to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeEX_Xchr")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 9:**For X-chromosomes variants, test for significantly different MAF between sexes in control samples. This test should be performed only for binary traits. We will flag the failed SNPs if any.
```{r}
finput <- "PostimputeEX_Xchr"
foutput <- "Test_output"
x <- MAFdiffSexControl(DataDir, ResultDir, finput,filterSNP = FALSE,foutput = foutput)
```

**Step 10:**Combine all the variant leveled QC-ed files i.e., autosomal and X chromosomes, and will proceed to the sex-specific QC for the samples.
```{r}
## Running the function
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput1 <- "PostimputeEX_Auto"
finput2 <- "PostimputeEX_Xchr"
foutput <- "PostimputeQC"
use_common_snps = FALSE
y <- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput, use_common_snps = FALSE)
```
```{r}
## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "PostimputeEX")
#invisible(file.remove(paste0(DataDir,"/",ftemp)))

ftemp <- list.files(paste0(ResultDir,"/"),pattern = "PostimputeQC")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 11:**Prepare prepare sex-specific plink binary files to perform sex-specific QC at sample level. 
```{r}
# Making female-specific plink files
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeQC"
foutput <- "Postimpute_Female"
sex <- "females"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)
```
```{r}
# Making male-specific plink files
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeQC"
foutput <- "Postimpute_Male"
sex <- "males"
x <- GetMFPlink(DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,sex = sex, xplink = FALSE, autoplink = FALSE)
```
```{r}
## Copying sex-specific plink files to DataDir
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "Postimpute_Female")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
ftemp <- list.files(paste0(ResultDir,"/"),pattern = "Postimpute_Male")
invisible(file.copy(paste0(ResultDir,"/",ftemp),DataDir))
```
**Step 12:**From each sex-specific plink files, remove individuals with missing genotype rate > 0.1, with absolute value of the heterozygosity F statistic  > 0.20 and IBD statistics pihat > 0.2.
For this, we will use QCsample2().

### <a id = "Function13"> </a> Function 13: QCsample2
```{r}
help(QCsample2, package = GXwasR)
```
#### <a id = "Running QCsample2"> </a> Running QCsample2
```{r}
## Running the function female-specific QC
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "Postimpute_Female"
foutput <- "Postimpute_Female1"
miss_geno = 0.1
het_Fstat = 0.2
small_sample_mod = TRUE
IBD = 0.2
ld_prunning = FALSE
keep = FALSE
x <- QCsample2(DataDir = DataDir,ResultDir = ResultDir, finput = finput,foutput = foutput,miss_geno = miss_geno, het_Fstat = het_Fstat, small_sample_mod = small_sample_mod, IBD = IBD, ld_prunning = ld_prunning, keep = keep)
```
```{r}
## Running the function male-specific QC
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "Postimpute_Male"
foutput <- "Postimpute_Male1"
miss_geno = 0.1
het_Fstat = 0.2
small_sample_mod = TRUE
IBD = 0.2
ld_prunning = FALSE
keep = FALSE
x <- QCsample2(DataDir = DataDir,ResultDir = ResultDir, finput = finput,foutput = foutput,miss_geno = miss_geno, het_Fstat = het_Fstat, small_sample_mod = small_sample_mod, IBD = IBD, ld_prunning = ld_prunning, keep = keep)
```
Since, there was no need for filtering out the samples, our final QC-ed file is "PostimputeQC".
In case of outlier samples, users need to merge the sex-specific datasets after filtering out the outliers to make a combined dataset and place that into DataDir for the next step.
```{r}
## Removing the not needed files from DataDir.
ftemp <- list.files(paste0(DataDir,"/"),pattern = "Postimpute_")
#invisible(file.remove(paste0(DataDir,"/",ftemp)))
```
# <c id = "Sex-combined and sex-stratified GWAS with XWAS:"> </c> Sex-combined and sex-stratified GWAS with XWAS:
Now, we will perform sex-aware GWAS with XWAS model.

### <a id = "Function14"> </a> Function 14: GXwas
```{r}
help(GXwas, package = GXwasR)
```
#### <a id = "Running GXwas"> </a> Running GXwas
```{r}
## Running
DataDir <- system.file("extdata", package = "GXwasR")
ResultDir = tempdir()
finput <- "PostimputeQC"
#finput <- "GXwasR_example_imputed"
ResultGXwas <- GXwas(DataDir = DataDir,ResultDir = ResultDir, finput = finput, xmodel = "FM02comb", covarfile = NULL, covartest = NULL, combtest = "fisher.method", snp_pval = 0.001, plot.jpeg = FALSE, suggestiveline = 3, genomewideline = 5.69897, ncores = 0)
```
```{r}
# Outputs
knitr::kable(ResultGXwas$CombinedWAS[1:5,], caption = 'Dataframe containing the result of FM02comb model')
knitr::kable(ResultGXwas$MaleWAS[1:5,], caption = 'Stratified GWAS result with male cohort.')
knitr::kable(ResultGXwas$FemaleWAS[1:5,], caption = 'Stratified GWAS result with female cohort.')

```
```{r}
## Top ten associations in female-specific study:
load(paste0(ResultDir,"/","FemaleWAS.Rda"))
x1 <- FemaleWAS
x2 <- x1[x1$TEST=="ADD",]
x3 <- x2[order(x2$P),]
x <- x3[1:10, -c(5:6)]
```
```{r}
## Top ten associations in male-specific study:
load(paste0(ResultDir,"/","MaleWAS.Rda"))
x1 <- MaleWAS
x2 <- x1[x1$TEST=="ADD",]
x3 <- x2[order(x2$P),]
x <- x3[1:10, -c(5:6)]
```
```{r}
## Top ten associations in FM02comb model:
load(paste0(ResultDir,"/","CombinedWAS.Rda"))
x1 <- CombinedWAS
x2 <- x1[order(x1$P),]
x <- x2[1:10,]
```
To know more about the function GXwas() to run different GWAS and XWAS models with different arguments, please follow this tutorial:**hhhdgdh**

# <c id = "Performing sex-differential test:"> </c> Performing sex-differential test:
Now, we will perform a sex-differential test:

### <a id = "Function15"> </a> Function 15: SexDiff
```{r}
help(SexDiff, package = GXwasR)
```
#### <a id = "Running SexDiff"> </a> Running SexDiff
```{r}
## Running
## Running the function
x1 <- MaleWAS
x1 <- x1[x1$TEST=="ADD",]
x1 <- x1[,c(1:4,7:8)]
colnames(x1)<- c("CHR","SNP","BP","A1","BETA_M","SE_M")
x2 <- FemaleWAS
x2 <- x2[x2$TEST=="ADD",]
x2 <- x2[,c(1:4,7:8)]
colnames(x2)<- c("CHR","SNP","BP","A1","BETA_F","SE_F")
Difftest <- SexDiff(Mfile=x1,Ffile=x2)
```
```{r}
## Significant SNPs with sex-differential effect
sig.snps <- Difftest[Difftest$adjP <0.05,]
colnames(sig.snps) <- c("SNP","CHR","BP","A1","T_statistic","Pvalue","Adjusted_Pvalue") 
knitr::kable(sig.snps, caption = 'SNPs with significant sex-differential effect.')
```
# <c id = "TutorialLinks:"> </c> Tutorial Links:

Please follow these tutorials to know more about the functionality of the package GXwasR.

Tutorial for sex-combined and sex-stratified GWAS with XWAS:**hdhghchjfvjh**

Tutorial for sex-differential test:**gdhgdcmh**

Tutorial for computing polygenic risk score:**jcjjcjh**

Tutorial for meta analysis:

Tutorial for heritability estimate and genetic correlation:

Tutorial for gene-based tests:


