<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>AncestryCheck: Evaluation of the samples' ancestry label. — AncestryCheck • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="AncestryCheck: Evaluation of the samples' ancestry label. — AncestryCheck"><meta name="description" content="This function displays the result of the ancestry analysis in a color-coded scatter plot of the first two principal
components for samples of the reference populations and the study population. Specifically, it compares the study samples'
ancestry labels to a panel representing a reference population, and it also flags the outlier samples with respect to a
chosen reference population.
The function first filters the reference and study data for non-A-T or G-C SNPs. It next conducts
LD pruning, fixes the chromosome mismatch between the reference and study datasets, checks for allele flips, updates the
locations, and flips the alleles. The two datasets are then joined, and the resulting genotype dataset is subjected to Principal
Component Analysis (PCA).
The detection of population structure down to the level of the reference dataset can then be accomplished
using PCA on this combined genotyping panel. For instance, the center of the European reference samples is determined using the
data from principal components 1 and 2 (median(PC1 europeanRef), median(PC2 europeanRef)). It determines the European reference
samples' maximum Euclidean distance (maxDist) from this center.
All study samples that are non-European, or outliers, are those whose Euclidean distances from the center are more than or
equal to the radius r= outlier threshold* maxDist. This function utilizes the HapMap phase 3 data in NCBI 36 and 1000GenomeIII
in CGRCh37. Both study and reference datasets should be of the same genome build. If not, users need to lift over one of the
datasets to the same build."><meta property="og:description" content="This function displays the result of the ancestry analysis in a color-coded scatter plot of the first two principal
components for samples of the reference populations and the study population. Specifically, it compares the study samples'
ancestry labels to a panel representing a reference population, and it also flags the outlier samples with respect to a
chosen reference population.
The function first filters the reference and study data for non-A-T or G-C SNPs. It next conducts
LD pruning, fixes the chromosome mismatch between the reference and study datasets, checks for allele flips, updates the
locations, and flips the alleles. The two datasets are then joined, and the resulting genotype dataset is subjected to Principal
Component Analysis (PCA).
The detection of population structure down to the level of the reference dataset can then be accomplished
using PCA on this combined genotyping panel. For instance, the center of the European reference samples is determined using the
data from principal components 1 and 2 (median(PC1 europeanRef), median(PC2 europeanRef)). It determines the European reference
samples' maximum Euclidean distance (maxDist) from this center.
All study samples that are non-European, or outliers, are those whose Euclidean distances from the center are more than or
equal to the radius r= outlier threshold* maxDist. This function utilizes the HapMap phase 3 data in NCBI 36 and 1000GenomeIII
in CGRCh37. Both study and reference datasets should be of the same genome build. If not, users need to lift over one of the
datasets to the same build."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_All_Functions.html">GXwasR Functions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>AncestryCheck: Evaluation of the samples' ancestry label.</h1>

      <div class="d-none name"><code>AncestryCheck.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function displays the result of the ancestry analysis in a color-coded scatter plot of the first two principal
components for samples of the reference populations and the study population. Specifically, it compares the study samples'
ancestry labels to a panel representing a reference population, and it also flags the outlier samples with respect to a
chosen reference population.</p>
<p>The function first filters the reference and study data for non-A-T or G-C SNPs. It next conducts
LD pruning, fixes the chromosome mismatch between the reference and study datasets, checks for allele flips, updates the
locations, and flips the alleles. The two datasets are then joined, and the resulting genotype dataset is subjected to Principal
Component Analysis (PCA).</p>
<p>The detection of population structure down to the level of the reference dataset can then be accomplished
using PCA on this combined genotyping panel. For instance, the center of the European reference samples is determined using the
data from principal components 1 and 2 (median(PC1 europeanRef), median(PC2 europeanRef)). It determines the European reference
samples' maximum Euclidean distance (maxDist) from this center.</p>
<p>All study samples that are non-European, or outliers, are those whose Euclidean distances from the center are more than or
equal to the radius r= outlier threshold* maxDist. This function utilizes the HapMap phase 3 data in NCBI 36 and 1000GenomeIII
in CGRCh37. Both study and reference datasets should be of the same genome build. If not, users need to lift over one of the
datasets to the same build.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">AncestryCheck</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  reference <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HapMapIII_NCBI36"</span>, <span class="st">"ThousandGenome"</span><span class="op">)</span>,</span>
<span>  filterSNP <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  studyLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  studyLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  studyLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  studyLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  referLD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  referLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  referLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  referLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  <span class="va">highLD_regions</span>,</span>
<span>  <span class="va">study_pop</span>,</span>
<span>  outlier <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  outlierOf <span class="op">=</span> <span class="st">"EUR"</span>,</span>
<span>  outlier_threshold <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the input PLINK binary files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files for the study samples.</p></dd>


<dt id="arg-reference">reference<a class="anchor" aria-label="anchor" href="#arg-reference"></a></dt>
<dd><p>Boolean value,'HapMapIII_NCBI36' and 'ThousandGenome', specifying Hapmap Phase3 (3 Consortium 2010)
 and 1000 Genomes
phase III (1000 Genomes Project Consortium 2015)
 reference population, respectively. The default is 'HapMapIII_NCBI36'.</p></dd>


<dt id="arg-filtersnp">filterSNP<a class="anchor" aria-label="anchor" href="#arg-filtersnp"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for filtering out the SNPs. The default is <code>TRUE</code>. We recommend setting it <code>FALSE</code>
only when the users are sure that they could join the study and reference samples directly.</p></dd>


<dt id="arg-studyld">studyLD<a class="anchor" aria-label="anchor" href="#arg-studyld"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for applying linkage disequilibrium (LD)-based filtering on study genotype data.</p></dd>


<dt id="arg-studyld-window-size">studyLD_window_size<a class="anchor" aria-label="anchor" href="#arg-studyld-window-size"></a></dt>
<dd><p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering of the variants for the study data.</p></dd>


<dt id="arg-studyld-step-size">studyLD_step_size<a class="anchor" aria-label="anchor" href="#arg-studyld-step-size"></a></dt>
<dd><p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for the study data.</p></dd>


<dt id="arg-studyld-r-threshold">studyLD_r2_threshold<a class="anchor" aria-label="anchor" href="#arg-studyld-r-threshold"></a></dt>
<dd><p>Numeric value between 0 to 1 of pairwise \(r^2\) threshold for LD-based filtering for the study data.</p></dd>


<dt id="arg-referld">referLD<a class="anchor" aria-label="anchor" href="#arg-referld"></a></dt>
<dd><p>Boolean value, 'TRUE' or 'FALSE' for applying linkage disequilibrium (LD)-based filtering on reference genotype data.</p></dd>


<dt id="arg-referld-window-size">referLD_window_size<a class="anchor" aria-label="anchor" href="#arg-referld-window-size"></a></dt>
<dd><p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering of the variants for the reference data.</p></dd>


<dt id="arg-referld-step-size">referLD_step_size<a class="anchor" aria-label="anchor" href="#arg-referld-step-size"></a></dt>
<dd><p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering for the reference data.</p></dd>


<dt id="arg-referld-r-threshold">referLD_r2_threshold<a class="anchor" aria-label="anchor" href="#arg-referld-r-threshold"></a></dt>
<dd><p>Numeric value between 0 to 1 of pairwise \(r^2\) threshold for LD-based filtering for the reference data.</p></dd>


<dt id="arg-highld-regions">highLD_regions<a class="anchor" aria-label="anchor" href="#arg-highld-regions"></a></dt>
<dd><p>A dataframe with known high LD regions (Anderson et al. 2010)
 is provided with the package.</p></dd>


<dt id="arg-study-pop">study_pop<a class="anchor" aria-label="anchor" href="#arg-study-pop"></a></dt>
<dd><p>A dataframe containing two columns for study in first column, sample ID (i.e., IID) and in second column, the ancestry label.</p></dd>


<dt id="arg-outlier">outlier<a class="anchor" aria-label="anchor" href="#arg-outlier"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying outlier detection will be performed or not.</p></dd>


<dt id="arg-outlierof">outlierOf<a class="anchor" aria-label="anchor" href="#arg-outlierof"></a></dt>
<dd><p>Chracter string, specifying the reference ancestry name for detecting outlier samples. The default is "outlierOf = "EUR".</p></dd>


<dt id="arg-outlier-threshold">outlier_threshold<a class="anchor" aria-label="anchor" href="#arg-outlier-threshold"></a></dt>
<dd><p>Numeric value, specifying the threshold to be be used to detect outlier samples. This threshold will be multiplied with the
Eucledean distance from the center of the PC 1 and PC2 to the maximum Euclidean distance of the reference samples. Study samples
outside this distance will be considered as outlier. The default is 3.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe with the IDs of non-European samples as outliers.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>1000 Genomes Project Consortium T (2015).
“A global reference for human genetic variation.”
<em>Nature</em>, <b>526</b>, 68–74.
<a href="https://doi.org/10.1038/nature15393" class="external-link">doi:10.1038/nature15393</a>
.<br><br> 3 Consortium TIH (2010).
“Integrating common and rare genetic variation in diverse human populations.”
<em>Nature</em>, <b>467</b>, 52–58.
<a href="https://doi.org/10.1038/nature09298" class="external-link">doi:10.1038/nature09298</a>
.<br><br> Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT (2010).
“Data quality control in genetic case-control association studies.”
<em>Nature Protocols</em>, <b>5</b>(9), 1564–1573.
<a href="https://doi.org/10.1038/nprot.2010.116" class="external-link">doi:10.1038/nprot.2010.116</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/21085122" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/21085122</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="st">"HapMapIII_NCBI36"</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"GXwasRData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">::</span><span class="va"><a href="highLD_hg19.html">highLD_hg19</a></span> <span class="co"># See: ?GXwasR::highLD_hg19</span></span></span>
<span class="r-in"><span><span class="va">study_pop</span> <span class="op">&lt;-</span> <span class="va">example_data_study_sample_ancestry</span> <span class="co"># PreimputeEX</span></span></span>
<span class="r-in"><span><span class="va">studyLD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span></span>
<span class="r-in"><span><span class="va">studyLD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="va">studyLD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span></span>
<span class="r-in"><span><span class="va">filterSNP</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">studyLD</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">referLD</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">referLD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span></span>
<span class="r-in"><span><span class="va">referLD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="va">referLD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span></span>
<span class="r-in"><span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">outlier_threshold</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">AncestryCheck</span><span class="op">(</span></span></span>
<span class="r-in"><span>  DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,</span></span>
<span class="r-in"><span>  reference <span class="op">=</span> <span class="st">"HapMapIII_NCBI36"</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>,</span></span>
<span class="r-in"><span>  study_pop <span class="op">=</span> <span class="va">study_pop</span>, studyLD <span class="op">=</span> <span class="va">studyLD</span>, referLD <span class="op">=</span> <span class="va">referLD</span>,</span></span>
<span class="r-in"><span>  outlierOf <span class="op">=</span> <span class="st">"EUR"</span>, outlier <span class="op">=</span> <span class="va">outlier</span>, outlier_threshold <span class="op">=</span> <span class="va">outlier_threshold</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Reference data 'HapMapIII_NCBI36' downloaded and extracted in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 4214 SNPs had 'A-T' and 'G-C' in study data. These SNPs were removed.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 111854 SNPs were 'A-T' and 'G-C' in reference data. These SNPs were removed.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> LD pruning was performed for study dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> LD pruning was performed for reference dataset.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> An error occurred: No common SNPs found between study and reference data. This analysis cannot be done.</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

