<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>ComputePRS: Computing polygenic risk score (PRS) — ComputePRS • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="ComputePRS: Computing polygenic risk score (PRS) — ComputePRS"><meta name="description" content="This function calculates the polygenic risk score, which is the total of allele counts (genotypes) weighted by estimated
effect sizes from genome-wide association studies. It uses C+T filtering techniques. The users could perform clumping procedure
choromosome-wise and genome-wide. Also, the function offers the choice of including several genetic principal components along with
other covariates. Using this function, users have the freedom to experiment with various clumping and thresholding arrangements to
test a wide range of various parameter values."><meta property="og:description" content="This function calculates the polygenic risk score, which is the total of allele counts (genotypes) weighted by estimated
effect sizes from genome-wide association studies. It uses C+T filtering techniques. The users could perform clumping procedure
choromosome-wise and genome-wide. Also, the function offers the choice of including several genetic principal components along with
other covariates. Using this function, users have the freedom to experiment with various clumping and thresholding arrangements to
test a wide range of various parameter values."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>ComputePRS: Computing polygenic risk score (PRS)</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>ComputePRS.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function calculates the polygenic risk score, which is the total of allele counts (genotypes) weighted by estimated
effect sizes from genome-wide association studies. It uses C+T filtering techniques. The users could perform clumping procedure
choromosome-wise and genome-wide. Also, the function offers the choice of including several genetic principal components along with
other covariates. Using this function, users have the freedom to experiment with various clumping and thresholding arrangements to
test a wide range of various parameter values.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">ComputePRS</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  <span class="va">summarystat</span>,</span>
<span>  <span class="va">phenofile</span>,</span>
<span>  covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  effectsize <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"BETA"</span>, <span class="st">"OR"</span><span class="op">)</span>,</span>
<span>  ldclump <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">LDreference</span>,</span>
<span>  <span class="va">clump_p1</span>,</span>
<span>  <span class="va">clump_p2</span>,</span>
<span>  <span class="va">clump_r2</span>,</span>
<span>  <span class="va">clump_kb</span>,</span>
<span>  byCHR <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  pthreshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.001</span>, <span class="fl">0.05</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.3</span>, <span class="fl">0.4</span>, <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>  <span class="va">highLD_regions</span>,</span>
<span>  ld_prunning <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  nPC <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  pheno_type <span class="op">=</span> <span class="st">"binary"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the all the input files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is tempdir().</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files for the genotype data i.e., the target data based on which
clumping procedure will be performed. This file needs to be in DataDir. If your target data are small (e.g. N &lt; 500) then you can use
the 1000 Genomes Project samples. Make sure to use the population that most closely reflects represents the base sample.</p></dd>


<dt id="arg-summarystat">summarystat<a class="anchor" aria-label="anchor" href="#arg-summarystat"></a></dt>
<dd><p>A dataframe object with GWAS summary statistics.</p>
<p>The mandatory column headers in this dataframe are:</p><ul><li><p><code>CHR</code>(Chromosome code)</p></li>
<li><p><code>BP</code>(Basepair position)</p></li>
<li><p><code>A1</code> (effect allele)</p></li>
<li><p><code>SNP</code> (i.e., SNP identifier)</p></li>
<li><p><code>BETA</code> or <code>OR</code> (i.e., effect-size or logarithm of odds ratio)</p></li>
<li><p><code>P</code> (i.e., p-values).</p></li>
</ul><p>Special Notes: The first three columns needed to be <code>SNP</code>, <code>A1</code> and <code>BETA</code> or <code>OR</code>.</p></dd>


<dt id="arg-phenofile">phenofile<a class="anchor" aria-label="anchor" href="#arg-phenofile"></a></dt>
<dd><p>A character string, specifying the name of the mandatory phenotype file. This is a plain text file with no header line; columns
family ID, individual ID and phenotype columns. For binary trait, the phenotypic value should be coded as 0 or 1, then it will be
recognized as a case-control study (0 for controls and 1 for cases). Missing value should be represented by "-9" or "NA". The
interested phenotype column should be labeled as "Pheno1". This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-covarfile">covarfile<a class="anchor" aria-label="anchor" href="#arg-covarfile"></a></dt>
<dd><p>A character string, specifying the name of the covariate file which is a plain .text file with no header line; columns are family ID,
individual ID and the covariates. The default is <code>NULL</code>. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-effectsize">effectsize<a class="anchor" aria-label="anchor" href="#arg-effectsize"></a></dt>
<dd><p>Boolean value, <code>BETA</code> or <code>OR</code>, specifying the type of the GWAS effectsize. The default is <code>BETA</code>.</p></dd>


<dt id="arg-ldclump">ldclump<a class="anchor" aria-label="anchor" href="#arg-ldclump"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to perform clumping or not.</p></dd>


<dt id="arg-ldreference">LDreference<a class="anchor" aria-label="anchor" href="#arg-ldreference"></a></dt>
<dd><p>A character string, specifying the  prefix of the PLINK files of the population reference panel of the same ancestry, and ideally
the one that was used for imputing your target dataset. These files should be in <code>DataDir</code>.</p></dd>


<dt id="arg-clump-p-">clump_p1<a class="anchor" aria-label="anchor" href="#arg-clump-p-"></a></dt>
<dd><p>Numeric value, specifying the significance threshold for index SNPs if <code>ldclump</code> was set to be <code>TRUE</code>. The default is 0.0001.</p></dd>


<dt id="arg-clump-p-">clump_p2<a class="anchor" aria-label="anchor" href="#arg-clump-p-"></a></dt>
<dd><p>Numeric value, specifying the secondary significance threshold for clumped SNPs if <code>ldclump</code> was set to be <code>TRUE</code>. The default is 0.01</p></dd>


<dt id="arg-clump-r-">clump_r2<a class="anchor" aria-label="anchor" href="#arg-clump-r-"></a></dt>
<dd><p>Numeric value, specifying the linkage disequilibrium (LD) threshold for clumping if <code>ldclump</code> was set to be <code>TRUE</code>. The default is 0.50.</p></dd>


<dt id="arg-clump-kb">clump_kb<a class="anchor" aria-label="anchor" href="#arg-clump-kb"></a></dt>
<dd><p>Integer value, specifying the physical distance threshold in base-pair for clumping if <code>ldclump</code> was set to be <code>TRUE</code>. The default is 250.</p></dd>


<dt id="arg-bychr">byCHR<a class="anchor" aria-label="anchor" href="#arg-bychr"></a></dt>
<dd><p>Boolean value, 'TRUE' or 'FALSE', specifying chromosome-wise clumping procedure if <code>ldclump</code> was set to be <code>TRUE</code>. The default is <code>TRUE</code></p></dd>


<dt id="arg-pthreshold">pthreshold<a class="anchor" aria-label="anchor" href="#arg-pthreshold"></a></dt>
<dd><p>Numeric vector, containing several p value thresholds to maximize predictive ability of the derived polygenic scores.</p></dd>


<dt id="arg-highld-regions">highLD_regions<a class="anchor" aria-label="anchor" href="#arg-highld-regions"></a></dt>
<dd><p>Character string, specifying the .txt file name with known genomic regions with high LD. The default is <code>NULL</code>.</p></dd>


<dt id="arg-ld-prunning">ld_prunning<a class="anchor" aria-label="anchor" href="#arg-ld-prunning"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for LD-based filtering for computing genetic PC as covariates.</p></dd>


<dt id="arg-window-size">window_size<a class="anchor" aria-label="anchor" href="#arg-window-size"></a></dt>
<dd><p>Integer value, specifying a window size in variant count or kilobase for LD-based filtering in computing genetic PC. The default is 50.</p></dd>


<dt id="arg-step-size">step_size<a class="anchor" aria-label="anchor" href="#arg-step-size"></a></dt>
<dd><p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering in computing genetic PCs. The default is 5.</p></dd>


<dt id="arg-r-threshold">r2_threshold<a class="anchor" aria-label="anchor" href="#arg-r-threshold"></a></dt>
<dd><p>Numeric value between 0 to 1 of pairwise \(r^2\) threshold for LD-based filtering in computing genetic PCs. The default is 0.02.</p></dd>


<dt id="arg-npc">nPC<a class="anchor" aria-label="anchor" href="#arg-npc"></a></dt>
<dd><p>Positive integer value, specifying the number of genetic PCs to be included as predictor in the PRS model fit. The default is 6.</p></dd>


<dt id="arg-pheno-type">pheno_type<a class="anchor" aria-label="anchor" href="#arg-pheno-type"></a></dt>
<dd><p>Boolean value, ‘binary’ or ‘quantitative’, specifying the type of the trait. The default is ‘binary’.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list object containing a dataframe a numeric value, a GeneticPC plot (if requested), and a PRS plot. The dataframe,PRS, contains four mandatory
columns, such as, IID (i.e., Individual ID), FID (i.e., Family ID), Pheno1 (i.e., the trait for PRS) and Score (i.e., the best PRS).
Other columns of covariates could be there. The numeric value, BestP contains the threshold of
of the best p-value for the best pRS model fit.</p>
<p>Also, the function produces several plots such as p-value thresholds vs PRS model fit and PRS distribution among male and females.
For case-control data, it shows PRS distribution among cases and controls and ROC curves as well.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex1"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_phenofile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_covarfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_pthresoldfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL</span></span></span>
<span class="r-in"><span><span class="co"># The interested phenotype column should be labeled as "Pheno1".</span></span></span>
<span class="r-in"><span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="va">Example_covarfile</span></span></span>
<span class="r-in"><span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span></span>
<span class="r-in"><span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span></span>
<span class="r-in"><span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span></span>
<span class="r-in"><span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span></span>
<span class="r-in"><span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span></span>
<span class="r-in"><span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span></span>
<span class="r-in"><span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span></span>
<span class="r-in"><span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span></span>
<span class="r-in"><span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span></span>
<span class="r-in"><span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PRSresult</span> <span class="op">&lt;-</span> <span class="fu">ComputePRS</span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>,</span></span>
<span class="r-in"><span>    effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">clump_p1</span>, <span class="va">clump_p2</span>,</span></span>
<span class="r-in"><span>    <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>,</span></span>
<span class="r-in"><span>    ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>,</span></span>
<span class="r-in"><span>    pheno_type <span class="op">=</span> <span class="st">"binary"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.001</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.001</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.3</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 0.5</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Computing PRS for threshold 0.5</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span></span>
<span class="r-in"><span><span class="va">PRS</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS</span></span></span>
<span class="r-in"><span><span class="va">PRS</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>, <span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        FID     IID Pheno1 AGE testcovar       PC1         PC2         PC3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  EUR_FIN HG00171      1  36         1 0.0964618 -0.03825140  0.00903796</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  EUR_FIN HG00173      1  81         1 0.0743932 -0.05672710  0.03969940</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  EUR_FIN HG00174      1  83         1 0.0645349 -0.05610350 -0.01457860</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  EUR_FIN HG00176      2  75         0 0.0832472 -0.05602080  0.03026540</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  EUR_FIN HG00177      1  88         1 0.0775687 -0.00946444  0.04517850</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  EUR_FIN HG00178      1  24         1 0.0661714 -0.05049060  0.02485390</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  EUR_FIN HG00179      2  78         1 0.0806110 -0.01154090 -0.00750723</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8  EUR_FIN HG00180      1  39         1 0.0840638  0.00117830  0.01286460</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  EUR_FIN HG00182      1  50         1 0.1029760 -0.04300460  0.01999710</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 EUR_FIN HG00183      1  58         0 0.0827103 -0.03865300  0.05093660</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>            PC4         PC5         PC6      SCORE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  -0.08952930 -0.02522910  0.01465430 0.04842120</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   0.02053430  0.02304100 -0.03603800 0.02937420</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3  -0.02643430 -0.00180949  0.01445210 0.00996742</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  -0.00110770 -0.05303710 -0.03358030 0.10420600</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  -0.03732150 -0.06174310 -0.07199510 0.03143030</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6  -0.08280660  0.04094180  0.00559525 0.07226890</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7  -0.03288230  0.03559730  0.04245120 0.11652400</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8   0.05915400  0.05476670 -0.01015950 0.06171590</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9  -0.03023650 -0.09019110 -0.03785150 0.02837030</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 10 -0.00153775 -0.10456100  0.06236340 0.03862970</span>
<span class="r-in"><span><span class="co">## The best threshold</span></span></span>
<span class="r-in"><span><span class="va">BestPvalue</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">BestP</span><span class="op">$</span><span class="va">Threshold</span></span></span>
<span class="r-in"><span><span class="va">BestPvalue</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.05</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

