<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>FilterRegion: Filter chromosomal regions. — FilterRegion • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="FilterRegion: Filter chromosomal regions. — FilterRegion"><meta name="description" content="Filtering Pseudo-Autosomal Region (PAR), X-transposed region (XTR), Ampliconic, filter based on chromosome code or
user-defined regions from input PLINK files. Only one type of filtering can be done from three types, either by region
(using regionfile = TRUE), by chromosome (filterCHR) or by any combination of these three, filterPAR,
filterXTR and filterAmpliconic."><meta property="og:description" content="Filtering Pseudo-Autosomal Region (PAR), X-transposed region (XTR), Ampliconic, filter based on chromosome code or
user-defined regions from input PLINK files. Only one type of filtering can be done from three types, either by region
(using regionfile = TRUE), by chromosome (filterCHR) or by any combination of these three, filterPAR,
filterXTR and filterAmpliconic."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>FilterRegion: Filter chromosomal regions.</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>FilterRegion.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Filtering Pseudo-Autosomal Region (PAR), X-transposed region (XTR), Ampliconic, filter based on chromosome code or
user-defined regions from input PLINK files. Only one type of filtering can be done from three types, either by region
(using <code>regionfile</code> = <code>TRUE</code>), by chromosome (<code>filterCHR</code>) or by any combination of these three, <code>filterPAR</code>,
<code>filterXTR</code> and <code>filterAmpliconic.</code></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">FilterRegion</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  <span class="va">ResultDir</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  <span class="va">foutput</span>,</span>
<span>  CHRX <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  CHRY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  filterPAR <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  filterXTR <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  filterAmpliconic <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  regionfile <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  filterCHR <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  Hg <span class="op">=</span> <span class="st">"19"</span>,</span>
<span>  exclude <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the input PLINK binary files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files.</p></dd>


<dt id="arg-foutput">foutput<a class="anchor" aria-label="anchor" href="#arg-foutput"></a></dt>
<dd><p>Character string, specifying the prefix of the output PLINK binary files if the filtering option for the SNPs is chosen.
The default is "FALSE".</p></dd>


<dt id="arg-chrx">CHRX<a class="anchor" aria-label="anchor" href="#arg-chrx"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter/flag regions from chromosome X. The default is <code>TRUE</code>. Note: <code>CHRX</code> only in effect if
one of <code>filterPAR</code>, <code>filterXTR</code> or <code>filterAmpliconic</code> filter is in effect.</p></dd>


<dt id="arg-chry">CHRY<a class="anchor" aria-label="anchor" href="#arg-chry"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter/flag regions from chromosome X. The default is <code>FALSE</code>. Note: CHRY only in effect
if one of <code>filterPAR</code>, <code>filterXTR</code> or <code>filterAmpliconic</code> filter is in effect.</p></dd>


<dt id="arg-filterpar">filterPAR<a class="anchor" aria-label="anchor" href="#arg-filterpar"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter out PARs from input PLINK file. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-filterxtr">filterXTR<a class="anchor" aria-label="anchor" href="#arg-filterxtr"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter out XTRs from input PLINK file. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-filterampliconic">filterAmpliconic<a class="anchor" aria-label="anchor" href="#arg-filterampliconic"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter out Ampliconic regions from input PLINK file. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-regionfile">regionfile<a class="anchor" aria-label="anchor" href="#arg-regionfile"></a></dt>
<dd><p>Character string, specifying the name of the .txt file containing the user-defined regions to be filtered out from input PLINK
file in bed format. The default is <code>FALSE</code>. If <code>regionfile</code> = <code>TRUE</code>, only this filtering will be in effect. Also, PAR, XTR and
Ampliconic SNPs from X-chomosome will be flagged and returned.</p></dd>


<dt id="arg-filterchr">filterCHR<a class="anchor" aria-label="anchor" href="#arg-filterchr"></a></dt>
<dd><p>Vector value with positive integer, specifying the chromosome code to filter/flag the SNPs. The default is 0, means no filtering
based on chromosome code. For non-zero values of this argument, the function will only consider the chromosome code to filter or
flag. All other filtering will not work. If filterCHR = TRUE, only this filtering will be in effect. Also, PAR, XTR and Ampliconic
SNPs from X-chomosome will be flagged and returned.</p></dd>


<dt id="arg-hg">Hg<a class="anchor" aria-label="anchor" href="#arg-hg"></a></dt>
<dd><p>Character value, '19', or '38', specifying which genome build to use for PAR, XTR and Ampliconic regions. The default is Hg = "19".</p></dd>


<dt id="arg-exclude">exclude<a class="anchor" aria-label="anchor" href="#arg-exclude"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> to filter and flag or only flag the SNPs. The default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of three dataframes: PAR containing SNPs from PAR regions; XTR containing SNPs from XTR region and Ampliconic containing
SNPs from Ampliconic region.</p>
<p>For non-zero value of <code>filterCHR</code>, a dataframe containing the excluded/flagged SNPs will be returned.</p>
<p>For <code>exclude</code> = <code>TRUE</code>, two sets of PLINK binary files will be produced in ResultDir. One set will have the remaining SNPs after
filtering and other one will have the discarded SNPs.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">FilterRegion</span><span class="op">(</span></span></span>
<span class="r-in"><span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span></span>
<span class="r-in"><span>    finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">TRUE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>    filterPAR <span class="op">=</span> <span class="cn">TRUE</span>, filterXTR <span class="op">=</span> <span class="cn">TRUE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="cn">NULL</span>, Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> chrX</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> There is no PAR region in the input data. Argument filterPAR cannot set to be TRUE.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Changing filterPAR to FALSE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> There is no XTR region in the input data. Argument filterXTR cannot set to be TRUE.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Changing filterXTR to FALSE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ampliconic SNPs:9</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 9 SNPs are discarded.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> PLINK files with passed SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpO7c0S8 prefixed as PostimputeEX_QC1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> PLINK files with discarded SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpO7c0S8 prefixed as PostimputeEX_QC1_snps_extracted</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

