<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>GeneticCorrBT: Computing genetic correlation between two traits. — GeneticCorrBT • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GeneticCorrBT: Computing genetic correlation between two traits. — GeneticCorrBT"><meta name="description" content="This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise."><meta property="og:description" content="This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_All_Functions.html">GXwasR Functions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>GeneticCorrBT: Computing genetic correlation between two traits.</h1>

      <div class="d-none name"><code>GeneticCorrBT.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">GeneticCorrBT</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  <span class="va">ResultDir</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  byCHR <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  REMLalgo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  nitr <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  <span class="va">phenofile</span>,</span>
<span>  cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  computeGRM <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  grmfile_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  partGRM <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  autosome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Xsome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  nGRM <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  cripticut <span class="op">=</span> <span class="fl">0.025</span>,</span>
<span>  minMAF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxMAF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  excludeResidual <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the all the input files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files for the genotype data. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-bychr">byCHR<a class="anchor" aria-label="anchor" href="#arg-bychr"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether the analysis will be performed chromosome wise or not. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-remlalgo">REMLalgo<a class="anchor" aria-label="anchor" href="#arg-remlalgo"></a></dt>
<dd><p>Integer value of 0, 1 or 2, specifying the algorithm to run REML iterations, 0 for average information (AI), 1 for Fisher-scoring and
2 for EM. The default option is 0, i.e. AI-REML (1).</p></dd>


<dt id="arg-nitr">nitr<a class="anchor" aria-label="anchor" href="#arg-nitr"></a></dt>
<dd><p>Integer value, specifying the number of iterations for performing the REML. The default is 100.</p></dd>


<dt id="arg-phenofile">phenofile<a class="anchor" aria-label="anchor" href="#arg-phenofile"></a></dt>
<dd><p>A dataframe for Bivar RELM has four columns <code>family ID</code>, <code>individual ID</code> and two trait columns. For binary trait, the phenotypic value
should be coded as 0 or 1, then it will be recognized as a case-control study (0 for controls and 1 for cases). Missing value should
be represented by "-9" or "NA".</p></dd>


<dt id="arg-cat-covarfile">cat_covarfile<a class="anchor" aria-label="anchor" href="#arg-cat-covarfile"></a></dt>
<dd><p>A character string, specifying the name of the categorical covariate file which is a plain text file with no header line; columns are
<code>family ID</code>, <code>individual ID</code> and discrete covariates. The default is <code>NULL</code>. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-quant-covarfile">quant_covarfile<a class="anchor" aria-label="anchor" href="#arg-quant-covarfile"></a></dt>
<dd><p>A character string, specifying the name of the quantitative covariate file which is a plain text file with no header line; columns
are <code>family ID</code>, <code>individual ID</code> and continuous covariates. The default is <code>NULL</code>. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-computegrm">computeGRM<a class="anchor" aria-label="anchor" href="#arg-computegrm"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to compute GRM matrices or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-grmfile-name">grmfile_name<a class="anchor" aria-label="anchor" href="#arg-grmfile-name"></a></dt>
<dd><p>A string of characters specifying the prefix of autosomal .grm.bin file. Users need to provide separate GRM files for autosomes
and X chromosome in <code>ResultDir</code>. The X chromosomal GRM file should have "x" added in the autosomal prefix as file name.</p>
<p>For instance, if autosomal file is "ABC.grm.bin", then X chromosomal file should be "xABC.grm.bim". If you are providing
chromosome-wise GRMs, then the prefix should add "ChrNumber_" at the starting of the prefix like, "Chr1_ABC.grm.bin".
The default is <code>NULL</code>.</p></dd>


<dt id="arg-partgrm">partGRM<a class="anchor" aria-label="anchor" href="#arg-partgrm"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether the GRM will be partitioned into n parts (by row) in GREML model. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-autosome">autosome<a class="anchor" aria-label="anchor" href="#arg-autosome"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether estimate of heritability will be done for autosomes or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-xsome">Xsome<a class="anchor" aria-label="anchor" href="#arg-xsome"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether estimate of heritability will be done for X chromosome or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-ngrm">nGRM<a class="anchor" aria-label="anchor" href="#arg-ngrm"></a></dt>
<dd><p>Integer value, specifying the number of the partition of the GRM in GREML model. The default is 3.</p></dd>


<dt id="arg-cripticut">cripticut<a class="anchor" aria-label="anchor" href="#arg-cripticut"></a></dt>
<dd><p>Numeric value, specifying the threshold to create a new GRM of "unrelated" individuals in GREML model. The default is arbitrary
chosen as 0.025 following (Yang et al. 2011)
.</p></dd>


<dt id="arg-minmaf">minMAF<a class="anchor" aria-label="anchor" href="#arg-minmaf"></a></dt>
<dd><p>Positive numeric value (&lt; maxMAF), specifying the minimum threshold for the MAF filter of the SNPs in the Bivariate GREML model.</p></dd>


<dt id="arg-maxmaf">maxMAF<a class="anchor" aria-label="anchor" href="#arg-maxmaf"></a></dt>
<dd><p>Positive numeric value (minMAF,1), specifying the maximum threshold for the MAF filter of the SNPs in the Bivariate GREML model.</p></dd>


<dt id="arg-excluderesidual">excludeResidual<a class="anchor" aria-label="anchor" href="#arg-excluderesidual"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to drop the residual covariance from the model. Recommended to set this <code>TRUE</code>
if the traits were measured on different individuals. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-ncores">ncores<a class="anchor" aria-label="anchor" href="#arg-ncores"></a></dt>
<dd><p>Integer value, specifying the number of cores to be used.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe with minimum three columns:</p><ul><li><p>Source" (i.e., source of heritability)</p></li>
<li><p>Variance" (i.e. estimated heritability)</p></li>
<li><p>SE" (i.e., standard error of the estimated heritability)</p></li>
</ul><p>Source column will have rows, such as V(G)_tr1 (genetic variance for trait 1), V(G)_tr2 (genetic variance for trait 2),
C(G)_tr12 (genetic covariance between traits 1 and 2),V(e)_tr1 (residual variance for trait 1), V(e)_tr2 (residual variance for trait 2),
C(e)_tr12 (residual covariance between traits 1 and 2), Vp_tr1 (proportion of variance explained by all SNPs for trait 1),
Vp_tr2 (proportion of variance explained by all SNPs for trait 2), V(G)/Vp_tr1 (phenotypic variance for trait 1),
V(G)/Vp_tr2 (phenotypic variance for trait 2), rG (genetic correlation) and n (sample size). In case of chromosome-wise
analysis, there will be 'chromosome' column for chromosome code.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Lee SH, Wray NR, Goddard ME, Visscher PM (2012).
“Estimation of pleiotropy between complex diseases using SNP-derived genomic relationships and restricted maximum likelihood.”
<em>Bioinformatics</em>, <b>28</b>, 2540–2542.
<a href="https://doi.org/10.1093/bioinformatics/bts474" class="external-link">doi:10.1093/bioinformatics/bts474</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/22843982" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/22843982</a>.<br><br> Yang J, Lee SH, Goddard ME, Visscher PM (2011).
“GCTA: a tool for Genome-wide Complex Trait Analysis.”
<em>American Journal of Human Genetics</em>, <b>88</b>(1), 76–82.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">REMLalgo</span> <span class="op">&lt;-</span> <span class="fl">0</span></span></span>
<span class="r-in"><span><span class="va">nitr</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"GXwasRData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL</span></span></span>
<span class="r-in"><span><span class="va">cat_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">quant_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">partGRM</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Partition the GRM into m parts (by row),</span></span></span>
<span class="r-in"><span><span class="va">autosome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">Xsome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">cripticut</span> <span class="op">&lt;-</span> <span class="fl">0.025</span></span></span>
<span class="r-in"><span><span class="va">minMAF</span> <span class="op">&lt;-</span> <span class="fl">0.01</span> <span class="co"># if MAF filter apply</span></span></span>
<span class="r-in"><span><span class="va">maxMAF</span> <span class="op">&lt;-</span> <span class="fl">0.04</span></span></span>
<span class="r-in"><span><span class="va">excludeResidual</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">genetic_correlation</span> <span class="op">&lt;-</span> <span class="fu">GeneticCorrBT</span><span class="op">(</span></span></span>
<span class="r-in"><span>  DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, byCHR <span class="op">=</span> <span class="va">byCHR</span>,</span></span>
<span class="r-in"><span>  REMLalgo <span class="op">=</span> <span class="fl">0</span>, nitr <span class="op">=</span> <span class="fl">10</span>, phenofile <span class="op">=</span> <span class="va">phenofile</span>, cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>, quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>  partGRM <span class="op">=</span> <span class="cn">FALSE</span>, autosome <span class="op">=</span> <span class="cn">TRUE</span>, Xsome <span class="op">=</span> <span class="cn">TRUE</span>, nGRM <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>  cripticut <span class="op">=</span> <span class="fl">0.025</span>, minMAF <span class="op">=</span> <span class="cn">NULL</span>, maxMAF <span class="op">=</span> <span class="cn">NULL</span>, excludeResidual <span class="op">=</span> <span class="cn">TRUE</span>, ncores <span class="op">=</span> <span class="va">ncores</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:01 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/1test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.124504 0.0220401 0.0229244  0.126658 0.0225898</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 330.05</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	334.26	0.08061	0.01023	0.01128	0.15742	0.03153	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	343.88	0.05138	0.00213	0.00469	0.18285	0.03878	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.62	0.03236	0.00002	0.00078	0.20155	0.04091	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	350.43	0.01988	0.00014	-0.00166	0.21472	0.04109	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	351.09	0.01092	0.00075	-0.00286	0.22398	0.04135	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	351.29	0.00000	0.00000	-0.00000	0.23846	0.03513	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	348.47	0.00000	0.00000	-0.00000	0.23783	0.02935	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	340.55	0.00000	0.00000	-0.00000	0.23783	0.02681	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	334.23	0.00000	0.00000	-0.00000	0.23783	0.02517	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:01 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/2test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123792 0.0218918 0.0230725  0.127096 0.0228552</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 327.422</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	331.72	0.07673	0.00572	0.01201	0.16079	0.03519	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	344.21	0.04528	0.00082	0.00608	0.18797	0.04005	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.77	0.02619	0.00030	0.00278	0.20702	0.04052	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	350.09	0.01424	0.00005	0.00084	0.22012	0.04062	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	350.64	0.00643	0.00002	-0.00035	0.22912	0.04056	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	350.86	0.00000	0.00000	-0.00000	0.23871	0.03045	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	342.66	0.00000	0.00000	-0.00000	0.23881	0.02720	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	335.34	0.00000	0.00000	-0.00000	0.23880	0.02534	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	329.39	0.00000	0.00000	-0.00000	0.23880	0.02405	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:02 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/3test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123913 0.0220185 0.0229477  0.127457 0.0222471</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 334.741</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	338.61	0.07354	0.01788	0.01222	0.16338	0.02448	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	346.39	0.04650	0.01572	0.00651	0.18726	0.02617	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	349.45	0.03044	0.01460	0.00304	0.20340	0.02725	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	350.74	0.02051	0.01404	0.00079	0.21430	0.02790	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	351.30	0.01423	0.01378	-0.00071	0.22162	0.02827	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	351.56	0.00148	0.01347	-0.00389	0.23714	0.02889	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	351.76	0.00272	0.01392	-0.00385	0.23632	0.02851	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	351.77	0.00286	0.01403	-0.00389	0.23620	0.02841	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	351.77	0.00291	0.01406	-0.00390	0.23615	0.02838	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 4</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:02 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/4test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.124421 0.0220363 0.0229478  0.127109 0.0223357</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 332.918</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	336.87	0.07372	0.01630	0.01199	0.16409	0.02604	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	344.74	0.04069	0.01256	0.00562	0.19310	0.02917	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.44	0.01864	0.01005	0.00160	0.21473	0.03158	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	350.23	0.00375	0.00834	-0.00103	0.23045	0.03337	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	351.10	0.00083	0.00308	-0.00160	0.23531	0.03145	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	348.39	0.00128	0.00244	-0.00177	0.23801	0.02243	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	328.18	0.00111	0.00228	-0.00159	0.23803	0.01658	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	291.74	0.00101	0.00235	-0.00154	0.23800	0.01077	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	206.43	0.00091	0.00278	-0.00159	0.23799	0.00327	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:02 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/5test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123941 0.0219362 0.0230565  0.127451 0.0226298</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 330.037</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	334.14	0.07744	0.01240	0.01206	0.16023	0.02952	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	343.54	0.04711	0.00633	0.00598	0.18671	0.03486	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	347.88	0.02702	0.00247	0.00238	0.20654	0.03871	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	349.87	0.01365	0.00002	0.00018	0.22080	0.04136	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	350.77	0.00441	0.00026	-0.00106	0.23088	0.04162	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	351.05	0.00000	0.00000	-0.00000	0.23855	0.02736	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	335.79	0.00000	0.00000	-0.00000	0.23850	0.02036	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	303.32	0.00000	0.00000	-0.00000	0.23849	0.01430	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	230.71	0.00000	0.00000	-0.00000	0.23849	0.00715	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:02 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/6test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.124004 0.0220242 0.0230103  0.128749 0.0223639</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 329.815</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	334.05	0.05978	0.01629	0.01133	0.17499	0.02597	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	343.90	0.02092	0.01238	0.00481	0.20929	0.02924	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.52	0.00009	0.00798	0.00086	0.23045	0.02999	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	349.79	0.00025	0.00679	-0.00129	0.23208	0.02361	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	342.61	0.00070	0.00590	-0.00203	0.23356	0.01812	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	324.03	0.00143	0.00521	-0.00273	0.23644	0.00074	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	-333.62	0.00087	0.00832	-0.00268	0.23658	0.00181	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	58.63	0.00085	0.01244	-0.00325	0.23660	0.00441	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	248.89	0.00112	0.00417	-0.00216	0.23722	0.00626	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 7</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:02 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/7test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.124442 0.0219988 0.0229822  0.127455 0.0227182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 327.728</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	332.17	0.07492	0.01130	0.01055	0.16196	0.03035	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	342.90	0.04212	0.00460	0.00347	0.19032	0.03621	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.19	0.02087	0.00049	-0.00076	0.21134	0.04034	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	350.68	0.00605	0.00097	-0.00243	0.22633	0.04112	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	351.34	0.00000	0.00000	-0.00000	0.23396	0.03904	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	350.51	0.00000	0.00000	-0.00000	0.23663	0.01653	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	265.73	0.00000	0.00000	-0.00000	0.23667	0.00093	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	-5212.73	0.00000	0.00000	-0.00000	0.23668	0.00185	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	-2222.98	0.00000	0.00000	-0.00000	0.23668	0.00364	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:03 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/8test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:   0.12313 0.0217658 0.0232932   0.12957 0.0230149</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 322.495</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	327.28	0.05765	0.00946	0.01244	0.17673	0.03182	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	341.27	0.01087	0.00261	0.00533	0.21702	0.03861	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.58	0.00000	0.00000	0.00000	0.22816	0.01999	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	300.37	0.00000	0.00000	-0.00000	0.22944	0.02424	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	324.95	0.00000	0.00000	-0.00000	0.23225	0.03887	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	350.43	0.00000	0.00000	-0.00000	0.23226	0.00992	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	101.50	0.00000	0.00000	-0.00000	0.23227	0.01255	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	191.95	0.00000	0.00000	-0.00000	0.23228	0.02253	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	316.83	0.00000	0.00000	-0.00000	0.23228	0.02702	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 9</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:03 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/9test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123436 0.0219414 0.0231149  0.129608 0.0226596</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 327.03</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	331.62	0.06501	0.01398	0.01097	0.16911	0.02797	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	342.84	0.03287	0.00896	0.00466	0.19717	0.03229	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	347.74	0.01424	0.00571	0.00106	0.21640	0.03552	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	349.88	0.00303	0.00353	-0.00113	0.22935	0.03788	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	350.84	0.00004	0.00004	-0.00004	0.23412	0.03764	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	350.01	0.00000	0.00000	-0.00000	0.23782	0.02216	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	314.85	0.00000	0.00000	-0.00000	0.23786	0.01250	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	190.64	0.00000	0.00000	-0.00000	0.23785	0.00111	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	-4249.75	0.00000	0.00000	-0.00000	0.23785	0.00219	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:03 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/10test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123226 0.0217355 0.0232332  0.129584 0.0228614</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 325.737</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	330.34	0.05536	0.01210	0.01202	0.17790	0.02943	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	343.11	0.01271	0.00711	0.00667	0.21588	0.03385	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.50	0.00113	0.00328	0.00193	0.22916	0.02920	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	345.17	0.00038	0.00229	0.00094	0.23154	0.02115	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	321.02	0.00003	0.00199	0.00023	0.23460	0.03184	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	347.21	0.00000	0.00112	-0.00001	0.23488	0.00308	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	-601.65	0.00000	0.00139	-0.00003	0.23489	0.00400	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	-323.56	0.00000	0.00238	-0.00009	0.23493	0.00769	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	101.47	0.00000	0.00274	-0.00012	0.23494	0.00975	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 23</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> GRM has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.bin]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Number of SNPs in each pair of individuals has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.N.bin]</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:03 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/23test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/xGXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.125247 0.0219977 0.0229266  0.121052 0.0231398</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 342.957</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	347.47	0.12865	0.01242	0.01066	0.10457	0.02908	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	358.03	0.13520	0.00746	0.00435	0.09522	0.03324	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	362.47	0.14145	0.00452	0.00046	0.08907	0.03622	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	364.51	0.14658	0.00268	-0.00211	0.08486	0.03834	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	365.47	0.15049	0.00148	-0.00386	0.08197	0.03984	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	365.93	0.15916	0.00037	-0.00763	0.07571	0.04217	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	366.30	0.15942	0.00037	-0.00767	0.07590	0.04232	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	366.30	0.15942	0.00037	-0.00767	0.07590	0.04231	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Log-likelihood ratio converged.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Summary result of REML analysis:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Source	Variance	SE</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(G)_tr1	0.159417	0.035638</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(G)_tr2	0.000369	0.005043</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> C(G)_tr12	-0.007666	0.005905</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(e)_tr1	0.075903	0.025711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(e)_tr2	0.042314	0.006211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Vp_tr1	0.235320	0.021614</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Vp_tr2	0.042683	0.003656</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(G)/Vp_tr1	0.677449	0.116797</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> V(G)/Vp_tr2	0.008637	0.118144</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rG	-1.000000	6.827093</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Sampling variance/covariance of the estimates of variance components:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1.270090e-03	2.392373e-07	-2.182865e-05	-7.319788e-04	6.440138e-07	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2.392373e-07	2.542983e-05	-2.155682e-06	2.959332e-07	-2.531750e-05	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -2.182865e-05	-2.155682e-06	3.486608e-05	5.544195e-06	-2.141574e-07	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -7.319788e-04	2.959332e-07	5.544195e-06	6.610370e-04	-5.885010e-07	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6.440138e-07	-2.531750e-05	-2.141574e-07	-5.885010e-07	3.857148e-05	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Summary result of REML analysis has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/23test_bireml.hsq].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis finished at 19:19:03 EDT on Wed Jun 18 2025</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Overall computational time: 0.05 sec.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 24</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Genome-wide Complex Trait Analysis (GCTA)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * version v1.94.4 Mac</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * (C) 2010-present, Yang Lab, Westlake University</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> * Please report bugs to Jian Yang &lt;jian.yang@westlake.edu.cn&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> *******************************************************************</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Analysis started at 19:19:03 EDT on Wed Jun 18 2025.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Hostname: DBMI-M-WCMF9QM</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Accepted options:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar 1 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-alg 0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-maxit 10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --grm /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --pheno /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --reml-bivar-nocove</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --thread-num 3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --out /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/24test_bireml</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: the program will be running on 3 threads.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading IDs of the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 IDs are read from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.id].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading the GRM from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRM for 276 individuals are included from [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp9Evg3L/GXwasR.grm.bin].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Reading phenotypes from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> There are 2 traits specified in the file [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Traits 1 and 2 are included in the bivariate analysis.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Non-missing phenotypes of 277 individuals are included from [/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp9Evg3L/GCphenofile].</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 individuals are in common in these files.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 276 non-missing phenotypes for trait #1 and 276 for trait #2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 108 cases and 168 controls for trait #1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: we can specify the disease prevalence by the option --reml-bivar-prevalence so that GCTA can transform the variance explained to the underlying liability scale.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Constructing the covariances</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Performing bivariate REML analysis ... (Note: may take hours depending on sample size).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 552 observations, 2 fixed effect(s), and 5 variance component(s)(including residual variance).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Calculating prior values of variance components by EM-REML ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Updated prior values:  0.123226 0.0217355 0.0232332  0.129584 0.0228614</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> logL: 325.737</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Running AI-REML algorithm ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Iter.	logL	V(G)_tr1	V(G)_tr2	C(G)_tr12	V(e)_tr1	V(e)_tr2	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1	330.34	0.05536	0.01210	0.01202	0.17790	0.02943	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2	343.11	0.01271	0.00711	0.00667	0.21588	0.03385	</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3	348.50	0.00113	0.00328	0.00193	0.22916	0.02920	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4	345.17	0.00038	0.00229	0.00094	0.23154	0.02115	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5	321.02	0.00003	0.00199	0.00023	0.23460	0.03184	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6	347.21	0.00000	0.00112	-0.00001	0.23488	0.00308	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 7	-601.65	0.00000	0.00139	-0.00003	0.23489	0.00400	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 8	-323.56	0.00000	0.00238	-0.00009	0.23493	0.00769	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9	101.47	0.00000	0.00274	-0.00012	0.23494	0.00975	(1 component(s) constrained)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> You can specify the option --reml-maxit to allow for more iterations.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> An error occurs, please check the options or data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Error: Log-likelihood not converged (stop after 10 iteractions). </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Convergence issue occurred, please:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set byCHR = TRUE</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - set different options</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   - verify SNP partitioning or quality of the data</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> The result will be provided for the last iteration.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

