<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>GeneticCorrBT: Computing genetic correlation between two traits. — GeneticCorrBT • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GeneticCorrBT: Computing genetic correlation between two traits. — GeneticCorrBT"><meta name="description" content="This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise."><meta property="og:description" content="This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>GeneticCorrBT: Computing genetic correlation between two traits.</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>GeneticCorrBT.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function computes genetic correlation, a quantitative genetic measure that describes the genetic link between two
traits and has been predicted to indicate pleiotropic gene activity or correlation between causative loci in two traits.
For example, it does a bivariate GREML analysis to determine the genetic association between two quantitative traits,
two binary disease traits from case-control studies, and between a quantitative trait and a binary disease trait
following (Yang et al. 2011; Lee et al. 2012)
. If users want, this function gives the flexibility to compute the genetic
correlation chromosome-wise.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">GeneticCorrBT</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  <span class="va">ResultDir</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  byCHR <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  REMLalgo <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span>  nitr <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  <span class="va">phenofile</span>,</span>
<span>  cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  computeGRM <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  grmfile_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  partGRM <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  autosome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Xsome <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  nGRM <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  cripticut <span class="op">=</span> <span class="fl">0.025</span>,</span>
<span>  minMAF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maxMAF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  excludeResidual <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the all the input files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files for the genotype data. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-bychr">byCHR<a class="anchor" aria-label="anchor" href="#arg-bychr"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether the analysis will be performed chromosome wise or not. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-remlalgo">REMLalgo<a class="anchor" aria-label="anchor" href="#arg-remlalgo"></a></dt>
<dd><p>Integer value of 0, 1 or 2, specifying the algorithm to run REML iterations, 0 for average information (AI), 1 for Fisher-scoring and
2 for EM. The default option is 0, i.e. AI-REML (1).</p></dd>


<dt id="arg-nitr">nitr<a class="anchor" aria-label="anchor" href="#arg-nitr"></a></dt>
<dd><p>Integer value, specifying the number of iterations for performing the REML. The default is 100.</p></dd>


<dt id="arg-phenofile">phenofile<a class="anchor" aria-label="anchor" href="#arg-phenofile"></a></dt>
<dd><p>A dataframe for Bivar RELM has four columns <code>family ID</code>, <code>individual ID</code> and two trait columns. For binary trait, the phenotypic value
should be coded as 0 or 1, then it will be recognized as a case-control study (0 for controls and 1 for cases). Missing value should
be represented by "-9" or "NA".</p></dd>


<dt id="arg-cat-covarfile">cat_covarfile<a class="anchor" aria-label="anchor" href="#arg-cat-covarfile"></a></dt>
<dd><p>A character string, specifying the name of the categorical covariate file which is a plain text file with no header line; columns are
<code>family ID</code>, <code>individual ID</code> and discrete covariates. The default is <code>NULL</code>. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-quant-covarfile">quant_covarfile<a class="anchor" aria-label="anchor" href="#arg-quant-covarfile"></a></dt>
<dd><p>A character string, specifying the name of the quantitative covariate file which is a plain text file with no header line; columns
are <code>family ID</code>, <code>individual ID</code> and continuous covariates. The default is <code>NULL</code>. This file needs to be in <code>DataDir</code>.</p></dd>


<dt id="arg-computegrm">computeGRM<a class="anchor" aria-label="anchor" href="#arg-computegrm"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to compute GRM matrices or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-grmfile-name">grmfile_name<a class="anchor" aria-label="anchor" href="#arg-grmfile-name"></a></dt>
<dd><p>A string of characters specifying the prefix of autosomal .grm.bin file. Users need to provide separate GRM files for autosomes
and X chromosome in <code>ResultDir</code>. The X chromosomal GRM file should have "x" added in the autosomal prefix as file name.</p>
<p>For instance, if autosomal file is "ABC.grm.bin", then X chromosomal file should be "xABC.grm.bim". If you are providing
chromosome-wise GRMs, then the prefix should add "ChrNumber_" at the starting of the prefix like, "Chr1_ABC.grm.bin".
The default is <code>NULL</code>.</p></dd>


<dt id="arg-partgrm">partGRM<a class="anchor" aria-label="anchor" href="#arg-partgrm"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether the GRM will be partitioned into n parts (by row) in GREML model. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-autosome">autosome<a class="anchor" aria-label="anchor" href="#arg-autosome"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether estimate of heritability will be done for autosomes or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-xsome">Xsome<a class="anchor" aria-label="anchor" href="#arg-xsome"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether estimate of heritability will be done for X chromosome or not. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-ngrm">nGRM<a class="anchor" aria-label="anchor" href="#arg-ngrm"></a></dt>
<dd><p>Integer value, specifying the number of the partition of the GRM in GREML model. The default is 3.</p></dd>


<dt id="arg-cripticut">cripticut<a class="anchor" aria-label="anchor" href="#arg-cripticut"></a></dt>
<dd><p>Numeric value, specifying the threshold to create a new GRM of "unrelated" individuals in GREML model. The default is arbitrary
chosen as 0.025 following (Yang et al. 2011)
.</p></dd>


<dt id="arg-minmaf">minMAF<a class="anchor" aria-label="anchor" href="#arg-minmaf"></a></dt>
<dd><p>Positive numeric value (&lt; maxMAF), specifying the minimum threshold for the MAF filter of the SNPs in the Bivariate GREML model.</p></dd>


<dt id="arg-maxmaf">maxMAF<a class="anchor" aria-label="anchor" href="#arg-maxmaf"></a></dt>
<dd><p>Positive numeric value (minMAF,1), specifying the maximum threshold for the MAF filter of the SNPs in the Bivariate GREML model.</p></dd>


<dt id="arg-excluderesidual">excludeResidual<a class="anchor" aria-label="anchor" href="#arg-excluderesidual"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to drop the residual covariance from the model. Recommended to set this <code>TRUE</code>
if the traits were measured on different individuals. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-ncores">ncores<a class="anchor" aria-label="anchor" href="#arg-ncores"></a></dt>
<dd><p>Integer value, specifying the number of cores to be used.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe with minimum three columns:</p><ul><li><p>Source" (i.e., source of heritability)</p></li>
<li><p>Variance" (i.e. estimated heritability)</p></li>
<li><p>SE" (i.e., standard error of the estimated heritability)</p></li>
</ul><p>Source column will have rows, such as V(G)_tr1 (genetic variance for trait 1), V(G)_tr2 (genetic variance for trait 2),
C(G)_tr12 (genetic covariance between traits 1 and 2),V(e)_tr1 (residual variance for trait 1), V(e)_tr2 (residual variance for trait 2),
C(e)_tr12 (residual covariance between traits 1 and 2), Vp_tr1 (proportion of variance explained by all SNPs for trait 1),
Vp_tr2 (proportion of variance explained by all SNPs for trait 2), V(G)/Vp_tr1 (phenotypic variance for trait 1),
V(G)/Vp_tr2 (phenotypic variance for trait 2), rG (genetic correlation) and n (sample size). In case of chromosome-wise
analysis, there will be 'chromosome' column for chromosome code.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Lee SH, Wray NR, Goddard ME, Visscher PM (2012).
“Estimation of pleiotropy between complex diseases using SNP-derived genomic relationships and restricted maximum likelihood.”
<em>Bioinformatics</em>, <b>28</b>, 2540–2542.
<a href="https://doi.org/10.1093/bioinformatics/bts474" class="external-link">doi:10.1093/bioinformatics/bts474</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/22843982" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/22843982</a>.<br><br> Yang J, Lee SH, Goddard ME, Visscher PM (2011).
“GCTA: a tool for Genome-wide Complex Trait Analysis.”
<em>American Journal of Human Genetics</em>, <b>88</b>(1), 76–82.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_phenofile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">REMLalgo</span> <span class="op">&lt;-</span> <span class="fl">0</span></span></span>
<span class="r-in"><span><span class="va">nitr</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL</span></span></span>
<span class="r-in"><span><span class="va">cat_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">quant_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">partGRM</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Partition the GRM into m parts (by row),</span></span></span>
<span class="r-in"><span><span class="va">autosome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">Xsome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">cripticut</span> <span class="op">&lt;-</span> <span class="fl">0.025</span></span></span>
<span class="r-in"><span><span class="va">minMAF</span> <span class="op">&lt;-</span> <span class="fl">0.01</span> <span class="co"># if MAF filter apply</span></span></span>
<span class="r-in"><span><span class="va">maxMAF</span> <span class="op">&lt;-</span> <span class="fl">0.04</span></span></span>
<span class="r-in"><span><span class="va">excludeResidual</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">genetic_correlation</span> <span class="op">&lt;-</span> <span class="fu">GeneticCorrBT</span><span class="op">(</span></span></span>
<span class="r-in"><span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, byCHR <span class="op">=</span> <span class="va">byCHR</span>,</span></span>
<span class="r-in"><span>    REMLalgo <span class="op">=</span> <span class="fl">0</span>, nitr <span class="op">=</span> <span class="fl">10</span>, phenofile <span class="op">=</span> <span class="va">phenofile</span>, cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>, quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>    partGRM <span class="op">=</span> <span class="cn">FALSE</span>, autosome <span class="op">=</span> <span class="cn">TRUE</span>, Xsome <span class="op">=</span> <span class="cn">TRUE</span>, nGRM <span class="op">=</span> <span class="fl">3</span>,</span></span>
<span class="r-in"><span>    cripticut <span class="op">=</span> <span class="fl">0.025</span>, minMAF <span class="op">=</span> <span class="cn">NULL</span>, maxMAF <span class="op">=</span> <span class="cn">NULL</span>, excludeResidual <span class="op">=</span> <span class="cn">TRUE</span>, ncores <span class="op">=</span> <span class="va">ncores</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Processing chromosome 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> An error occurred: An error occurred while executing GCTA: GCTA binary not found.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'GCTA_PATH' environment variable and 'gcta64' in system PATH.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure GCTA is available and executable.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'gcta64' by running:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: GCTA_PATH=/full/path/to/gcta64</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in value[[3L]](cond):</span> An error occurred while executing GCTA: GCTA binary not found.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'GCTA_PATH' environment variable and 'gcta64' in system PATH.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure GCTA is available and executable.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'gcta64' by running:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: GCTA_PATH=/full/path/to/gcta64</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

