<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="all-functions">All functions<a class="anchor" aria-label="anchor" href="#all-functions"></a></h2>




    </div><div class="section level2">




      <dl><dt>

          <code><a href="AncestryCheck.html">AncestryCheck()</a></code>

        </dt>
        <dd>AncestryCheck: Evaluation of the samples' ancestry label.</dd>
      </dl><dl><dt>

          <code><a href="ClumpLD.html">ClumpLD()</a></code>

        </dt>
        <dd>ClumpLD: Clumping SNPs using linkage disequilibrium between SNPs</dd>
      </dl><dl><dt>

          <code><a href="ComputeGeneticPC.html">ComputeGeneticPC()</a></code>

        </dt>
        <dd>ComputeGeneticPC: Computing principal components from genetic relationship matrix</dd>
      </dl><dl><dt>

          <code><a href="ComputeLD.html">ComputeLD()</a></code>

        </dt>
        <dd>ComputeLD: Compute Linkage Disequilibrium (LD) for SNP Data</dd>
      </dl><dl><dt>

          <code><a href="ComputePRS.html">ComputePRS()</a></code>

        </dt>
        <dd>ComputePRS: Computing polygenic risk score (PRS)</dd>
      </dl><dl><dt>

          <code><a href="DiffZeroOne.html">DiffZeroOne()</a></code>

        </dt>
        <dd>DiffZeroOne: Assessing the Z-score for deviation from one and zero.</dd>
      </dl><dl><dt>

          <code><a href="DummyCovar.html">DummyCovar()</a></code>

        </dt>
        <dd>DummyCovar: Recode a categorical variable to a set of binary dummy variables.</dd>
      </dl><dl><dt>

          <code><a href="EstimateHerit.html">EstimateHerit()</a></code>

        </dt>
        <dd>EstimateHerit: Computing SNP heritability i.e., the proportion of phenotypic variance explained by SNPs.</dd>
      </dl><dl><dt>

          <code><a href="Example_covarfile.html">Example_covarfile</a></code>

        </dt>
        <dd>Example of covariate dataframe</dd>
      </dl><dl><dt>

          <code><a href="Example_h2data.html">Example_h2data</a></code>

        </dt>
        <dd>Example dataset for h2 estimation between male and female for eight different traits</dd>
      </dl><dl><dt>

          <code><a href="Example_phenofile.html">Example_phenofile</a></code>

        </dt>
        <dd>Example dataset for phenotype file</dd>
      </dl><dl><dt>

          <code><a href="Example_pthresoldfile.html">Example_pthresoldfile</a></code>

        </dt>
        <dd>Example dataset with p-value threshold</dd>
      </dl><dl><dt>

          <code><a href="Example_rgdata.html">Example_rgdata</a></code>

        </dt>
        <dd>Example dataset for running DiffZerOne function</dd>
      </dl><dl><dt>

          <code><a href="Ffile.html">Ffile</a></code>

        </dt>
        <dd>Example dataset for FemaleWAS</dd>
      </dl><dl><dt>

          <code><a href="FilterAllele.html">FilterAllele()</a></code>

        </dt>
        <dd>FilterAllele: Filtering out the multi-allelic variants</dd>
      </dl><dl><dt>

          <code><a href="FilterPlinkSample.html">FilterPlinkSample()</a></code>

        </dt>
        <dd>FilterPlinkSample: Making PLINK files with desired samples.</dd>
      </dl><dl><dt>

          <code><a href="FilterRegion.html">FilterRegion()</a></code>

        </dt>
        <dd>FilterRegion: Filter chromosomal regions.</dd>
      </dl><dl><dt>

          <code><a href="FilterSNP.html">FilterSNP()</a></code>

        </dt>
        <dd>FilterSNP: Filter out SNPs.</dd>
      </dl><dl><dt>

          <code><a href="GXWASmiami.html">GXWASmiami()</a></code>

        </dt>
        <dd>Miami plot</dd>
      </dl><dl><dt>

          <code><a href="GXwas.html">GXwas()</a></code>

        </dt>
        <dd>GXwas: Running genome-wide association study (GWAS) and X-chromosome-wide association study (XWAS) models.</dd>
      </dl><dl><dt>

          <code><a href="GXwasR_data.html">GXwasR_data()</a></code>

        </dt>
        <dd>GXwasR Data</dd>
      </dl><dl><dt>

          <code><a href="GeneticCorrBT.html">GeneticCorrBT()</a></code>

        </dt>
        <dd>GeneticCorrBT: Computing genetic correlation between two traits.</dd>
      </dl><dl><dt>

          <code><a href="GetMFPlink.html">GetMFPlink()</a></code>

        </dt>
        <dd>GetMFPlink: Getting male and female PLINK binary files.</dd>
      </dl><dl><dt>

          <code><a href="LDPrune.html">LDPrune()</a></code>

        </dt>
        <dd>LDPrune: Performs LD pruning on SNP data using PLINK</dd>
      </dl><dl><dt>

          <code><a href="MAFdiffSexControl.html">MAFdiffSexControl()</a></code>

        </dt>
        <dd>MAFdiffSexControl: Test for significantly different minor allele frequency (MAF) between sexes in control samples</dd>
      </dl><dl><dt>

          <code><a href="MergeRegion.html">MergeRegion()</a></code>

        </dt>
        <dd>MergeRegion: Merging two sets of plink binary files.</dd>
      </dl><dl><dt>

          <code><a href="MetaGWAS.html">MetaGWAS()</a></code>

        </dt>
        <dd>MetaGWAS: Combining summary-level results from two or more GWA studies into a single estimate.</dd>
      </dl><dl><dt>

          <code><a href="Mfile.html">Mfile</a></code>

        </dt>
        <dd>Example dataset for MaleWAS</dd>
      </dl><dl><dt>

          <code><a href="PlinkSummary.html">PlinkSummary()</a></code>

        </dt>
        <dd>PlinkSummary: Summary of plink format genotype dataset</dd>
      </dl><dl><dt>

          <code><a href="PvalComb.html">PvalComb()</a></code>

        </dt>
        <dd>PvalComb</dd>
      </dl><dl><dt>

          <code><a href="QCsample.html">QCsample()</a></code>

        </dt>
        <dd>QCsample: Quality control for samples in the plink binary files.</dd>
      </dl><dl><dt>

          <code><a href="QCsnp.html">QCsnp()</a></code>

        </dt>
        <dd>QCsnp: Quality control (QC) for SNPs.</dd>
      </dl><dl><dt>

          <code><a href="Regression_Ex.html">Regression_Ex</a></code>

        </dt>
        <dd>Example dataset for regression</dd>
      </dl><dl><dt>

          <code><a href="SexCheck.html">SexCheck()</a></code>

        </dt>
        <dd>SexCheck: Compare sex assignments in the input plink files with those imputed from X chromosome inbreeding coefficients</dd>
      </dl><dl><dt>

          <code><a href="SexDiff.html">SexDiff()</a></code>

        </dt>
        <dd>SexDiff: Sex difference in effect size for each SNP using t-test.</dd>
      </dl><dl><dt>

          <code><a href="SexDiffZscore.html">SexDiffZscore()</a></code>

        </dt>
        <dd>SexDiffZscore: Z-score-based sex difference test.</dd>
      </dl><dl><dt>

          <code><a href="SexRegress.html">SexRegress()</a></code>

        </dt>
        <dd>SexRegress: Performing linear regression analysis with quantitative response variable.</dd>
      </dl><dl><dt>

          <code><a href="Summary_Stat_Ex1.html">Summary_Stat_Ex1</a></code>

        </dt>
        <dd>Example dataset for GWAS summary statistics</dd>
      </dl><dl><dt>

          <code><a href="Summary_Stat_Ex2.html">Summary_Stat_Ex2</a></code>

        </dt>
        <dd>Example dataset for GWAS summary statistics</dd>
      </dl><dl><dt>

          <code><a href="SumstatGenCorr.html">SumstatGenCorr()</a></code>

        </dt>
        <dd>SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics</dd>
      </dl><dl><dt>

          <code><a href="TestXGene.html">TestXGene()</a></code>

        </dt>
        <dd>TestXGene: Performing gene-based association test using GWAS/XWAS summary statistics.</dd>
      </dl><dl><dt>

          <code><a href="XWAS_Summary_Example.html">XWAS_Summary_Example</a></code>

        </dt>
        <dd>Example dataset for XWAS summary statistics</dd>
      </dl><dl><dt>

          <code><a href="Xhwe.html">Xhwe()</a></code>

        </dt>
        <dd>Xhwe: Filter X-chromosome variants for HWE in females.</dd>
      </dl><dl><dt>

          <code><a href="example_data_study_sample_ancestry.html">example_data_study_sample_ancestry</a></code>

        </dt>
        <dd>Example of ancestry dataframe of samples</dd>
      </dl><dl><dt>

          <code><a href="executePlinkMAF.html">executePlinkMAF()</a></code>

        </dt>
        <dd>executePlinkMAF: Execute PLINK to Calculate Minor Allele Frequencies (MAF)</dd>
      </dl><dl><dt>

          <code><a href="highLD_hg19.html">highLD_hg19</a></code>

        </dt>
        <dd>Example dataset for high LD regions in hg19 built</dd>
      </dl><dl><dt>

          <code><a href="plinkVCF.html">plinkVCF()</a></code>

        </dt>
        <dd>plinkVCF: Converting VCF files to plink binary files and vice-versa.</dd>
      </dl><dl><dt>

          <code><a href="validate_reference_data.html">validate_reference_data()</a></code>

        </dt>
        <dd>Validate Path to Reference Data Set</dd>
      </dl></div>
  </main></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

