<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>PvalComb — PvalComb • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PvalComb — PvalComb"><meta name="description" content="This function combines the p-values of two separate GWAS summary statistics (for instance male and female populations),
merges them, and then applies various statistical methods (like Stouffer's method, Fisher's method) to integrate the
p-values. It also includes functionality for generating plots like Manhattan plots and Q-Q plots."><meta property="og:description" content="This function combines the p-values of two separate GWAS summary statistics (for instance male and female populations),
merges them, and then applies various statistical methods (like Stouffer's method, Fisher's method) to integrate the
p-values. It also includes functionality for generating plots like Manhattan plots and Q-Q plots."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>PvalComb</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>PvalComb.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function combines the p-values of two separate GWAS summary statistics (for instance male and female populations),
merges them, and then applies various statistical methods (like Stouffer's method, Fisher's method) to integrate the
p-values. It also includes functionality for generating plots like Manhattan plots and Q-Q plots.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PvalComb</span><span class="op">(</span></span>
<span>  <span class="va">SumstatMale</span>,</span>
<span>  <span class="va">SumstatFemale</span>,</span>
<span>  <span class="va">combtest</span>,</span>
<span>  MF.p.corr <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  MF.zero.sub <span class="op">=</span> <span class="fl">1e-05</span>,</span>
<span>  MF.na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  MF.mc.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  B <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  plot.jpeg <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  plotname <span class="op">=</span> <span class="st">"GXwas.plot"</span>,</span>
<span>  PlotDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">snp_pval</span>,</span>
<span>  annotateTopSnp <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  suggestiveline <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  genomewideline <span class="op">=</span> <span class="fl">7.3</span>,</span>
<span>  ncores <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-sumstatmale">SumstatMale<a class="anchor" aria-label="anchor" href="#arg-sumstatmale"></a></dt>
<dd><p>R dataframe object of summary statistics of male GWAS with five mandatory columns:</p><ul><li><p><code>CHR</code> (numeric chromosome code)</p></li>
<li><p><code>SNP</code> (variant id)</p></li>
<li><p><code>A1</code> (allele)</p></li>
<li><p><code>POS</code> (base-pair position)</p></li>
<li><p><code>P</code> (p-value).</p></li>
</ul><p>Other columns may present.</p></dd>


<dt id="arg-sumstatfemale">SumstatFemale<a class="anchor" aria-label="anchor" href="#arg-sumstatfemale"></a></dt>
<dd><p>R dataframe object of summary statistics of female GWAS with five mandatory columns:</p><ul><li><p><code>SNP</code></p></li>
<li><p><code>A1</code></p></li>
<li><p><code>TEST</code></p></li>
<li><p><code>POS</code></p></li>
<li><p><code>P</code></p></li>
</ul><p>Other columns may present.</p></dd>


<dt id="arg-combtest">combtest<a class="anchor" aria-label="anchor" href="#arg-combtest"></a></dt>
<dd><p>Character vector specifying method for combining p-values for stratified GWAS models. Choices are “stouffer.method”,
"fisher.method" and "fisher.method.perm". For fisher.method the function for combining p-values uses a statistic,
\(S = -2 ∑^k /log p\), which follows a \(χ^2\) distribution with 2k degrees of freedom (Fisher 1925)
.
For fisher.method.perm, using p-values from stratified tests, the summary statistic for combining p-values
is \(S = -2 ∑ /log p\). A p-value for this statistic can be derived by randomly generating summary statistics (Rhodes 2002)
.
Therefore, a p-value is randomly sampled from each contributing study, and a random statistic is calculated. The
fraction of random statistics greater or equal to S then gives the final p-value.</p></dd>


<dt id="arg-mf-p-corr">MF.p.corr<a class="anchor" aria-label="anchor" href="#arg-mf-p-corr"></a></dt>
<dd><p>Character vector specifying method for correcting the summary p-values for FMfcomb and FMscomb models. Choices are
"bonferroni", "BH" and "none" for Bonferroni,  Benjamini-Hochberg and none, respectively. The default is "none".</p></dd>


<dt id="arg-mf-zero-sub">MF.zero.sub<a class="anchor" aria-label="anchor" href="#arg-mf-zero-sub"></a></dt>
<dd><p>Small numeric value for substituting p-values of 0 in GWAS summary statistics. The default is 0.00001. As \(log(0)\)
results in Inf this replaces p-values of 0 by default with a small float.</p></dd>


<dt id="arg-mf-na-rm">MF.na.rm<a class="anchor" aria-label="anchor" href="#arg-mf-na-rm"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for removing p-values of NA in stratified GWAS summary satistics in case of using Fisher’s
and Stouffer’s methods. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-mf-mc-cores">MF.mc.cores<a class="anchor" aria-label="anchor" href="#arg-mf-mc-cores"></a></dt>
<dd><p>Number of cores used for fisher.method.perm for combining p-values. The default is 1.</p></dd>


<dt id="arg-b">B<a class="anchor" aria-label="anchor" href="#arg-b"></a></dt>
<dd><p>Integer value specifying the number of permutation in case of using fisher.method.perm method. The default is 10000.</p></dd>


<dt id="arg-plot-jpeg">plot.jpeg<a class="anchor" aria-label="anchor" href="#arg-plot-jpeg"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for saving the plots in .jpeg file. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-plotname">plotname<a class="anchor" aria-label="anchor" href="#arg-plotname"></a></dt>
<dd><p>A character string specifying the prefix of the file for plots. This file will be saved in DataDir.
The default is "GXwas.plot".</p></dd>


<dt id="arg-plotdir">PlotDir<a class="anchor" aria-label="anchor" href="#arg-plotdir"></a></dt>
<dd><p>A character string specifying the path of the directory where the plots will be saved. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-snp-pval">snp_pval<a class="anchor" aria-label="anchor" href="#arg-snp-pval"></a></dt>
<dd><p>Numeric value as p-value threshold for annotation. SNPs below this p-value will be annotated on the plot. The default is 1e-08.</p></dd>


<dt id="arg-annotatetopsnp">annotateTopSnp<a class="anchor" aria-label="anchor" href="#arg-annotatetopsnp"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE.</code> If <code>TRUE</code>, it only annotates the top hit on each chromosome that is below the snp_pval threshold. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-suggestiveline">suggestiveline<a class="anchor" aria-label="anchor" href="#arg-suggestiveline"></a></dt>
<dd><p>Numeric value for suggestive cut-off line in GWAS manhattan plot. The default is 5 (for p-value 1e-05).</p></dd>


<dt id="arg-genomewideline">genomewideline<a class="anchor" aria-label="anchor" href="#arg-genomewideline"></a></dt>
<dd><p>Numeric value for genome-wide significant cut-off line in GWAS manhattan plot. The default is 7.3 (for p-value 5e-08).</p></dd>


<dt id="arg-ncores">ncores<a class="anchor" aria-label="anchor" href="#arg-ncores"></a></dt>
<dd><p>Integer value, specifying the number of cores for parallel processing. The default is 0 (no parallel computation).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe with GWAS summary statistics (with XWAS for X-chromosomal variants) along with Manhattan and Q-Q plots.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Fisher RA (1925).
<em>Statistical Methods for Research Workers</em>.
Oliver and Boyd, Edinburgh.<br><br> Rhodes DR (2002).
“Meta-analysis of microarrays: interstudy validation of gene expression profiles reveals pathway dysregulation in prostate cancer.”
<em>Cancer Research</em>, <b>62</b>(15), 4427–4433.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Mfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Ffile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">SumstatMale</span> <span class="op">&lt;-</span> <span class="va">Mfile</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SumstatMale</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"POS"</span></span></span>
<span class="r-in"><span><span class="va">SumstatFemale</span> <span class="op">&lt;-</span> <span class="va">Ffile</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SumstatFemale</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"POS"</span></span></span>
<span class="r-in"><span><span class="va">PvalComb_Result</span> <span class="op">&lt;-</span> <span class="fu">PvalComb</span><span class="op">(</span></span></span>
<span class="r-in"><span>    SumstatMale <span class="op">=</span> <span class="va">SumstatMale</span>, SumstatFemale <span class="op">=</span> <span class="va">SumstatFemale</span>,</span></span>
<span class="r-in"><span>    combtest <span class="op">=</span> <span class="st">"fisher.method"</span>, MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, snp_pval <span class="op">=</span> <span class="fl">0.001</span>, plot.jpeg <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>    suggestiveline <span class="op">=</span> <span class="fl">3</span>, genomewideline <span class="op">=</span> <span class="fl">5.69897</span>, ncores <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Saving plot to /tmp/RtmpyJFifX/Stratified_GWAS.png</span>
<span class="r-plt img"><img src="PvalComb-1.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Saving plot to /tmp/RtmpyJFifX/Stratified_XWAS.png</span>
<span class="r-plt img"><img src="PvalComb-2.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="PvalComb-3.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="PvalComb-4.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="PvalComb-5.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="PvalComb-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

