<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>QCsnp: Quality control (QC) for SNPs. — QCsnp • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="QCsnp: Quality control (QC) for SNPs. — QCsnp"><meta name="description" content="This function performs quality control on genetic variant data from PLINK binary files. It filters variants based on parameters
like minor allele frequency, Hardy-Weinberg equilibrium, call rate, and differential missingness between cases and controls.
It also handles monomorphic SNPs and applies linkage disequilibrium-based filtering if specified."><meta property="og:description" content="This function performs quality control on genetic variant data from PLINK binary files. It filters variants based on parameters
like minor allele frequency, Hardy-Weinberg equilibrium, call rate, and differential missingness between cases and controls.
It also handles monomorphic SNPs and applies linkage disequilibrium-based filtering if specified."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>QCsnp: Quality control (QC) for SNPs.</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>QCsnp.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs quality control on genetic variant data from PLINK binary files. It filters variants based on parameters
like minor allele frequency, Hardy-Weinberg equilibrium, call rate, and differential missingness between cases and controls.
It also handles monomorphic SNPs and applies linkage disequilibrium-based filtering if specified.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">QCsnp</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  foutput <span class="op">=</span> <span class="st">"FALSE"</span>,</span>
<span>  casecontrol <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  hweCase <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hweControl <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  hwe <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  maf <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  geno <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  monomorphicSNPs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  caldiffmiss <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  diffmissFilter <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dmissX <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  dmissAutoY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  highLD_regions <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  ld_prunning <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  r2_threshold <span class="op">=</span> <span class="fl">0.02</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the input PLINK binary files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files with both male and female samples. This file needs
to be in <code>DataDir</code>.</p></dd>


<dt id="arg-foutput">foutput<a class="anchor" aria-label="anchor" href="#arg-foutput"></a></dt>
<dd><p>Character string, specifying the prefix of the output PLINK binary files if the filtering option for the SNPs is chosen.
The default is "FALSE".</p></dd>


<dt id="arg-casecontrol">casecontrol<a class="anchor" aria-label="anchor" href="#arg-casecontrol"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> indicating if the input plink files has cases-control status or not.
The default is <code>FALSE</code>.</p></dd>


<dt id="arg-hwecase">hweCase<a class="anchor" aria-label="anchor" href="#arg-hwecase"></a></dt>
<dd><p>Numeric value between 0 to 1 or <code>NULL</code> for removing SNPs which fail Hardy-Weinberg equilibrium for cases.
The default is <code>NULL</code>.</p></dd>


<dt id="arg-hwecontrol">hweControl<a class="anchor" aria-label="anchor" href="#arg-hwecontrol"></a></dt>
<dd><p>Numeric value between 0 to 1 or <code>NULL</code> for removing SNPs which fail Hardy-Weinberg equilibrium for controls.
The default is <code>NULL</code>.</p></dd>


<dt id="arg-hwe">hwe<a class="anchor" aria-label="anchor" href="#arg-hwe"></a></dt>
<dd><p>Numeric value between 0 to 1 or <code>NULL</code> for removing SNPs which fail Hardy-Weinberg equilibrium for entire dataset.
The default is <code>NULL</code>.</p></dd>


<dt id="arg-maf">maf<a class="anchor" aria-label="anchor" href="#arg-maf"></a></dt>
<dd><p>Numeric value between 0 to 1 for removing SNPs with minor allele frequency less than the specified threshold.
The default is 0.05.</p></dd>


<dt id="arg-geno">geno<a class="anchor" aria-label="anchor" href="#arg-geno"></a></dt>
<dd><p>Numeric value between 0 to 1 for removing SNPs that have less than the specified call rate. The default is 0.05.</p>
<p>Users can set this as <code>NULL</code> to not apply this filter.</p></dd>


<dt id="arg-monomorphicsnps">monomorphicSNPs<a class="anchor" aria-label="anchor" href="#arg-monomorphicsnps"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for filtering out monomorphic SNP. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-caldiffmiss">caldiffmiss<a class="anchor" aria-label="anchor" href="#arg-caldiffmiss"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to compute differential missingness between cases and controls
for each SNP (threshold is \(0.05/length(unique(No. of. SNPs in the test))\)). The default is <code>TRUE.</code></p></dd>


<dt id="arg-diffmissfilter">diffmissFilter<a class="anchor" aria-label="anchor" href="#arg-diffmissfilter"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether to filter out the SNPs or only flagged them for differential
missingness in cases vs controls. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-dmissx">dmissX<a class="anchor" aria-label="anchor" href="#arg-dmissx"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for computing differential missingness between cases and controls for X chromosome
SNPs only. The default is <code>FALSE</code>. The diffmissFilter will work for all these SNPs.</p></dd>


<dt id="arg-dmissautoy">dmissAutoY<a class="anchor" aria-label="anchor" href="#arg-dmissautoy"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for computing differential missingness between cases and controls for SNPs on autosomes
and Y chromosome only. The default is <code>FALSE</code>.</p>
<p>If <code>dmissX</code> and <code>dmissAutoY</code> are both <code>FALSE</code>, then this will be computed genome-wide. The <code>diffmissFilter</code> will work
for all these SNPs.</p></dd>


<dt id="arg-highld-regions">highLD_regions<a class="anchor" aria-label="anchor" href="#arg-highld-regions"></a></dt>
<dd><p>A dataframe with known high LD regions (Anderson et al. 2010)
 is provided with the package.</p></dd>


<dt id="arg-ld-prunning">ld_prunning<a class="anchor" aria-label="anchor" href="#arg-ld-prunning"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for applying linkage disequilibrium (LD)-based filtering.</p></dd>


<dt id="arg-window-size">window_size<a class="anchor" aria-label="anchor" href="#arg-window-size"></a></dt>
<dd><p>Integer value, specifying a window size in the variant counts for LD-based filtering. The default is 50.</p></dd>


<dt id="arg-step-size">step_size<a class="anchor" aria-label="anchor" href="#arg-step-size"></a></dt>
<dd><p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering. The default is 5.</p></dd>


<dt id="arg-r-threshold">r2_threshold<a class="anchor" aria-label="anchor" href="#arg-r-threshold"></a></dt>
<dd><p>Numeric value between 0 to 1 of pairwise \(r^2\) threshold for LD-based filtering. The default is 0.02.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list of two objects, namely, <code>MonomorSNPs</code> and <code>DiffMissSNPs</code> containing monomorphic SNPs and SNPs with differential missingness
in cases vs controls, respectively. Output plink binary files in the working directory.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP, Zondervan KT (2010).
“Data quality control in genetic case-control association studies.”
<em>Nature Protocols</em>, <b>5</b>(9), 1564–1573.
<a href="https://doi.org/10.1038/nprot.2010.116" class="external-link">doi:10.1038/nprot.2010.116</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/21085122" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/21085122</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span></span>
<span class="r-in"><span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span></span>
<span class="r-in"><span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">QCsnp</span><span class="op">(</span></span></span>
<span class="r-in"><span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>,</span></span>
<span class="r-in"><span>    geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>,</span></span>
<span class="r-in"><span>    ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>,</span></span>
<span class="r-in"><span>    caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> An error occurred: An error occurred while executing Plink: PLINK binary not found.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'PLINK_PATH' environment variable and 'plink' in system PATH.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure PLINK is available and executable (bioinformatics version).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'plink' by running:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: PLINK_PATH=/full/path/to/plink</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

