<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>SexCheck: Compare sex assignments in the input PLINK files with those imputed from X chromosome inbreeding coefficients — SexCheck • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SexCheck: Compare sex assignments in the input PLINK files with those imputed from X chromosome inbreeding coefficients — SexCheck"><meta name="description" content="This function compares sex assignments in the input dataset with those predicted from X chromosome inbreeding coefficients (Purcell et al. 2007)
,
and gives the option to convert the sex assignments to the predicted values. Implicitly, this function computes observed and expected autosomal homozygous
genotype counts for each sample and reports method-of-moments F coefficient estimates (i.e., observed hom. \(count - expected count) / (total observations - expected count)\)).
The expected counts will be based on loaded or imputed minor allele frequencies.  Since imputed MAFs are highly inaccurate when there are few samples,
the 'compute freq' parameter should be set to TRUE to compute MAF implicitly.
Due to the use of allele frequencies, if a cohort is comprised of individuals of different ancestries, users may need to process any samples with rare
ancestry individually if the dataset has a very unbalanced ancestry distribution. It is advised to run this function with all the parameters set to zero,
then examine the distribution of the F estimates (there should be a clear gap between a very tight male clump on the right side of the distribution and the
females everywhere else). Then, rerun the function with the parameters that correspond to this gap."><meta property="og:description" content="This function compares sex assignments in the input dataset with those predicted from X chromosome inbreeding coefficients (Purcell et al. 2007)
,
and gives the option to convert the sex assignments to the predicted values. Implicitly, this function computes observed and expected autosomal homozygous
genotype counts for each sample and reports method-of-moments F coefficient estimates (i.e., observed hom. \(count - expected count) / (total observations - expected count)\)).
The expected counts will be based on loaded or imputed minor allele frequencies.  Since imputed MAFs are highly inaccurate when there are few samples,
the 'compute freq' parameter should be set to TRUE to compute MAF implicitly.
Due to the use of allele frequencies, if a cohort is comprised of individuals of different ancestries, users may need to process any samples with rare
ancestry individually if the dataset has a very unbalanced ancestry distribution. It is advised to run this function with all the parameters set to zero,
then examine the distribution of the F estimates (there should be a clear gap between a very tight male clump on the right side of the distribution and the
females everywhere else). Then, rerun the function with the parameters that correspond to this gap."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>SexCheck: Compare sex assignments in the input PLINK files with those imputed from X chromosome inbreeding coefficients</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>SexCheck.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function compares sex assignments in the input dataset with those predicted from X chromosome inbreeding coefficients (Purcell et al. 2007)
,
and gives the option to convert the sex assignments to the predicted values. Implicitly, this function computes observed and expected autosomal homozygous
genotype counts for each sample and reports method-of-moments F coefficient estimates (i.e., observed hom. \(count - expected count) / (total observations - expected count)\)).
The expected counts will be based on loaded or imputed minor allele frequencies.  Since imputed MAFs are highly inaccurate when there are few samples,
the 'compute freq' parameter should be set to TRUE to compute MAF implicitly.</p>
<p>Due to the use of allele frequencies, if a cohort is comprised of individuals of different ancestries, users may need to process any samples with rare
ancestry individually if the dataset has a very unbalanced ancestry distribution. It is advised to run this function with all the parameters set to zero,
then examine the distribution of the F estimates (there should be a clear gap between a very tight male clump on the right side of the distribution and the
females everywhere else). Then, rerun the function with the parameters that correspond to this gap.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SexCheck</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  impute_sex <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  compute_freq <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  LD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  LD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  LD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  LD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>  fmax_F <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  mmin_F <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>Character string for the file path of the input PLINK binary files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files. Note: Input dataset should contain X and Y regions.</p></dd>


<dt id="arg-impute-sex">impute_sex<a class="anchor" aria-label="anchor" href="#arg-impute-sex"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying sex to be imputed or not. If <code>TRUE</code> then sex-imputed PLINK files, prefixed, 'seximputed_plink', will
be produced in <code>DataDir</code>.</p></dd>


<dt id="arg-compute-freq">compute_freq<a class="anchor" aria-label="anchor" href="#arg-compute-freq"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying minor allele frequency (MAF). This function requires reasonable MAF estimates, so it is essential
to use <code>compute_freq</code> = <code>TRUE</code> for computing MAF from an input PLINK file if there are very few samples in the input dataset. The default is <code>FALSE</code>.</p></dd>


<dt id="arg-ld">LD<a class="anchor" aria-label="anchor" href="#arg-ld"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> for applying linkage disequilibrium (LD)-based filtering. The default is <code>TRUE</code>.</p></dd>


<dt id="arg-ld-window-size">LD_window_size<a class="anchor" aria-label="anchor" href="#arg-ld-window-size"></a></dt>
<dd><p>Integer value, specifying a window size in variant count for LD-based filtering. The default is 50.</p></dd>


<dt id="arg-ld-step-size">LD_step_size<a class="anchor" aria-label="anchor" href="#arg-ld-step-size"></a></dt>
<dd><p>Integer value, specifying a variant count to shift the window at the end of each step for LD filtering. The default is 5.</p></dd>


<dt id="arg-ld-r-threshold">LD_r2_threshold<a class="anchor" aria-label="anchor" href="#arg-ld-r-threshold"></a></dt>
<dd><p>Numeric value between 0 to 1 of pairwise \(r^2\) threshold for LD-based filtering. The default is 0.02.</p></dd>


<dt id="arg-fmax-f">fmax_F<a class="anchor" aria-label="anchor" href="#arg-fmax-f"></a></dt>
<dd><p>Numeric value between 0 to 1. Samples with F estimates smaller than this value will be labeled as females. The default is 0.2.</p></dd>


<dt id="arg-mmin-f">mmin_F<a class="anchor" aria-label="anchor" href="#arg-mmin-f"></a></dt>
<dd><p>Numeric value between 0 to 1. Samples with F estimates larger than this value will be labeled as males. The default is 0.8.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A dataframe with six columns:</p><ul><li><p><code>FID</code> (Family ID)</p></li>
<li><p><code>IID</code> (Individual ID)</p></li>
<li><p><code>PEDSEX </code>(Sex as determined in pedigree file (1=male, 2=female))</p></li>
<li><p><code>SNPSEX</code> (Sex as determined by X chromosome)</p></li>
<li><p><code>STATUS</code> (Displays "PROBLEM" or "OK" for each individual)</p></li>
<li><p><code>F</code> (The actual X chromosome inbreeding (homozygosity) estimate)</p></li>
</ul><p>A PROBLEM arises if the two sexes do not match, or if the SNP data or pedigree data are ambiguous with regard to sex.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).
“PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.”
<em>The American Journal of Human Genetics</em>, <b>81</b>(3), 559–575.
<a href="https://doi.org/10.1086/519795" class="external-link">doi:10.1086/519795</a>
.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span><span class="va">LD</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="va">LD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span></span>
<span class="r-in"><span><span class="va">LD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span></span>
<span class="r-in"><span><span class="va">LD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span></span>
<span class="r-in"><span><span class="va">fmax_F</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span></span>
<span class="r-in"><span><span class="va">mmin_F</span> <span class="op">&lt;-</span> <span class="fl">0.8</span></span></span>
<span class="r-in"><span><span class="va">impute_sex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span><span class="va">compute_freq</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">SexCheck</span><span class="op">(</span></span></span>
<span class="r-in"><span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, impute_sex <span class="op">=</span> <span class="va">impute_sex</span>,</span></span>
<span class="r-in"><span>    compute_freq <span class="op">=</span> <span class="va">compute_freq</span>, LD_window_size <span class="op">=</span> <span class="va">LD_window_size</span>, LD_step_size <span class="op">=</span> <span class="va">LD_step_size</span>,</span></span>
<span class="r-in"><span>    LD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, fmax_F <span class="op">=</span> <span class="fl">0.2</span>, mmin_F <span class="op">=</span> <span class="fl">0.8</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> An error occurred: An error occurred while executing Plink: PLINK binary not found.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'PLINK_PATH' environment variable and 'plink' in system PATH.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure PLINK is available and executable (bioinformatics version).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'plink' by running:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: PLINK_PATH=/full/path/to/plink</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Checking if there is any wrong sex assignment</span></span></span>
<span class="r-in"><span><span class="va">problematic_sex</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">STATUS</span> <span class="op">!=</span> <span class="st">"OK"</span>, <span class="op">]</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

