<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>SexDiffZscore: Z-score-based sex difference test. — SexDiffZscore • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SexDiffZscore: Z-score-based sex difference test. — SexDiffZscore"><meta name="description" content="This function calculates the difference in any kind of measured entities,(example: including SNP heritability estimate,
genetic correlation, and GWAS β values) between sexes using a Z-score and its associated p-value statistic.
When STAT/SE is normally distributed and the test statistics are independent in sex, the test is well calibrated. If
the statistics are positively correlated, this test is conservative (1).
We could define SNPs with SDEs as those variants at the extreme ends of the distribution with an absolute value of the
Z-score greater than 3(|Z-score| &amp;gt; 3), which is roughly equivalent to p &amp;lt;10−3, and represents 0.3% of all tested SNPs.
The input dataframes should only include X-chromosome in order to obtain results for sex differences based solely on
X-linked loci."><meta property="og:description" content="This function calculates the difference in any kind of measured entities,(example: including SNP heritability estimate,
genetic correlation, and GWAS β values) between sexes using a Z-score and its associated p-value statistic.
When STAT/SE is normally distributed and the test statistics are independent in sex, the test is well calibrated. If
the statistics are positively correlated, this test is conservative (1).
We could define SNPs with SDEs as those variants at the extreme ends of the distribution with an absolute value of the
Z-score greater than 3(|Z-score| &amp;gt; 3), which is roughly equivalent to p &amp;lt;10−3, and represents 0.3% of all tested SNPs.
The input dataframes should only include X-chromosome in order to obtain results for sex differences based solely on
X-linked loci."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_All_Functions.html">GXwasR Functions</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>SexDiffZscore: Z-score-based sex difference test.</h1>

      <div class="d-none name"><code>SexDiffZscore.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function calculates the difference in any kind of measured entities,(example: including SNP heritability estimate,
genetic correlation, and GWAS β values) between sexes using a Z-score and its associated p-value statistic.
When STAT/SE is normally distributed and the test statistics are independent in sex, the test is well calibrated. If
the statistics are positively correlated, this test is conservative (1).</p>
<p>We could define SNPs with SDEs as those variants at the extreme ends of the distribution with an absolute value of the
Z-score greater than 3(|Z-score| &gt; 3), which is roughly equivalent to p &lt;10−3, and represents 0.3% of all tested SNPs.
The input dataframes should only include X-chromosome in order to obtain results for sex differences based solely on
X-linked loci.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SexDiffZscore</span><span class="op">(</span><span class="va">inputdata</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-inputdata">inputdata<a class="anchor" aria-label="anchor" href="#arg-inputdata"></a></dt>
<dd><p>A dataframe with five columns:</p><ul><li><p><code>ID</code> (i.e., SNP ID or the phenotype of interest, etc.)</p></li>
<li><p><code>Fstat</code> (i.e., the measured statistics in females)</p></li>
<li><p><code>Fse</code> (i.e., the standard error of the measured statistics in females)</p></li>
<li><p><code>Mstat</code> (i.e., the measured statistics in males)</p></li>
<li><p><code>Mse</code> (i.e., the standard error of the measured statistics in males)</p></li>
</ul></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Original input dataframe with:</p><ul><li><p><code>Zscore</code> (i.e., Z-score),</p></li>
<li><p><code>p</code> (i.e., p-value) and</p></li>
<li><p><code>adjP</code> (i.e., Bonferroni corrected p-value)
columns added.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"GXwasRData"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">inputdata</span> <span class="op">&lt;-</span> <span class="va">Example_h2data</span></span></span>
<span class="r-in"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">SexDiffZscore</span><span class="op">(</span><span class="va">inputdata</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

