<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics — SumstatGenCorr • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics — SumstatGenCorr"><meta name="description" content="This function calculates the genetic correlation between two summary statistics
using a specified reference linkage disequilibrium (LD) matrix from the UK Biobank."><meta property="og:description" content="This function calculates the genetic correlation between two summary statistics
using a specified reference linkage disequilibrium (LD) matrix from the UK Biobank."><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary Statistics</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>SumstatGenCorr.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function calculates the genetic correlation between two summary statistics
using a specified reference linkage disequilibrium (LD) matrix from the UK Biobank.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SumstatGenCorr</span><span class="op">(</span></span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">referenceLD</span>,</span>
<span>  <span class="va">sumstat1</span>,</span>
<span>  <span class="va">sumstat2</span>,</span>
<span>  Nref <span class="op">=</span> <span class="fl">335265</span>,</span>
<span>  N0 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="va">sumstat1</span><span class="op">$</span><span class="va">N</span><span class="op">)</span>,</span>
<span>  eigen.cut <span class="op">=</span> <span class="st">"automatic"</span>,</span>
<span>  lim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="fl">18</span><span class="op">)</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  numCores <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>Directory where results should be saved.</p></dd>


<dt id="arg-referenceld">referenceLD<a class="anchor" aria-label="anchor" href="#arg-referenceld"></a></dt>
<dd><p>Reference LD matrix identifier. These are the LD matrices and their eigen-decomposition from 335,265 genomic
British UK Biobank individuals. Two sets of reference panel are provided:</p><ol><li><p>307,519 QCed UK Biobank Axiom Array SNPs. The size is about 7.5 GB after unzipping.</p></li>
<li><p>1,029,876 QCed UK Biobank imputed SNPs. The size is about 31 GB after unzipping. Although it takes more time,
using the imputed panel provides more accurate estimates of genetic correlations.
Therefore if the GWAS includes most of the HapMap3 SNPs, then it is recommend using the imputed reference panel.</p></li>
</ol></dd>


<dt id="arg-sumstat-">sumstat1<a class="anchor" aria-label="anchor" href="#arg-sumstat-"></a></dt>
<dd><p>Data frame for the first set of summary statistics.
The input data frame should include following columns: SNP, SNP ID; A1, effect allele; A2, reference allele;
N, sample size; Z, z-score; If Z is not given, alternatively, you may provide: b, estimate of marginal effect in GWAS; se,
standard error of the estimates of marginal effects in GWAS.</p></dd>


<dt id="arg-sumstat-">sumstat2<a class="anchor" aria-label="anchor" href="#arg-sumstat-"></a></dt>
<dd><p>Data frame for the second set of summary statistics.
The input data frame should include following columns: SNP, SNP ID; A1, effect allele; A2, reference allele;
N, sample size; Z, z-score; If Z is not given, alternatively, you may provide: b, estimate of marginal effect in GWAS; se,
standard error of the estimates of marginal effects in GWAS.</p></dd>


<dt id="arg-nref">Nref<a class="anchor" aria-label="anchor" href="#arg-nref"></a></dt>
<dd><p>Sample size of the reference sample where LD is computed. If the default UK Biobank reference sample is used, Nref = 335265</p></dd>


<dt id="arg-n-">N0<a class="anchor" aria-label="anchor" href="#arg-n-"></a></dt>
<dd><p>Number of individuals included in both cohorts. The estimated genetic correlation is usually robust against misspecified N0.
If not given, the default value is set to the minimum sample size across all SNPs in cohort 1 and cohort 2.</p></dd>


<dt id="arg-eigen-cut">eigen.cut<a class="anchor" aria-label="anchor" href="#arg-eigen-cut"></a></dt>
<dd><p>Which eigenvalues and eigenvectors in each LD score matrix should be used for HDL.
Users are allowed to specify a numeric value between 0 and 1 for eigen.cut. For example, eigen.cut = 0.99 means using the
leading eigenvalues explaining 99% of the variance
and their correspondent eigenvectors. If the default 'automatic' is used, the eigen.cut gives the most stable heritability
estimates will be used.</p></dd>


<dt id="arg-lim">lim<a class="anchor" aria-label="anchor" href="#arg-lim"></a></dt>
<dd><p>Tolerance limitation, default lim = exp(-18).</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>Boolean value, TRUE or FALSE for whether to perform parallel computation. The default is FALSE</p></dd>


<dt id="arg-numcores">numCores<a class="anchor" aria-label="anchor" href="#arg-numcores"></a></dt>
<dd><p>The number of cores to be used. The default is 2.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A list is returned with:</p><ul><li><p><code>rg</code>: The estimated genetic correlation.</p></li>
<li><p><code>rg.se</code>: The standard error of the estimated genetic correlation.</p></li>
<li><p><code>P</code>: P-value based on Wald test.</p></li>
<li><p><code>estimates.df</code>: A detailed matrix includes the estimates and standard errors of heritabilities, genetic covariance
and genetic correlation.</p></li>
<li><p><code>eigen.use</code>: The eigen.cut used in computation.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function requires access to the reference LD data via an environment variable.
You must set one of the following environment variables to the appropriate directory:</p><ul><li><p><code>UKB_ARRAY_PATH</code> for the Axiom Array reference (<code>UKB_array_SVD_eigen90_extraction</code>)</p></li>
<li><p><code>UKB_IMPUTED_PATH</code> for the full imputed reference (<code>UKB_imputed_SVD_eigen99_extraction</code>)</p></li>
<li><p><code>UKB_IMPUTED_HAPMAP2_PATH</code> for the imputed HapMap2 subset (<code>UKB_imputed_hapmap2_SVD_eigen99_extraction</code>)</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Ning Z, Pawitan Y, Shen X (2020).
“High-definition likelihood inference of genetic correlations across human complex traits.”
<em>Nature Genetics</em>, <b>52</b>, 859–864.
<a href="https://doi.org/10.1038/s41588-020-0676-4" class="external-link">doi:10.1038/s41588-020-0676-4</a>
.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">sumstat1</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="simulateSumstats.html">simulateSumstats</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">sumstat2</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="simulateSumstats.html">simulateSumstats</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nzchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"UKB_IMPUTED_HAPMAP2_PATH"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">SumstatGenCorr</span><span class="op">(</span></span></span>
<span class="r-in"><span>        ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>        referenceLD <span class="op">=</span> <span class="st">"UKB_imputed_hapmap2_SVD_eigen99_extraction"</span>,</span></span>
<span class="r-in"><span>        sumstat1 <span class="op">=</span> <span class="va">sumstat1</span>,</span></span>
<span class="r-in"><span>        sumstat2 <span class="op">=</span> <span class="va">sumstat2</span>,</span></span>
<span class="r-in"><span>        parallel <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysis starts on Fri Aug  8 10:35:38 2025</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 9276 out of 769306 (1.21%) SNPs in reference panel are available in GWAS 1.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 9458 out of 769306 (1.23%) SNPs in reference panel are available in GWAS 2.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Warning: More than 1% SNPs in reference panel are missed in GWAS 1. This may generate bias in estimation. Please make sure that you are using the correct reference panel.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Warning: More than 1% SNPs in reference panel are missed in GWAS 2. This may generate bias in estimation. Please make sure that you are using the correct reference panel.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   2%  |                                                                              |==                                                                    |   3%  |                                                                              |===                                                                   |   5%  |                                                                              |=====                                                                 |   6%  |                                                                              |======                                                                |   8%  |                                                                              |=======                                                               |  10%  |                                                                              |========                                                              |  11%  |                                                                              |=========                                                             |  13%  |                                                                              |==========                                                            |  15%  |                                                                              |===========                                                           |  16%  |                                                                              |============                                                          |  18%  |                                                                              |==============                                                        |  19%  |                                                                              |===============                                                       |  21%  |                                                                              |================                                                      |  23%  |                                                                              |=================                                                     |  24%  |                                                                              |==================                                                    |  26%  |                                                                              |===================                                                   |  27%  |                                                                              |====================                                                  |  29%  |                                                                              |=====================                                                 |  31%  |                                                                              |=======================                                               |  32%  |                                                                              |========================                                              |  34%  |                                                                              |=========================                                             |  35%  |                                                                              |==========================                                            |  37%  |                                                                              |===========================                                           |  39%  |                                                                              |============================                                          |  40%  |                                                                              |=============================                                         |  42%  |                                                                              |==============================                                        |  44%  |                                                                              |================================                                      |  45%  |                                                                              |=================================                                     |  47%  |                                                                              |==================================                                    |  48%  |                                                                              |===================================                                   |  50%  |                                                                              |====================================                                  |  52%  |                                                                              |=====================================                                 |  53%  |                                                                              |======================================                                |  55%  |                                                                              |========================================                              |  56%  |                                                                              |=========================================                             |  58%  |                                                                              |==========================================                            |  60%  |                                                                              |===========================================                           |  61%  |                                                                              |============================================                          |  63%  |                                                                              |=============================================                         |  65%  |                                                                              |==============================================                        |  66%  |                                                                              |===============================================                       |  68%  |                                                                              |=================================================                     |  69%  |                                                                              |==================================================                    |  71%  |                                                                              |===================================================                   |  73%  |                                                                              |====================================================                  |  74%  |                                                                              |=====================================================                 |  76%  |                                                                              |======================================================                |  77%  |                                                                              |=======================================================               |  79%  |                                                                              |========================================================              |  81%  |                                                                              |==========================================================            |  82%  |                                                                              |===========================================================           |  84%  |                                                                              |============================================================          |  85%  |                                                                              |=============================================================         |  87%  |                                                                              |==============================================================        |  89%  |                                                                              |===============================================================       |  90%  |                                                                              |================================================================      |  92%  |                                                                              |=================================================================     |  94%  |                                                                              |===================================================================   |  95%  |                                                                              |====================================================================  |  97%  |                                                                              |===================================================================== |  98%  |                                                                              |======================================================================| 100%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Integrating piecewise results</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Point estimates:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Heritability of phenotype 1:  0.00e+00</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Heritability of phenotype 2:  0.00e+00</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Genetic Covariance:  -1.46e-05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Genetic Correlation:  -Inf</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Warning: Heritability of one trait was estimated to be 0, which may be due to:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   1) The true heritability is very small;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   2) The sample size is too small;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   3) Many SNPs in the chosen reference panel are missing in the GWAS;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   4) There is a severe mismatch between the GWAS population and the population for computing the reference panel</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Continuing computing standard error with jackknife</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |=                                                                     |   2%</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Heritability of phenotype 1: 0.00e+00 ( 0.00e+00 )</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Heritability of phenotype 2: 0.00e+00 ( 0.00e+00 )</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Genetic Covariance: -1.46e-05 ( 0.00e+00 )</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> Genetic Correlation: -Inf ( NA )</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> P: NA</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BBBB00;">!</span> Warning: Heritability of one trait was estimated to be 0, which may be due to:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   1) The true heritability is very small;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   2) The sample size is too small;</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   3) Many SNPs in the chosen reference panel are missing in the GWAS.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Analysis finished at Fri Aug  8 10:36:10 2025</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

