<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>TestXGene: Performing gene-based association test using GWAS/XWAS summary statistics. — TestXGene • GXwasR</title><!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png"><link rel="icon" type="”image/svg+xml”" href="../favicon.svg"><link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png"><link rel="icon" sizes="any" href="../favicon.ico"><link rel="manifest" href="../site.webmanifest"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="TestXGene: Performing gene-based association test using GWAS/XWAS summary statistics. — TestXGene"><meta name="description" content='This function performs gene-based association tests using GWAS/XWAS summary statistics and SNP-SNP correlation
matrices. For  SNP-SNP correlation matrices, users have the flexibility to use either the base genotype data or 1000 Genomes
Phase 3 reference genotype data. Users also have options to define the regional positions of genes to include the SNPs according
to their investigation.
This function computes gene-wise SNP-SNP correlation matrices and can perform nine different gene-based tests, such as, “BT" (burden test),
"SKAT" (sequence kernel association test), "SKATO" (combination of BT and SKAT), "sumchi" (sum of χ2-statistics), "ACAT" (aggregated
Cauchy association test for combining P values), "PCA"(principal component approach), "FLM"( functional multiple linear regression model),
"simpleM" (Bonferroni correction test), "minp" (minimum P-value) leveraging PLINK1.9 (Purcell et al. 2007)
 and sumFREGAT
(Svishcheva et al. 2019; Belonogova et al. 2022)
 tools.
Though this function implicitly performs X-linked gene-based test, it is flexible to perform this analysis genome-wide.
For the details about the different tests, please follow the associated paper.'><meta property="og:description" content='This function performs gene-based association tests using GWAS/XWAS summary statistics and SNP-SNP correlation
matrices. For  SNP-SNP correlation matrices, users have the flexibility to use either the base genotype data or 1000 Genomes
Phase 3 reference genotype data. Users also have options to define the regional positions of genes to include the SNPs according
to their investigation.
This function computes gene-wise SNP-SNP correlation matrices and can perform nine different gene-based tests, such as, “BT" (burden test),
"SKAT" (sequence kernel association test), "SKATO" (combination of BT and SKAT), "sumchi" (sum of χ2-statistics), "ACAT" (aggregated
Cauchy association test for combining P values), "PCA"(principal component approach), "FLM"( functional multiple linear regression model),
"simpleM" (Bonferroni correction test), "minp" (minimum P-value) leveraging PLINK1.9 (Purcell et al. 2007)
 and sumFREGAT
(Svishcheva et al. 2019; Belonogova et al. 2022)
 tools.
Though this function implicitly performs X-linked gene-based test, it is flexible to perform this analysis genome-wide.
For the details about the different tests, please follow the associated paper.'><meta property="og:image" content="/logo.svg"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>TestXGene: Performing gene-based association test using GWAS/XWAS summary statistics.</h1>
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/R/GXwasR_main_functions.R" class="external-link"><code>R/GXwasR_main_functions.R</code></a></small>
      <div class="d-none name"><code>TestXGene.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function performs gene-based association tests using GWAS/XWAS summary statistics and SNP-SNP correlation
matrices. For  SNP-SNP correlation matrices, users have the flexibility to use either the base genotype data or 1000 Genomes
Phase 3 reference genotype data. Users also have options to define the regional positions of genes to include the SNPs according
to their investigation.</p>
<p>This function computes gene-wise SNP-SNP correlation matrices and can perform nine different gene-based tests, such as, “BT" (burden test),
"SKAT" (sequence kernel association test), "SKATO" (combination of BT and SKAT), "sumchi" (sum of χ2-statistics), "ACAT" (aggregated
Cauchy association test for combining P values), "PCA"(principal component approach), "FLM"( functional multiple linear regression model),
"simpleM" (Bonferroni correction test), "minp" (minimum P-value) leveraging PLINK1.9 (Purcell et al. 2007)
 and sumFREGAT
(Svishcheva et al. 2019; Belonogova et al. 2022)
 tools.</p>
<p>Though this function implicitly performs X-linked gene-based test, it is flexible to perform this analysis genome-wide.
For the details about the different tests, please follow the associated paper.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">TestXGene</span><span class="op">(</span></span>
<span>  <span class="va">DataDir</span>,</span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  <span class="va">finput</span>,</span>
<span>  <span class="va">sumstat</span>,</span>
<span>  <span class="va">gene_file</span>,</span>
<span>  gene_range <span class="op">=</span> <span class="fl">5e+05</span>,</span>
<span>  <span class="va">score_file</span>,</span>
<span>  ref_data <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  max_gene <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sample_size <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  genebasedTest <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SKAT"</span>, <span class="st">"SKATO"</span>, <span class="st">"sumchi"</span>, <span class="st">"ACAT"</span>, <span class="st">"BT"</span>, <span class="st">"PCA"</span>, <span class="st">"FLM"</span>, <span class="st">"simpleM"</span>,</span>
<span>    <span class="st">"minp"</span><span class="op">)</span>,</span>
<span>  gene_approximation <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">beta_par</span>,</span>
<span>  <span class="va">weights_function</span>,</span>
<span>  <span class="va">geno_variance_weights</span>,</span>
<span>  kernel_p_method <span class="op">=</span> <span class="st">"kuonen"</span>,</span>
<span>  acc_devies <span class="op">=</span> <span class="fl">1e-08</span>,</span>
<span>  lim_devies <span class="op">=</span> <span class="fl">1e+06</span>,</span>
<span>  rho <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skato_p_threshold <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  anno_type <span class="op">=</span> <span class="st">""</span>,</span>
<span>  <span class="va">mac_threshold</span>,</span>
<span>  <span class="va">reference_matrix_used</span>,</span>
<span>  <span class="va">regularize_fun</span>,</span>
<span>  pca_var_fraction <span class="op">=</span> <span class="fl">0.85</span>,</span>
<span>  flm_basis_function <span class="op">=</span> <span class="st">"fourier"</span>,</span>
<span>  flm_num_basis <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  flm_poly_order <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  flip_genotypes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  omit_linear_variant <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-datadir">DataDir<a class="anchor" aria-label="anchor" href="#arg-datadir"></a></dt>
<dd><p>A character string for the file path of the all the input files.</p></dd>


<dt id="arg-resultdir">ResultDir<a class="anchor" aria-label="anchor" href="#arg-resultdir"></a></dt>
<dd><p>A character string for the file path where all output files will be stored. The default is <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir()</a></code>.</p></dd>


<dt id="arg-finput">finput<a class="anchor" aria-label="anchor" href="#arg-finput"></a></dt>
<dd><p>Character string, specifying the prefix of the input PLINK binary files for the genotype data. This file is used to compute the
correlation between the SNPs. This file needs to be in DataDir. If the base genotype data is unavailable, then users can use the
1000 Genomes Project samples. Users should use the population that most closely represents the base sample. For ACAT model, this
parameter is not mandatory and could be set <code>NULL</code>.</p></dd>


<dt id="arg-sumstat">sumstat<a class="anchor" aria-label="anchor" href="#arg-sumstat"></a></dt>
<dd><p>A dataframe object with GWAS summary statistics. When the base-genotype data is used to compute genetic correlations, the mandatory
columns are:</p><ul><li><p>Column 1: <code>CHROM</code> (i.e., chromosome code),</p></li>
<li><p>Column 2: <code>POS</code> (i.e., base-pair position),</p></li>
<li><p>Column 3: <code>ID</code> (i.e. SNP IDs),</p></li>
<li><p>Column 4: <code>P</code> (i.e., p-values),</p></li>
<li><p>Column 5: <code>BETA</code> (i.e., effect-size),</p></li>
<li><p>Column 6: <code>A1</code> (i.e., effect allele),</p></li>
<li><p>Column 7: <code>A2</code> (i.e., alternative allele) and</p></li>
<li><p>Column 8: <code>EAF</code> (i.e., the effect allele frequency)</p></li>
</ul><p>These are mandatory when base-genotype data is used to compute genetic correlations. Otherwise, if the users are using reference data,
then columns 5 to 8 are optional. Also, in that case, columns, such as <code>REF</code> (i.e., reference allele), and <code>ALT</code> (i.e., alternative allele)
could be present to compare alleles with those in the reference file and exclude genetic variants if alleles do not match. There could be an
additional column, <code>ANNO</code> with functional annotations (like "intron_variant", "synonymous", "missense" etc.)</p></dd>


<dt id="arg-gene-file">gene_file<a class="anchor" aria-label="anchor" href="#arg-gene-file"></a></dt>
<dd><p>Character string, specifying the prefix of the name of a .txt file listing genes in refFlat format. This file needs to be in <code>DataDir</code>. The
X-linked gene files, "Xlinkedgenes_hg19.txt" and "Xlinkedgenes_hg38.txt" and autosomal gene files, “Autosomes_hg19.txt” and “Autosomes_hg38.txt”
can be specified. The default is "Xlinkedgenes_hg19.txt". The genome built should be in agreement with the analysis.</p></dd>


<dt id="arg-gene-range">gene_range<a class="anchor" aria-label="anchor" href="#arg-gene-range"></a></dt>
<dd><p>Integer value, specifying the up stream and down stream range (in kilo base) of a gene for SNPs to be considered. The default is 500000.</p></dd>


<dt id="arg-score-file">score_file<a class="anchor" aria-label="anchor" href="#arg-score-file"></a></dt>
<dd><p>Character string, specifying the prefix of a file which will be used to produce score files with Z scores from P values and beta input
from GWAS summary statistics.</p></dd>


<dt id="arg-ref-data">ref_data<a class="anchor" aria-label="anchor" href="#arg-ref-data"></a></dt>
<dd><p>Character string, specifying the path to a reference dataframe with additional data needed to recode user data according to correlation matrices
that will be used. It contains <code>ID</code> column with names of  SNPs,  <code>REF</code> and <code>ALT</code> columns with alleles that were coded as 0 and 1, respectively.
Effect sizes from data will be inverted for variants with effect alleles different from <code>ALT</code> alleles in reference data. If presented, <code>REF</code> and
<code>ALT</code> columns from the input data will be used to sort out variants with alleles different from those in reference data. This dataframe  can also
be a source of map data and allele frequencies if they are not present in data. <code>AF</code> column in the reference file represents the allele frequency
of <code>ALT</code> allele. The default is "ref1KG.MAC5.EUR_AF.RData".</p></dd>


<dt id="arg-max-gene">max_gene<a class="anchor" aria-label="anchor" href="#arg-max-gene"></a></dt>
<dd><p>Positive integer value, specifying the number of genes for which the gene-based test will be performed. The default is NULL to consider
all the genes.</p></dd>


<dt id="arg-sample-size">sample_size<a class="anchor" aria-label="anchor" href="#arg-sample-size"></a></dt>
<dd><p>Positive integer value, specifying the sample size of the GWAS. Only needed for FLM and PCA models.</p></dd>


<dt id="arg-genebasedtest">genebasedTest<a class="anchor" aria-label="anchor" href="#arg-genebasedtest"></a></dt>
<dd><p>Character string, specifying the name of the gene-based test. Nine different tests can be specified, "SKAT","SKATO","sumchi","ACAT","BT","PCA",
"FLM","simpleM","minp". The default is "SKAT".</p></dd>


<dt id="arg-gene-approximation">gene_approximation<a class="anchor" aria-label="anchor" href="#arg-gene-approximation"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether approximation for large genes (&gt;= 500 SNPs) should be used. Applicable for SKAT,
SKATO, sumchi, PCA, FLM. The default is <code>TRUE</code> for these methods).</p></dd>


<dt id="arg-beta-par">beta_par<a class="anchor" aria-label="anchor" href="#arg-beta-par"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code>, specifying whether approximation for large genes (&gt;= 500 SNPs) should be used. Applicable for SKAT, SKATO,
sumchi, PCA, FLM (default = <code>TRUE</code> for these methods).</p></dd>


<dt id="arg-weights-function">weights_function<a class="anchor" aria-label="anchor" href="#arg-weights-function"></a></dt>
<dd><p>A function of MAF to assign weights for each genetic variant. By default is <code>NULL</code>. In this case the weights will be calculated using
the beta distribution.</p></dd>


<dt id="arg-geno-variance-weights">geno_variance_weights<a class="anchor" aria-label="anchor" href="#arg-geno-variance-weights"></a></dt>
<dd><p>Character string, indicating whether scores should be weighted by the variance of genotypes: "none" (i.e., no weights applied, resulting
in a sum chi-square test); "se.beta" (i.e., scores weighted by variance of genotypes estimated from P values and effect sizes); "af"
(i.e., scores weighted by variance of genotypes calculated as \(AF * (1 - AF)\), where AF is allele frequency.</p></dd>


<dt id="arg-kernel-p-method">kernel_p_method<a class="anchor" aria-label="anchor" href="#arg-kernel-p-method"></a></dt>
<dd><p>Character string, specifying the method for computing P value in kernel-based tests, such as SKAT, SKATO and sumchi. Available methods
are "kuonen" (Belonogova et al. 2022)
 "davies" (Belonogova et al. 2022)
 and "hybrid" (Belonogova et al. 2022)
.
The default is "kuonen".</p></dd>


<dt id="arg-acc-devies">acc_devies<a class="anchor" aria-label="anchor" href="#arg-acc-devies"></a></dt>
<dd><p>Positive numeric value, specifying the accuracy parameter for "davies" method. The default is 1e-8.</p></dd>


<dt id="arg-lim-devies">lim_devies<a class="anchor" aria-label="anchor" href="#arg-lim-devies"></a></dt>
<dd><p>Positive numeric value, specifying the limit parameter for "davies" method. The default is 1e+6.</p></dd>


<dt id="arg-rho">rho<a class="anchor" aria-label="anchor" href="#arg-rho"></a></dt>
<dd><p>Logical value, 'TRUE' or 'FALSE' or can be a vector of grid values from 0 to 1. If TRUE, the optimal test (SKAT-O) is performed (12).
The default grid is c(0, 0.1^2, 0.2^2, 0.3^2, 0.4^2, 0.5^2, 0.5, 1).</p></dd>


<dt id="arg-skato-p-threshold">skato_p_threshold<a class="anchor" aria-label="anchor" href="#arg-skato-p-threshold"></a></dt>
<dd><p>Positive numeric value, specifying the largest P value that will be considered as important when performing computational optimization in
SKAT-O. All P values larger than skato_p_threshold will be processed via burden test. The default is 0.8</p></dd>


<dt id="arg-anno-type">anno_type<a class="anchor" aria-label="anchor" href="#arg-anno-type"></a></dt>
<dd><p>A character (or character vector) indicating annotation types to be used. The default is "" (i.e, nothing).</p></dd>


<dt id="arg-mac-threshold">mac_threshold<a class="anchor" aria-label="anchor" href="#arg-mac-threshold"></a></dt>
<dd><p>Integer value, specifying the threshold of MACs (Minor allele content) calculated from MAFs. In ACAT, scores with MAC &lt;= 10 will be
combined using Burden test.</p></dd>


<dt id="arg-reference-matrix-used">reference_matrix_used<a class="anchor" aria-label="anchor" href="#arg-reference-matrix-used"></a></dt>
<dd><p>Boolean value, <code>TRUE</code> or <code>FALSE</code> logical indicating whether the correlation matrices were generated using the reference matrix. The
default is <code>FALSE</code>. If  <code>TRUE</code>, regularization algorithms will be applied to ensure the invertibility and numerical stability of
the matrices.</p></dd>


<dt id="arg-regularize-fun">regularize_fun<a class="anchor" aria-label="anchor" href="#arg-regularize-fun"></a></dt>
<dd><p>Character string, specifying the one of two regularization algorithms if ‘reference_matrix’ is TRUE:  'LH' (default) or 'derivLH'.
Currently, both give similar results.</p></dd>


<dt id="arg-pca-var-fraction">pca_var_fraction<a class="anchor" aria-label="anchor" href="#arg-pca-var-fraction"></a></dt>
<dd><p>P
ositive numeric value, specifying the minimal proportion of genetic variance within the region that should be explained by principal
components used in PCA method. This is also valid in 'simpleM'. The default is 0.85.</p></dd>


<dt id="arg-flm-basis-function">flm_basis_function<a class="anchor" aria-label="anchor" href="#arg-flm-basis-function"></a></dt>
<dd><p>Character string, specifying the name of a basis function type for beta-smooth in FLM method. Can be set to "bspline" (B-spline basis)
or "fourier" (Fourier basis, default).</p></dd>


<dt id="arg-flm-num-basis">flm_num_basis<a class="anchor" aria-label="anchor" href="#arg-flm-num-basis"></a></dt>
<dd><p>Positive integer value, specifying the number of basis functions to be used for beta-smooth in FLM method. The default is 25.</p></dd>


<dt id="arg-flm-poly-order">flm_poly_order<a class="anchor" aria-label="anchor" href="#arg-flm-poly-order"></a></dt>
<dd><p>Positive integer value, specifying the polynomial order to be used in "bspline" for FLM model. The default = 4, which corresponds to
the cubic B-splines. This has no effect if only Fourier bases are used</p></dd>


<dt id="arg-flip-genotypes">flip_genotypes<a class="anchor" aria-label="anchor" href="#arg-flip-genotypes"></a></dt>
<dd><p>L
ogical value, <code>TRUE</code> or <code>FALSE</code>, indicating whether the genotypes of some genetic variants should be flipped (relabelled) for their
better functional representation (13). The default is <code>FALSE</code>.</p></dd>


<dt id="arg-omit-linear-variant">omit_linear_variant<a class="anchor" aria-label="anchor" href="#arg-omit-linear-variant"></a></dt>
<dd><p>Logical value, <code>TRUE</code> or <code>FALSE</code>, indicating whether to omit linearly dependent genetic variants. It was done in the FLM test (4).
The default is <code>FALSE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame with columns:</p><ul><li><p>gene</p></li>
<li><p>chrom</p></li>
<li><p>start</p></li>
<li><p>end</p></li>
<li><p>markers (i.e., numbers of SNPs),</p></li>
<li><p>filtered.markers (i.e. filtered SNPs)</p></li>
<li><p>pvalue (i.e., p-value).</p></li>
</ul><p>Additionally, for  “BT”, there will be “beta” (i.e., gene-level estimates of betas) and “beta.se” (i.e., standard errors of betas).
For “FLM”, there will be the “model” column with the names of the functional models used for each region. Names shortly describe the
functional basis and the number of basis functions used. E.g., "F25" means 25 Fourier basis functions, "B15" means 15 B-spline basis
functions. For “PCA”, there will be the “ncomponents” (the number of the principal components used for each region) and
“explained.variance.fraction” (i.e., the proportion of genetic variance they make up) columns.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Belonogova NM, Svishcheva GR, Kirichenko AV, Zorkoltseva IV, Tsepilov YA, Axenovich TI (2022).
“sumSTAAR: A flexible framework for gene-based association studies using GWAS summary statistics.”
<em>PLoS Computational Biology</em>, <b>18</b>(6), e1010172.
<a href="https://doi.org/10.1371/journal.pcbi.1010172" class="external-link">doi:10.1371/journal.pcbi.1010172</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/35653402" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/35653402</a>.<br><br> Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).
“PLINK: A Tool Set for Whole-Genome Association and Population-Based Linkage Analyses.”
<em>The American Journal of Human Genetics</em>, <b>81</b>(3), 559–575.
<a href="https://doi.org/10.1086/519795" class="external-link">doi:10.1086/519795</a>
.<br><br> Svishcheva GR, Belonogova NM, Zorkoltseva IV, Kirichenko AV, Axenovich TI (2019).
“Gene-based association tests using GWAS summary statistics.”
<em>Bioinformatics</em>, <b>35</b>(19), 3701–3708.
<a href="https://doi.org/10.1093/bioinformatics/btz172" class="external-link">doi:10.1093/bioinformatics/btz172</a>
, <a href="http://www.ncbi.nlm.nih.gov/pubmed/30860568" class="external-link">http://www.ncbi.nlm.nih.gov/pubmed/30860568</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Banabithi Bose</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Sys.getenv.html" class="external-link">Sys.getenv</a></span><span class="op">(</span><span class="st">"CI"</span><span class="op">)</span> <span class="op">==</span> <span class="st">"true"</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/Sys.info.html" class="external-link">Sys.info</a></span><span class="op">(</span><span class="op">)</span><span class="op">[[</span><span class="st">"sysname"</span><span class="op">]</span><span class="op">]</span> <span class="op">==</span> <span class="st">"Darwin"</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"XWAS_Summary_Example"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></span>
<span class="r-in"><span>    <span class="va">sumstat</span> <span class="op">&lt;-</span> <span class="va">XWAS_Summary_Example</span></span></span>
<span class="r-in"><span>    <span class="va">ref_data</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span>    <span class="va">gene_file</span> <span class="op">&lt;-</span> <span class="st">"Xlinkedgenes_hg19.txt"</span></span></span>
<span class="r-in"><span>    <span class="va">gene_range</span> <span class="op">&lt;-</span> <span class="fl">500000</span></span></span>
<span class="r-in"><span>    <span class="va">max_gene</span> <span class="op">&lt;-</span> <span class="fl">10</span></span></span>
<span class="r-in"><span>    <span class="va">gene_approximation</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>    <span class="va">beta_par</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">25</span><span class="op">)</span></span></span>
<span class="r-in"><span>    <span class="va">weights_function</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span>    <span class="va">geno_variance_weights</span> <span class="op">&lt;-</span> <span class="st">"se.beta"</span></span></span>
<span class="r-in"><span>    <span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"kuonen"</span></span></span>
<span class="r-in"><span>    <span class="va">acc_devies</span> <span class="op">&lt;-</span> <span class="fl">1e-8</span></span></span>
<span class="r-in"><span>    <span class="va">lim_devies</span> <span class="op">&lt;-</span> <span class="fl">1e+6</span></span></span>
<span class="r-in"><span>    <span class="va">rho</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span>    <span class="va">skato_p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.8</span></span></span>
<span class="r-in"><span>    <span class="va">mac_threshold</span> <span class="op">&lt;-</span> <span class="fl">3</span></span></span>
<span class="r-in"><span>    <span class="va">sample_size</span> <span class="op">&lt;-</span> <span class="fl">4000</span></span></span>
<span class="r-in"><span>    <span class="va">reference_matrix_used</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span>    <span class="va">regularize_fun</span> <span class="op">&lt;-</span> <span class="st">"LH"</span></span></span>
<span class="r-in"><span>    <span class="va">pca_var_fraction</span> <span class="op">&lt;-</span> <span class="fl">0.85</span></span></span>
<span class="r-in"><span>    <span class="va">flm_basis_function</span> <span class="op">&lt;-</span> <span class="st">"fourier"</span></span></span>
<span class="r-in"><span>    <span class="va">flm_num_basis</span> <span class="op">&lt;-</span> <span class="fl">25</span></span></span>
<span class="r-in"><span>    <span class="va">flm_poly_order</span> <span class="op">&lt;-</span> <span class="fl">4</span></span></span>
<span class="r-in"><span>    <span class="va">flip_genotypes</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span>    <span class="va">omit_linear_variant</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span></span>
<span class="r-in"><span>    <span class="va">kernel_p_method</span> <span class="op">&lt;-</span> <span class="st">"kuonen"</span></span></span>
<span class="r-in"><span>    <span class="va">anno_type</span> <span class="op">&lt;-</span> <span class="st">""</span></span></span>
<span class="r-in"><span>    <span class="va">GenetestResult</span> <span class="op">&lt;-</span> <span class="fu">TestXGene</span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">sumstat</span>, <span class="va">gene_file</span>,</span></span>
<span class="r-in"><span>        <span class="va">gene_range</span>, <span class="va">score_file</span>, <span class="va">ref_data</span>, <span class="va">max_gene</span>, <span class="va">sample_size</span>,</span></span>
<span class="r-in"><span>        genebasedTest <span class="op">=</span> <span class="st">"SKAT"</span>,</span></span>
<span class="r-in"><span>        <span class="va">gene_approximation</span>, <span class="va">beta_par</span>, <span class="va">weights_function</span>, <span class="va">geno_variance_weights</span>,</span></span>
<span class="r-in"><span>        <span class="va">kernel_p_method</span>, <span class="va">acc_devies</span>, <span class="va">lim_devies</span>, <span class="va">rho</span>, <span class="va">skato_p_threshold</span>, <span class="va">anno_type</span>,</span></span>
<span class="r-in"><span>        <span class="va">mac_threshold</span>, <span class="va">reference_matrix_used</span>, <span class="va">regularize_fun</span>, <span class="va">pca_var_fraction</span>,</span></span>
<span class="r-in"><span>        <span class="va">flm_basis_function</span>, <span class="va">flm_num_basis</span>, <span class="va">flm_poly_order</span>, <span class="va">flip_genotypes</span>,</span></span>
<span class="r-in"><span>        <span class="va">omit_linear_variant</span></span></span>
<span class="r-in"><span>    <span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 80 of 80 variants found in reference</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Effect sizes recoded for 0 variant(s)</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> Allele frequencies found and linked</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BB00;">✔</span> File /tmp/RtmpteYOxY/gene.test.score.file.vcf.gz has been created and indexed</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> 157 genes are having 321 SNPs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> SNP-SNP correlation matrices are being created...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> An error occurred: PLINK binary not found.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'PLINK_PATH' environment variable and 'plink' in system PATH.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure PLINK is available and executable (bioinformatics version).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'plink' by running:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: PLINK_PATH=/full/path/to/plink</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in verifyPlink():</span> PLINK binary not found.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #BB0000;">✖</span> Attempted to locate the 'PLINK_PATH' environment variable and 'plink' in system PATH.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> Ensure PLINK is available and executable (bioinformatics version).</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span style="color: #0000BB;">ℹ</span> You can permanently set the path to 'plink' by running:</span>
<span class="r-err co"><span class="r-pr">#&gt;</span>   usethis::edit_r_environ()  # then add a line like: PLINK_PATH=/full/path/to/plink</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

