<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">8 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/vignettes/decoding_ancestry.Rmd" class="external-link"><code>vignettes/decoding_ancestry.Rmd</code></a></small>
      <div class="d-none name"><code>decoding_ancestry.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The pursuit of ancestral origins using genetic data has become
increasingly accessible and sophisticated with the advent of genomics
technology. Our R package GXwasR enables the identification of
individuals of divergent ancestry through genotype datasets. This
vignette provides an in-depth look at how GXwasR facilitates ancestry
estimation, particularly focusing on its <code>AncestryCheck</code>()
function.</p>
</div>
<div class="section level2">
<h2 id="identifying-divergent-ancestry">Identifying Divergent Ancestry<a class="anchor" aria-label="anchor" href="#identifying-divergent-ancestry"></a>
</h2>
<p>The core of ancestry estimation lies in combining the genotypes of a
study population with those from a reference dataset, such as the HapMap
or 1000 Genomes studies. This integration allows for the detection of
population structure down to the level of the reference dataset,
revealing continental-scale ancestry differences.</p>
</div>
<div class="section level2">
<h2 id="accessing-gxwasr-and-simulated-genotype-datasets">Accessing GXwasR and Simulated Genotype Datasets<a class="anchor" aria-label="anchor" href="#accessing-gxwasr-and-simulated-genotype-datasets"></a>
</h2>
<p>GXwasR is readily available on Bioconductor and can be installed
using BiocManager::install(“GXwasR”). The package includes a simulated
genotype dataset prefixed as <strong>GXwasR_example</strong> as plink
bed files (i.e., .bed, .fam, and .bim), ideal for users to experiment
with. This simulated dataset consists of genotypes from 276 individuals
across 26,527 genetic markers, providing a realistic scenario for
testing the functionality.</p>
<p>Let’s see more details of this genotype dataset that we will call
‘Study data’.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GXwasR: Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data</span></span>
<span><span class="co">#&gt; Version 0.99.0 () installed</span></span>
<span><span class="co">#&gt; Author: c(</span></span>
<span><span class="co">#&gt;     person(given = "Banabithi",</span></span>
<span><span class="co">#&gt;            family = "Bose",</span></span>
<span><span class="co">#&gt;            role = c("cre", "aut"),</span></span>
<span><span class="co">#&gt;            email = "banabithi.bose@gmail.com",</span></span>
<span><span class="co">#&gt;            comment = c(ORCID = "0000-0003-0842-8768"))</span></span>
<span><span class="co">#&gt;            )</span></span>
<span><span class="co">#&gt; Maintainer: Banabithi Bose &lt;banabithi.bose@gmail.com&gt;</span></span>
<span><span class="co">#&gt; Tutorial: https://github.com</span></span>
<span><span class="co">#&gt; Use citation("GXwasR") to know how to cite this work.</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ancestrycheck-function-overview">AncestryCheck Function Overview<a class="anchor" aria-label="anchor" href="#ancestrycheck-function-overview"></a>
</h2>
<p><code>AncestryCheck</code> is a comprehensive R function designed for
comprehensive ancestry analysis in genetic studies. By leveraging plink
software via R platform,it seamlessly integrates several critical steps:
SNP filtering for data integrity, linkage disequilibrium (LD) pruning to
reduce data complexity, correction of chromosome mismatches, and
checking for allele flips to ensure data accuracy. Additionally, it
adeptly merges study and reference datasets, conducts Principal
Component Analysis (PCA) to unravel population structures, and
incorporates outlier detection and visualization. This function is
pivotal in comparing study samples’ self-reported ancestry against
reference populations, enabling a detailed analysis of population
structure and the identification of divergent ancestries.</p>
</div>
<div class="section level2">
<h2 id="implemented-workflow-of-ancestry-estimation-in-ancestrycheck">Implemented workflow of ancestry estimation in AncestryCheck<a class="anchor" aria-label="anchor" href="#implemented-workflow-of-ancestry-estimation-in-ancestrycheck"></a>
</h2>
<p>The <code>AncestryCheck</code>() function in the GXwasR package is a
sophisticated tool designed for ancestry analysis in genetic studies. It
provides a comprehensive approach, from data preparation to advanced
statistical analysis, for estimating ancestral backgrounds. Let’s delve
into the technical aspects and workflow of this function:</p>
<div class="section level3">
<h3 id="data-preparation">Data Preparation:<a class="anchor" aria-label="anchor" href="#data-preparation"></a>
</h3>
<ol style="list-style-type: decimal">
<li><p><strong>Directories and initial quality check:</strong> The
‘<code>AncestryCheck</code>’ function requires specific directories for
inputting PLINK binary files and for saving output files. At this stage,
the input genotype dataset, i.e., the study dataset, should have already
undergone thorough quality control. GXwasR offers a range of functions
to facilitate this preliminary quality check (QC) of genotype datasets,
essential for accurate ancestry estimation. This QC process typically
includes filtering out multi-allelic variants, selecting SNPs based on
genotype call rate, filtering for specific chromosomes, SNP filtering
grounded in minor allele frequency, adherence to Hardy-Weinberg
equilibrium, and sample selection based on criteria like call rate, low
missing rate, and heterozygosity thresholds. It’s important to apply
these filters in the appropriate sequence for optimal dataset
preparation. For a more detailed understanding of this QC pipeline,
users can refer to the comprehensive guide available at <a href="https://boseb.github.io/GXwasR/articles/preimputationQC.html" class="external-link">https://boseb.github.io/GXwasR/articles/preimputationQC.html</a>.</p></li>
<li><p><strong>Genomic Build and X-Chromosome Compatibility:</strong>
Users of the <code>AncestryCheck</code> function must ensure that both
their study and reference datasets are aligned on the same genomic
build, with consistent chromosome coding across the entire genome. This
alignment is crucial, particularly since sex chromosomes (X and Y) often
have different encoding conventions in genomic datasets. The
<code>AncestryCheck</code> function is equipped to implicitly handle
variations in sex chromosome encoding. However, users are advised to
employ tools like Liftover or CrossMap for verifying and aligning
genomic builds and chromosome codes between their study and reference
datasets, thus ensuring compatibility and accuracy in the ancestry
estimation process.</p></li>
<li><p><strong>Reference Dataset Selection:</strong> The function offers
users the flexibility to select from two established reference
populations, such as, <strong>HapMap phase III</strong>[2] or the
<strong>1000 Genomes phase 3</strong>[3] Projects. It is specifically
designed to work seamlessly with the preprocessed genotype datasets of
these reference panels. For users seeking to download and process these
datasets, a comprehensive guidance was provided through a dedicated
vignette. This detailed resource, which can be accessed at <a href="https://boseb.github.io/Processing_1000Genome/tutorial.html" class="external-link">https://boseb.github.io/Processing_1000Genome/tutorial.html</a>,
offers step-by-step instructions on how to effectively prepare these
reference datasets for use in ancestry analysis, ensuring users can make
the most out of their genetic studies with the selected reference
populations.</p></li>
</ol>
</div>
<div class="section level3">
<h3 id="snp-filtering">SNP Filtering:<a class="anchor" aria-label="anchor" href="#snp-filtering"></a>
</h3>
<p>The function incorporates a crucial step of SNP filtering for both
study and reference datasets. This filtering process is particularly
focused on excluding SNPs that do not conform to A-T or G-C base
pairings. By implementing this filtering, <code>AncestryCheck</code>
ensures the overall quality and reliability of the genetic data from
both study and reference datasets. This step is vital in maintaining the
integrity of the analysis, as it helps in eliminating potentially
problematic SNPs that could skew the ancestry estimation results.</p>
</div>
<div class="section level3">
<h3 id="ld-pruning">LD Pruning:<a class="anchor" aria-label="anchor" href="#ld-pruning"></a>
</h3>
<p>The function includes a critical step of LD-based filtering for both
the study and reference datasets, which is essential in simplifying the
genetic data by removing variants in high linkage disequilibrium. This
process is especially significant for conducting accurate PCA on the
study dataset SNPs. The default setting targets variants in LD with an
<code>r2</code> value greater than 0.2 within a 50kb window. However,
<code>AncestryCheck</code> offers users the flexibility to adjust these
parameters according to their specific research needs. Users can choose
whether or not to apply LD-based filtering and can modify parameters
like the <code>r2</code> threshold and the <code>window size</code>.
This adaptability ensures that the function can be tailored to a variety
of study designs and research objectives, making it a versatile tool for
ancestry analysis.</p>
<p>The ’<code>AncestryCheck</code> then utilizes the list of pruned
variants from the study data to proportionately adjust the reference
dataset, aligning it to the scale of the study data. This harmonization
is crucial for accurate comparative analysis and ensures consistency in
the ancestral estimation process.</p>
</div>
<div class="section level3">
<h3 id="filter-for-known-high-ld-structure-">Filter for known high-LD structure.<a class="anchor" aria-label="anchor" href="#filter-for-known-high-ld-structure-"></a>
</h3>
<p>The ‘<code>AncestryCheck</code>’ provides an additional layer of
refinement by offering the option to filter out genomic ranges
characterized by high-LD [4]. This feature is particularly beneficial
for genomic builds hg19 and hg38, as it targets and removes regions
within these builds that are known for their high-LD structures. These
specific high-LD region files are already incorporated into the GXwasR
package, readily accessible for use in <code>AncestryCheck</code>. This
targeted filtering is crucial in further enhancing the quality and
accuracy of the genetic data being analyzed, ensuring that the resulting
ancestry estimation is as precise and reliable as possible.</p>
</div>
<div class="section level3">
<h3 id="genotype-data-processing-for-merge">Genotype Data Processing for Merge:<a class="anchor" aria-label="anchor" href="#genotype-data-processing-for-merge"></a>
</h3>
<p>The function includes a meticulous process to prepare genotype data
for merging, addressing key issues that could otherwise compromise the
accuracy of the analysis:</p>
<p><strong>Correcting Mismatches:</strong> The function systematically
corrects mismatches in chromosome IDs, positions, and allele
annotations. This step is essential to ensure that the study and
reference data align accurately, providing a reliable basis for
subsequent analysis. By rectifying these discrepancies,
<code>AncestryCheck</code> safeguards against potential errors that
could arise from inconsistencies in the datasets.</p>
<p><strong>Handling Allele Flips:</strong> Allele flips are a frequent
complication in genotype data, where alleles may be reported in reverse
order. The function is adept at identifying and rectifying these flips,
thereby preserving the integrity of the genetic information. Given the
importance of accurately matching SNPs, the ability to correct allele
flips is crucial for error-free dataset merging.</p>
<p><strong>Dealing with Sex Chromosomes:</strong> The encoding of sex
chromosomes often differs from autosomal chromosomes, which can
introduce additional complexity in matching study and reference
datasets. <code>AncestryCheck</code> is designed to handle these
variations effectively.</p>
<p><strong>Removing Mismatches Post-Flipping:</strong> After the allele
flipping process, the function performs a thorough check to identify any
remaining mismatches. Any alleles that are still not aligned correctly
after the flipping process are identified and removed from the reference
dataset. This final step ensures that only perfectly matched SNPs are
retained for the merged dataset, thereby enhancing the accuracy of the
ancestral analysis.</p>
<p>Through these comprehensive steps, <code>AncestryCheck</code> ensures
that the genotype datasets are optimally prepared for merging, laying a
solid foundation for accurate and reliable ancestry estimation.</p>
</div>
<div class="section level3">
<h3 id="principal-component-analysis-pca">Principal Component Analysis (PCA):<a class="anchor" aria-label="anchor" href="#principal-component-analysis-pca"></a>
</h3>
<p><strong>Merging Datasets:</strong> After preprocessing, the study and
reference datasets are merged.</p>
<p><strong>Executing PCA:</strong> PCA is then performed on the combined
dataset, providing insights into the genetic variance and ancestral
components.</p>
<p>By implementing PCA on the harmonized genetic data,
<code>AncestryCheck</code> effectively translates the nuanced genetic
variations into interpretable patterns, facilitating a deeper
understanding of the ancestral backgrounds inherent in the study
population. This step is fundamental in transforming raw genetic data
into meaningful insights about human ancestry and genetic diversity.</p>
</div>
<div class="section level3">
<h3 id="reference-populations">Reference Populations:<a class="anchor" aria-label="anchor" href="#reference-populations"></a>
</h3>
<p><strong>HapMap:</strong> We combined and used HapMap data for the
following populations: AFR = ASW (African Ancestry in SW USA), LWK
(Luhya in Webuye, Kenya), MKK (Maasai in Kinyawa, Kenya), and YRI
(Yoruba in Ibadan, Nigeria). EAS = CHB (Han Chinese in Beijing, China),
CHD (Chinese in Metropolitan Denver, CO, USA), and JPT (Japanese in
Tokyo, Japan). EUR = CEU (CEPH/Utah Collection) and TSI (Toscani in
Italia). SAS = GIH (Gujarati Indians in Houston, Texas, USA).</p>
<p><strong>1000Genomes:</strong> We used 1000Genomes data for the
following super populations: AFR = ASW (African Ancestry in Southwest
US), ACB (African Caribbean in Barbados), ESN (Esan in Nigeria), GWD
(Gambian in Western Division, The Gambia), LWK (Luhya in Webuye, Kenya),
MSL (Mende in Sierra Leone), and YRI (Yoruba in Ibadan, Nigeria). EAS =
CDX (Chinese Dai in Xishuangbanna, China), CHB (Han Chinese in Bejing,
China), JPT (Japanese in Tokyo, Japan), KHV (Kinh in Ho Chi Minh City,
Vietnam), and CHS (Southern Han Chinese, China). EUR = GBR (British in
England and Scotland), FIN (Finnish in Finland), IBS (Iberian
populations in Spain), TSI (Toscani in Italy), and CEU (Utah residents
with Northern and Western European ancestry). SAS = BEB (Bengali in
Bangladesh), GIH (Gujarati Indian in Houston,TX), ITU (Indian Telugu in
the UK), PJL (Punjabi in Lahore,Pakistan), and STU (Sri Lankan Tamil in
the UK).</p>
<p>More information can be found here: <a href="http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv" class="external-link uri">http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/20131219.populations.tsv</a></p>
</div>
<div class="section level3">
<h3 id="outlier-detection-and-ancestry-assignment-analysis">Outlier Detection and Ancestry Assignment Analysis:<a class="anchor" aria-label="anchor" href="#outlier-detection-and-ancestry-assignment-analysis"></a>
</h3>
<p><strong>Flexibility in Reference Population Selection:</strong> A key
feature of the function is its versatility in handling various reference
populations. Users have the freedom to select a specific reference
population against which the study samples’ ancestry labels are
compared. For instance, setting the parameter <code>outlierOf</code> to
EUR (i.e., <code>Ref_EUR</code>) enables the identification of
non-European outliers in a European context. Other reference populations
like AFR (African), EAS (East Asian), and SAS (South Asian) can also be
used, allowing for a wide range of comparative ancestral analyses. This
flexibility ensures that <code>AncestryCheck</code> can cater to diverse
research needs and objectives.</p>
<p><strong>Setting Outlier/Prediction Thresholds:</strong> The function
allows users to define specific thresholds for outlier
detection/ancestry assignement, adding a layer of customization to the
analysis. It employs a methodical approach by calculating PC1 and PC2
Z-scores for each subject in relation to mean and standard deviation of
PC1 and PC2 for each reference population. Samples with PC1 and PC2
Z-scores within the user-defined threshold are predicted as having the
same ancestry as that reference population. For example, in the case of
European reference samples, the function computes the mean and standard
deviation of the first two principal components
(<code>mean(PC1 of Ref_EUR)</code>, <code>sd(PC1 of Ref_EUR</code>),
<code>mean(PC2 of Ref_EUR)</code>, <code>sd(PC2 of Ref_EUR)</code>). It
then calculates Z-scores for PC1 and PC2
(<code>Z-score PC1 of Ref_EUR &lt;- (PC1 - mean(PC1 of Ref_EUR))/sd(PC1 of Ref_EUR)</code>,
<code>Z-score PC2 of Ref_EUR &lt;- (PC2 - mean(PC2 of Ref_EUR))/sd(PC2 of Ref_EUR)</code>
) for the samples in relation to the European Reference. Any study
sample whose PC1 and PC2 Z-score is considered an outlier of the
European population. This approach is particularly effective in
pinpointing samples with ancestral backgrounds that significantly
deviate from the selected reference population, providing a clear and
quantifiable method to identify and analyze outliers.</p>
</div>
<div class="section level3">
<h3 id="visualization-and-reporting">Visualization and Reporting:<a class="anchor" aria-label="anchor" href="#visualization-and-reporting"></a>
</h3>
<p><strong>PCA Plot Visualization:</strong> One of the most insightful
aspects of <code>AncestryCheck</code> is its ability to visually present
the results of the PCA. This is done through a color-coded scatter plot
that displays the first two principal components. This graphical
representation is instrumental in highlighting the genetic similarities
and differences among the analyzed populations. By plotting individuals
on this scatter plot, researchers can easily discern the underlying
genetic structure and observe how study samples cluster with or diverge
from various reference populations.</p>
<p><strong>Resultant Dataframe:</strong> Following the PCA and outlier
analysis, <code>AncestryCheck</code> generates a dataframe that lists
the identified outlier samples. This output is particularly useful for
researchers, as it provides a clear and organized view of the samples
that significantly differ genetically from the chosen reference
population. This information is crucial for further detailed
investigations and interpretations of genetic ancestry.</p>
<p><strong>Comprehensive Reporting:</strong> In addition to the visual
and tabular outputs, the function also produces detailed reports
covering various aspects of the analysis. These reports include
information on allele flips and a thorough breakdown of the outlier
analysis. Such comprehensive reporting enhances the overall
understanding of the genetic structure and ancestry estimations,
providing a multi-dimensional view of the results. This feature of
<code>AncestryCheck</code> is particularly valuable for researchers who
require in-depth documentation and interpretation of their genetic
data.</p>
</div>
</div>
<div class="section level2">
<h2 id="ancestrycheck-parameters-in-brief">
<code>AncestryCheck</code> parameters in brief<a class="anchor" aria-label="anchor" href="#ancestrycheck-parameters-in-brief"></a>
</h2>
<p>The <code>AncestryCheck</code> function takes several arguments to
customize its operations:</p>
<ul>
<li>
<code>DataDir</code>: Directory containing input PLINK files.</li>
<li>
<code>ResultDir</code>: Directory for storing output files; defaults
to a temporary directory.</li>
<li>
<code>finput</code>: Prefix of PLINK binary files for study
samples.</li>
<li>
<code>reference</code>: Choice of reference population data, either
‘HapMapIII_NCBI36’ or ‘ThousandGenome’.</li>
<li>
<code>filterSNP</code>: Flag to enable or disable SNP
filtering.</li>
<li>
<code>studyLD and referLD</code>: Flags for applying LD-based
filtering on study and reference data, respectively.</li>
<li>
<code>studyLD_window_size</code>, <code>studyLD_step_size</code>,
<code>studyLD_r2_threshold</code>, <code>referLD_window_size</code>,
<code>referLD_step_size</code>, <code>referLD_r2_threshold</code>:
Parameters for LD pruning.</li>
<li>
<code>highLD_regions</code>: Dataframe of known high LD
regions.</li>
<li>
<code>study_pop</code>: Dataframe containing sample IDs and ancestry
labels for study samples.</li>
<li>
<code>outlier</code>: Flag to enable or disable outlier
detection.</li>
<li>
<code>outlierOf</code>: Reference ancestry name for detecting
outlier samples.</li>
<li>
<code>outlier_threshold</code>: Numeric threshold for outlier
detection.</li>
</ul>
</div>
<div class="section level2">
<h2 id="detailed-parameter-description">Detailed parameter description<a class="anchor" aria-label="anchor" href="#detailed-parameter-description"></a>
</h2>
<p>Before exploring the practical application of the
<code>AncestryCheck</code> function, it is essential to understand the
details of the above arguments and their implications:</p>
<ul>
<li>
<code>DataDir</code>: This parameter requires a character string
specifying the file path for the input PLINK binary files. It is
essential for indicating where the study’s genetic data is stored.</li>
<li>
<code>ResultDir</code> (default = tempdir()): This argument is a
character string for the file path where all output files will be
stored. If not specified, it defaults to R’s temporary directory.</li>
<li>
<code>finput</code>: This parameter specifies the prefix of the
input PLINK binary files for the study samples.</li>
<li>
<code>reference</code> (default = c(“HapMapIII_NCBI36”,
“ThousandGenome”)): A vector indicating the choice of reference
population. The options include HapMap Phase 3 and 1000 Genomes Phase
III.</li>
<li>
<code>filterSNP</code> (default = TRUE): Indicates whether to filter
out SNPs. It’s set to TRUE by default but can be changed to FALSE if
direct joining of study and reference samples is possible.</li>
<li>
<code>studyLD</code> (default = TRUE): This Boolean value determines
whether LD-based filtering is applied to study genotype data.</li>
<li>
<code>studyLD_window_size</code> (default = 50): An integer defining
the window size in variant count or kilobase for LD-based filtering of
the variants for the study data.</li>
<li>
<code>studyLD_step_size</code> (default = 5): Specifies the variant
count to shift the window at the end of each step for LD filtering of
the study data.</li>
<li>
<code>studyLD_r2_threshold</code> (default = 0.02): A numeric value
between 0 and 1 setting the pairwise r2 threshold for LD-based filtering
of the study data.</li>
<li>
<code>referLD</code> (default = FALSE): Determines whether LD-based
filtering is applied to reference genotype data.</li>
<li>
<code>referLD_window_size</code> (default = 50), referLD_step_size
(default = 5), referLD_r2_threshold (default = 0.02): These parameters
define the window size, step size, and r2 threshold for LD filtering of
the reference data, similar to the study data settings.</li>
<li>
<code>highLD_regions</code>: This parameter requires a dataframe
containing known high LD regions. A dataframe from Anderson et. al.[4],
‘highLD_hg19’ from hg19 genome built is provided in the package.</li>
<li>
<code>study_pop</code>: A dataframe containing two columns for the
study: sample ID (IID) and the ancestry label. An example dataframe
prefixed as ‘example_data_study_sample_ancestry’ for ‘GXwasR_example’
data.</li>
<li>
<code>outlier</code> (default = FALSE): A Boolean value specifying
whether outlier detection will be performed.</li>
<li>
<code>outlierOf</code> (default = “EUR”): This parameter specifies
the reference ancestry name for detecting outlier samples.</li>
<li>
<code>outlier_threshold</code> (default = 3): Specifies the numeric
threshold for outlier detection. This threshold is multiplied with the
Euclidean distance from the center of the PC1 and PC2 to the maximum
Euclidean distance of the reference samples to determine outliers.</li>
</ul>
<p>These parameters and their default values allow for a significant
degree of customization in the <code>AncestryCheck</code> function,
enabling users to tailor the analysis to their specific research
needs.</p>
</div>
<div class="section level2">
<h2 id="ancestrycheck-usage-using-hapmapiii-reference">AncestryCheck Usage using HapMapIII Reference<a class="anchor" aria-label="anchor" href="#ancestrycheck-usage-using-hapmapiii-reference"></a>
</h2>
<p>The function can be used in the following manner with our study data
and reference data as HapMapIII:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define parameters</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_data_study_sample_ancestry"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Call the AncestryCheck function</span></span>
<span><span class="va">hapmap_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AncestryCheck.html">AncestryCheck</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    finput <span class="op">=</span> <span class="st">"GXwasR_example"</span>,</span>
<span>    reference <span class="op">=</span> <span class="st">"HapMapIII_NCBI36"</span>,</span>
<span>    filterSNP <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    studyLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    studyLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    studyLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    studyLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    referLD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    referLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    referLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    referLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    highLD_regions <span class="op">=</span> <span class="va">highLD_hg19</span>,</span>
<span>    study_pop <span class="op">=</span> <span class="va">example_data_study_sample_ancestry</span>,</span>
<span>    outlier <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    outlierOf <span class="op">=</span> <span class="st">"EUR"</span>,</span>
<span>    outlier_threshold <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 'HapMapIII_NCBI36' reference data found at /Users/mayerdav/Downloads/HapMapIII_NCBI36.</span></span>
<span><span class="co">#&gt; Using PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)</span></span>
<span><span class="co">#&gt; 4214 ambiguous SNPs removed from study data.</span></span>
<span><span class="co">#&gt; 111854 ambiguous SNPs removed from reference data.</span></span>
<span><span class="co">#&gt; ! LD pruning is recommended for reference dataset. Set referLD = TRUE.</span></span>
<span><span class="co">#&gt; ✔ LD pruning was performed for study dataset.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Number of overlapping SNPs between study and reference data using rsID: 1096</span></span>
<span><span class="co">#&gt; ℹ No allele flips between study and reference data.</span></span>
<span><span class="co">#&gt; ℹ 9 samples are outliers of selected reference population.</span></span>
<span><span class="co">#&gt; ℹ 179 samples are NOT outliers of selected reference population.</span></span>
<span><span class="co">#&gt; This message is displayed once every 8 hours.</span></span>
<span><span class="va">hapmap_results</span><span class="op">$</span><span class="va">pca_plot</span></span></code></pre></div>
<div class="float">
<img src="hapmap_ex-1.png" alt="plot of chunk hapmap_ex"><div class="figcaption">plot of chunk hapmap_ex</div>
</div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Samples_with_predicted_ancestry</span> <span class="op">&lt;-</span> <span class="va">hapmap_results</span><span class="op">[[</span><span class="st">"Samples_with_predicted_ancestry"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="va">hapmap_results</span><span class="op">[[</span><span class="st">"Outlier_samples"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">outlier</span></span>
<span><span class="co">#&gt;       FID     IID</span></span>
<span><span class="co">#&gt; 1 EUR_FIN HG00189</span></span>
<span><span class="co">#&gt; 2 EUR_FIN HG00280</span></span>
<span><span class="co">#&gt; 3 EUR_FIN HG00315</span></span>
<span><span class="co">#&gt; 4 EUR_FIN HG00332</span></span>
<span><span class="co">#&gt; 5 EUR_FIN HG00349</span></span>
<span><span class="co">#&gt; 6 EUR_FIN HG00350</span></span>
<span><span class="co">#&gt; 7 EUR_FIN HG00377</span></span>
<span><span class="co">#&gt; 8 EUR_FIN HG00383</span></span>
<span><span class="co">#&gt; 9 EUR_TSI NA20537</span></span>
<span></span>
<span><span class="va">Samples_with_predicted_ancestry</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;       FID     IID Predicted Ancestry</span></span>
<span><span class="co">#&gt; 1 EUR_GBR HG00096                EUR</span></span>
<span><span class="co">#&gt; 2 EUR_GBR HG00097                EUR</span></span>
<span><span class="co">#&gt; 3 EUR_GBR HG00099                EUR</span></span>
<span><span class="co">#&gt; 4 EUR_GBR HG00100                EUR</span></span>
<span><span class="co">#&gt; 5 EUR_GBR HG00101                EUR</span></span></code></pre></div>
<p>The plot shows the distribution of individual genetic samples along
the first two principal components (PC1 and PC2). Each point represents
a genetic sample, and they are color-coded based on reference
populations (<code>Ref_ASIAN</code>, <code>Ref_CEU</code>, and
<code>Ref_YRI</code>) and study populations (<code>Study_EUR_FIN</code>,
<code>Study_EUR_GBR</code>, and <code>Study_EUR_TSI</code>).</p>
<p>The points are spread across the PCA plot, with each cluster likely
representing genetic similarity within a group and differences between
the groups. The reference populations, <code>Ref_AFR</code>,
<code>Ref_EAS</code>, <code>Ref_EUR</code>, and <code>Ref_SAS</code>
clustered distinctively, suggesting clear genetic differentiation
between these populations. The study populations overlap with the
<code>Ref_EUR</code> (<code>Euoprean</code>) reference population,
indicating European ancestry, with each study subgroup
(<code>FIN, GBR, TSI</code>) forming subclusters, suggesting finer-scale
differences in ancestry within Europe.</p>
<p>Since the study dataset was simulated from CEU 1000 Genome dataset,
it is not surprising that we do see some outlier samples. Specifically,
TSI population is not highly similar to CEU.</p>
</div>
<div class="section level2">
<h2 id="ancestrycheck-using-thousandgenome-reference">AncestryCheck Using ThousandGenome Reference<a class="anchor" aria-label="anchor" href="#ancestrycheck-using-thousandgenome-reference"></a>
</h2>
<p>The function can be used in the following manner with our study data
and reference data as ThousandGenome:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define parameters</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_data_study_sample_ancestry"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">thousandgenome_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AncestryCheck.html">AncestryCheck</a></span><span class="op">(</span></span>
<span>  DataDir <span class="op">=</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>  ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>  finput <span class="op">=</span> <span class="st">"GXwasR_example"</span>, </span>
<span>  reference <span class="op">=</span> <span class="st">"ThousandGenome"</span>, </span>
<span>  highLD_regions <span class="op">=</span> <span class="va">highLD_hg19</span>, </span>
<span>  study_pop <span class="op">=</span> <span class="va">example_data_study_sample_ancestry</span>, </span>
<span>  studyLD <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  referLD <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  outlierOf <span class="op">=</span> <span class="st">"EUR"</span>, </span>
<span>  outlier <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>  outlier_threshold <span class="op">=</span> <span class="fl">5</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; 4214 ambiguous SNPs removed from study data.</span></span>
<span><span class="co">#&gt; No A-T or G-C SNPs found in reference data.</span></span>
<span><span class="co">#&gt; ✔ LD pruning was performed for reference dataset.</span></span>
<span><span class="co">#&gt; ✔ LD pruning was performed for study dataset.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Number of overlapping SNPs between study and reference data using rsID: 1005</span></span>
<span><span class="co">#&gt; ℹ No allele flips between study and reference data.</span></span>
<span><span class="co">#&gt; ✖ There is no outlier sample.</span></span>
<span><span class="co">#&gt; ✖ There are no non-outlier samples.</span></span>
<span>  <span class="va">thousandgenome_results</span><span class="op">$</span><span class="va">pca_plot</span></span></code></pre></div>
<div class="float">
<img src="thousandgenome_ex-1.png" alt="plot of chunk thousandgenome_ex"><div class="figcaption">plot of chunk thousandgenome_ex</div>
</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Samples_with_predicted_ancestry</span> <span class="op">&lt;-</span> <span class="va">thousandgenome_results</span><span class="op">[[</span><span class="st">"Samples_with_predicted_ancestry"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="va">thousandgenome_results</span><span class="op">[[</span><span class="st">"Outlier_samples"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">outlier</span></span>
<span><span class="co">#&gt; [1] FID IID</span></span>
<span><span class="co">#&gt; &lt;0 rows&gt; (or 0-length row.names)</span></span>
<span></span>
<span><span class="va">Samples_with_predicted_ancestry</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;       FID  IID Predicted Ancestry</span></span>
<span><span class="co">#&gt; NA   &lt;NA&gt; &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; NA.1 &lt;NA&gt; &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; NA.2 &lt;NA&gt; &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; NA.3 &lt;NA&gt; &lt;NA&gt;               &lt;NA&gt;</span></span>
<span><span class="co">#&gt; NA.4 &lt;NA&gt; &lt;NA&gt;               &lt;NA&gt;</span></span></code></pre></div>
<p>The lack of outlier samples—those that would be significantly
distanced from the reference group—suggests homogeneity in the ancestry
of the study population. Given that our study data was simulated based
on the EUR population from the 1000 Genomes project data, the absence of
outliers aligns with expectations.</p>
<p>The precise clustering and absence of outliers not only reaffirm the
simulation parameters but also shows the reliability of the ancestry
estimation methods employed. Such accuracy is crucial in genetic
studies, as it reinforces confidence in the PCA’s ability to discern and
delineate ancestry with precision.</p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>The combination of genotype datasets from study populations with
comprehensive reference datasets, followed by PCA, provides a nuanced
view of ancestry. The GXwasR package simplifies this process, offering
researchers and enthusiasts a robust tool for ancestry estimation. This
efficient approach not only enhances our understanding of genetic
diversity but also strengthens our connection to our ancestral past.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ol style="list-style-type: decimal">
<li><p>Purcell et. al.,(2007). PLINK: A Tool Set for Whole-Genome
Association and Population-Based Linkage Analyses. The American Journal
of Human Genetics, 81(3), 559-575. <a href="https://doi.org/10.1086/519795" class="external-link uri">https://doi.org/10.1086/519795</a>.</p></li>
<li><p>The International HapMap 3 Consortium. Integrating common and
rare genetic variation in diverse human populations. Nature 467, 52–58
(2010). <a href="https://doi.org/10.1038/nature09298" class="external-link uri">https://doi.org/10.1038/nature09298</a></p></li>
<li><p>The 1000 Genomes Project Consortium. A global reference for human
genetic variation. Nature 526, 68–74 (2015). <a href="https://doi.org/10.1038/nature15393" class="external-link uri">https://doi.org/10.1038/nature15393</a></p></li>
<li><p>Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP,
Zondervan KT. Data quality control in genetic case-control association
studies. Nat Protoc. 2010 Sep;5(9):1564-73. doi: 10.1038/nprot.2010.116.
Epub 2010 Aug 26. PMID: 21085122; PMCID: PMC3025522.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis LK, Stranger BE (2025). "GXwasR: A Toolkit</span></span>
<span><span class="co">#&gt;   for Investigating Sex-Differentiated Genetic Effects on Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim, Winters, Actkins, Mayer, Congivaram,
Niarchou, Edwards, Davis, and Stranger, 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.6</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-08</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.33 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-08 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown              2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
<div class="section level3">
<h3 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h3>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
