<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genetic Correlation using GWAS Summary Statistics • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Genetic Correlation using GWAS Summary Statistics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Genetic Correlation using GWAS Summary Statistics</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">6 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/genetic_correlation_sumstat.Rmd" class="external-link"><code>vignettes/genetic_correlation_sumstat.Rmd</code></a></small>
      <div class="d-none name"><code>genetic_correlation_sumstat.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="gxwasr-function-sumstatgencorr">GXwasR function SumstatGenCorr()<a class="anchor" aria-label="anchor" href="#gxwasr-function-sumstatgencorr"></a>
</h2>
<p><strong>SumstatGenCorr: Genetic Correlation Calculation from GWAS
Summary Statistics</strong></p>
<p>This function calculates the genetic correlation between two summary
statistics using a specified reference LD matrix from the UK Biobank or
Hapmap2 following Ning Z, Pawitan Y, Shen X (2020). High-definition
likelihood inference of genetic correlations across human complex
traits, Nature Genetics.</p>
<p>The function utilizes the precomputed eigenvalues and eigenvectors of
LD correlation matrices for European ancestry population.</p>
<p>These are the LD matrices and their eigen-decomposition from three
different sets:</p>
<p><strong>1) UKB_imputed_hapmap2_SVD_eigen99_extraction:</strong>
769,306 QCed UK Biobank imputed HapMap2 SNPs: If one of your GWAS
includes most of the HapMap 2 SNPs, but many SNPs (more than 1%) in the
above HapMap 3 reference panel are absent, then this HapMap2 panel is
more proper to be used for computing genetic correlation. The size is
about 18 GB after unzipping</p>
<p><strong>2) UKB_imputed_SVD_eigen99_extraction:</strong> 1,029,876
QCed UK Biobank imputed SNPs. The size is about 31 GB after unzipping.
Although it takes more time, using the imputed panel provides more
accurate estimates of genetic correlations. The reference panels with
imputed SNPs are based on genotypes in UK Biobank, which were imputed to
HRC and UK10K + 1000 Genomes. Therefore if the GWAS includes most of the
HapMap3 SNPs, then it is recommend using the imputed reference
panel.</p>
<p><strong>3) UKB_array_SVD_eigen90_extraction:</strong> 307,519 QCed UK
Biobank Axiom Array SNPs. The size is about 7.5 GB after unzipping.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GXwasR: Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data</span></span>
<span><span class="co">#&gt; Version 0.99.0 () installed</span></span>
<span><span class="co">#&gt; Author: c(</span></span>
<span><span class="co">#&gt;     person(given = "Banabithi",</span></span>
<span><span class="co">#&gt;            family = "Bose",</span></span>
<span><span class="co">#&gt;            role = c("cre", "aut"),</span></span>
<span><span class="co">#&gt;            email = "banabithi.bose@gmail.com",</span></span>
<span><span class="co">#&gt;            comment = c(ORCID = "0000-0003-0842-8768"))</span></span>
<span><span class="co">#&gt;            )</span></span>
<span><span class="co">#&gt; Maintainer: Banabithi Bose &lt;banabithi.bose@gmail.com&gt;</span></span>
<span><span class="co">#&gt; Tutorial: https://github.com</span></span>
<span><span class="co">#&gt; Use citation("GXwasR") to know how to cite this work.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SumstatGenCorr</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-ukb-imputed-hapmap2-data-neale-lab">Example: UKB Imputed HapMap2 Data (Neale Lab)<a class="anchor" aria-label="anchor" href="#example-ukb-imputed-hapmap2-data-neale-lab"></a>
</h2>
<p>Birth weight and type 2 diabetes based on the summary statistics with
around 20,000 individuals.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>ResultDir <span class="ot">&lt;-</span> <span class="fu">tempdir</span>()</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>sumstat1 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"gwas1.imputed.example.rds"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>sumstat2 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"gwas2.imputed.example.rds"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>referenceLD <span class="ot">=</span> <span class="st">"UKB_imputed_hapmap2_SVD_eigen99_extraction"</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">SumstatGenCorr</span>(<span class="at">ResultDir=</span>ResultDir, <span class="at">referenceLD =</span>referenceLD, <span class="at">sumstat1=</span>sumstat1, <span class="at">sumstat2=</span>sumstat2)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a><span class="co">#&gt; Analysis starts on Wed Aug  6 11:23:15 2025</span></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co">#&gt; ℹ 0 SNPs were removed in GWAS 1 due to missing N or missing test statistic.</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="co">#&gt; ℹ 0 SNPs were removed in GWAS 2 due to missing N or missing test statistic.</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt; ℹ 769306 out of 769306 (100%) SNPs in reference panel are available in GWAS 1.</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt; ℹ 769306 out of 769306 (100%) SNPs in reference panel are available in GWAS 2.</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a>■■                                 <span class="dv">3</span>% (<span class="dv">2</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a>■■                                 <span class="dv">5</span>% (<span class="dv">3</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a>■■■                                <span class="dv">6</span>% (<span class="dv">4</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a>■■■                                <span class="dv">8</span>% (<span class="dv">5</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a>■■■■                              <span class="dv">10</span>% (<span class="dv">6</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-22"><a href="#cb2-22" tabindex="-1"></a>■■■■                              <span class="dv">11</span>% (<span class="dv">7</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-23"><a href="#cb2-23" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-24"><a href="#cb2-24" tabindex="-1"></a>■■■■■                             <span class="dv">13</span>% (<span class="dv">8</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-25"><a href="#cb2-25" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-26"><a href="#cb2-26" tabindex="-1"></a>■■■■■                             <span class="dv">15</span>% (<span class="dv">9</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-27"><a href="#cb2-27" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-28"><a href="#cb2-28" tabindex="-1"></a>■■■■■■                            <span class="dv">16</span>% (<span class="dv">10</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-29"><a href="#cb2-29" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-30"><a href="#cb2-30" tabindex="-1"></a>■■■■■■                            <span class="dv">18</span>% (<span class="dv">11</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-31"><a href="#cb2-31" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-32"><a href="#cb2-32" tabindex="-1"></a>■■■■■■■                           <span class="dv">19</span>% (<span class="dv">12</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-33"><a href="#cb2-33" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-34"><a href="#cb2-34" tabindex="-1"></a>■■■■■■■                           <span class="dv">21</span>% (<span class="dv">13</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-35"><a href="#cb2-35" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-36"><a href="#cb2-36" tabindex="-1"></a>■■■■■■■■                          <span class="dv">23</span>% (<span class="dv">14</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-37"><a href="#cb2-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-38"><a href="#cb2-38" tabindex="-1"></a>■■■■■■■■                          <span class="dv">24</span>% (<span class="dv">15</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-39"><a href="#cb2-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-40"><a href="#cb2-40" tabindex="-1"></a>■■■■■■■■■                         <span class="dv">26</span>% (<span class="dv">16</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-41"><a href="#cb2-41" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-42"><a href="#cb2-42" tabindex="-1"></a>■■■■■■■■■                         <span class="dv">27</span>% (<span class="dv">17</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-43"><a href="#cb2-43" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-44"><a href="#cb2-44" tabindex="-1"></a>■■■■■■■■■■                        <span class="dv">29</span>% (<span class="dv">18</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-45"><a href="#cb2-45" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-46"><a href="#cb2-46" tabindex="-1"></a>■■■■■■■■■■                        <span class="dv">31</span>% (<span class="dv">19</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-47"><a href="#cb2-47" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-48"><a href="#cb2-48" tabindex="-1"></a>■■■■■■■■■■■                       <span class="dv">32</span>% (<span class="dv">20</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-49"><a href="#cb2-49" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-50"><a href="#cb2-50" tabindex="-1"></a>■■■■■■■■■■■                       <span class="dv">34</span>% (<span class="dv">21</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-51"><a href="#cb2-51" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-52"><a href="#cb2-52" tabindex="-1"></a>■■■■■■■■■■■■                      <span class="dv">35</span>% (<span class="dv">22</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-53"><a href="#cb2-53" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-54"><a href="#cb2-54" tabindex="-1"></a>■■■■■■■■■■■■                      <span class="dv">37</span>% (<span class="dv">23</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-55"><a href="#cb2-55" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-56"><a href="#cb2-56" tabindex="-1"></a>■■■■■■■■■■■■■                     <span class="dv">39</span>% (<span class="dv">24</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-57"><a href="#cb2-57" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-58"><a href="#cb2-58" tabindex="-1"></a>■■■■■■■■■■■■■                     <span class="dv">40</span>% (<span class="dv">25</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-59"><a href="#cb2-59" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-60"><a href="#cb2-60" tabindex="-1"></a>■■■■■■■■■■■■■■                    <span class="dv">42</span>% (<span class="dv">26</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-61"><a href="#cb2-61" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-62"><a href="#cb2-62" tabindex="-1"></a>■■■■■■■■■■■■■■                    <span class="dv">44</span>% (<span class="dv">27</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-63"><a href="#cb2-63" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-64"><a href="#cb2-64" tabindex="-1"></a>■■■■■■■■■■■■■■■                   <span class="dv">45</span>% (<span class="dv">28</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-65"><a href="#cb2-65" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-66"><a href="#cb2-66" tabindex="-1"></a>■■■■■■■■■■■■■■■                   <span class="dv">47</span>% (<span class="dv">29</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-67"><a href="#cb2-67" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-68"><a href="#cb2-68" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">48</span>% (<span class="dv">30</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-69"><a href="#cb2-69" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-70"><a href="#cb2-70" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">50</span>% (<span class="dv">31</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-71"><a href="#cb2-71" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-72"><a href="#cb2-72" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">52</span>% (<span class="dv">32</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-73"><a href="#cb2-73" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-74"><a href="#cb2-74" tabindex="-1"></a>■■■■■■■■■■■■■■■■■                 <span class="dv">53</span>% (<span class="dv">33</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-75"><a href="#cb2-75" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-76"><a href="#cb2-76" tabindex="-1"></a>■■■■■■■■■■■■■■■■■                 <span class="dv">55</span>% (<span class="dv">34</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-77"><a href="#cb2-77" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-78"><a href="#cb2-78" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■                <span class="dv">56</span>% (<span class="dv">35</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-79"><a href="#cb2-79" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-80"><a href="#cb2-80" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■                <span class="dv">58</span>% (<span class="dv">36</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-81"><a href="#cb2-81" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-82"><a href="#cb2-82" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■               <span class="dv">60</span>% (<span class="dv">37</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-83"><a href="#cb2-83" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-84"><a href="#cb2-84" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■               <span class="dv">61</span>% (<span class="dv">38</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-85"><a href="#cb2-85" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-86"><a href="#cb2-86" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■              <span class="dv">63</span>% (<span class="dv">39</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-87"><a href="#cb2-87" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-88"><a href="#cb2-88" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■              <span class="dv">65</span>% (<span class="dv">40</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-89"><a href="#cb2-89" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-90"><a href="#cb2-90" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■             <span class="dv">66</span>% (<span class="dv">41</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-91"><a href="#cb2-91" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-92"><a href="#cb2-92" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■             <span class="dv">68</span>% (<span class="dv">42</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-93"><a href="#cb2-93" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-94"><a href="#cb2-94" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■            <span class="dv">69</span>% (<span class="dv">43</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-95"><a href="#cb2-95" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-96"><a href="#cb2-96" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■            <span class="dv">71</span>% (<span class="dv">44</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-97"><a href="#cb2-97" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-98"><a href="#cb2-98" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■           <span class="dv">73</span>% (<span class="dv">45</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-99"><a href="#cb2-99" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-100"><a href="#cb2-100" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■           <span class="dv">74</span>% (<span class="dv">46</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-101"><a href="#cb2-101" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-102"><a href="#cb2-102" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■          <span class="dv">76</span>% (<span class="dv">47</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-103"><a href="#cb2-103" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-104"><a href="#cb2-104" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■          <span class="dv">77</span>% (<span class="dv">48</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-105"><a href="#cb2-105" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-106"><a href="#cb2-106" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■         <span class="dv">79</span>% (<span class="dv">49</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-107"><a href="#cb2-107" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-108"><a href="#cb2-108" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■         <span class="dv">81</span>% (<span class="dv">50</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-109"><a href="#cb2-109" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-110"><a href="#cb2-110" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■        <span class="dv">82</span>% (<span class="dv">51</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-111"><a href="#cb2-111" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-112"><a href="#cb2-112" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■        <span class="dv">84</span>% (<span class="dv">52</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-113"><a href="#cb2-113" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-114"><a href="#cb2-114" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■       <span class="dv">85</span>% (<span class="dv">53</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-115"><a href="#cb2-115" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-116"><a href="#cb2-116" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■       <span class="dv">87</span>% (<span class="dv">54</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-117"><a href="#cb2-117" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-118"><a href="#cb2-118" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■      <span class="dv">89</span>% (<span class="dv">55</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-119"><a href="#cb2-119" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-120"><a href="#cb2-120" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■      <span class="dv">90</span>% (<span class="dv">56</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-121"><a href="#cb2-121" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-122"><a href="#cb2-122" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■     <span class="dv">94</span>% (<span class="dv">58</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-123"><a href="#cb2-123" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-124"><a href="#cb2-124" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■    <span class="dv">95</span>% (<span class="dv">59</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-125"><a href="#cb2-125" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-126"><a href="#cb2-126" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■    <span class="dv">97</span>% (<span class="dv">60</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-127"><a href="#cb2-127" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-128"><a href="#cb2-128" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■   <span class="dv">98</span>% (<span class="dv">61</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-129"><a href="#cb2-129" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-130"><a href="#cb2-130" tabindex="-1"></a>                                             </span>
<span id="cb2-131"><a href="#cb2-131" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-132"><a href="#cb2-132" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-133"><a href="#cb2-133" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-134"><a href="#cb2-134" tabindex="-1"></a><span class="co">#&gt; Integrating piecewise results</span></span>
<span id="cb2-135"><a href="#cb2-135" tabindex="-1"></a><span class="co">#&gt; Point estimates:</span></span>
<span id="cb2-136"><a href="#cb2-136" tabindex="-1"></a><span class="co">#&gt; • Heritability of phenotype 1:  0.1008</span></span>
<span id="cb2-137"><a href="#cb2-137" tabindex="-1"></a><span class="co">#&gt; • Heritability of phenotype 2:  0.0066</span></span>
<span id="cb2-138"><a href="#cb2-138" tabindex="-1"></a><span class="co">#&gt; • Genetic Covariance:  -0.006</span></span>
<span id="cb2-139"><a href="#cb2-139" tabindex="-1"></a><span class="co">#&gt; • Genetic Correlation:  -0.2318</span></span>
<span id="cb2-140"><a href="#cb2-140" tabindex="-1"></a><span class="co">#&gt; ℹ Continuing computing standard error with jackknife</span></span>
<span id="cb2-141"><a href="#cb2-141" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-142"><a href="#cb2-142" tabindex="-1"></a>■■                                 <span class="dv">3</span>% (<span class="dv">2</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-143"><a href="#cb2-143" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-144"><a href="#cb2-144" tabindex="-1"></a>■■                                 <span class="dv">5</span>% (<span class="dv">3</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-145"><a href="#cb2-145" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-146"><a href="#cb2-146" tabindex="-1"></a>■■■                                <span class="dv">6</span>% (<span class="dv">4</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-147"><a href="#cb2-147" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-148"><a href="#cb2-148" tabindex="-1"></a>■■■                                <span class="dv">8</span>% (<span class="dv">5</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-149"><a href="#cb2-149" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-150"><a href="#cb2-150" tabindex="-1"></a>■■■■                              <span class="dv">10</span>% (<span class="dv">6</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-151"><a href="#cb2-151" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-152"><a href="#cb2-152" tabindex="-1"></a>■■■■                              <span class="dv">11</span>% (<span class="dv">7</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-153"><a href="#cb2-153" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-154"><a href="#cb2-154" tabindex="-1"></a>■■■■■                             <span class="dv">13</span>% (<span class="dv">8</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-155"><a href="#cb2-155" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-156"><a href="#cb2-156" tabindex="-1"></a>■■■■■                             <span class="dv">15</span>% (<span class="dv">9</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-157"><a href="#cb2-157" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-158"><a href="#cb2-158" tabindex="-1"></a>■■■■■■                            <span class="dv">16</span>% (<span class="dv">10</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-159"><a href="#cb2-159" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-160"><a href="#cb2-160" tabindex="-1"></a>■■■■■■                            <span class="dv">18</span>% (<span class="dv">11</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-161"><a href="#cb2-161" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-162"><a href="#cb2-162" tabindex="-1"></a>■■■■■■■                           <span class="dv">19</span>% (<span class="dv">12</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-163"><a href="#cb2-163" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-164"><a href="#cb2-164" tabindex="-1"></a>■■■■■■■                           <span class="dv">21</span>% (<span class="dv">13</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-165"><a href="#cb2-165" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-166"><a href="#cb2-166" tabindex="-1"></a>■■■■■■■■                          <span class="dv">23</span>% (<span class="dv">14</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-167"><a href="#cb2-167" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-168"><a href="#cb2-168" tabindex="-1"></a>■■■■■■■■                          <span class="dv">24</span>% (<span class="dv">15</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-169"><a href="#cb2-169" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-170"><a href="#cb2-170" tabindex="-1"></a>■■■■■■■■■                         <span class="dv">26</span>% (<span class="dv">16</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-171"><a href="#cb2-171" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-172"><a href="#cb2-172" tabindex="-1"></a>■■■■■■■■■                         <span class="dv">27</span>% (<span class="dv">17</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-173"><a href="#cb2-173" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-174"><a href="#cb2-174" tabindex="-1"></a>■■■■■■■■■■                        <span class="dv">29</span>% (<span class="dv">18</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-175"><a href="#cb2-175" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-176"><a href="#cb2-176" tabindex="-1"></a>■■■■■■■■■■                        <span class="dv">31</span>% (<span class="dv">19</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-177"><a href="#cb2-177" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-178"><a href="#cb2-178" tabindex="-1"></a>■■■■■■■■■■■                       <span class="dv">32</span>% (<span class="dv">20</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-179"><a href="#cb2-179" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-180"><a href="#cb2-180" tabindex="-1"></a>■■■■■■■■■■■                       <span class="dv">34</span>% (<span class="dv">21</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-181"><a href="#cb2-181" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-182"><a href="#cb2-182" tabindex="-1"></a>■■■■■■■■■■■■                      <span class="dv">35</span>% (<span class="dv">22</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-183"><a href="#cb2-183" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-184"><a href="#cb2-184" tabindex="-1"></a>■■■■■■■■■■■■                      <span class="dv">37</span>% (<span class="dv">23</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-185"><a href="#cb2-185" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-186"><a href="#cb2-186" tabindex="-1"></a>■■■■■■■■■■■■■                     <span class="dv">39</span>% (<span class="dv">24</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-187"><a href="#cb2-187" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-188"><a href="#cb2-188" tabindex="-1"></a>■■■■■■■■■■■■■                     <span class="dv">40</span>% (<span class="dv">25</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-189"><a href="#cb2-189" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-190"><a href="#cb2-190" tabindex="-1"></a>■■■■■■■■■■■■■■                    <span class="dv">42</span>% (<span class="dv">26</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-191"><a href="#cb2-191" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-192"><a href="#cb2-192" tabindex="-1"></a>■■■■■■■■■■■■■■                    <span class="dv">44</span>% (<span class="dv">27</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-193"><a href="#cb2-193" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-194"><a href="#cb2-194" tabindex="-1"></a>■■■■■■■■■■■■■■■                   <span class="dv">45</span>% (<span class="dv">28</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-195"><a href="#cb2-195" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-196"><a href="#cb2-196" tabindex="-1"></a>■■■■■■■■■■■■■■■                   <span class="dv">47</span>% (<span class="dv">29</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-197"><a href="#cb2-197" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-198"><a href="#cb2-198" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">48</span>% (<span class="dv">30</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-199"><a href="#cb2-199" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-200"><a href="#cb2-200" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">50</span>% (<span class="dv">31</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-201"><a href="#cb2-201" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-202"><a href="#cb2-202" tabindex="-1"></a>■■■■■■■■■■■■■■■■                  <span class="dv">52</span>% (<span class="dv">32</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-203"><a href="#cb2-203" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-204"><a href="#cb2-204" tabindex="-1"></a>■■■■■■■■■■■■■■■■■                 <span class="dv">53</span>% (<span class="dv">33</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-205"><a href="#cb2-205" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-206"><a href="#cb2-206" tabindex="-1"></a>■■■■■■■■■■■■■■■■■                 <span class="dv">55</span>% (<span class="dv">34</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-207"><a href="#cb2-207" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-208"><a href="#cb2-208" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■                <span class="dv">56</span>% (<span class="dv">35</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-209"><a href="#cb2-209" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-210"><a href="#cb2-210" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■                <span class="dv">58</span>% (<span class="dv">36</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-211"><a href="#cb2-211" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-212"><a href="#cb2-212" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■               <span class="dv">60</span>% (<span class="dv">37</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-213"><a href="#cb2-213" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-214"><a href="#cb2-214" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■               <span class="dv">61</span>% (<span class="dv">38</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-215"><a href="#cb2-215" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-216"><a href="#cb2-216" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■              <span class="dv">63</span>% (<span class="dv">39</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-217"><a href="#cb2-217" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-218"><a href="#cb2-218" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■              <span class="dv">65</span>% (<span class="dv">40</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-219"><a href="#cb2-219" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-220"><a href="#cb2-220" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■             <span class="dv">66</span>% (<span class="dv">41</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-221"><a href="#cb2-221" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-222"><a href="#cb2-222" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■             <span class="dv">68</span>% (<span class="dv">42</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-223"><a href="#cb2-223" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-224"><a href="#cb2-224" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■            <span class="dv">69</span>% (<span class="dv">43</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-225"><a href="#cb2-225" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-226"><a href="#cb2-226" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■            <span class="dv">71</span>% (<span class="dv">44</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-227"><a href="#cb2-227" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-228"><a href="#cb2-228" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■           <span class="dv">73</span>% (<span class="dv">45</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-229"><a href="#cb2-229" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-230"><a href="#cb2-230" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■           <span class="dv">74</span>% (<span class="dv">46</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-231"><a href="#cb2-231" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-232"><a href="#cb2-232" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■          <span class="dv">76</span>% (<span class="dv">47</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-233"><a href="#cb2-233" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-234"><a href="#cb2-234" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■          <span class="dv">77</span>% (<span class="dv">48</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-235"><a href="#cb2-235" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-236"><a href="#cb2-236" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■         <span class="dv">79</span>% (<span class="dv">49</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-237"><a href="#cb2-237" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-238"><a href="#cb2-238" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■         <span class="dv">81</span>% (<span class="dv">50</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-239"><a href="#cb2-239" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-240"><a href="#cb2-240" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■        <span class="dv">82</span>% (<span class="dv">51</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-241"><a href="#cb2-241" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-242"><a href="#cb2-242" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■        <span class="dv">84</span>% (<span class="dv">52</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-243"><a href="#cb2-243" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-244"><a href="#cb2-244" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■       <span class="dv">85</span>% (<span class="dv">53</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-245"><a href="#cb2-245" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-246"><a href="#cb2-246" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■       <span class="dv">87</span>% (<span class="dv">54</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-247"><a href="#cb2-247" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-248"><a href="#cb2-248" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■      <span class="dv">89</span>% (<span class="dv">55</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-249"><a href="#cb2-249" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-250"><a href="#cb2-250" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■      <span class="dv">90</span>% (<span class="dv">56</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-251"><a href="#cb2-251" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-252"><a href="#cb2-252" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■     <span class="dv">92</span>% (<span class="dv">57</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-253"><a href="#cb2-253" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-254"><a href="#cb2-254" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■     <span class="dv">94</span>% (<span class="dv">58</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-255"><a href="#cb2-255" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-256"><a href="#cb2-256" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■    <span class="dv">95</span>% (<span class="dv">59</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-257"><a href="#cb2-257" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-258"><a href="#cb2-258" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■    <span class="dv">97</span>% (<span class="dv">60</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-259"><a href="#cb2-259" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-260"><a href="#cb2-260" tabindex="-1"></a>■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■■   <span class="dv">98</span>% (<span class="dv">61</span><span class="sc">/</span><span class="dv">62</span>)</span>
<span id="cb2-261"><a href="#cb2-261" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-262"><a href="#cb2-262" tabindex="-1"></a>                                             </span>
<span id="cb2-263"><a href="#cb2-263" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-264"><a href="#cb2-264" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-265"><a href="#cb2-265" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-266"><a href="#cb2-266" tabindex="-1"></a><span class="co">#&gt; • Heritability of phenotype 1: 0.1008 ( 0.0046 )</span></span>
<span id="cb2-267"><a href="#cb2-267" tabindex="-1"></a><span class="co">#&gt; • Heritability of phenotype 2: 0.0066 ( 8e-04 )</span></span>
<span id="cb2-268"><a href="#cb2-268" tabindex="-1"></a><span class="co">#&gt; • Genetic Covariance: -0.006 ( 0.0011 )</span></span>
<span id="cb2-269"><a href="#cb2-269" tabindex="-1"></a><span class="co">#&gt; • Genetic Correlation: -0.2318 ( 0.0428 )</span></span>
<span id="cb2-270"><a href="#cb2-270" tabindex="-1"></a><span class="co">#&gt; • P: 6.19e-08</span></span>
<span id="cb2-271"><a href="#cb2-271" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-272"><a href="#cb2-272" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-273"><a href="#cb2-273" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb2-274"><a href="#cb2-274" tabindex="-1"></a><span class="co">#&gt; Analysis finished at Wed Aug  6 11:24:55 2025</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-hdl-sample-data-20k-snps-ukb-array-svd-eigen90">Example: HDL Sample Data (20K SNPs, UKB Array SVD Eigen90)<a class="anchor" aria-label="anchor" href="#example-hdl-sample-data-20k-snps-ukb-array-svd-eigen90"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gwas1.example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'gwas1.array.example.rds'</span><span class="op">)</span></span>
<span><span class="va">gwas2.example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'gwas2.array.example.rds'</span><span class="op">)</span></span>
<span><span class="va">referenceLD</span> <span class="op">=</span> <span class="st">"UKB_array_SVD_eigen90_extraction"</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SumstatGenCorr.html">SumstatGenCorr</a></span><span class="op">(</span>ResultDir<span class="op">=</span><span class="va">ResultDir</span>, referenceLD <span class="op">=</span><span class="va">referenceLD</span>, sumstat1<span class="op">=</span><span class="va">gwas1.example</span>, sumstat2<span class="op">=</span><span class="va">gwas2.example</span><span class="op">)</span></span>
<span><span class="co">#&gt; Analysis starts on Wed Aug  6 11:24:56 2025</span></span>
<span><span class="co">#&gt; Error in estimating Genetic Correlation: attempt to set an attribute on NULL</span></span></code></pre></div>
<p><strong>Small number of SNPs anyways not good.</strong></p>
<p><strong>Note:</strong> Singularities or ill-conditioned matrices can
occur due to <strong>collinear SNPs</strong>, insufficient variation, or
extremely small values, causing numerical instability and
non-convergence.</p>
<div class="section level3">
<h3 id="hdl-decoded">HDL decoded<a class="anchor" aria-label="anchor" href="#hdl-decoded"></a>
</h3>
<p>Checking the provided likelihood functions (<strong>llfun and
llfun.gcov.part.2</strong>), it’s possible that very similar summary
statistics (e.g., identical Z-scores) for two traits can lead to
convergence issues in this algorithm due to several reasons:</p>
<p><strong>1. Collinearity and Identical Data:</strong></p>
<p>If the summary statistics are identical between two traits:</p>
<p><strong>Collinearity:</strong> The covariance matrix may become
singular or nearly singular, which can make the optimization
unstable.</p>
<p>Example snippet where identical Z-scores may cause collinearity
issues</p>
<pre><code><span><span class="va">llfun.gcov.part.2</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">param</span>, <span class="va">h11</span>, <span class="va">h22</span>, <span class="va">rho12</span>, <span class="va">M</span>, <span class="va">N1</span>, <span class="va">N2</span>, <span class="va">N0</span>, <span class="va">Nref</span>,</span>
<span>                              <span class="va">lam0</span>, <span class="va">lam1</span>, <span class="va">lam2</span>, <span class="va">bstar1</span>, <span class="va">bstar2</span>, <span class="va">lim</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>   <span class="co"># Estimating covariance matrices</span></span>
<span>   <span class="va">lam11</span> <span class="op">=</span> <span class="va">h11</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">/</span> <span class="va">M</span> <span class="op">*</span> <span class="va">lam1</span><span class="op">^</span><span class="fl">2</span> <span class="op">-</span> <span class="va">h11</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">*</span> <span class="va">lam1</span> <span class="op">/</span> <span class="va">Nref</span> <span class="op">+</span> <span class="va">h11</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">*</span> <span class="va">lam1</span> <span class="op">/</span> <span class="va">N1</span></span>
<span>   <span class="va">lam22</span> <span class="op">=</span> <span class="va">h22</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">/</span> <span class="va">M</span> <span class="op">*</span> <span class="va">lam2</span><span class="op">^</span><span class="fl">2</span> <span class="op">-</span> <span class="va">h22</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">*</span> <span class="va">lam2</span> <span class="op">/</span> <span class="va">Nref</span> <span class="op">+</span> <span class="va">h22</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">*</span> <span class="va">lam2</span> <span class="op">/</span> <span class="va">N2</span></span>
<span>   <span class="va">lam12</span> <span class="op">=</span> <span class="kw">if</span> <span class="op">(</span><span class="va">N0</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span> <span class="va">h12</span> <span class="op">/</span> <span class="va">M</span> <span class="op">*</span> <span class="va">lam1</span> <span class="op">*</span> <span class="va">lam2</span> <span class="op">+</span> <span class="va">N0</span> <span class="op">/</span> <span class="op">(</span><span class="va">N1</span> <span class="op">*</span> <span class="va">N2</span><span class="op">)</span> <span class="op">*</span> <span class="va">int</span> <span class="op">*</span> <span class="va">lam1</span> <span class="kw">else</span> <span class="va">h12</span> <span class="op">/</span> <span class="va">M</span> <span class="op">*</span> <span class="va">lam1</span> <span class="op">*</span> <span class="va">lam2</span></span>
<span></span>
<span>   <span class="co"># High collinearity leads to potential singularity</span></span>
<span>   <span class="kw">if</span> <span class="op">(</span><span class="fu">is.singular</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">lam11</span>, <span class="va">lam12</span>, <span class="va">lam12</span>, <span class="va">lam22</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>       <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Singular matrix due to collinearity"</span><span class="op">)</span></span>
<span>   <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre>
<p><strong>Identical Statistics:</strong> The functions rely on
variability between the traits, and identical statistics can lead to a
flat or undefined likelihood surface, causing numerical instability.</p>
<p>Example where identical summary stats lead to flat likelihood
surface</p>
<pre><code><span><span class="va">llfun</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">param</span>, <span class="va">N</span>, <span class="va">M</span>, <span class="va">Nref</span> <span class="op">=</span> <span class="fl">1000</span>, <span class="va">lam</span>, <span class="va">bstar</span>, <span class="va">lim</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">exp</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>   <span class="va">h2</span> <span class="op">=</span> <span class="va">param</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span>   <span class="va">int</span> <span class="op">=</span> <span class="va">param</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span>   <span class="va">lamh2</span> <span class="op">=</span> <span class="va">h2</span> <span class="op">/</span> <span class="va">M</span> <span class="op">*</span> <span class="va">lam</span><span class="op">^</span><span class="fl">2</span> <span class="op">-</span> <span class="va">h2</span> <span class="op">*</span> <span class="va">lam</span> <span class="op">/</span> <span class="va">Nref</span> <span class="op">+</span> <span class="va">int</span> <span class="op">*</span> <span class="va">lam</span> <span class="op">/</span> <span class="va">N</span></span>
<span></span>
<span>   <span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="va">bstar</span> <span class="op">==</span> <span class="va">bstar</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>       <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Identical statistics detected, leading to undefined likelihood surface"</span><span class="op">)</span></span>
<span>   <span class="op">}</span></span>
<span></span>
<span>   <span class="va">lamh2</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="va">lamh2</span> <span class="op">&lt;</span> <span class="va">lim</span>, <span class="va">lim</span>, <span class="va">lamh2</span><span class="op">)</span></span>
<span>   <span class="va">ll</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log</a></span><span class="op">(</span><span class="va">lamh2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">bstar</span><span class="op">^</span><span class="fl">2</span> <span class="op">/</span> <span class="op">(</span><span class="va">lamh2</span><span class="op">)</span><span class="op">)</span></span>
<span>   <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">ll</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
<p><strong>2. Flat Likelihood Surface:</strong></p>
<p>If the data is identical, the likelihood surface might be flat,
meaning that the algorithm has difficulty finding a unique solution due
to lack of gradients. In llfun.gcov.part.2, the covariance (h12) may end
up zero or undefined if there’s no variance in the data.</p>
<p><strong>3. Optimization Method Sensitivity:</strong></p>
<p>The numerical optimization (‘optim’) used for finding the best fit is
sensitive to data quality. Identical data can result in values that
don’t fit the assumptions or cause overflows in likelihood calculations.
This may results in <strong>Algorithm failed to converge after trying
different initial values.</strong></p>
<p><strong>4. Variance Estimates:</strong></p>
<p>The variance estimates (lamh2, lam22.1) are sensitive to the
differences in summary statistics. Identical data may result in variance
estimates being too small or large, leading to numerical instability in
the log-likelihood calculations.</p>
</div>
<div class="section level3">
<h3 id="why-ukb-imputed-eigen99-fails-to-converge-while-ukb-array-eigen90-succeeds">Why UKB Imputed (Eigen99) Fails to Converge While UKB Array
(Eigen90) Succeeds<a class="anchor" aria-label="anchor" href="#why-ukb-imputed-eigen99-fails-to-converge-while-ukb-array-eigen90-succeeds"></a>
</h3>
<p><strong>Expected Genetic Correlation:</strong> When using identical
summary statistics, the genetic correlation is theoretically expected to
be very close to 1, as the traits are the same.</p>
<p>However, this can vary in practice due to the quality of the data and
the specific nuances of the LD reference panel.</p>
<p><strong>Sensitivity to LD Reference Panels:</strong> Different LD
panels can have distinct effects on the computation, especially in cases
of identical summary statistics:</p>
<p><strong>LD Structure:</strong> Panels with better-defined LD
structures (like the Axiom array) may produce more stable covariance
matrices, leading to better convergence.</p>
<p><strong>Imputation Accuracy:</strong> LD panels with more imputed
SNPs (like the HRC, 1000 Genomes, HapMap3 panel) may introduce noise,
making convergence more challenging.</p>
<p><strong>Effect of Imputation and Rare Variants:</strong> Imputed
panels often have a larger number of SNPs, including rare variants,
which can introduce noise and increase the complexity of the covariance
structure.</p>
<p>The additional noise and potential artifacts from imputation could
make it difficult for the algorithm to converge, especially when working
with identical summary statistics.</p>
<p><strong>Impact of Multicollinearity:</strong> Identical summary
statistics can lead to high multicollinearity, which may be mitigated
differently depending on the LD reference panel.</p>
<p>High multicollinearity can destabilize the covariance matrix
calculations, resulting in non-convergence for one panel but not the
other.</p>
<p>In summary, the differences in convergence for the two LD panels when
computing genetic correlation with identical summary statistics
highlight the importance of:</p>
<ul>
<li>LD structure quality</li>
<li>Imputation noise</li>
<li>The inherent properties of the SNPs in the panels</li>
<li>The Axiom array may offer a cleaner, more stable set of SNPs with
well-defined LD, which is critical for achieving convergence in the
genetic correlation computation</li>
</ul>
</div>
<div class="section level3">
<h3 id="trying-the-genomicsem-package">Trying the GenomicSEM Package<a class="anchor" aria-label="anchor" href="#trying-the-genomicsem-package"></a>
</h3>
<p>763976 out of 769306 (99.31%) SNPs in reference panel are available
in the GWAS of Data1.txt</p>
<p>Estimation for cell: 1 out of: 3 cells is ongoing … 100%</p>
<p>Estimation for cell: 2 out of: 3 cells is ongoing … 100%</p>
<p>301051 out of 769306 (39.13%) SNPs in reference panel are available
in the GWAS of Data2.txt</p>
<p>Warning: More than 1% SNPs in reference panel are missed in the GWAS.
This may generate bias in estimation.</p>
<p>Please make sure that you are using correct reference panel.</p>
<p>Estimation for cell: 3 out of: 3 cells is ongoing … 100%Warning
message:</p>
<p>In cov2cor(V) : diag(.) had 0 or NA entries; non-finite result is
doubtful</p>
<p><strong>I thought this is happening because DBgap summary statistics
is in hg18, whereas ours is in hg19, also hm3 is in hg19.</strong></p>
<p>In cov2cor(V) : diag(.) had 0 or NA entries; non-finite result is
doubtful</p>
<p>Hence, I updated the SNPs using proper bim file but with no use.</p>
<p><strong>I realized this is happening because DBgap’s summary
statistics is not imputed.</strong></p>
<p><strong>I used .Bed file made from summary statistics</strong> and
processed via Crossmap to update the coordinates but got lower overlap
between DBgap and reference panel Hapmap3 provided by the
GenomicSem.</p>
</div>
<div class="section level3">
<h3 id="can-ld-pruned-snps-be-used-in-this-computation">Can LD-Pruned SNPs Be Used in This Computation?<a class="anchor" aria-label="anchor" href="#can-ld-pruned-snps-be-used-in-this-computation"></a>
</h3>
<p><strong>The answer is NO.</strong></p>
<p>if we use LD pruned SNPs:</p>
<p>32641 out of 769306 (4.24%) SNPs in reference panel are available in
the GWAS of Data1.txt</p>
<p>Warning: More than 1% SNPs in reference panel are missed in the GWAS.
This may generate bias in estimation. Please make sure that you are
using correct reference panel.</p>
</div>
<div class="section level3">
<h3 id="what-happened-when-correlating-the-same-dataset-in-genomicsem">What Happened When Correlating the Same Dataset in GenomicSEM<a class="anchor" aria-label="anchor" href="#what-happened-when-correlating-the-same-dataset-in-genomicsem"></a>
</h3>
<p><strong>It did not converge.</strong></p>
<p>If two summary statistics are exactly the same, the covariance
between them will be equal to the variance of either one of the
statistics.</p>
<p>The presence of the same value for variance in both 𝐷1 and 𝐷2 could
suggest similar traits or measurements under the same conditions but
with different outcomes based on the conditions or methods of
measurement leading to a negative covariance.</p>
</div>
<div class="section level3">
<h3 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h3>
<p><strong>The genetic correlation analysis is sensitive to the number
of SNPs in the summary statistics and the quality of the LD reference
panel.</strong></p>
<p>In Panscan datasets, the result is:</p>
<p>Heritability of phenotype 1: 0.00e+00 (0.00e+00)</p>
<p>Heritability of phenotype 2: 0.00e+00 (0.00e+00)</p>
<p>Genetic Covariance: -0.1536 (0.1436)</p>
<p>Genetic Correlation: -Inf (NA)</p>
<p><strong>About Warning: Heritability of one trait was estimated to be
0, which may due to:</strong></p>
<pre><code>      1) The true heritability is very small;
      
      2) The sample size is too small;
      
      3) Many SNPs in the chosen reference panel misses in the GWAS.
      
      4) There is a severe mismatch between the GWAS population and the population for computing reference panel</code></pre>
<p><strong>Note:</strong> True heritability in pancreatic cancer is 21%.
Summary statistics may not give this estimate.</p>
<p><strong>Potential Calculation Issues</strong></p>
<p><strong>HDL Package:</strong> It uses likelihood-based methods which
can also run into numerical difficulties with very low or zero
heritability data, possibly leading to non-finite results.</p>
<p><strong>Relevant Sections in the HDL Algorithm</strong>:</p>
<p><strong>Heritability Calculation:</strong></p>
<p>The heritability of each trait is calculated using linear models
(lm()) on squared z-scores against LD scores. Later, likelihood-based
methods (optim()) refine these estimates. <strong>Here, having small
number of SNPs with a few significant hits may lead to low to no
variation that leads towards zero heritability estimate.</strong></p>
<p><strong>Genetic Covariance:</strong></p>
<p>Genetic covariance between traits is calculated using a linear model
(lm()) on the product of z-scores.</p>
<p><strong>Genetic Correlation:</strong></p>
<p>The genetic correlation (rg) is derived by dividing the genetic
covariance by the square root of the product of heritability values: The
genetic correlation
(<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>r</mi><mi>g</mi></msub><annotation encoding="application/x-tex">r_g</annotation></semantics></math>)
can be calculated as shown below:</p>
<p><math display="block" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><msub><mi>r</mi><mi>g</mi></msub><mo>=</mo><mfrac><msub><mi>h</mi><mn>12</mn></msub><msqrt><mrow><msub><mi>h</mi><mn>11</mn></msub><mo>⋅</mo><msub><mi>h</mi><mn>22</mn></msub></mrow></msqrt></mfrac></mrow><annotation encoding="application/x-tex">
r_g = \frac{h_{12}}{\sqrt{h_{11} \cdot h_{22}}}
</annotation></semantics></math></p>
<p>Where: -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>h</mi><mn>11</mn></msub><annotation encoding="application/x-tex">h_{11}</annotation></semantics></math>
is the heritability of Trait 1 -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>h</mi><mn>22</mn></msub><annotation encoding="application/x-tex">h_{22}</annotation></semantics></math>
is the heritability of Trait 2 -
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><msub><mi>h</mi><mn>12</mn></msub><annotation encoding="application/x-tex">h_{12}</annotation></semantics></math>
is the genetic covariance between the two traits</p>
<p><strong>Impact of Zero Heritability</strong></p>
<p><strong>Zero Heritability:</strong></p>
<p>If the heritability for one of the traits is zero, it implies that
genetic factors don’t explain any variance in that trait.</p>
<p><strong>Mathematical Impact:</strong></p>
<p>In the calculation for h11 or h22 is zero, the denominator becomes
zero. This results in division by zero, which in mathematical operations
typically yields NaN, Inf, or -Inf.</p>
<p><strong>Analysis and Impact</strong></p>
<p>The number of SNPs and their overlap with the reference LD panel and
the <strong>quality</strong> of the reference significantly impact
genetic correlation computation.</p>
<p><strong>What is the best LD panel:</strong> Best LD reference panel
for Panscan seems to be UKB_array_SVD_eigen90_extraction due to accurate
heritability estimate. This UK Biobank Axiom Array has extensive genomic
coverage, including common and rare variants than hapmap or 1000 genome.
The quality of the SNPs is higher in Axiom array. Also it is meant for
only EUR in UKB.</p>
<p><strong>Number of SNPs:</strong></p>
<p>More SNPs in the summary statistics generate more variation, leading
to more accurate results. LD Reference Panel: The choice of the LD
reference panel is crucial for accurate computation, as the algorithm is
sensitive to <strong>LD information quality</strong> and
<strong>overlap</strong> with the GWAS data.</p>
<p><strong>Other factors:</strong></p>
<p><strong>1. Difficulty in Detecting Genetic Associations</strong></p>
<p>The combination of a <strong>small sample size</strong> and low
heritability makes it challenging to detect genetic associations
accurately. Genetic factors may only account for a small portion of the
trait’s variability, which is further overshadowed by environmental
factors or measurement errors. This challenge becomes more pronounced
with the reliance on summary statistics.</p>
<p><strong>2. Larger Sample Sizes and Imputed Summary Statistics are
Required</strong></p>
<p>Increasing the sample size is necessary to enhance statistical power,
making it easier to detect even minor genetic influences on pancreatic
cancer. However, the limited sample size in the current dataset hinders
this approach, making it challenging to achieve sufficient power for
reliable results.</p>
<p><strong>3. Statistical Significance and Noise</strong></p>
<p>With a limited sample size, the signal-to-noise ratio in the data is
diminished. This makes it difficult to distinguish true genetic effects
from random fluctuations, leading to a higher likelihood of false
positives (type I errors) or missing true effects (type II errors). This
instability is further exacerbated by using summary statistics.</p>
<p><strong>4. Polygenic Traits</strong></p>
<p>Pancreatic cancer’s genetic basis likely involves many genes with
small effects, underscoring its polygenic nature. This dilution of
effect sizes complicates the identification of significant genetic
variants in underpowered datasets.</p>
<p><strong>5. Impact on Heritability Estimates</strong></p>
<p>The small sample size and low heritability bias the estimates, and
the genetic architecture of pancreatic cancer includes non-additive
factors like gene-gene and gene-environment interactions.</p>
<p><strong>6. Meta-Analyses and Collaborative Efforts</strong></p>
<p>Given the difficulties with small sample size and low heritability,
GWAS on pancreatic cancer often require meta-analyses and collaborations
to pool data and resources, boosting statistical power and
reliability.</p>
</div>
<div class="section level3">
<h3 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h3>
<p>In summary, the genetic correlation analysis is highly sensitive to
the number of SNPs and the quality of the LD reference panel. More SNPs
in the summary statistics provide more variation and yield better
results, while good overlap with the reference LD panel enhances the
quality of genetic correlation computations. Also, good sample size is
required to have power in this estimation.</p>
</div>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits." _medRxiv 2025.06.10.25329327_.</span></span>
<span><span class="co">#&gt;   doi:10.1101/2025.06.10.25329327 &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim, Winters, Actkins, Mayer, Congivaram,
Niarchou, Edwards, Davis, and Stranger, 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-06</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-06 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown              2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpJ93upV/temp_libpathcfe02f0c39d5</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
