<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Running GWAS, XWAS and Sex-differential tests Using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Running GWAS, XWAS and Sex-differential tests Using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Running GWAS, XWAS and Sex-differential tests Using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">8 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/vignettes/gwas_models.Rmd" class="external-link"><code>vignettes/gwas_models.Rmd</code></a></small>
      <div class="d-none name"><code>gwas_models.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This document contains an example of running a GWAS with XWAS model
and Sex-differential tests.</p>
<p>Users must perform pre-imputation and post-imputation quality control
(QC) on genotype dataset and then proceed with this tutorial in actual
scenario. Please follow the detailed QC-ed steps provided in: <a href="https://boseb.github.io/GXwasR/articles/preimputationQC.html" class="external-link">GXwasR_Preimputation</a>
and <a href="https://boseb.github.io/GXwasR/articles/postimputationQC.html" class="external-link">GXwasR_Postimputation</a>.</p>
</div>
<div class="section level2">
<h2 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a>
</h2>
<p>The PLINK bed, bim, and fam files are the three mandatory files
representing a genotype dataset to run this tutorial. To know about
these file extensions, please check <a href="https://www.cog-genomics.org/plink/1.9/formats" class="external-link uri">https://www.cog-genomics.org/plink/1.9/formats</a>.</p>
<p><strong>GXwasR_example:</strong></p>
<ul>
<li><p>GXwasR_example.bed</p></li>
<li><p>GXwasR_example.bim</p></li>
<li><p>GXwasR_example.fam</p></li>
</ul>
<p>These plink files contain genotypes for 276 individuals (males and
females) simulated from 1000Genome from European decent with 26515
variants across twelve chromosomes (1-10,23,24). This dataset contains
125 males, 151 females, 108 cases and 168 controls. We will utilize this
set of plink files as proxy of pre-imputated genotype data.</p>
</div>
<div class="section level2">
<h2 id="loading-the-gxwasr-library">Loading the GXwasR library<a class="anchor" aria-label="anchor" href="#loading-the-gxwasr-library"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'printr':</span></span>
<span><span class="co">#&gt;   method                from     </span></span>
<span><span class="co">#&gt;   knit_print.data.frame rmarkdown</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="learn-about-all-the-gxwasr-functions">Learn about all the GXwasR functions<a class="anchor" aria-label="anchor" href="#learn-about-all-the-gxwasr-functions"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-dataset-summary">Example Dataset Summary<a class="anchor" aria-label="anchor" href="#example-dataset-summary"></a>
</h2>
<p><strong>Dataset: GXwasR_example</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sex-combined-and-sex-stratified-gwas-with-xwas">Sex-combined and sex-stratified GWAS with XWAS<a class="anchor" aria-label="anchor" href="#sex-combined-and-sex-stratified-gwas-with-xwas"></a>
</h2>
<div class="section level3">
<h3 id="function-gxwas">Function: GXwas()<a class="anchor" aria-label="anchor" href="#function-gxwas"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXwas</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; GXwas: Running genome-wide association study (GWAS) and</span></span>
<span><span class="co">#&gt; X-chromosome-wide association study (XWAS) models.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function runs GWAS models in autosomes with several</span></span>
<span><span class="co">#&gt;      alternative XWAS models. Models such as</span></span>
<span><span class="co">#&gt;      "FMcombx01","FMcombx02",and "FMstratified" can be applied to both</span></span>
<span><span class="co">#&gt;      binary and quantitative traits, while "GWAcxci" can only be</span></span>
<span><span class="co">#&gt;      applied to a binary trait.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For binary and quantitative features, this function uses logistic</span></span>
<span><span class="co">#&gt;      and linear regression, allowing for multiple covariates and the</span></span>
<span><span class="co">#&gt;      interactions with those covariates in a multiple-regression</span></span>
<span><span class="co">#&gt;      approach. These models are all run using the additive effects of</span></span>
<span><span class="co">#&gt;      SNPs, and each additional minor allele's influence is represented</span></span>
<span><span class="co">#&gt;      by the direction of the regression coefficient.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function attempts to identify the multi-collinearity among</span></span>
<span><span class="co">#&gt;      predictors by displaying NA for the test statistic and a p-value</span></span>
<span><span class="co">#&gt;      for all terms in the model. The more terms you add, the more</span></span>
<span><span class="co">#&gt;      likely you are to run into issues.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For details about the different XWAS model, please follow the</span></span>
<span><span class="co">#&gt;      associated publication.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      GXwas(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        trait = c("binary", "quantitative"),</span></span>
<span><span class="co">#&gt;        standard_beta = TRUE,</span></span>
<span><span class="co">#&gt;        xmodel = c("FMcombx01", "FMcombx02", "FMstratified", "GWAScxci"),</span></span>
<span><span class="co">#&gt;        sex = FALSE,</span></span>
<span><span class="co">#&gt;        xsex = FALSE,</span></span>
<span><span class="co">#&gt;        covarfile = NULL,</span></span>
<span><span class="co">#&gt;        interaction = FALSE,</span></span>
<span><span class="co">#&gt;        covartest = c("ALL"),</span></span>
<span><span class="co">#&gt;        Inphenocov = c("ALL"),</span></span>
<span><span class="co">#&gt;        combtest = c("fisher.method", "fisher.method.perm", "stouffer.method"),</span></span>
<span><span class="co">#&gt;        MF.zero.sub = 1e-05,</span></span>
<span><span class="co">#&gt;        B = 10000,</span></span>
<span><span class="co">#&gt;        MF.mc.cores = 1,</span></span>
<span><span class="co">#&gt;        MF.na.rm = FALSE,</span></span>
<span><span class="co">#&gt;        MF.p.corr = "none",</span></span>
<span><span class="co">#&gt;        plot.jpeg = FALSE,</span></span>
<span><span class="co">#&gt;        plotname = "GXwas.plot",</span></span>
<span><span class="co">#&gt;        snp_pval = 1e-08,</span></span>
<span><span class="co">#&gt;        annotateTopSnp = FALSE,</span></span>
<span><span class="co">#&gt;        suggestiveline = 5,</span></span>
<span><span class="co">#&gt;        genomewideline = 7.3,</span></span>
<span><span class="co">#&gt;        ncores = 0</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the folder path where the outputs will</span></span>
<span><span class="co">#&gt;           be saved.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files with both male and female samples. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: Case/control phenotypes are expected to be encoded as</span></span>
<span><span class="co">#&gt;           1=unaffected (control), 2=affected (case); 0 is accepted as</span></span>
<span><span class="co">#&gt;           an alternate missing value encoding. The missing case/control</span></span>
<span><span class="co">#&gt;           or quantitative phenotypes are expected to be encoded as</span></span>
<span><span class="co">#&gt;           'NA'/'nan' (any capitalization) or -9.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    trait: Boolean value, 'binary' or 'quantitative' for the phenotype</span></span>
<span><span class="co">#&gt;           i.e. the trait.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; standard_beta: Boolean value, 'TRUE' or 'FALSE' in case of quantitative</span></span>
<span><span class="co">#&gt;           trait for standardizing the trait or phenotype values (mean</span></span>
<span><span class="co">#&gt;           0, unit variance), so the resulting coefficients will be</span></span>
<span><span class="co">#&gt;           standardized. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   xmodel: Models "FMcombx01","FMcombx02",and "FMstratified" can be</span></span>
<span><span class="co">#&gt;           chosen for both binary and quantitative traits while</span></span>
<span><span class="co">#&gt;           "GWAcxci" can only apply to the binary trait. These models</span></span>
<span><span class="co">#&gt;           take care of the X-chromosomal marker. Three female genotypes</span></span>
<span><span class="co">#&gt;           are coded by 0, 1, and 2 in FM01 and FM02. The two genotypes</span></span>
<span><span class="co">#&gt;           of males that follow the X-chromosome inactivation (XCI)</span></span>
<span><span class="co">#&gt;           pattern as random (XCI-R) in the FM01 model are coded by 0</span></span>
<span><span class="co">#&gt;           and 1, while the two genotypes that follow the XCI is escaped</span></span>
<span><span class="co">#&gt;           (XCI-E) in the FM02 model are coded by 0 and 1. To reflect</span></span>
<span><span class="co">#&gt;           the dose compensation connection between the sexes, FM02</span></span>
<span><span class="co">#&gt;           treats men as homozygous females.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           In the "FMstratified" associations are tested separately for</span></span>
<span><span class="co">#&gt;           males and females, and then the combined p values are</span></span>
<span><span class="co">#&gt;           computed the Fisher's method, Fisher's method with</span></span>
<span><span class="co">#&gt;           permutation, or Stouffer's method(1,3-7]. An X-chromosome</span></span>
<span><span class="co">#&gt;           inactivation (XCI) pattern, or coding technique for</span></span>
<span><span class="co">#&gt;           X-chromosomal genotypes between sexes, is not required for</span></span>
<span><span class="co">#&gt;           the XCGA. By simultaneously accounting for four distinct XCI</span></span>
<span><span class="co">#&gt;           patterns, namely XCI-R, XCI-E, XCI-SN (XCI fully toward</span></span>
<span><span class="co">#&gt;           normal allele), and XCI-SR (XCI fully toward risk allele),</span></span>
<span><span class="co">#&gt;           this model may maintain a respectably high power (Su et al.</span></span>
<span><span class="co">#&gt;           2022).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: 'sex' shouldn't be provided as a covariate in the XCGA</span></span>
<span><span class="co">#&gt;           model.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      sex: Boolean value, 'TRUE' or 'FALSE' for using sex as covariate</span></span>
<span><span class="co">#&gt;           in association test. It is applicable genome-wide.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     xsex: Boolean value, 'TRUE' or 'FALSE' for using sex as covariate</span></span>
<span><span class="co">#&gt;           in association test for X-chromosomal SNPs. The default is</span></span>
<span><span class="co">#&gt;           FALSE. This will overwrite 'sex' argument for X-chromosome.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covarfile: Character string for the full name of the covariate file in</span></span>
<span><span class="co">#&gt;           .txt format. This file should be placed in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note about the covariate file: The first column of this file</span></span>
<span><span class="co">#&gt;           should be 'FID', the second column should be 'IID' and the</span></span>
<span><span class="co">#&gt;           other columns should be covariates. The primary header line</span></span>
<span><span class="co">#&gt;           should be there starting with “FID”, and “IID” followed by</span></span>
<span><span class="co">#&gt;           covariate names. If an individual is not present in the</span></span>
<span><span class="co">#&gt;           covariate file, or if the individual has a missing phenotype</span></span>
<span><span class="co">#&gt;           value (i.e. -9 by default) for the covariate, then that</span></span>
<span><span class="co">#&gt;           individual is set to missing (i.e. will be excluded from</span></span>
<span><span class="co">#&gt;           association analysis). It is important to note that for</span></span>
<span><span class="co">#&gt;           stratified GWAS model, if PCs are included as covar then it</span></span>
<span><span class="co">#&gt;           should be generated separately for each cohort and then</span></span>
<span><span class="co">#&gt;           included in the covarfile. Use the function 'DummyCovar' to</span></span>
<span><span class="co">#&gt;           generate a new covariate file with categorical variables</span></span>
<span><span class="co">#&gt;           down-coded as binary dummy variables for the covariate file</span></span>
<span><span class="co">#&gt;           with categorical variables. For instance, if a variable has K</span></span>
<span><span class="co">#&gt;           categories, K-1 new dummy variables are constructed, and the</span></span>
<span><span class="co">#&gt;           original covariate is now estimated with a coefficient for</span></span>
<span><span class="co">#&gt;           each category.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; interaction: Boolean value, 'TRUE' or 'FALSE' for including SNP x</span></span>
<span><span class="co">#&gt;           covariate interaction term/terms from the association</span></span>
<span><span class="co">#&gt;           analysis. The default is 'FALSE'. If a permutation procedure</span></span>
<span><span class="co">#&gt;           is chosen then the interaction will be automatically 'FALSE'.</span></span>
<span><span class="co">#&gt;           For the interaction with the two covariates COV1 and COV2,</span></span>
<span><span class="co">#&gt;           the model will look like: Y = b0 + b1.ADD + b2.COV1 + b3.COV2</span></span>
<span><span class="co">#&gt;           + b4.ADD x COV1 + b5.ADD x COV2 + e. When interaction factors</span></span>
<span><span class="co">#&gt;           are incorporated into the model, the main effects'</span></span>
<span><span class="co">#&gt;           significance is not always determined simply; rather, it</span></span>
<span><span class="co">#&gt;           depends on the arbitrary coding of the variables. To put it</span></span>
<span><span class="co">#&gt;           another way, you should probably just interpret the p-value</span></span>
<span><span class="co">#&gt;           for the interaction. Also, The p-values for the covariates do</span></span>
<span><span class="co">#&gt;           not represent the test for the SNP-phenotype association</span></span>
<span><span class="co">#&gt;           after controlling for the covariate. That is the first row</span></span>
<span><span class="co">#&gt;           (ADD). Rather, the covariate term is the test associated with</span></span>
<span><span class="co">#&gt;           the covariate-phenotype association. These p-values might be</span></span>
<span><span class="co">#&gt;           extremely significant (e.g. if one covaries for smoking in an</span></span>
<span><span class="co">#&gt;           analysis of heart disease, etc) but this does not mean that</span></span>
<span><span class="co">#&gt;           the SNP has a highly significant effect necessarily. Note</span></span>
<span><span class="co">#&gt;           that, this feature is not valid for XCGA model for XWAS part.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covartest: Vector value with 'NULL',"ALL" or covariate name/names to be</span></span>
<span><span class="co">#&gt;           included in the test. The default is 'NULL.' For instance,</span></span>
<span><span class="co">#&gt;           the user can choose “AGE” and “SEX” as covartest = c(“AGE”,</span></span>
<span><span class="co">#&gt;           “SEX”) or all the covariates as covartest = c(“ALL”).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inphenocov: Vector of integer values starting from 1 to extract the</span></span>
<span><span class="co">#&gt;           terms which user wants from the above model: Y = b0 + b1.ADD</span></span>
<span><span class="co">#&gt;           + b2.COV1 + b3.COV2 + b4.ADDxCOV1 + b5.ADDxCOV2 + e. The</span></span>
<span><span class="co">#&gt;           terms will appear in order as (Purcell et al. 2007) for ADD,</span></span>
<span><span class="co">#&gt;           (Su et al. 2022) for COV1, (Rhodes 2002) for ADD x COV1, and</span></span>
<span><span class="co">#&gt;           (Moreau and others 2003) for ADD x COV2. If the user wants to</span></span>
<span><span class="co">#&gt;           extract the terms for COV1 and ADD x COV1, they need to</span></span>
<span><span class="co">#&gt;           specify it as c(2,4). The default is c(“ALL”).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: This feature is not valid for the XCGA model for the</span></span>
<span><span class="co">#&gt;           XWAS part.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; combtest: Character vector specifying method for combining p-values</span></span>
<span><span class="co">#&gt;           after stratified GWAS/XWAS models. Choices are</span></span>
<span><span class="co">#&gt;           “stouffer.method”, "fisher.method" and "fisher.method.perm".</span></span>
<span><span class="co">#&gt;           For fisher.method the function for combining p-values uses a</span></span>
<span><span class="co">#&gt;           statistic, S = -2 ∑^k /log p, which follows a χ^2</span></span>
<span><span class="co">#&gt;           distribution with 2k degrees of freedom (Fisher 1925).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For fisher.method.perm, using p-values from stratified tests,</span></span>
<span><span class="co">#&gt;           the summary statistic for combining p-values is S = -2 ∑ /log</span></span>
<span><span class="co">#&gt;           p. A p-value for this statistic can be derived by randomly</span></span>
<span><span class="co">#&gt;           generating summary statistics (Rhodes 2002). Therefore, a</span></span>
<span><span class="co">#&gt;           p-value is randomly sampled from each contributing study, and</span></span>
<span><span class="co">#&gt;           a random statistic is calculated. The fraction of random</span></span>
<span><span class="co">#&gt;           statistics greater or equal to S then gives the final</span></span>
<span><span class="co">#&gt;           p-value.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For stouffer.method ,the function applies Stouffer’s method</span></span>
<span><span class="co">#&gt;           (Stouffer et al. 1949) to the p-values assuming that the</span></span>
<span><span class="co">#&gt;           p-values to be combined are independent. Letting p1, p2, . .</span></span>
<span><span class="co">#&gt;           . , pk denote the individual (one- or two-sided) p-values of</span></span>
<span><span class="co">#&gt;           the k hypothesis tests to be combined, the test statistic is</span></span>
<span><span class="co">#&gt;           then computed with $z = ∑^k_{1}frac{z_{i}}{sqrt(k)}$ where</span></span>
<span><span class="co">#&gt;           $z_{i}$ = Φ−1 (1 – $p_{i}$) and Φ −1 (·) denotes the inverse</span></span>
<span><span class="co">#&gt;           of the cumulative distribution function of a standard normal</span></span>
<span><span class="co">#&gt;           distribution. Under the joint null hypothesis, the test</span></span>
<span><span class="co">#&gt;           statistic follows a standard normal distribution which is</span></span>
<span><span class="co">#&gt;           used to compute the combined p-value. This functionality is</span></span>
<span><span class="co">#&gt;           taken from the R package poolr (Cinar and Viechtbauer 2022).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note that only p-values between 0 and 1 are allowed to be</span></span>
<span><span class="co">#&gt;           passed to these methods.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: Though this parameter is enabled for both autosome GWAS</span></span>
<span><span class="co">#&gt;           and XWAS, the combining pvalue after sex-stratified test is</span></span>
<span><span class="co">#&gt;           recommended to ChrX only.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.zero.sub: Small numeric value for substituting p-values of 0 in in</span></span>
<span><span class="co">#&gt;           stratified GWAS with FM01comb and FM02comb XWAS models. The</span></span>
<span><span class="co">#&gt;           default is 0.00001. As log(0) results in Inf this replaces</span></span>
<span><span class="co">#&gt;           p-values of 0 by default with a small float.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;        B: Integer value specifying the number of permutation in case of</span></span>
<span><span class="co">#&gt;           using fisher.method.perm method in stratified GWAS with</span></span>
<span><span class="co">#&gt;           FM01comb and FM02comb XWAS models. The default is 10000.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.mc.cores: Number of cores used for fisher.method.perm in stratified</span></span>
<span><span class="co">#&gt;           GWAS with FM01comb and FM02comb XWAS models.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.na.rm: Boolean value, 'TRUE' or 'FALSE' for removing p-values of NA</span></span>
<span><span class="co">#&gt;           in stratified GWAS with FM01comb and FM02comb XWAS in case of</span></span>
<span><span class="co">#&gt;           using Fisher’s and Stouffer’s methods. The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.p.corr: Character vector specifying method for correcting the</span></span>
<span><span class="co">#&gt;           summary p-values for FMfcomb and FMscomb models. Choices are</span></span>
<span><span class="co">#&gt;           "bonferroni", "BH" and "none" for Bonferroni,</span></span>
<span><span class="co">#&gt;           Benjamini-Hochberg and none, respectively. The default is</span></span>
<span><span class="co">#&gt;           "none".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plot.jpeg: Boolean value, 'TRUE' or 'FALSE' for saving the plots in</span></span>
<span><span class="co">#&gt;           .jpeg file. The default is TRUE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotname: A character string specifying the prefix of the file for</span></span>
<span><span class="co">#&gt;           plots. This file will be saved in DataDir. The default is</span></span>
<span><span class="co">#&gt;           "GXwas.plot".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; snp_pval: Numeric value as p-value threshold for annotation. SNPs below</span></span>
<span><span class="co">#&gt;           this p-value will be annotated on the plot. The default is</span></span>
<span><span class="co">#&gt;           1e-08.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; annotateTopSnp: Boolean value, 'TRUE' or 'FALSE. If TRUE, it only</span></span>
<span><span class="co">#&gt;           annotates the top hit on each chromosome that is below the</span></span>
<span><span class="co">#&gt;           snp_pval threshold. The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; suggestiveline: The default is 5 (for p-value 1e-05).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; genomewideline: The default is 7.3 (for p-value 5e-08).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   ncores: Integer value, specifying the number of cores for parallel</span></span>
<span><span class="co">#&gt;           processing. The default is 0 (no parallel computation).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with GWAS (with XWAS for X-chromosomal variants) along</span></span>
<span><span class="co">#&gt;      with Manhattan and Q-Q plots. In the case of the stratified test,</span></span>
<span><span class="co">#&gt;      the return is a list containing three dataframes, namely, FWAS,</span></span>
<span><span class="co">#&gt;      MWAS, and MFWAS with association results in only female, only</span></span>
<span><span class="co">#&gt;      male, and both cohorts, respectively. This will be accompanied by</span></span>
<span><span class="co">#&gt;      Miami and Q-Q plots. The individual manhattan and Q-Q-plots for</span></span>
<span><span class="co">#&gt;      stratified tests prefixed with xmodel type will be in the DataDir.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Cinar O, Viechtbauer W (2022). "The poolr package for combining</span></span>
<span><span class="co">#&gt;      independent and dependent p values." _Journal of Statistical</span></span>
<span><span class="co">#&gt;      Software_, *101*(1), 1-42. doi:10.18637/jss.v101.i01</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.18637/jss.v101.i01&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Fisher RA (1925). _Statistical Methods for Research Workers_.</span></span>
<span><span class="co">#&gt;      Oliver and Boyd, Edinburgh.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Moreau Y, others (2003). "Comparison and meta-analysis of</span></span>
<span><span class="co">#&gt;      microarray data: from the bench to the computer desk." _Trends in</span></span>
<span><span class="co">#&gt;      Genetics_, *19*(10), 570-577.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Rhodes DR (2002). "Meta-analysis of microarrays: interstudy</span></span>
<span><span class="co">#&gt;      validation of gene expression profiles reveals pathway</span></span>
<span><span class="co">#&gt;      dysregulation in prostate cancer." _Cancer Research_, *62*(15),</span></span>
<span><span class="co">#&gt;      4427-4433.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Stouffer SA, Suchman EA, DeVinney LC, Star SA, Williams RM (1949).</span></span>
<span><span class="co">#&gt;      _The American Soldier: Adjustment During Army Life_, volume 1.</span></span>
<span><span class="co">#&gt;      Princeton University Press, Princeton, NJ.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Su Y, Hu J, Yin P, Jiang H, Chen S, Dai M, Chen Z, Wang P (2022).</span></span>
<span><span class="co">#&gt;      "XCMAX4: A Robust X Chromosomal Genetic Association Test</span></span>
<span><span class="co">#&gt;      Accounting for Covariates." _Genes (Basel)_, *13*(5), 847.</span></span>
<span><span class="co">#&gt;      doi:10.3390/genes13050847 &lt;https://doi.org/10.3390/genes13050847&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/35627231&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      standard_beta &lt;- TRUE</span></span>
<span><span class="co">#&gt;      xsex &lt;- FALSE</span></span>
<span><span class="co">#&gt;      sex &lt;- TRUE</span></span>
<span><span class="co">#&gt;      Inphenocov &lt;- NULL</span></span>
<span><span class="co">#&gt;      covartest &lt;- NULL</span></span>
<span><span class="co">#&gt;      interaction &lt;- FALSE</span></span>
<span><span class="co">#&gt;      MF.na.rm &lt;- FALSE</span></span>
<span><span class="co">#&gt;      B &lt;- 10000</span></span>
<span><span class="co">#&gt;      MF.zero.sub &lt;- 0.00001</span></span>
<span><span class="co">#&gt;      trait &lt;- "binary"</span></span>
<span><span class="co">#&gt;      xmodel &lt;- "FMcombx02"</span></span>
<span><span class="co">#&gt;      combtest &lt;- "fisher.method"</span></span>
<span><span class="co">#&gt;      snp_pval &lt;- 1e-08</span></span>
<span><span class="co">#&gt;      covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      ncores &lt;- 0</span></span>
<span><span class="co">#&gt;      MF.mc.cores &lt;- 1</span></span>
<span><span class="co">#&gt;      ResultGXwas &lt;- GXwas(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, xmodel = xmodel, trait = trait, covarfile = covarfile,</span></span>
<span><span class="co">#&gt;          sex = sex, xsex = xsex, combtest = combtest, MF.p.corr = "none",</span></span>
<span><span class="co">#&gt;          snp_pval = snp_pval, plot.jpeg = TRUE, suggestiveline = 5, genomewideline = 7.3,</span></span>
<span><span class="co">#&gt;          MF.mc.cores = 1, ncores = ncores</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-gxwas">
<a id="Running GXwas"> </a> Running GXwas<a class="anchor" aria-label="anchor" href="#running-gxwas"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">standard_beta</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">xsex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">Inphenocov</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">covartest</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">interaction</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.na.rm</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span><span class="va">MF.zero.sub</span> <span class="op">&lt;-</span> <span class="fl">0.00001</span></span>
<span><span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">xmodel</span> <span class="op">&lt;-</span> <span class="st">"FMstratified"</span></span>
<span><span class="va">combtest</span> <span class="op">&lt;-</span> <span class="st">"fisher.method"</span></span>
<span><span class="va">snp_pval</span> <span class="op">&lt;-</span> <span class="fl">1e-08</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">plot.jpeg</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.mc.cores</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">genomewideline</span> <span class="op">&lt;-</span> <span class="fl">7.3</span></span>
<span><span class="va">suggestiveline</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"GXwas.plot"</span></span>
<span><span class="va">annotateTopSnp</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.na.rm</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span><span class="va">MF.zero.sub</span> <span class="op">&lt;-</span> <span class="fl">0.00001</span></span>
<span><span class="va">MF.p.corr</span> <span class="op">&lt;-</span> <span class="st">"none"</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"GXwas.plot"</span></span>
<span><span class="va">ResultGXwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GXwas.html">GXwas</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, xmodel <span class="op">=</span> <span class="va">xmodel</span>, trait <span class="op">=</span> <span class="va">trait</span>, covarfile <span class="op">=</span> <span class="va">covarfile</span>, sex <span class="op">=</span> <span class="va">sex</span>, xsex <span class="op">=</span> <span class="va">xsex</span>, combtest <span class="op">=</span> <span class="va">combtest</span>, MF.p.corr <span class="op">=</span> <span class="st">"none"</span>, snp_pval <span class="op">=</span> <span class="va">snp_pval</span>, plot.jpeg <span class="op">=</span> <span class="va">plot.jpeg</span>, suggestiveline <span class="op">=</span> <span class="fl">5</span>, genomewideline <span class="op">=</span> <span class="fl">7.3</span>, MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, ncores <span class="op">=</span> <span class="va">ncores</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Running FMstratified model</span></span>
<span><span class="co">#&gt; • Stratified test is running for females</span></span>
<span><span class="co">#&gt; • Stratified test is running for males</span></span>
<span><span class="co">#&gt; ℹ If you want parallel computation, please provide non-zero value for argument ncores.</span></span>
<span><span class="co">#&gt; ℹ Plots are initiated.</span></span>
<span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ/Stratified_GWAS.png</span></span></code></pre></div>
<div class="float">
<img src="gwas_models-gxwas-1.png" alt="plot of chunk gwas_models-gxwas"><div class="figcaption">plot of chunk gwas_models-gxwas</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ/Stratified_XWAS.png</span></span></code></pre>
<p><img src="gwas_models-gxwas-2.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-3.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-4.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-5.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-6.png" alt="plot of chunk gwas_models-gxwas"></p>
<pre><code><span><span class="co">#&gt; Three data frames have been created and saved:</span></span>
<span><span class="co">#&gt; • CombinedWAS</span></span>
<span><span class="co">#&gt; • MaleWAS</span></span>
<span><span class="co">#&gt; • FemaleWAS</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Outputs</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">CombinedWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Dataframe containing the result of stratified model"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Dataframe containing the result of stratified model</caption>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs10000405</td>
<td align="right">4</td>
<td align="right">47716881</td>
<td align="right">0.9768464</td>
</tr>
<tr class="even">
<td align="left">rs10000452</td>
<td align="right">4</td>
<td align="right">63234460</td>
<td align="right">0.6969710</td>
</tr>
<tr class="odd">
<td align="left">rs10000465</td>
<td align="right">4</td>
<td align="right">120835814</td>
<td align="right">0.5378310</td>
</tr>
<tr class="even">
<td align="left">rs10000605</td>
<td align="right">4</td>
<td align="right">13875675</td>
<td align="right">0.7221672</td>
</tr>
<tr class="odd">
<td align="left">rs10000675</td>
<td align="right">4</td>
<td align="right">121520624</td>
<td align="right">0.8246220</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">MaleWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Stratified GWAS result with male cohort."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Stratified GWAS result with male cohort.</caption>
<colgroup>
<col width="4%">
<col width="14%">
<col width="9%">
<col width="3%">
<col width="5%">
<col width="7%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.0789</td>
<td align="right">0.2643</td>
<td align="right">-0.5968</td>
<td align="right">0.4390</td>
<td align="right">-0.2986</td>
<td align="right">0.76530</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.3959</td>
<td align="right">1.2380</td>
<td align="right">-2.8230</td>
<td align="right">2.0310</td>
<td align="right">-0.3197</td>
<td align="right">0.74920</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="right">863863</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">1.0500</td>
<td align="right">0.8858</td>
<td align="right">-0.6864</td>
<td align="right">2.7860</td>
<td align="right">1.1850</td>
<td align="right">0.23600</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs12041521</td>
<td align="right">1109154</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.4451</td>
<td align="right">0.3387</td>
<td align="right">-1.1090</td>
<td align="right">0.2188</td>
<td align="right">-1.3140</td>
<td align="right">0.18890</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs148527527</td>
<td align="right">1127860</td>
<td align="left">G</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">1.6770</td>
<td align="right">0.6864</td>
<td align="right">0.3317</td>
<td align="right">3.0220</td>
<td align="right">2.4430</td>
<td align="right">0.01456</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">FemaleWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Stratified GWAS result with female cohort."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Stratified GWAS result with female cohort.</caption>
<colgroup>
<col width="4%">
<col width="13%">
<col width="9%">
<col width="3%">
<col width="5%">
<col width="6%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="10%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.3459</td>
<td align="right">0.2818</td>
<td align="right">-0.2064</td>
<td align="right">0.89820</td>
<td align="right">1.2270</td>
<td align="right">0.21970</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.1568</td>
<td align="right">0.9290</td>
<td align="right">-1.6640</td>
<td align="right">1.97800</td>
<td align="right">0.1688</td>
<td align="right">0.86590</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="right">863863</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">-0.1446</td>
<td align="right">0.7282</td>
<td align="right">-1.5720</td>
<td align="right">1.28300</td>
<td align="right">-0.1985</td>
<td align="right">0.84260</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs115490086</td>
<td align="right">928969</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.5649</td>
<td align="right">1.4240</td>
<td align="right">-2.2270</td>
<td align="right">3.35700</td>
<td align="right">0.3966</td>
<td align="right">0.69170</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs12041521</td>
<td align="right">1109154</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">-0.6697</td>
<td align="right">0.3314</td>
<td align="right">-1.3190</td>
<td align="right">-0.02008</td>
<td align="right">-2.0210</td>
<td align="right">0.04332</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in female-specific study:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"FemaleWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">FemaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in male-specific study:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"MaleWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">MaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in FM02comb model:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"CombinedWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">CombinedWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<p>To know more about the function GXwas() to run different GWAS and
XWAS models with different arguments, please follow this tutorial: <a href="https://boseb.github.io/GXwasR/articles/GXwasR_overview.html" class="external-link">https://boseb.github.io/GXwasR/articles/GXwasR_overview.html</a></p>
</div>
</div>
<div class="section level2">
<h2 id="performing-sex-differential-test">Performing sex-differential test<a class="anchor" aria-label="anchor" href="#performing-sex-differential-test"></a>
</h2>
<p>Now, we will perform a sex-differential test:</p>
<div class="section level3">
<h3 id="function-sexdiff">Function: SexDiff()<a class="anchor" aria-label="anchor" href="#function-sexdiff"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiff</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexDiff: Sex difference in effect size for each SNP using t-test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function uses the GWAS summary statistics from sex-stratified</span></span>
<span><span class="co">#&gt;      tests like "FMstratified", to evaluate the difference in effect</span></span>
<span><span class="co">#&gt;      size between males and females at each SNP using a t-test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The input dataframes should only include X-chromosome in order to</span></span>
<span><span class="co">#&gt;      obtain results for sex differences based solely on X-linked loci.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexDiff(Mfile, Ffile)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Mfile: R dataframe of summary statistics of GWAS or XWAS of male</span></span>
<span><span class="co">#&gt;           samples with six mandatory columns,</span></span>
<span><span class="co">#&gt;           SNP(Variant),CHR(Chromosome number), BP(Base pair</span></span>
<span><span class="co">#&gt;           position),A1(Minor allele),BETA_M(Effect size) and</span></span>
<span><span class="co">#&gt;           SE_M(Standard error). This can be generated by running</span></span>
<span><span class="co">#&gt;           FM01comb or "FMstratified" model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Ffile: R dataframe of summary statistics of GWAS or XWAS of male</span></span>
<span><span class="co">#&gt;           samples with six mandatory columns,</span></span>
<span><span class="co">#&gt;           SNP(Variant),CHR(Chromosome number), BP(Base pair</span></span>
<span><span class="co">#&gt;           position),A1(Minor allele),BETA_F(Effect size) and</span></span>
<span><span class="co">#&gt;           SE_F(Standard error). This can be generated by running</span></span>
<span><span class="co">#&gt;           FM01comb or "FMstratified" model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      R dataframe with seven columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SNP' (Variant)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'CHR' (Chromosome number)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'BP' (Base pair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'A1' (Minor allele)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'tstat' (t-statistics for effect-size test)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P' (p-value) and</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'adjP' (Bonferroni corrected p-value)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Mfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Ffile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      Difftest &lt;- SexDiff(Mfile, Ffile)</span></span>
<span><span class="co">#&gt;      significant_snps &lt;- Difftest[Difftest$adjP &lt; 0.05, ]</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-sexdiff">Running SexDiff<a class="anchor" aria-label="anchor" href="#running-sexdiff"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running</span></span>
<span><span class="co">## Running the function</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">MaleWAS</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR"</span>, <span class="st">"SNP"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"BETA_M"</span>, <span class="st">"SE_M"</span><span class="op">)</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">FemaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="va">x2</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR"</span>, <span class="st">"SNP"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"BETA_F"</span>, <span class="st">"SE_F"</span><span class="op">)</span></span>
<span><span class="va">Difftest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexDiff.html">SexDiff</a></span><span class="op">(</span>Mfile <span class="op">=</span> <span class="va">x1</span>, Ffile <span class="op">=</span> <span class="va">x2</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="sex-differentiated-effect-1.png" alt="plot of chunk sex-differentiated-effect"><div class="figcaption">plot of chunk sex-differentiated-effect</div>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Significant SNPs with sex-differential effect</span></span>
<span><span class="va">sig.snps</span> <span class="op">&lt;-</span> <span class="va">Difftest</span><span class="op">[</span><span class="va">Difftest</span><span class="op">$</span><span class="va">adjP</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sig.snps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"T_statistic"</span>, <span class="st">"Pvalue"</span>, <span class="st">"Adjusted_Pvalue"</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sig.snps</span>, caption <span class="op">=</span> <span class="st">"SNPs with significant sex-differential effect."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>SNPs with significant sex-differential effect.</caption>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="right">T_statistic</th>
<th align="right">Pvalue</th>
<th align="right">Adjusted_Pvalue</th>
</tr></thead>
<tbody></tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.6</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-08</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.33 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-08 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
