<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Running GWAS, XWAS and Sex-differential tests Using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Running GWAS, XWAS and Sex-differential tests Using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Running GWAS, XWAS and Sex-differential tests Using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">5 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/gwas_models.Rmd" class="external-link"><code>vignettes/gwas_models.Rmd</code></a></small>
      <div class="d-none name"><code>gwas_models.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This document contains an example of running a GWAS with XWAS model
and Sex-differential tests.</p>
<p>Users must perform pre-imputation and post-imputation quality control
(QC) on genotype dataset and then proceed with this tutorial in actual
scenario. Please follow the detailed QC-ed steps provided in: <a href="https://boseb.github.io/GXwasR/articles/preimputationQC.html" class="external-link">GXwasR_Preimputation</a>
and <a href="https://boseb.github.io/GXwasR/articles/postimputationQC.html" class="external-link">GXwasR_Postimputation</a>.</p>
</div>
<div class="section level2">
<h2 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a>
</h2>
<p>The PLINK bed, bim, and fam files are the three mandatory files
representing a genotype dataset to run this tutorial. To know about
these file extensions, please check <a href="https://www.cog-genomics.org/plink/1.9/formats" class="external-link uri">https://www.cog-genomics.org/plink/1.9/formats</a>.</p>
<p><strong>GXwasR_example:</strong></p>
<ul>
<li><p>GXwasR_example.bed</p></li>
<li><p>GXwasR_example.bim</p></li>
<li><p>GXwasR_example.fam</p></li>
</ul>
<p>These plink files contain genotypes for 276 individuals (males and
females) simulated from 1000Genome from European decent with 26515
variants across twelve chromosomes (1-10,23,24). This dataset contains
125 males, 151 females, 108 cases and 168 controls. We will utilize this
set of plink files as proxy of pre-imputated genotype data.</p>
</div>
<div class="section level2">
<h2 id="loading-the-gxwasr-library">Loading the GXwasR library<a class="anchor" aria-label="anchor" href="#loading-the-gxwasr-library"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="learn-about-all-the-gxwasr-functions">Learn about all the GXwasR functions<a class="anchor" aria-label="anchor" href="#learn-about-all-the-gxwasr-functions"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-dataset-summary">Example Dataset Summary<a class="anchor" aria-label="anchor" href="#example-dataset-summary"></a>
</h2>
<p><strong>Dataset: GXwasR_example</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="sex-combined-and-sex-stratified-gwas-with-xwas">Sex-combined and sex-stratified GWAS with XWAS<a class="anchor" aria-label="anchor" href="#sex-combined-and-sex-stratified-gwas-with-xwas"></a>
</h2>
<div class="section level3">
<h3 id="function-gxwas">Function: GXwas()<a class="anchor" aria-label="anchor" href="#function-gxwas"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXwas</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "GXwas"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-gxwas">
<a id="Running GXwas"> </a> Running GXwas<a class="anchor" aria-label="anchor" href="#running-gxwas"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">standard_beta</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">xsex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">Inphenocov</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">covartest</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">interaction</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.na.rm</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span><span class="va">MF.zero.sub</span> <span class="op">&lt;-</span> <span class="fl">0.00001</span></span>
<span><span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">xmodel</span> <span class="op">&lt;-</span> <span class="st">"FMstratified"</span></span>
<span><span class="va">combtest</span> <span class="op">&lt;-</span> <span class="st">"fisher.method"</span></span>
<span><span class="va">snp_pval</span> <span class="op">&lt;-</span> <span class="fl">1e-08</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">plot.jpeg</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.mc.cores</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">genomewideline</span> <span class="op">&lt;-</span> <span class="fl">7.3</span></span>
<span><span class="va">suggestiveline</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"GXwas.plot"</span></span>
<span><span class="va">annotateTopSnp</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.na.rm</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span><span class="va">MF.zero.sub</span> <span class="op">&lt;-</span> <span class="fl">0.00001</span></span>
<span><span class="va">MF.p.corr</span> <span class="op">&lt;-</span> <span class="st">"none"</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"GXwas.plot"</span></span>
<span><span class="va">ResultGXwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GXwas.html">GXwas</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, xmodel <span class="op">=</span> <span class="va">xmodel</span>, trait <span class="op">=</span> <span class="va">trait</span>, covarfile <span class="op">=</span> <span class="va">covarfile</span>, sex <span class="op">=</span> <span class="va">sex</span>, xsex <span class="op">=</span> <span class="va">xsex</span>, combtest <span class="op">=</span> <span class="va">combtest</span>, MF.p.corr <span class="op">=</span> <span class="st">"none"</span>, snp_pval <span class="op">=</span> <span class="va">snp_pval</span>, plot.jpeg <span class="op">=</span> <span class="va">plot.jpeg</span>, suggestiveline <span class="op">=</span> <span class="fl">5</span>, genomewideline <span class="op">=</span> <span class="fl">7.3</span>, MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, ncores <span class="op">=</span> <span class="va">ncores</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Running FMstratified model</span></span>
<span><span class="co">#&gt; • Stratified test is running for females</span></span>
<span><span class="co">#&gt; • Stratified test is running for males</span></span>
<span><span class="co">#&gt; ℹ Plots are initiated.</span></span>
<span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez/Stratified_GWAS.png</span></span></code></pre></div>
<div class="float">
<img src="gwas_models-gxwas-1.png" alt="plot of chunk gwas_models-gxwas"><div class="figcaption">plot of chunk gwas_models-gxwas</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez/Stratified_XWAS.png</span></span></code></pre>
<p><img src="gwas_models-gxwas-2.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-3.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-4.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-5.png" alt="plot of chunk gwas_models-gxwas"><img src="gwas_models-gxwas-6.png" alt="plot of chunk gwas_models-gxwas"></p>
<pre><code><span><span class="co">#&gt; Three data frames have been created and saved:</span></span>
<span><span class="co">#&gt; • CombinedWAS</span></span>
<span><span class="co">#&gt; • MaleWAS</span></span>
<span><span class="co">#&gt; • FemaleWAS</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Outputs</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">CombinedWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Dataframe containing the result of stratified model"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Dataframe containing the result of stratified model</caption>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs10000405</td>
<td align="right">4</td>
<td align="right">47716881</td>
<td align="right">0.9768464</td>
</tr>
<tr class="even">
<td align="left">rs10000452</td>
<td align="right">4</td>
<td align="right">63234460</td>
<td align="right">0.6969710</td>
</tr>
<tr class="odd">
<td align="left">rs10000465</td>
<td align="right">4</td>
<td align="right">120835814</td>
<td align="right">0.5378310</td>
</tr>
<tr class="even">
<td align="left">rs10000605</td>
<td align="right">4</td>
<td align="right">13875675</td>
<td align="right">0.7221672</td>
</tr>
<tr class="odd">
<td align="left">rs10000675</td>
<td align="right">4</td>
<td align="right">121520624</td>
<td align="right">0.8246220</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">MaleWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Stratified GWAS result with male cohort."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Stratified GWAS result with male cohort.</caption>
<colgroup>
<col width="4%">
<col width="14%">
<col width="9%">
<col width="3%">
<col width="5%">
<col width="7%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.0789</td>
<td align="right">0.2643</td>
<td align="right">-0.5968</td>
<td align="right">0.4390</td>
<td align="right">-0.2986</td>
<td align="right">0.76530</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.3959</td>
<td align="right">1.2380</td>
<td align="right">-2.8230</td>
<td align="right">2.0310</td>
<td align="right">-0.3197</td>
<td align="right">0.74920</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="right">863863</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">1.0500</td>
<td align="right">0.8858</td>
<td align="right">-0.6864</td>
<td align="right">2.7860</td>
<td align="right">1.1850</td>
<td align="right">0.23600</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs12041521</td>
<td align="right">1109154</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.4451</td>
<td align="right">0.3387</td>
<td align="right">-1.1090</td>
<td align="right">0.2188</td>
<td align="right">-1.3140</td>
<td align="right">0.18890</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs148527527</td>
<td align="right">1127860</td>
<td align="left">G</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">1.6770</td>
<td align="right">0.6864</td>
<td align="right">0.3317</td>
<td align="right">3.0220</td>
<td align="right">2.4430</td>
<td align="right">0.01456</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">ResultGXwas</span><span class="op">$</span><span class="va">FemaleWAS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Stratified GWAS result with female cohort."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Stratified GWAS result with female cohort.</caption>
<colgroup>
<col width="4%">
<col width="13%">
<col width="9%">
<col width="3%">
<col width="5%">
<col width="6%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="10%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.3459</td>
<td align="right">0.2818</td>
<td align="right">-0.2064</td>
<td align="right">0.89820</td>
<td align="right">1.2270</td>
<td align="right">0.21970</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.1568</td>
<td align="right">0.9290</td>
<td align="right">-1.6640</td>
<td align="right">1.97800</td>
<td align="right">0.1688</td>
<td align="right">0.86590</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="right">863863</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">-0.1446</td>
<td align="right">0.7282</td>
<td align="right">-1.5720</td>
<td align="right">1.28300</td>
<td align="right">-0.1985</td>
<td align="right">0.84260</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs115490086</td>
<td align="right">928969</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">0.5649</td>
<td align="right">1.4240</td>
<td align="right">-2.2270</td>
<td align="right">3.35700</td>
<td align="right">0.3966</td>
<td align="right">0.69170</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs12041521</td>
<td align="right">1109154</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">151</td>
<td align="right">-0.6697</td>
<td align="right">0.3314</td>
<td align="right">-1.3190</td>
<td align="right">-0.02008</td>
<td align="right">-2.0210</td>
<td align="right">0.04332</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in female-specific study:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"FemaleWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">FemaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in male-specific study:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"MaleWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">MaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x3</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x2</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span><span class="op">:</span><span class="fl">6</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Top ten associations in FM02comb model:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"CombinedWAS.Rda"</span><span class="op">)</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">CombinedWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<p>To know more about the function GXwas() to run different GWAS and
XWAS models with different arguments, please follow this tutorial: <a href="https://boseb.github.io/GXwasR/articles/GXwasR_overview.html" class="external-link">https://boseb.github.io/GXwasR/articles/GXwasR_overview.html</a></p>
</div>
</div>
<div class="section level2">
<h2 id="performing-sex-differential-test">Performing sex-differential test<a class="anchor" aria-label="anchor" href="#performing-sex-differential-test"></a>
</h2>
<p>Now, we will perform a sex-differential test:</p>
<div class="section level3">
<h3 id="function-sexdiff">Function: SexDiff()<a class="anchor" aria-label="anchor" href="#function-sexdiff"></a>
</h3>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiff</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SexDiff"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-sexdiff">Running SexDiff<a class="anchor" aria-label="anchor" href="#running-sexdiff"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running</span></span>
<span><span class="co">## Running the function</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">MaleWAS</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="va">x1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR"</span>, <span class="st">"SNP"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"BETA_M"</span>, <span class="st">"SE_M"</span><span class="op">)</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">FemaleWAS</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span><span class="va">x2</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="op">]</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fl">7</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CHR"</span>, <span class="st">"SNP"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"BETA_F"</span>, <span class="st">"SE_F"</span><span class="op">)</span></span>
<span><span class="va">Difftest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexDiff.html">SexDiff</a></span><span class="op">(</span>Mfile <span class="op">=</span> <span class="va">x1</span>, Ffile <span class="op">=</span> <span class="va">x2</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="sex-differentiated-effect-1.png" alt="plot of chunk sex-differentiated-effect"><div class="figcaption">plot of chunk sex-differentiated-effect</div>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Significant SNPs with sex-differential effect</span></span>
<span><span class="va">sig.snps</span> <span class="op">&lt;-</span> <span class="va">Difftest</span><span class="op">[</span><span class="va">Difftest</span><span class="op">$</span><span class="va">adjP</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">sig.snps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"BP"</span>, <span class="st">"A1"</span>, <span class="st">"T_statistic"</span>, <span class="st">"Pvalue"</span>, <span class="st">"Adjusted_Pvalue"</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">sig.snps</span>, caption <span class="op">=</span> <span class="st">"SNPs with significant sex-differential effect."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>SNPs with significant sex-differential effect.</caption>
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="right">T_statistic</th>
<th align="right">Pvalue</th>
<th align="right">Adjusted_Pvalue</th>
</tr></thead>
<tbody></tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-05</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  ! package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;    abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    commonmark             2.0.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    desc                   1.4.3      2023-12-10 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;    GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  P GXwasR               * 0.99.0     2025-08-05 [?] Bioconductor</span></span>
<span><span class="co">#&gt;    hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    hunspell               3.0.6      2025-03-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;    pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rcmdcheck              1.4.0      2021-09-27 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    roxygen2               7.3.2      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rstudioapi             0.17.1     2024-10-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    spelling               2.3.1      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    xopen                  1.0.1      2024-04-25 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpdyNNez/temp_libpath5481679d3a1d</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt;  P ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
