<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>GXwasR Overview • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GXwasR Overview">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>GXwasR Overview</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">6 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/GXwasR_overview.Rmd" class="external-link"><code>vignettes/GXwasR_overview.Rmd</code></a></small>
      <div class="d-none name"><code>GXwasR_overview.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="gxwasr-description">
<a id="GXwasR Description"></a> GXwasR Description:<a class="anchor" aria-label="anchor" href="#gxwasr-description"></a>
</h2>
<p>This package implements various statistical genetics models for
Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA)
analyses in a sex-combined or sex-stratified way considering
X-Chromosome Inactivation (XCI) pattern. In addition to association
analysis, the package also enables testing for sex differences in
genetic effects, including the implementation of specific models and
applying best practices for additional quality control (QC) of genetic
data required for these tests. The package includes thirty different
functions in six different categories (A-F) which enable a comprehensive
pipeline for sex-aware genetic association analysis of common variants
with unrelated individuals.</p>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-gxwasr">Install <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#install-gxwasr"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em> is
a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em> by
using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="functions-overview">Functions Overview<a class="anchor" aria-label="anchor" href="#functions-overview"></a>
</h2>
<p>This document lists all the functions in GXwasR package.It offers
thirty-three distinct functions, which are organized into six main
categories: A) Pre-imputation QC, B) Post-imputation QC, C) Sex-combined
and sex-stratified GWAS with specialized analysis for XWAS, D)
Sex-differential test, E) High level analysis and a range of F) Utility
Functions. These categories and their respective functions are detailed
in the below Table.</p>
<div class="section level3">
<h3 id="table-of-functions">Table of Functions<a class="anchor" aria-label="anchor" href="#table-of-functions"></a>
</h3>
<table class="table">
<colgroup>
<col width="10%">
<col width="66%">
<col width="22%">
</colgroup>
<thead><tr class="header">
<th>Function</th>
<th>Description</th>
<th>Category</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>QCsnp()</td>
<td>Performs quality control for SNPs in PLINK binary files.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="even">
<td>QCsample()</td>
<td>Identifies outlier individuals based on heterozygosity and missing
genotype rates.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="odd">
<td>AncestryCheck()</td>
<td>Evaluates samples’ ancestry and flags outliers using PCA.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="even">
<td>SexCheck()</td>
<td>Compares sex assignments with predictions from X chromosome
inbreeding coefficients.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="odd">
<td>Xhwe()</td>
<td>Filters X-chromosome variants violating Hardy-Weinberg Equilibrium
in females.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="even">
<td>MAFdiffSexControl()</td>
<td>Tests for significant MAF differences between sexes in control
samples.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="odd">
<td>FilterRegion()</td>
<td>Filters out specific chromosomal regions from input PLINK
files.</td>
<td>Post-imputation QC, Utility Functions</td>
</tr>
<tr class="even">
<td>GXwas()</td>
<td>Runs GWAS models in autosomes with XWAS models like “FMcomb01”,
“FMcomb02”, “FMstratified” for binary and quantitative traits, and
“GWAScxci” for binary traits. Focuses on additive SNP effects,
multi-collinearity issues, and includes multiple covariates and their
interactions.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="odd">
<td>PvalComb()</td>
<td>Combines p-values from separate GWAS using various statistical
methods and corrects summary p-values.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="even">
<td>SexDiff()</td>
<td>Evaluates sex differences in genetic effect size for each SNP.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>SexDiffZscore()</td>
<td>Presumably analyzes sex differences using Z-score methodology,
comparing genetic effect sizes between males and females.</td>
<td>Sex-differential test</td>
</tr>
<tr class="even">
<td>DiffZeroOne()</td>
<td>Assesses Z-scores for deviation from one and zero for statistics
like genetic correlation.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>TestXGene()</td>
<td>Performs gene-based association tests using GWAS/XWAS summary
statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>MetaGWAS()</td>
<td>Combines summary-level GWAS results using fixed-effect and
random-effect models.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>ComputePRS()</td>
<td>Calculates polygenic risk scores from GWAS summary statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>GeneticCorrBT()</td>
<td>Computes genetic correlation between to traits.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>EstimateHerit()</td>
<td>Computes SNP heritability using GREML or LDSC models.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>SexRegress()</td>
<td>Not previously described. Presumably involves regression analyses
specific to sex-stratified data.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>FilterPlinkSample()</td>
<td>Prepares PLINK binary files with desired samples based on specified
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>ComputeGeneticPC()</td>
<td>Computes principal components from a genetic relationship matrix for
population stratification correction.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>ClumpLD()</td>
<td>Performs linkage disequilibrium clumping of SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>GetMFPlink()</td>
<td>Prepares separate male and female PLINK binary files from combined
files.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>plinkVCF()</td>
<td>Converts VCF files to plink binary formats and vice versa, including
creation of dummy FAM files.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>MergeRegion()</td>
<td>Combines two genotype datasets based on common or all SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterAllele()</td>
<td>Filters out multi-allelic variants from the genetic dataset,
essential for maintaining dataset integrity and simplifying genetic
analyses.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>PlinkSummary()</td>
<td>Provides a summary of genotype datasets in plink format.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterSNP()</td>
<td>Filters out specific SNPs from the dataset based on user-defined
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>DummyCovar()</td>
<td>Recodes a categorical covariate into binary dummy variables for
statistical analysis in GXwasR.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>GXWASmiami()</td>
<td>Generates Miami plots for GWAS and XWAS.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>SumstatGenCorr()</td>
<td>Genetic Correlation Calculation from GWAS Summary Statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>LDPrune()</td>
<td>Performs LD pruning on SNP data.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>executePlinkMAF()</td>
<td>Calculate minor allele frequencies.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>ComputeLD()</td>
<td>Calculate LD matrix.</td>
<td>Utility Functions</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; GXwasR: Genome-wide and x-chromosome wide association analyses applying best practices of quality control over genetic data</span></span>
<span><span class="co">#&gt; Version 0.99.0 () installed</span></span>
<span><span class="co">#&gt; Author: c(</span></span>
<span><span class="co">#&gt;     person(given = "Banabithi",</span></span>
<span><span class="co">#&gt;            family = "Bose",</span></span>
<span><span class="co">#&gt;            role = c("cre", "aut"),</span></span>
<span><span class="co">#&gt;            email = "banabithi.bose@gmail.com",</span></span>
<span><span class="co">#&gt;            comment = c(ORCID = "0000-0003-0842-8768"))</span></span>
<span><span class="co">#&gt;            )</span></span>
<span><span class="co">#&gt; Maintainer: Banabithi Bose &lt;banabithi.bose@gmail.com&gt;</span></span>
<span><span class="co">#&gt; Tutorial: https://github.com</span></span>
<span><span class="co">#&gt; Use citation("GXwasR") to know how to cite this work.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'magrittr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:testthat':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     equals, is_less_than, not</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'printr':</span></span>
<span><span class="co">#&gt;   method                from     </span></span>
<span><span class="co">#&gt;   knit_print.data.frame rmarkdown</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="function-1">Function 1<a class="anchor" aria-label="anchor" href="#function-1"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">QCsnp</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; QCsnp: Quality control (QC) for SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function performs QC of genotype data from PLINK binary</span></span>
<span><span class="co">#&gt;      files. It can filter based on minor allele frequency,</span></span>
<span><span class="co">#&gt;      Hardy-Weinberg equilibrium, call rate, and differential</span></span>
<span><span class="co">#&gt;      missingness between cases and controls. It can also perform</span></span>
<span><span class="co">#&gt;      linkage disequilibrium-based filtering.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      QCsnp(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput = "FALSE",</span></span>
<span><span class="co">#&gt;        casecontrol = TRUE,</span></span>
<span><span class="co">#&gt;        hweCase = NULL,</span></span>
<span><span class="co">#&gt;        hweControl = NULL,</span></span>
<span><span class="co">#&gt;        hwe = NULL,</span></span>
<span><span class="co">#&gt;        maf = 0.05,</span></span>
<span><span class="co">#&gt;        geno = 0.1,</span></span>
<span><span class="co">#&gt;        monomorphicSNPs = FALSE,</span></span>
<span><span class="co">#&gt;        caldiffmiss = FALSE,</span></span>
<span><span class="co">#&gt;        diffmissFilter = FALSE,</span></span>
<span><span class="co">#&gt;        dmissX = FALSE,</span></span>
<span><span class="co">#&gt;        dmissAutoY = FALSE,</span></span>
<span><span class="co">#&gt;        highLD_regions = NULL,</span></span>
<span><span class="co">#&gt;        ld_prunning = FALSE,</span></span>
<span><span class="co">#&gt;        window_size = 50,</span></span>
<span><span class="co">#&gt;        step_size = 5,</span></span>
<span><span class="co">#&gt;        r2_threshold = 0.02</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files with both male and female samples. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if the filtering option for the SNPs is chosen.</span></span>
<span><span class="co">#&gt;           The default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; casecontrol: Boolean value, 'TRUE' or 'FALSE' indicating if the input</span></span>
<span><span class="co">#&gt;           PLINK files has cases-control status or not. The default is</span></span>
<span><span class="co">#&gt;           'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  hweCase: Numeric value between 0 to 1 or 'NULL' for removing SNPs</span></span>
<span><span class="co">#&gt;           which fail Hardy-Weinberg equilibrium for cases. The default</span></span>
<span><span class="co">#&gt;           is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; hweControl: Numeric value between 0 to 1 or 'NULL' for removing SNPs</span></span>
<span><span class="co">#&gt;           which fail Hardy-Weinberg equilibrium for controls. The</span></span>
<span><span class="co">#&gt;           default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      hwe: Numeric value between 0 to 1 or 'NULL' for removing SNPs</span></span>
<span><span class="co">#&gt;           which fail Hardy-Weinberg equilibrium for entire dataset. The</span></span>
<span><span class="co">#&gt;           default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      maf: Numeric value between 0 to 1 for removing SNPs with minor</span></span>
<span><span class="co">#&gt;           allele frequency less than the specified threshold. The</span></span>
<span><span class="co">#&gt;           default is 0.05.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     geno: Numeric value between 0 to 1 for removing SNPs that have less</span></span>
<span><span class="co">#&gt;           than the specified call rate. The default is 0.05.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Users can set this as 'NULL' to not apply this filter.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; monomorphicSNPs: Boolean value, 'TRUE' or 'FALSE' for filtering out</span></span>
<span><span class="co">#&gt;           monomorphic SNP. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; caldiffmiss: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           compute differential missingness between cases and controls</span></span>
<span><span class="co">#&gt;           for each SNP (threshold is 0.05/length(unique(No. of. SNPs in</span></span>
<span><span class="co">#&gt;           the test))). The default is 'TRUE.'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; diffmissFilter: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           filter out the SNPs or only flagged them for differential</span></span>
<span><span class="co">#&gt;           missingness in cases vs controls. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   dmissX: Boolean value, 'TRUE' or 'FALSE' for computing differential</span></span>
<span><span class="co">#&gt;           missingness between cases and controls for X chromosome SNPs</span></span>
<span><span class="co">#&gt;           only. The default is 'FALSE'. The diffmissFilter will work</span></span>
<span><span class="co">#&gt;           for all these SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; dmissAutoY: Boolean value, 'TRUE' or 'FALSE' for computing differential</span></span>
<span><span class="co">#&gt;           missingness between cases and controls for SNPs on autosomes</span></span>
<span><span class="co">#&gt;           and Y chromosome only. The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           If 'dmissX' and 'dmissAutoY' are both 'FALSE', then this will</span></span>
<span><span class="co">#&gt;           be computed genome-wide. The 'diffmissFilter' will work for</span></span>
<span><span class="co">#&gt;           all these SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: A dataframe with known high LD regions (Anderson et al.</span></span>
<span><span class="co">#&gt;           2010) is provided with the package.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ld_prunning: Boolean value, 'TRUE' or 'FALSE' for applying linkage</span></span>
<span><span class="co">#&gt;           disequilibrium (LD)-based filtering.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; window_size: Integer value, specifying a window size in the variant</span></span>
<span><span class="co">#&gt;           counts for LD-based filtering. The default is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; step_size: Integer value, specifying a variant count to shift the</span></span>
<span><span class="co">#&gt;           window at the end of each step for LD filtering. The default</span></span>
<span><span class="co">#&gt;           is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; r2_threshold: Numeric value between 0 to 1 of pairwise r^2 threshold</span></span>
<span><span class="co">#&gt;           for LD-based filtering. The default is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list of two objects, namely, 'MonomorSNPs' and 'DiffMissSNPs'</span></span>
<span><span class="co">#&gt;      containing monomorphic SNPs and SNPs with differential missingness</span></span>
<span><span class="co">#&gt;      in cases vs controls, respectively. Output PLINK binary files in</span></span>
<span><span class="co">#&gt;      the working directory.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP,</span></span>
<span><span class="co">#&gt;      Zondervan KT (2010). "Data quality control in genetic case-control</span></span>
<span><span class="co">#&gt;      association studies." _Nature Protocols_, *5*(9), 1564-1573.</span></span>
<span><span class="co">#&gt;      doi:10.1038/nprot.2010.116</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1038/nprot.2010.116&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/21085122&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      geno &lt;- NULL</span></span>
<span><span class="co">#&gt;      maf &lt;- 0.05</span></span>
<span><span class="co">#&gt;      casecontrol &lt;- FALSE</span></span>
<span><span class="co">#&gt;      hweCase &lt;- NULL</span></span>
<span><span class="co">#&gt;      hweControl &lt;- NULL</span></span>
<span><span class="co">#&gt;      hweCase &lt;- NULL</span></span>
<span><span class="co">#&gt;      monomorphicSNPs &lt;- FALSE</span></span>
<span><span class="co">#&gt;      caldiffmiss &lt;- FALSE</span></span>
<span><span class="co">#&gt;      ld_prunning &lt;- FALSE</span></span>
<span><span class="co">#&gt;      x &lt;- QCsnp(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput, foutput = foutput,</span></span>
<span><span class="co">#&gt;          geno = geno, maf = maf, hweCase = hweCase, hweControl = hweControl,</span></span>
<span><span class="co">#&gt;          ld_prunning = ld_prunning, casecontrol = casecontrol, monomorphicSNPs = monomorphicSNPs,</span></span>
<span><span class="co">#&gt;          caldiffmiss = caldiffmiss</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-2">Function 2<a class="anchor" aria-label="anchor" href="#function-2"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">QCsample</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; QCsample: Quality control for samples in the PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function identifies outlier individuals for heterozygosity</span></span>
<span><span class="co">#&gt;      and/or missing genotype rates, which aids in the detection of</span></span>
<span><span class="co">#&gt;      samples with subpar DNA quality and/or concentration that should</span></span>
<span><span class="co">#&gt;      be removed from the study. Individuals missing more than 3-7% of</span></span>
<span><span class="co">#&gt;      their genotype calls are often excluded from the analysis.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Having the correct designation of sex is important to obtain</span></span>
<span><span class="co">#&gt;      accurate genotype rate estimates, or avoid incorrectly removing</span></span>
<span><span class="co">#&gt;      samples, etc. Details can be accessed from the paper.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      QCsample(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput = NULL,</span></span>
<span><span class="co">#&gt;        imiss,</span></span>
<span><span class="co">#&gt;        het,</span></span>
<span><span class="co">#&gt;        small_sample_mod = FALSE,</span></span>
<span><span class="co">#&gt;        IBD,</span></span>
<span><span class="co">#&gt;        IBDmatrix = FALSE,</span></span>
<span><span class="co">#&gt;        ambi_out = TRUE,</span></span>
<span><span class="co">#&gt;        legend_text_size = 8,</span></span>
<span><span class="co">#&gt;        legend_title_size = 7,</span></span>
<span><span class="co">#&gt;        axis_text_size = 5,</span></span>
<span><span class="co">#&gt;        axis_title_size = 7,</span></span>
<span><span class="co">#&gt;        title_size = 9,</span></span>
<span><span class="co">#&gt;        filterSample = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string, specifying the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files. The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files with both male and female samples. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if filtering option for the samples is chosen.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    imiss: Numeric value between 0 to 1 for removing samples that have</span></span>
<span><span class="co">#&gt;           more than the specified missingness. The default is 0.03.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      het: Positive numeric value, specifying the standard deviation</span></span>
<span><span class="co">#&gt;           from the mean heterozygosity rate. The samples whose rates</span></span>
<span><span class="co">#&gt;           are more than the specified sd from the mean heterozygosity</span></span>
<span><span class="co">#&gt;           rate are removed. The default is 3. With this default value,</span></span>
<span><span class="co">#&gt;           outlying heterozygosity rates would remove individuals who</span></span>
<span><span class="co">#&gt;           are three sd away from the mean rate (1).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; small_sample_mod: Boolean value indicating whether to apply</span></span>
<span><span class="co">#&gt;           modifications for small sample sizes. Default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      IBD: Numeric value for setting the threshold for Identity by</span></span>
<span><span class="co">#&gt;           Descent (IBD) analysis. Default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; IBDmatrix: Boolean value indicating whether to generate an entire IBD</span></span>
<span><span class="co">#&gt;           matrix. Default is 'FALSE'. In this case filtered IBD matrix</span></span>
<span><span class="co">#&gt;           will be stored.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ambi_out: Boolean value indicating whether to process ambiguous</span></span>
<span><span class="co">#&gt;           samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; legend_text_size: Integer, specifying the size for legend text in the</span></span>
<span><span class="co">#&gt;           plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; legend_title_size: Integer, specifying the size for the legend title in</span></span>
<span><span class="co">#&gt;           the plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; axis_text_size: Integer, specifying the size for axis text in the plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; axis_title_size: Integer, specifying the size for the axis title in the</span></span>
<span><span class="co">#&gt;           plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; title_size: Integer, specifying the size of the title of the plot</span></span>
<span><span class="co">#&gt;           heterozygosity estimate vs missingness across samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterSample: Boolean value, 'TRUE' or 'FALSE' for filtering out the</span></span>
<span><span class="co">#&gt;           samples or not (i.e., only flagged). The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A plot of heterogysity estimate vs missingness across sample and a</span></span>
<span><span class="co">#&gt;      list containing five R dataframe objects, namely, 'HM' (samples</span></span>
<span><span class="co">#&gt;      with outlying heterozygosity and/or missing genotype rates),</span></span>
<span><span class="co">#&gt;      'Failed_Missingness' (samples with missing genotype rates),</span></span>
<span><span class="co">#&gt;      'Failed_heterozygosity' (samples with outlying heterozygosity),</span></span>
<span><span class="co">#&gt;      'Missingness_results' (missingness results) and</span></span>
<span><span class="co">#&gt;      'Heterozygosity_results' (heterozygosity results) with output</span></span>
<span><span class="co">#&gt;      PLINK files in ResultDir if filtering out the samples option is</span></span>
<span><span class="co">#&gt;      chosen.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'Missingness_results' contains missingness results for each</span></span>
<span><span class="co">#&gt;      individual, with six columns as 'FID', 'IID', 'MISS_PHENO',</span></span>
<span><span class="co">#&gt;      'N_MISS', 'N_GENO' and 'F_MISS' for Family ID, Within-family ID,</span></span>
<span><span class="co">#&gt;      Phenotype missing? (Y/N), Number of missing genotype call(s), not</span></span>
<span><span class="co">#&gt;      including obligatory missings or heterozygous haploids, number of</span></span>
<span><span class="co">#&gt;      potentially valid call(s), and missing call rate, respectively.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'Heterozygosity_results' contains heterozygosity results for each</span></span>
<span><span class="co">#&gt;      individual, with six columns as 'FID', 'IID', 'O(HOM)', 'E(HOM)',</span></span>
<span><span class="co">#&gt;      'N(NM)', and 'F' for Family ID, Within-family ID, Observed number</span></span>
<span><span class="co">#&gt;      of homozygotes, Expected number of homozygotes, Number of</span></span>
<span><span class="co">#&gt;      (non-missing, non-monomorphic) autosomal genotype observations</span></span>
<span><span class="co">#&gt;      and, Method-of-moments F coefficient estimate, respectively.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      imiss &lt;- 0.01</span></span>
<span><span class="co">#&gt;      het &lt;- 2</span></span>
<span><span class="co">#&gt;      small_sample_mod &lt;- FALSE</span></span>
<span><span class="co">#&gt;      IBD &lt;- 0.2</span></span>
<span><span class="co">#&gt;      IBDmatrix &lt;- FALSE</span></span>
<span><span class="co">#&gt;      ambi_out &lt;- TRUE</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      x &lt;- QCsample(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput,</span></span>
<span><span class="co">#&gt;          foutput = foutput, imiss = imiss, het = het, IBD = IBD,</span></span>
<span><span class="co">#&gt;          ambi_out = ambi_out</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-3">Function 3<a class="anchor" aria-label="anchor" href="#function-3"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">AncestryCheck</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; AncestryCheck: Evaluation of the samples' ancestry label.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function displays the result of the ancestry analysis in a</span></span>
<span><span class="co">#&gt;      color-coded scatter plot of the first two principal components for</span></span>
<span><span class="co">#&gt;      samples of the reference populations and the study population.</span></span>
<span><span class="co">#&gt;      Specifically, it compares the study samples' ancestry labels to a</span></span>
<span><span class="co">#&gt;      panel representing a reference population, and it also flags the</span></span>
<span><span class="co">#&gt;      outlier samples with respect to a chosen reference population.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Users are required to provide SNPs ids or rsids in the input PLINK</span></span>
<span><span class="co">#&gt;      files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The function first filters the reference and study data for</span></span>
<span><span class="co">#&gt;      non-A-T or G-C SNPs. It next conducts LD pruning, fixes the</span></span>
<span><span class="co">#&gt;      chromosome mismatch between the reference and study datasets,</span></span>
<span><span class="co">#&gt;      checks for allele flips, updates the locations, and flips the</span></span>
<span><span class="co">#&gt;      alleles. The two datasets are then joined, and the resulting</span></span>
<span><span class="co">#&gt;      genotype dataset is subjected to Principal Component Analysis</span></span>
<span><span class="co">#&gt;      (PCA).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The detection of population structure down to the level of the</span></span>
<span><span class="co">#&gt;      reference dataset can then be accomplished using PCA on this</span></span>
<span><span class="co">#&gt;      combined genotyping panel. For instance, the center of the</span></span>
<span><span class="co">#&gt;      European reference samples is determined using the data from</span></span>
<span><span class="co">#&gt;      principal components 1 and 2 (median(PC1 europeanRef), median(PC2</span></span>
<span><span class="co">#&gt;      europeanRef)). It determines the European reference samples'</span></span>
<span><span class="co">#&gt;      maximum Euclidean distance (maxDist) from this center.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      All study samples that are non-European, or outliers, are those</span></span>
<span><span class="co">#&gt;      whose Euclidean distances from the center are more than or equal</span></span>
<span><span class="co">#&gt;      to the radius r= outlier threshold* maxDist. This function</span></span>
<span><span class="co">#&gt;      utilizes the HapMap phase 3 data in NCBI 36 and 1000GenomeIII in</span></span>
<span><span class="co">#&gt;      CGRCh37. Both study and reference datasets should be of the same</span></span>
<span><span class="co">#&gt;      genome build. If not, users need to lift over one of the datasets</span></span>
<span><span class="co">#&gt;      to the same build.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      AncestryCheck(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        reference = c("HapMapIII_NCBI36", "ThousandGenome"),</span></span>
<span><span class="co">#&gt;        filterSNP = TRUE,</span></span>
<span><span class="co">#&gt;        studyLD = TRUE,</span></span>
<span><span class="co">#&gt;        studyLD_window_size = 50,</span></span>
<span><span class="co">#&gt;        studyLD_step_size = 5,</span></span>
<span><span class="co">#&gt;        studyLD_r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        referLD = FALSE,</span></span>
<span><span class="co">#&gt;        referLD_window_size = 50,</span></span>
<span><span class="co">#&gt;        referLD_step_size = 5,</span></span>
<span><span class="co">#&gt;        referLD_r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        highLD_regions,</span></span>
<span><span class="co">#&gt;        study_pop,</span></span>
<span><span class="co">#&gt;        outlier = FALSE,</span></span>
<span><span class="co">#&gt;        outlierOf = "EUR",</span></span>
<span><span class="co">#&gt;        outlier_threshold = 3</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the study samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; reference: Boolean value,'HapMapIII_NCBI36' and 'ThousandGenome',</span></span>
<span><span class="co">#&gt;           specifying Hapmap Phase3 (3 Consortium 2010) and 1000 Genomes</span></span>
<span><span class="co">#&gt;           phase III (1000 Genomes Project Consortium 2015) reference</span></span>
<span><span class="co">#&gt;           population, respectively. The default is 'HapMapIII_NCBI36'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterSNP: Boolean value, 'TRUE' or 'FALSE' for filtering out the SNPs.</span></span>
<span><span class="co">#&gt;           The default is 'TRUE'. We recommend setting it 'FALSE' only</span></span>
<span><span class="co">#&gt;           when the users are sure that they could join the study and</span></span>
<span><span class="co">#&gt;           reference samples directly.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  studyLD: Boolean value, 'TRUE' or 'FALSE' for applying linkage</span></span>
<span><span class="co">#&gt;           disequilibrium (LD)-based filtering on study genotype data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; studyLD_window_size: Integer value, specifying a window size in variant</span></span>
<span><span class="co">#&gt;           count or kilobase for LD-based filtering of the variants for</span></span>
<span><span class="co">#&gt;           the study data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; studyLD_step_size: Integer value, specifying a variant count to shift</span></span>
<span><span class="co">#&gt;           the window at the end of each step for LD filtering for the</span></span>
<span><span class="co">#&gt;           study data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; studyLD_r2_threshold: Numeric value between 0 to 1 of pairwise r^2</span></span>
<span><span class="co">#&gt;           threshold for LD-based filtering for the study data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  referLD: Boolean value, 'TRUE' or 'FALSE' for applying linkage</span></span>
<span><span class="co">#&gt;           disequilibrium (LD)-based filtering on reference genotype</span></span>
<span><span class="co">#&gt;           data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; referLD_window_size: Integer value, specifying a window size in variant</span></span>
<span><span class="co">#&gt;           count or kilobase for LD-based filtering of the variants for</span></span>
<span><span class="co">#&gt;           the reference data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; referLD_step_size: Integer value, specifying a variant count to shift</span></span>
<span><span class="co">#&gt;           the window at the end of each step for LD filtering for the</span></span>
<span><span class="co">#&gt;           reference data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; referLD_r2_threshold: Numeric value between 0 to 1 of pairwise r^2</span></span>
<span><span class="co">#&gt;           threshold for LD-based filtering for the reference data.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: A dataframe with known high LD regions (Anderson et al.</span></span>
<span><span class="co">#&gt;           2010) is provided with the package.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; study_pop: A dataframe containing two columns for study in first</span></span>
<span><span class="co">#&gt;           column, sample ID (i.e., IID) and in second column, the</span></span>
<span><span class="co">#&gt;           ancestry label.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  outlier: Boolean value, 'TRUE' or 'FALSE', specifying outlier</span></span>
<span><span class="co">#&gt;           detection will be performed or not.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; outlierOf: Chracter string, specifying the reference ancestry name for</span></span>
<span><span class="co">#&gt;           detecting outlier samples. The default is "outlierOf = "EUR".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; outlier_threshold: Numeric value, specifying the threshold to be be</span></span>
<span><span class="co">#&gt;           used to detect outlier samples. This threshold will be</span></span>
<span><span class="co">#&gt;           multiplied with the Eucledean distance from the center of the</span></span>
<span><span class="co">#&gt;           PC1 and PC2 to the maximum Euclidean distance of the</span></span>
<span><span class="co">#&gt;           reference samples. Study samples outside this distance will</span></span>
<span><span class="co">#&gt;           be considered as outlier. The default is 3.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list containing two data frames: one with the IDs of outlier</span></span>
<span><span class="co">#&gt;      samples (Outlier_samples), and another with samples annotated with</span></span>
<span><span class="co">#&gt;      predicted ancestry (Samples_with_predicted_ancestry).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      1000 Genomes Project Consortium T (2015). "A global reference for</span></span>
<span><span class="co">#&gt;      human genetic variation." _Nature_, *526*, 68-74.</span></span>
<span><span class="co">#&gt;      doi:10.1038/nature15393 &lt;https://doi.org/10.1038/nature15393&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      3 Consortium TIH (2010). "Integrating common and rare genetic</span></span>
<span><span class="co">#&gt;      variation in diverse human populations." _Nature_, *467*, 52-58.</span></span>
<span><span class="co">#&gt;      doi:10.1038/nature09298 &lt;https://doi.org/10.1038/nature09298&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Anderson CA, Pettersson FH, Clarke GM, Cardon LR, Morris AP,</span></span>
<span><span class="co">#&gt;      Zondervan KT (2010). "Data quality control in genetic case-control</span></span>
<span><span class="co">#&gt;      association studies." _Nature Protocols_, *5*(9), 1564-1573.</span></span>
<span><span class="co">#&gt;      doi:10.1038/nprot.2010.116</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1038/nprot.2010.116&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/21085122&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("highLD_hg19", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("example_data_study_sample_ancestry", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      reference &lt;- "HapMapIII_NCBI36"</span></span>
<span><span class="co">#&gt;      highLD_regions &lt;- highLD_hg19</span></span>
<span><span class="co">#&gt;      study_pop &lt;- example_data_study_sample_ancestry # PreimputeEX</span></span>
<span><span class="co">#&gt;      studyLD_window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      studyLD_step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      studyLD_r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      filterSNP &lt;- TRUE</span></span>
<span><span class="co">#&gt;      studyLD &lt;- FALSE</span></span>
<span><span class="co">#&gt;      referLD &lt;- FALSE</span></span>
<span><span class="co">#&gt;      referLD_window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      referLD_step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      referLD_r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      outlier &lt;- TRUE</span></span>
<span><span class="co">#&gt;      outlier_threshold &lt;- 3</span></span>
<span><span class="co">#&gt;      x &lt;- AncestryCheck(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput,</span></span>
<span><span class="co">#&gt;          reference = reference, highLD_regions = highLD_regions,</span></span>
<span><span class="co">#&gt;          study_pop = study_pop, studyLD = studyLD, referLD = referLD,</span></span>
<span><span class="co">#&gt;          outlierOf = "EUR", outlier = outlier, outlier_threshold = outlier_threshold</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-4">Function 4<a class="anchor" aria-label="anchor" href="#function-4"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexCheck</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexCheck: Compare sex assignments in the input PLINK files with those</span></span>
<span><span class="co">#&gt; imputed from X chromosome inbreeding coefficients</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function compares sex assignments in the input dataset with</span></span>
<span><span class="co">#&gt;      those predicted from X chromosome inbreeding coefficients (Purcell</span></span>
<span><span class="co">#&gt;      et al. 2007), and gives the option to convert the sex assignments</span></span>
<span><span class="co">#&gt;      to the predicted values. Implicitly, this function computes</span></span>
<span><span class="co">#&gt;      observed and expected autosomal homozygous genotype counts for</span></span>
<span><span class="co">#&gt;      each sample and reports method-of-moments F coefficient estimates</span></span>
<span><span class="co">#&gt;      (i.e., observed hom. count - expected count) / (total observations</span></span>
<span><span class="co">#&gt;      - expected count)). The expected counts will be based on loaded or</span></span>
<span><span class="co">#&gt;      imputed minor allele frequencies.  Since imputed MAFs are highly</span></span>
<span><span class="co">#&gt;      inaccurate when there are few samples, the 'compute freq'</span></span>
<span><span class="co">#&gt;      parameter should be set to TRUE to compute MAF implicitly.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Due to the use of allele frequencies, if a cohort is comprised of</span></span>
<span><span class="co">#&gt;      individuals of different ancestries, users may need to process any</span></span>
<span><span class="co">#&gt;      samples with rare ancestry individually if the dataset has a very</span></span>
<span><span class="co">#&gt;      unbalanced ancestry distribution. It is advised to run this</span></span>
<span><span class="co">#&gt;      function with all the parameters set to zero, then examine the</span></span>
<span><span class="co">#&gt;      distribution of the F estimates (there should be a clear gap</span></span>
<span><span class="co">#&gt;      between a very tight male clump on the right side of the</span></span>
<span><span class="co">#&gt;      distribution and the females everywhere else). Then, rerun the</span></span>
<span><span class="co">#&gt;      function with the parameters that correspond to this gap.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexCheck(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        impute_sex = FALSE,</span></span>
<span><span class="co">#&gt;        compute_freq = FALSE,</span></span>
<span><span class="co">#&gt;        LD = TRUE,</span></span>
<span><span class="co">#&gt;        LD_window_size = 50,</span></span>
<span><span class="co">#&gt;        LD_step_size = 5,</span></span>
<span><span class="co">#&gt;        LD_r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        fmax_F = 0.2,</span></span>
<span><span class="co">#&gt;        mmin_F = 0.8</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files. Note: Input dataset should contain X and Y</span></span>
<span><span class="co">#&gt;           regions.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; impute_sex: Boolean value, 'TRUE' or 'FALSE', specifying sex to be</span></span>
<span><span class="co">#&gt;           imputed or not. If 'TRUE' then sex-imputed PLINK files,</span></span>
<span><span class="co">#&gt;           prefixed, 'seximputed_plink', will be produced in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; compute_freq: Boolean value, 'TRUE' or 'FALSE', specifying minor allele</span></span>
<span><span class="co">#&gt;           frequency (MAF). This function requires reasonable MAF</span></span>
<span><span class="co">#&gt;           estimates, so it is essential to use 'compute_freq' = 'TRUE'</span></span>
<span><span class="co">#&gt;           for computing MAF from an input PLINK file if there are very</span></span>
<span><span class="co">#&gt;           few samples in the input dataset. The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       LD: Boolean value, 'TRUE' or 'FALSE' for applying linkage</span></span>
<span><span class="co">#&gt;           disequilibrium (LD)-based filtering. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LD_window_size: Integer value, specifying a window size in variant</span></span>
<span><span class="co">#&gt;           count for LD-based filtering. The default is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LD_step_size: Integer value, specifying a variant count to shift the</span></span>
<span><span class="co">#&gt;           window at the end of each step for LD filtering. The default</span></span>
<span><span class="co">#&gt;           is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LD_r2_threshold: Numeric value between 0 to 1 of pairwise r^2 threshold</span></span>
<span><span class="co">#&gt;           for LD-based filtering. The default is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   fmax_F: Numeric value between 0 to 1. Samples with F estimates</span></span>
<span><span class="co">#&gt;           smaller than this value will be labeled as females. The</span></span>
<span><span class="co">#&gt;           default is 0.2.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   mmin_F: Numeric value between 0 to 1. Samples with F estimates larger</span></span>
<span><span class="co">#&gt;           than this value will be labeled as males. The default is 0.8.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with six columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'FID' (Family ID)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'IID' (Individual ID)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'PEDSEX '(Sex as determined in pedigree file (1=male,</span></span>
<span><span class="co">#&gt;           2=female))</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SNPSEX' (Sex as determined by X chromosome)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'STATUS' (Displays "PROBLEM" or "OK" for each individual)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'F' (The actual X chromosome inbreeding (homozygosity)</span></span>
<span><span class="co">#&gt;           estimate)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A PROBLEM arises if the two sexes do not match, or if the SNP data</span></span>
<span><span class="co">#&gt;      or pedigree data are ambiguous with regard to sex.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      LD &lt;- TRUE</span></span>
<span><span class="co">#&gt;      LD_window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      LD_step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      LD_r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      fmax_F &lt;- 0.2</span></span>
<span><span class="co">#&gt;      mmin_F &lt;- 0.8</span></span>
<span><span class="co">#&gt;      impute_sex &lt;- FALSE</span></span>
<span><span class="co">#&gt;      compute_freq &lt;- FALSE</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      x &lt;- SexCheck(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput, impute_sex = impute_sex,</span></span>
<span><span class="co">#&gt;          compute_freq = compute_freq, LD_window_size = LD_window_size, LD_step_size = LD_step_size,</span></span>
<span><span class="co">#&gt;          LD_r2_threshold = 0.02, fmax_F = 0.2, mmin_F = 0.8</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      # Checking if there is any wrong sex assignment</span></span>
<span><span class="co">#&gt;      problematic_sex &lt;- x[x$STATUS != "OK", ]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-5">Function 5<a class="anchor" aria-label="anchor" href="#function-5"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">Xhwe</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Xhwe: Filter X-chromosome variants for HWE in females.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function is a part of the post-imputation quality control</span></span>
<span><span class="co">#&gt;      process prior to GWAS. This tests for Hardy-Weinberg Equilibrium</span></span>
<span><span class="co">#&gt;      (HWE) for X-chromosome variants in females. Males' hemizygous X</span></span>
<span><span class="co">#&gt;      chromosome prevents testing for HWE on their haploid X calls, and</span></span>
<span><span class="co">#&gt;      testing for HWE across all samples would have a high failure rate.</span></span>
<span><span class="co">#&gt;      This function will check for HWE across the X in females (cases</span></span>
<span><span class="co">#&gt;      and controls combined), following the recommendation in Khramtsova</span></span>
<span><span class="co">#&gt;      et al., 2023, and can remove these regions from analysis in all</span></span>
<span><span class="co">#&gt;      samples. The p-value threshold for filtering out SNPs is</span></span>
<span><span class="co">#&gt;      0.05/no.of. X-chromosome variants.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Xhwe(DataDir, ResultDir = tempdir(), finput, filterSNP = TRUE, foutput)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterSNP: Boolean value, 'TRUE' or 'FALSE' for filtering out the</span></span>
<span><span class="co">#&gt;           X-chromosome variants i.e., SNPs from the input file or not</span></span>
<span><span class="co">#&gt;           (i.e., only flagged). The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if filtering option for the SNPs is chosen. The</span></span>
<span><span class="co">#&gt;           default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list object containing SNPs. If 'filterSNP' = 'TRUE', the output</span></span>
<span><span class="co">#&gt;      filtered PLINK binary files will be produced inside 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      x &lt;- Xhwe(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, foutput = foutput, filterSNP = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      x</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-6">Function 6<a class="anchor" aria-label="anchor" href="#function-6"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MAFdiffSexControl</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; MAFdiffSexControl: Test for significantly different minor allele</span></span>
<span><span class="co">#&gt; frequency (MAF) between sexes in control samples</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      With parameters to filter out SNPs and/or flag the SNPs, this</span></span>
<span><span class="co">#&gt;      function tests for significantly different MAF (p-value &lt; 0.05/no.</span></span>
<span><span class="co">#&gt;      of SNPs) between sexes in control samples solely for binary</span></span>
<span><span class="co">#&gt;      phenotypes. Since the disparities may be caused by technical</span></span>
<span><span class="co">#&gt;      confounding or sample biases for the research cohorts, it is</span></span>
<span><span class="co">#&gt;      advised that any SNPs in the controls with a sex difference in MAF</span></span>
<span><span class="co">#&gt;      be carefully evaluated and identified for further examination</span></span>
<span><span class="co">#&gt;      (Khramtsova et. al., 2023). In autosomal allele frequencies, sex</span></span>
<span><span class="co">#&gt;      differences are not anticipated.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      MAFdiffSexControl(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        filterSNP = FALSE,</span></span>
<span><span class="co">#&gt;        foutput = NULL</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files with both male and female samples. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterSNP: Boolean value, 'TRUE' or 'FALSE' for filtering out the SNPs</span></span>
<span><span class="co">#&gt;           or not (i.e., only flagged). The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if filtering option for the SNPs is chosen. The</span></span>
<span><span class="co">#&gt;           default is NULL.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list object containing excluded or flagged SNPs. If 'filterSNP'</span></span>
<span><span class="co">#&gt;      = 'TRUE', the output filtered PLINK binary files will be produced</span></span>
<span><span class="co">#&gt;      inside 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      x &lt;- MAFdiffSexControl(DataDir, ResultDir, finput, filterSNP = TRUE, foutput = foutput)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-7">Function 7<a class="anchor" aria-label="anchor" href="#function-7"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterRegion</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; FilterRegion: Filter chromosomal regions.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Filtering Pseudo-Autosomal Region (PAR), X-transposed region</span></span>
<span><span class="co">#&gt;      (XTR), Ampliconic, filter based on chromosome code or user-defined</span></span>
<span><span class="co">#&gt;      regions from input PLINK files. Only one type of filtering can be</span></span>
<span><span class="co">#&gt;      done from three types, either by region (using 'regionfile' =</span></span>
<span><span class="co">#&gt;      'TRUE'), by chromosome ('filterCHR') or by any combination of</span></span>
<span><span class="co">#&gt;      these three, 'filterPAR', 'filterXTR' and 'filterAmpliconic.'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      FilterRegion(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput,</span></span>
<span><span class="co">#&gt;        CHRX = TRUE,</span></span>
<span><span class="co">#&gt;        CHRY = FALSE,</span></span>
<span><span class="co">#&gt;        filterPAR = TRUE,</span></span>
<span><span class="co">#&gt;        filterXTR = TRUE,</span></span>
<span><span class="co">#&gt;        filterAmpliconic = TRUE,</span></span>
<span><span class="co">#&gt;        regionfile = FALSE,</span></span>
<span><span class="co">#&gt;        filterCHR = NULL,</span></span>
<span><span class="co">#&gt;        Hg = "19",</span></span>
<span><span class="co">#&gt;        exclude = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if the filtering option for the SNPs is chosen.</span></span>
<span><span class="co">#&gt;           The default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     CHRX: Boolean value, 'TRUE' or 'FALSE' to filter/flag regions from</span></span>
<span><span class="co">#&gt;           chromosome X. The default is 'TRUE'. Note: 'CHRX' only in</span></span>
<span><span class="co">#&gt;           effect if one of 'filterPAR', 'filterXTR' or</span></span>
<span><span class="co">#&gt;           'filterAmpliconic' filter is in effect.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     CHRY: Boolean value, 'TRUE' or 'FALSE' to filter/flag regions from</span></span>
<span><span class="co">#&gt;           chromosome X. The default is 'FALSE'. Note: CHRY only in</span></span>
<span><span class="co">#&gt;           effect if one of 'filterPAR', 'filterXTR' or</span></span>
<span><span class="co">#&gt;           'filterAmpliconic' filter is in effect.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterPAR: Boolean value, 'TRUE' or 'FALSE' to filter out PARs from</span></span>
<span><span class="co">#&gt;           input PLINK file. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterXTR: Boolean value, 'TRUE' or 'FALSE' to filter out XTRs from</span></span>
<span><span class="co">#&gt;           input PLINK file. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterAmpliconic: Boolean value, 'TRUE' or 'FALSE' to filter out</span></span>
<span><span class="co">#&gt;           Ampliconic regions from input PLINK file. The default is</span></span>
<span><span class="co">#&gt;           'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; regionfile: Character string, specifying the name of the .txt file</span></span>
<span><span class="co">#&gt;           containing the user-defined regions to be filtered out from</span></span>
<span><span class="co">#&gt;           input PLINK file in bed format. The default is 'FALSE'. If</span></span>
<span><span class="co">#&gt;           'regionfile' = 'TRUE', only this filtering will be in effect.</span></span>
<span><span class="co">#&gt;           Also, PAR, XTR and Ampliconic SNPs from X-chomosome will be</span></span>
<span><span class="co">#&gt;           flagged and returned.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filterCHR: Vector value with positive integer, specifying the</span></span>
<span><span class="co">#&gt;           chromosome code to filter/flag the SNPs. The default is 0,</span></span>
<span><span class="co">#&gt;           means no filtering based on chromosome code. For non-zero</span></span>
<span><span class="co">#&gt;           values of this argument, the function will only consider the</span></span>
<span><span class="co">#&gt;           chromosome code to filter or flag. All other filtering will</span></span>
<span><span class="co">#&gt;           not work. If filterCHR = TRUE, only this filtering will be in</span></span>
<span><span class="co">#&gt;           effect. Also, PAR, XTR and Ampliconic SNPs from X-chomosome</span></span>
<span><span class="co">#&gt;           will be flagged and returned.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       Hg: Character value, '19', or '38', specifying which genome build</span></span>
<span><span class="co">#&gt;           to use for PAR, XTR and Ampliconic regions. The default is Hg</span></span>
<span><span class="co">#&gt;           = "19".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  exclude: Boolean value, 'TRUE' or 'FALSE' to filter and flag or only</span></span>
<span><span class="co">#&gt;           flag the SNPs. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list of three dataframes: PAR containing SNPs from PAR regions;</span></span>
<span><span class="co">#&gt;      XTR containing SNPs from XTR region and Ampliconic containing SNPs</span></span>
<span><span class="co">#&gt;      from Ampliconic region.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For non-zero value of 'filterCHR', a dataframe containing the</span></span>
<span><span class="co">#&gt;      excluded/flagged SNPs will be returned.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For 'exclude' = 'TRUE', two sets of PLINK binary files will be</span></span>
<span><span class="co">#&gt;      produced in ResultDir. One set will have the remaining SNPs after</span></span>
<span><span class="co">#&gt;      filtering and other one will have the discarded SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "PostimputeEX_QC1"</span></span>
<span><span class="co">#&gt;      x &lt;- FilterRegion(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, foutput = foutput, CHRX = TRUE, CHRY = FALSE,</span></span>
<span><span class="co">#&gt;          filterPAR = TRUE, filterXTR = TRUE, filterAmpliconic = TRUE,</span></span>
<span><span class="co">#&gt;          regionfile = FALSE, filterCHR = NULL, Hg = "38", exclude = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-8">Function 8<a class="anchor" aria-label="anchor" href="#function-8"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXwas</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GXwas: Running genome-wide association study (GWAS) and</span></span>
<span><span class="co">#&gt; X-chromosome-wide association study (XWAS) models.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function runs GWAS models in autosomes with several</span></span>
<span><span class="co">#&gt;      alternative XWAS models. Models such as</span></span>
<span><span class="co">#&gt;      "FMcombx01","FMcombx02",and "FMstratified" can be applied to both</span></span>
<span><span class="co">#&gt;      binary and quantitative traits, while "GWAcxci" can only be</span></span>
<span><span class="co">#&gt;      applied to a binary trait.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For binary and quantitative features, this function uses logistic</span></span>
<span><span class="co">#&gt;      and linear regression, allowing for multiple covariates and the</span></span>
<span><span class="co">#&gt;      interactions with those covariates in a multiple-regression</span></span>
<span><span class="co">#&gt;      approach. These models are all run using the additive effects of</span></span>
<span><span class="co">#&gt;      SNPs, and each additional minor allele's influence is represented</span></span>
<span><span class="co">#&gt;      by the direction of the regression coefficient.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function attempts to identify the multi-collinearity among</span></span>
<span><span class="co">#&gt;      predictors by displaying NA for the test statistic and a p-value</span></span>
<span><span class="co">#&gt;      for all terms in the model. The more terms you add, the more</span></span>
<span><span class="co">#&gt;      likely you are to run into issues.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For details about the different XWAS model, please follow the</span></span>
<span><span class="co">#&gt;      associated publication.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      GXwas(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        trait = c("binary", "quantitative"),</span></span>
<span><span class="co">#&gt;        standard_beta = TRUE,</span></span>
<span><span class="co">#&gt;        xmodel = c("FMcombx01", "FMcombx02", "FMstratified", "GWAScxci"),</span></span>
<span><span class="co">#&gt;        sex = FALSE,</span></span>
<span><span class="co">#&gt;        xsex = FALSE,</span></span>
<span><span class="co">#&gt;        covarfile = NULL,</span></span>
<span><span class="co">#&gt;        interaction = FALSE,</span></span>
<span><span class="co">#&gt;        covartest = c("ALL"),</span></span>
<span><span class="co">#&gt;        Inphenocov = c("ALL"),</span></span>
<span><span class="co">#&gt;        combtest = c("fisher.method", "fisher.method.perm", "stouffer.method"),</span></span>
<span><span class="co">#&gt;        MF.zero.sub = 1e-05,</span></span>
<span><span class="co">#&gt;        B = 10000,</span></span>
<span><span class="co">#&gt;        MF.mc.cores = 1,</span></span>
<span><span class="co">#&gt;        MF.na.rm = FALSE,</span></span>
<span><span class="co">#&gt;        MF.p.corr = "none",</span></span>
<span><span class="co">#&gt;        plot.jpeg = FALSE,</span></span>
<span><span class="co">#&gt;        plotname = "GXwas.plot",</span></span>
<span><span class="co">#&gt;        snp_pval = 1e-08,</span></span>
<span><span class="co">#&gt;        annotateTopSnp = FALSE,</span></span>
<span><span class="co">#&gt;        suggestiveline = 5,</span></span>
<span><span class="co">#&gt;        genomewideline = 7.3,</span></span>
<span><span class="co">#&gt;        ncores = 0</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the folder path where the outputs will</span></span>
<span><span class="co">#&gt;           be saved.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files with both male and female samples. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: Case/control phenotypes are expected to be encoded as</span></span>
<span><span class="co">#&gt;           1=unaffected (control), 2=affected (case); 0 is accepted as</span></span>
<span><span class="co">#&gt;           an alternate missing value encoding. The missing case/control</span></span>
<span><span class="co">#&gt;           or quantitative phenotypes are expected to be encoded as</span></span>
<span><span class="co">#&gt;           'NA'/'nan' (any capitalization) or -9.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    trait: Boolean value, 'binary' or 'quantitative' for the phenotype</span></span>
<span><span class="co">#&gt;           i.e. the trait.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; standard_beta: Boolean value, 'TRUE' or 'FALSE' in case of quantitative</span></span>
<span><span class="co">#&gt;           trait for standardizing the trait or phenotype values (mean</span></span>
<span><span class="co">#&gt;           0, unit variance), so the resulting coefficients will be</span></span>
<span><span class="co">#&gt;           standardized. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   xmodel: Models "FMcombx01","FMcombx02",and "FMstratified" can be</span></span>
<span><span class="co">#&gt;           chosen for both binary and quantitative traits while</span></span>
<span><span class="co">#&gt;           "GWAcxci" can only apply to the binary trait. These models</span></span>
<span><span class="co">#&gt;           take care of the X-chromosomal marker. Three female genotypes</span></span>
<span><span class="co">#&gt;           are coded by 0, 1, and 2 in FM01 and FM02. The two genotypes</span></span>
<span><span class="co">#&gt;           of males that follow the X-chromosome inactivation (XCI)</span></span>
<span><span class="co">#&gt;           pattern as random (XCI-R) in the FM01 model are coded by 0</span></span>
<span><span class="co">#&gt;           and 1, while the two genotypes that follow the XCI is escaped</span></span>
<span><span class="co">#&gt;           (XCI-E) in the FM02 model are coded by 0 and 1. To reflect</span></span>
<span><span class="co">#&gt;           the dose compensation connection between the sexes, FM02</span></span>
<span><span class="co">#&gt;           treats men as homozygous females.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           In the "FMstratified" associations are tested separately for</span></span>
<span><span class="co">#&gt;           males and females, and then the combined p values are</span></span>
<span><span class="co">#&gt;           computed the Fisher's method, Fisher's method with</span></span>
<span><span class="co">#&gt;           permutation, or Stouffer's method(1,3-7]. An X-chromosome</span></span>
<span><span class="co">#&gt;           inactivation (XCI) pattern, or coding technique for</span></span>
<span><span class="co">#&gt;           X-chromosomal genotypes between sexes, is not required for</span></span>
<span><span class="co">#&gt;           the XCGA. By simultaneously accounting for four distinct XCI</span></span>
<span><span class="co">#&gt;           patterns, namely XCI-R, XCI-E, XCI-SN (XCI fully toward</span></span>
<span><span class="co">#&gt;           normal allele), and XCI-SR (XCI fully toward risk allele),</span></span>
<span><span class="co">#&gt;           this model may maintain a respectably high power (Su et al.</span></span>
<span><span class="co">#&gt;           2022).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: 'sex' shouldn't be provided as a covariate in the XCGA</span></span>
<span><span class="co">#&gt;           model.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      sex: Boolean value, 'TRUE' or 'FALSE' for using sex as covariate</span></span>
<span><span class="co">#&gt;           in association test. It is applicable genome-wide.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     xsex: Boolean value, 'TRUE' or 'FALSE' for using sex as covariate</span></span>
<span><span class="co">#&gt;           in association test for X-chromosomal SNPs. The default is</span></span>
<span><span class="co">#&gt;           FALSE. This will overwrite 'sex' argument for X-chromosome.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covarfile: Character string for the full name of the covariate file in</span></span>
<span><span class="co">#&gt;           .txt format. This file should be placed in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note about the covariate file: The first column of this file</span></span>
<span><span class="co">#&gt;           should be 'FID', the second column should be 'IID' and the</span></span>
<span><span class="co">#&gt;           other columns should be covariates. The primary header line</span></span>
<span><span class="co">#&gt;           should be there starting with “FID”, and “IID” followed by</span></span>
<span><span class="co">#&gt;           covariate names. If an individual is not present in the</span></span>
<span><span class="co">#&gt;           covariate file, or if the individual has a missing phenotype</span></span>
<span><span class="co">#&gt;           value (i.e. -9 by default) for the covariate, then that</span></span>
<span><span class="co">#&gt;           individual is set to missing (i.e. will be excluded from</span></span>
<span><span class="co">#&gt;           association analysis). It is important to note that for</span></span>
<span><span class="co">#&gt;           stratified GWAS model, if PCs are included as covar then it</span></span>
<span><span class="co">#&gt;           should be generated separately for each cohort and then</span></span>
<span><span class="co">#&gt;           included in the covarfile. Use the function 'DummyCovar' to</span></span>
<span><span class="co">#&gt;           generate a new covariate file with categorical variables</span></span>
<span><span class="co">#&gt;           down-coded as binary dummy variables for the covariate file</span></span>
<span><span class="co">#&gt;           with categorical variables. For instance, if a variable has K</span></span>
<span><span class="co">#&gt;           categories, K-1 new dummy variables are constructed, and the</span></span>
<span><span class="co">#&gt;           original covariate is now estimated with a coefficient for</span></span>
<span><span class="co">#&gt;           each category.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; interaction: Boolean value, 'TRUE' or 'FALSE' for including SNP x</span></span>
<span><span class="co">#&gt;           covariate interaction term/terms from the association</span></span>
<span><span class="co">#&gt;           analysis. The default is 'FALSE'. If a permutation procedure</span></span>
<span><span class="co">#&gt;           is chosen then the interaction will be automatically 'FALSE'.</span></span>
<span><span class="co">#&gt;           For the interaction with the two covariates COV1 and COV2,</span></span>
<span><span class="co">#&gt;           the model will look like: Y = b0 + b1.ADD + b2.COV1 + b3.COV2</span></span>
<span><span class="co">#&gt;           + b4.ADD x COV1 + b5.ADD x COV2 + e. When interaction factors</span></span>
<span><span class="co">#&gt;           are incorporated into the model, the main effects'</span></span>
<span><span class="co">#&gt;           significance is not always determined simply; rather, it</span></span>
<span><span class="co">#&gt;           depends on the arbitrary coding of the variables. To put it</span></span>
<span><span class="co">#&gt;           another way, you should probably just interpret the p-value</span></span>
<span><span class="co">#&gt;           for the interaction. Also, The p-values for the covariates do</span></span>
<span><span class="co">#&gt;           not represent the test for the SNP-phenotype association</span></span>
<span><span class="co">#&gt;           after controlling for the covariate. That is the first row</span></span>
<span><span class="co">#&gt;           (ADD). Rather, the covariate term is the test associated with</span></span>
<span><span class="co">#&gt;           the covariate-phenotype association. These p-values might be</span></span>
<span><span class="co">#&gt;           extremely significant (e.g. if one covaries for smoking in an</span></span>
<span><span class="co">#&gt;           analysis of heart disease, etc) but this does not mean that</span></span>
<span><span class="co">#&gt;           the SNP has a highly significant effect necessarily. Note</span></span>
<span><span class="co">#&gt;           that, this feature is not valid for XCGA model for XWAS part.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covartest: Vector value with 'NULL',"ALL" or covariate name/names to be</span></span>
<span><span class="co">#&gt;           included in the test. The default is 'NULL.' For instance,</span></span>
<span><span class="co">#&gt;           the user can choose “AGE” and “SEX” as covartest = c(“AGE”,</span></span>
<span><span class="co">#&gt;           “SEX”) or all the covariates as covartest = c(“ALL”).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Inphenocov: Vector of integer values starting from 1 to extract the</span></span>
<span><span class="co">#&gt;           terms which user wants from the above model: Y = b0 + b1.ADD</span></span>
<span><span class="co">#&gt;           + b2.COV1 + b3.COV2 + b4.ADDxCOV1 + b5.ADDxCOV2 + e. The</span></span>
<span><span class="co">#&gt;           terms will appear in order as (Purcell et al. 2007) for ADD,</span></span>
<span><span class="co">#&gt;           (Su et al. 2022) for COV1, (Rhodes 2002) for ADD x COV1, and</span></span>
<span><span class="co">#&gt;           (Moreau and others 2003) for ADD x COV2. If the user wants to</span></span>
<span><span class="co">#&gt;           extract the terms for COV1 and ADD x COV1, they need to</span></span>
<span><span class="co">#&gt;           specify it as c(2,4). The default is c(“ALL”).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: This feature is not valid for the XCGA model for the</span></span>
<span><span class="co">#&gt;           XWAS part.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; combtest: Character vector specifying method for combining p-values</span></span>
<span><span class="co">#&gt;           after stratified GWAS/XWAS models. Choices are</span></span>
<span><span class="co">#&gt;           “stouffer.method”, "fisher.method" and "fisher.method.perm".</span></span>
<span><span class="co">#&gt;           For fisher.method the function for combining p-values uses a</span></span>
<span><span class="co">#&gt;           statistic, S = -2 ∑^k /log p, which follows a χ^2</span></span>
<span><span class="co">#&gt;           distribution with 2k degrees of freedom (Fisher 1925).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For fisher.method.perm, using p-values from stratified tests,</span></span>
<span><span class="co">#&gt;           the summary statistic for combining p-values is S = -2 ∑ /log</span></span>
<span><span class="co">#&gt;           p. A p-value for this statistic can be derived by randomly</span></span>
<span><span class="co">#&gt;           generating summary statistics (Rhodes 2002). Therefore, a</span></span>
<span><span class="co">#&gt;           p-value is randomly sampled from each contributing study, and</span></span>
<span><span class="co">#&gt;           a random statistic is calculated. The fraction of random</span></span>
<span><span class="co">#&gt;           statistics greater or equal to S then gives the final</span></span>
<span><span class="co">#&gt;           p-value.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For stouffer.method ,the function applies Stouffer’s method</span></span>
<span><span class="co">#&gt;           (Stouffer et al. 1949) to the p-values assuming that the</span></span>
<span><span class="co">#&gt;           p-values to be combined are independent. Letting p1, p2, . .</span></span>
<span><span class="co">#&gt;           . , pk denote the individual (one- or two-sided) p-values of</span></span>
<span><span class="co">#&gt;           the k hypothesis tests to be combined, the test statistic is</span></span>
<span><span class="co">#&gt;           then computed with $z = ∑^k_{1}frac{z_{i}}{sqrt(k)}$ where</span></span>
<span><span class="co">#&gt;           $z_{i}$ = Φ−1 (1 – $p_{i}$) and Φ −1 (·) denotes the inverse</span></span>
<span><span class="co">#&gt;           of the cumulative distribution function of a standard normal</span></span>
<span><span class="co">#&gt;           distribution. Under the joint null hypothesis, the test</span></span>
<span><span class="co">#&gt;           statistic follows a standard normal distribution which is</span></span>
<span><span class="co">#&gt;           used to compute the combined p-value. This functionality is</span></span>
<span><span class="co">#&gt;           taken from the R package poolr (Cinar and Viechtbauer 2022).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note that only p-values between 0 and 1 are allowed to be</span></span>
<span><span class="co">#&gt;           passed to these methods.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: Though this parameter is enabled for both autosome GWAS</span></span>
<span><span class="co">#&gt;           and XWAS, the combining pvalue after sex-stratified test is</span></span>
<span><span class="co">#&gt;           recommended to ChrX only.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.zero.sub: Small numeric value for substituting p-values of 0 in in</span></span>
<span><span class="co">#&gt;           stratified GWAS with FM01comb and FM02comb XWAS models. The</span></span>
<span><span class="co">#&gt;           default is 0.00001. As log(0) results in Inf this replaces</span></span>
<span><span class="co">#&gt;           p-values of 0 by default with a small float.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;        B: Integer value specifying the number of permutation in case of</span></span>
<span><span class="co">#&gt;           using fisher.method.perm method in stratified GWAS with</span></span>
<span><span class="co">#&gt;           FM01comb and FM02comb XWAS models. The default is 10000.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.mc.cores: Number of cores used for fisher.method.perm in stratified</span></span>
<span><span class="co">#&gt;           GWAS with FM01comb and FM02comb XWAS models.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.na.rm: Boolean value, 'TRUE' or 'FALSE' for removing p-values of NA</span></span>
<span><span class="co">#&gt;           in stratified GWAS with FM01comb and FM02comb XWAS in case of</span></span>
<span><span class="co">#&gt;           using Fisher’s and Stouffer’s methods. The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.p.corr: Character vector specifying method for correcting the</span></span>
<span><span class="co">#&gt;           summary p-values for FMfcomb and FMscomb models. Choices are</span></span>
<span><span class="co">#&gt;           "bonferroni", "BH" and "none" for Bonferroni,</span></span>
<span><span class="co">#&gt;           Benjamini-Hochberg and none, respectively. The default is</span></span>
<span><span class="co">#&gt;           "none".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plot.jpeg: Boolean value, 'TRUE' or 'FALSE' for saving the plots in</span></span>
<span><span class="co">#&gt;           .jpeg file. The default is TRUE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotname: A character string specifying the prefix of the file for</span></span>
<span><span class="co">#&gt;           plots. This file will be saved in DataDir. The default is</span></span>
<span><span class="co">#&gt;           "GXwas.plot".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; snp_pval: Numeric value as p-value threshold for annotation. SNPs below</span></span>
<span><span class="co">#&gt;           this p-value will be annotated on the plot. The default is</span></span>
<span><span class="co">#&gt;           1e-08.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; annotateTopSnp: Boolean value, 'TRUE' or 'FALSE. If TRUE, it only</span></span>
<span><span class="co">#&gt;           annotates the top hit on each chromosome that is below the</span></span>
<span><span class="co">#&gt;           snp_pval threshold. The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; suggestiveline: The default is 5 (for p-value 1e-05).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; genomewideline: The default is 7.3 (for p-value 5e-08).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   ncores: Integer value, specifying the number of cores for parallel</span></span>
<span><span class="co">#&gt;           processing. The default is 0 (no parallel computation).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with GWAS (with XWAS for X-chromosomal variants) along</span></span>
<span><span class="co">#&gt;      with Manhattan and Q-Q plots. In the case of the stratified test,</span></span>
<span><span class="co">#&gt;      the return is a list containing three dataframes, namely, FWAS,</span></span>
<span><span class="co">#&gt;      MWAS, and MFWAS with association results in only female, only</span></span>
<span><span class="co">#&gt;      male, and both cohorts, respectively. This will be accompanied by</span></span>
<span><span class="co">#&gt;      Miami and Q-Q plots. The individual manhattan and Q-Q-plots for</span></span>
<span><span class="co">#&gt;      stratified tests prefixed with xmodel type will be in the DataDir.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Cinar O, Viechtbauer W (2022). "The poolr package for combining</span></span>
<span><span class="co">#&gt;      independent and dependent p values." _Journal of Statistical</span></span>
<span><span class="co">#&gt;      Software_, *101*(1), 1-42. doi:10.18637/jss.v101.i01</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.18637/jss.v101.i01&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Fisher RA (1925). _Statistical Methods for Research Workers_.</span></span>
<span><span class="co">#&gt;      Oliver and Boyd, Edinburgh.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Moreau Y, others (2003). "Comparison and meta-analysis of</span></span>
<span><span class="co">#&gt;      microarray data: from the bench to the computer desk." _Trends in</span></span>
<span><span class="co">#&gt;      Genetics_, *19*(10), 570-577.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Rhodes DR (2002). "Meta-analysis of microarrays: interstudy</span></span>
<span><span class="co">#&gt;      validation of gene expression profiles reveals pathway</span></span>
<span><span class="co">#&gt;      dysregulation in prostate cancer." _Cancer Research_, *62*(15),</span></span>
<span><span class="co">#&gt;      4427-4433.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Stouffer SA, Suchman EA, DeVinney LC, Star SA, Williams RM (1949).</span></span>
<span><span class="co">#&gt;      _The American Soldier: Adjustment During Army Life_, volume 1.</span></span>
<span><span class="co">#&gt;      Princeton University Press, Princeton, NJ.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Su Y, Hu J, Yin P, Jiang H, Chen S, Dai M, Chen Z, Wang P (2022).</span></span>
<span><span class="co">#&gt;      "XCMAX4: A Robust X Chromosomal Genetic Association Test</span></span>
<span><span class="co">#&gt;      Accounting for Covariates." _Genes (Basel)_, *13*(5), 847.</span></span>
<span><span class="co">#&gt;      doi:10.3390/genes13050847 &lt;https://doi.org/10.3390/genes13050847&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/35627231&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      standard_beta &lt;- TRUE</span></span>
<span><span class="co">#&gt;      xsex &lt;- FALSE</span></span>
<span><span class="co">#&gt;      sex &lt;- TRUE</span></span>
<span><span class="co">#&gt;      Inphenocov &lt;- NULL</span></span>
<span><span class="co">#&gt;      covartest &lt;- NULL</span></span>
<span><span class="co">#&gt;      interaction &lt;- FALSE</span></span>
<span><span class="co">#&gt;      MF.na.rm &lt;- FALSE</span></span>
<span><span class="co">#&gt;      B &lt;- 10000</span></span>
<span><span class="co">#&gt;      MF.zero.sub &lt;- 0.00001</span></span>
<span><span class="co">#&gt;      trait &lt;- "binary"</span></span>
<span><span class="co">#&gt;      xmodel &lt;- "FMcombx02"</span></span>
<span><span class="co">#&gt;      combtest &lt;- "fisher.method"</span></span>
<span><span class="co">#&gt;      snp_pval &lt;- 1e-08</span></span>
<span><span class="co">#&gt;      covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      ncores &lt;- 0</span></span>
<span><span class="co">#&gt;      MF.mc.cores &lt;- 1</span></span>
<span><span class="co">#&gt;      ResultGXwas &lt;- GXwas(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, xmodel = xmodel, trait = trait, covarfile = covarfile,</span></span>
<span><span class="co">#&gt;          sex = sex, xsex = xsex, combtest = combtest, MF.p.corr = "none",</span></span>
<span><span class="co">#&gt;          snp_pval = snp_pval, plot.jpeg = TRUE, suggestiveline = 5, genomewideline = 7.3,</span></span>
<span><span class="co">#&gt;          MF.mc.cores = 1, ncores = ncores</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-9">Function 9<a class="anchor" aria-label="anchor" href="#function-9"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">PvalComb</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; PvalComb</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function combines the p-values of two separate GWAS summary</span></span>
<span><span class="co">#&gt;      statistics (for instance male and female populations), merges</span></span>
<span><span class="co">#&gt;      them, and then applies various statistical methods (like</span></span>
<span><span class="co">#&gt;      Stouffer's method, Fisher's method) to integrate the p-values. It</span></span>
<span><span class="co">#&gt;      also includes functionality for generating plots like Manhattan</span></span>
<span><span class="co">#&gt;      plots and Q-Q plots.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      PvalComb(</span></span>
<span><span class="co">#&gt;        SumstatMale,</span></span>
<span><span class="co">#&gt;        SumstatFemale,</span></span>
<span><span class="co">#&gt;        combtest,</span></span>
<span><span class="co">#&gt;        MF.p.corr = "none",</span></span>
<span><span class="co">#&gt;        MF.zero.sub = 1e-05,</span></span>
<span><span class="co">#&gt;        MF.na.rm = TRUE,</span></span>
<span><span class="co">#&gt;        MF.mc.cores = 1,</span></span>
<span><span class="co">#&gt;        B = 1000,</span></span>
<span><span class="co">#&gt;        plot.jpeg = TRUE,</span></span>
<span><span class="co">#&gt;        plotname = "GXwas.plot",</span></span>
<span><span class="co">#&gt;        PlotDir = tempdir(),</span></span>
<span><span class="co">#&gt;        snp_pval,</span></span>
<span><span class="co">#&gt;        annotateTopSnp = FALSE,</span></span>
<span><span class="co">#&gt;        suggestiveline = 5,</span></span>
<span><span class="co">#&gt;        genomewideline = 7.3,</span></span>
<span><span class="co">#&gt;        ncores = 0</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SumstatMale: R dataframe object of summary statistics of male GWAS with</span></span>
<span><span class="co">#&gt;           five mandatory columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'CHR' (numeric chromosome code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'SNP' (variant id)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'A1' (allele)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'POS' (base-pair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'P' (p-value).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Other columns may present.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SumstatFemale: R dataframe object of summary statistics of female GWAS</span></span>
<span><span class="co">#&gt;           with five mandatory columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'SNP'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'A1'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'TEST'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'POS'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'P'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Other columns may present.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; combtest: Character vector specifying method for combining p-values for</span></span>
<span><span class="co">#&gt;           stratified GWAS models. Choices are “stouffer.method”,</span></span>
<span><span class="co">#&gt;           "fisher.method" and "fisher.method.perm". For fisher.method</span></span>
<span><span class="co">#&gt;           the function for combining p-values uses a statistic, S = -2</span></span>
<span><span class="co">#&gt;           ∑^k /log p, which follows a χ^2 distribution with 2k degrees</span></span>
<span><span class="co">#&gt;           of freedom (Fisher 1925). For fisher.method.perm, using</span></span>
<span><span class="co">#&gt;           p-values from stratified tests, the summary statistic for</span></span>
<span><span class="co">#&gt;           combining p-values is S = -2 ∑ /log p. A p-value for this</span></span>
<span><span class="co">#&gt;           statistic can be derived by randomly generating summary</span></span>
<span><span class="co">#&gt;           statistics (Rhodes 2002). Therefore, a p-value is randomly</span></span>
<span><span class="co">#&gt;           sampled from each contributing study, and a random statistic</span></span>
<span><span class="co">#&gt;           is calculated. The fraction of random statistics greater or</span></span>
<span><span class="co">#&gt;           equal to S then gives the final p-value.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.p.corr: Character vector specifying method for correcting the</span></span>
<span><span class="co">#&gt;           summary p-values for FMfcomb and FMscomb models. Choices are</span></span>
<span><span class="co">#&gt;           "bonferroni", "BH" and "none" for Bonferroni,</span></span>
<span><span class="co">#&gt;           Benjamini-Hochberg and none, respectively. The default is</span></span>
<span><span class="co">#&gt;           "none".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.zero.sub: Small numeric value for substituting p-values of 0 in GWAS</span></span>
<span><span class="co">#&gt;           summary statistics. The default is 0.00001. As log(0) results</span></span>
<span><span class="co">#&gt;           in Inf this replaces p-values of 0 by default with a small</span></span>
<span><span class="co">#&gt;           float.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.na.rm: Boolean value, 'TRUE' or 'FALSE' for removing p-values of NA</span></span>
<span><span class="co">#&gt;           in stratified GWAS summary satistics in case of using</span></span>
<span><span class="co">#&gt;           Fisher’s and Stouffer’s methods. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; MF.mc.cores: Number of cores used for fisher.method.perm for combining</span></span>
<span><span class="co">#&gt;           p-values. The default is 1.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;        B: Integer value specifying the number of permutation in case of</span></span>
<span><span class="co">#&gt;           using fisher.method.perm method. The default is 10000.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plot.jpeg: Boolean value, 'TRUE' or 'FALSE' for saving the plots in</span></span>
<span><span class="co">#&gt;           .jpeg file. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotname: A character string specifying the prefix of the file for</span></span>
<span><span class="co">#&gt;           plots. This file will be saved in DataDir. The default is</span></span>
<span><span class="co">#&gt;           "GXwas.plot".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  PlotDir: A character string specifying the path of the directory where</span></span>
<span><span class="co">#&gt;           the plots will be saved. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; snp_pval: Numeric value as p-value threshold for annotation. SNPs below</span></span>
<span><span class="co">#&gt;           this p-value will be annotated on the plot. The default is</span></span>
<span><span class="co">#&gt;           1e-08.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; annotateTopSnp: Boolean value, 'TRUE' or 'FALSE.' If 'TRUE', it only</span></span>
<span><span class="co">#&gt;           annotates the top hit on each chromosome that is below the</span></span>
<span><span class="co">#&gt;           snp_pval threshold. The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; suggestiveline: Numeric value for suggestive cut-off line in GWAS</span></span>
<span><span class="co">#&gt;           manhattan plot. The default is 5 (for p-value 1e-05).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; genomewideline: Numeric value for genome-wide significant cut-off line</span></span>
<span><span class="co">#&gt;           in GWAS manhattan plot. The default is 7.3 (for p-value</span></span>
<span><span class="co">#&gt;           5e-08).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   ncores: Integer value, specifying the number of cores for parallel</span></span>
<span><span class="co">#&gt;           processing. The default is 0 (no parallel computation).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with GWAS summary statistics (with XWAS for</span></span>
<span><span class="co">#&gt;      X-chromosomal variants) along with Manhattan and Q-Q plots.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Fisher RA (1925). _Statistical Methods for Research Workers_.</span></span>
<span><span class="co">#&gt;      Oliver and Boyd, Edinburgh.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Rhodes DR (2002). "Meta-analysis of microarrays: interstudy</span></span>
<span><span class="co">#&gt;      validation of gene expression profiles reveals pathway</span></span>
<span><span class="co">#&gt;      dysregulation in prostate cancer." _Cancer Research_, *62*(15),</span></span>
<span><span class="co">#&gt;      4427-4433.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Mfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Ffile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      SumstatMale &lt;- Mfile</span></span>
<span><span class="co">#&gt;      colnames(SumstatMale)[3] &lt;- "POS"</span></span>
<span><span class="co">#&gt;      SumstatFemale &lt;- Ffile</span></span>
<span><span class="co">#&gt;      colnames(SumstatFemale)[3] &lt;- "POS"</span></span>
<span><span class="co">#&gt;      PvalComb_Result &lt;- PvalComb(</span></span>
<span><span class="co">#&gt;          SumstatMale = SumstatMale, SumstatFemale = SumstatFemale,</span></span>
<span><span class="co">#&gt;          combtest = "fisher.method", MF.mc.cores = 1, snp_pval = 0.001, plot.jpeg = FALSE,</span></span>
<span><span class="co">#&gt;          suggestiveline = 3, genomewideline = 5.69897, ncores = 1</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-10">Function 10<a class="anchor" aria-label="anchor" href="#function-10"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiff</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexDiff: Sex difference in effect size for each SNP using t-test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function uses the GWAS summary statistics from sex-stratified</span></span>
<span><span class="co">#&gt;      tests like "FMstratified", to evaluate the difference in effect</span></span>
<span><span class="co">#&gt;      size between males and females at each SNP using a t-test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The input dataframes should only include X-chromosome in order to</span></span>
<span><span class="co">#&gt;      obtain results for sex differences based solely on X-linked loci.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexDiff(Mfile, Ffile)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Mfile: R dataframe of summary statistics of GWAS or XWAS of male</span></span>
<span><span class="co">#&gt;           samples with six mandatory columns,</span></span>
<span><span class="co">#&gt;           SNP(Variant),CHR(Chromosome number), BP(Base pair</span></span>
<span><span class="co">#&gt;           position),A1(Minor allele),BETA_M(Effect size) and</span></span>
<span><span class="co">#&gt;           SE_M(Standard error). This can be generated by running</span></span>
<span><span class="co">#&gt;           FM01comb or "FMstratified" model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Ffile: R dataframe of summary statistics of GWAS or XWAS of male</span></span>
<span><span class="co">#&gt;           samples with six mandatory columns,</span></span>
<span><span class="co">#&gt;           SNP(Variant),CHR(Chromosome number), BP(Base pair</span></span>
<span><span class="co">#&gt;           position),A1(Minor allele),BETA_F(Effect size) and</span></span>
<span><span class="co">#&gt;           SE_F(Standard error). This can be generated by running</span></span>
<span><span class="co">#&gt;           FM01comb or "FMstratified" model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      R dataframe with seven columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SNP' (Variant)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'CHR' (Chromosome number)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'BP' (Base pair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'A1' (Minor allele)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'tstat' (t-statistics for effect-size test)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P' (p-value) and</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'adjP' (Bonferroni corrected p-value)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Mfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Ffile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      Difftest &lt;- SexDiff(Mfile, Ffile)</span></span>
<span><span class="co">#&gt;      significant_snps &lt;- Difftest[Difftest$adjP &lt; 0.05, ]</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-11">Function 11<a class="anchor" aria-label="anchor" href="#function-11"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiffZscore</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexDiffZscore: Z-score-based sex difference test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function calculates the difference in any kind of measured</span></span>
<span><span class="co">#&gt;      entities,(example: including SNP heritability estimate, genetic</span></span>
<span><span class="co">#&gt;      correlation, and GWAS β values) between sexes using a Z-score and</span></span>
<span><span class="co">#&gt;      its associated p-value statistic. When STAT/SE is normally</span></span>
<span><span class="co">#&gt;      distributed and the test statistics are independent in sex, the</span></span>
<span><span class="co">#&gt;      test is well calibrated. If the statistics are positively</span></span>
<span><span class="co">#&gt;      correlated, this test is conservative (1).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      We could define SNPs with SDEs as those variants at the extreme</span></span>
<span><span class="co">#&gt;      ends of the distribution with an absolute value of the Z-score</span></span>
<span><span class="co">#&gt;      greater than 3(|Z-score| &gt; 3), which is roughly equivalent to p</span></span>
<span><span class="co">#&gt;      &lt;10−3, and represents 0.3% of all tested SNPs. The input</span></span>
<span><span class="co">#&gt;      dataframes should only include X-chromosome in order to obtain</span></span>
<span><span class="co">#&gt;      results for sex differences based solely on X-linked loci.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexDiffZscore(inputdata)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; inputdata: A dataframe with five columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'ID' (i.e., SNP ID or the phenotype of interest, etc.)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Fstat' (i.e., the measured statistics in females)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Fse' (i.e., the standard error of the measured</span></span>
<span><span class="co">#&gt;               statistics in females)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Mstat' (i.e., the measured statistics in males)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Mse' (i.e., the standard error of the measured</span></span>
<span><span class="co">#&gt;               statistics in males)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Original input dataframe with:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'Zscore' (i.e., Z-score),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'p' (i.e., p-value) and</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'adjP' (i.e., Bonferroni corrected p-value) columns added.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Example_h2data", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      inputdata &lt;- Example_h2data</span></span>
<span><span class="co">#&gt;      x &lt;- SexDiffZscore(inputdata)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-12">Function 12<a class="anchor" aria-label="anchor" href="#function-12"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">DiffZeroOne</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; DiffZeroOne: Assessing the Z-score for deviation from one and zero.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function tests the null hypothesis that a measured statistics</span></span>
<span><span class="co">#&gt;      (example: genetic correlation, rg for a trait) &lt; 1 using a</span></span>
<span><span class="co">#&gt;      1-tailed test compared with a normal distribution (z = (1 −</span></span>
<span><span class="co">#&gt;      measure statistics)/Standard error). For multiple tests, users are</span></span>
<span><span class="co">#&gt;      encouraged to apply a Bonferroni multiple-testing correction.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DiffZeroOne(inputdata, diffzero = TRUE, diffone = TRUE)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; inputdata: A dataframe object, containing three columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Trait' (i.e., the phenotype of interest)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'Stat' (i.e., the measured statistics)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'SE' (i.e., the standard error of the measured</span></span>
<span><span class="co">#&gt;               statistics)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; diffzero: Boolean value, 'TRUE' or 'FALSE', specifying to perform</span></span>
<span><span class="co">#&gt;           deviation from 0 test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  diffone: Boolean value, 'TRUE' or 'FALSE', specifying to perform</span></span>
<span><span class="co">#&gt;           deviation from 1 test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'Trait'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'Stat'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SE'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P0' (i.e, p-value for deviation from zero test)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P1' (i.e., p-value for deviation from 1 test)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Example_rgdata", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      inputdata &lt;- Example_rgdata</span></span>
<span><span class="co">#&gt;      colnames(inputdata) &lt;- c("Trait", "Stat", "SE")</span></span>
<span><span class="co">#&gt;      x &lt;- DiffZeroOne(inputdata, FALSE, TRUE)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-13">Function 13<a class="anchor" aria-label="anchor" href="#function-13"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">TestXGene</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; TestXGene: Performing gene-based association test using GWAS/XWAS</span></span>
<span><span class="co">#&gt; summary statistics.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function performs gene-based association tests using</span></span>
<span><span class="co">#&gt;      GWAS/XWAS summary statistics and SNP-SNP correlation matrices. For</span></span>
<span><span class="co">#&gt;      SNP-SNP correlation matrices, users have the flexibility to use</span></span>
<span><span class="co">#&gt;      either the base genotype data or 1000 Genomes Phase 3 reference</span></span>
<span><span class="co">#&gt;      genotype data. Users also have options to define the regional</span></span>
<span><span class="co">#&gt;      positions of genes to include the SNPs according to their</span></span>
<span><span class="co">#&gt;      investigation.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function computes gene-wise SNP-SNP correlation matrices and</span></span>
<span><span class="co">#&gt;      can perform nine different gene-based tests, such as, “BT" (burden</span></span>
<span><span class="co">#&gt;      test), "SKAT" (sequence kernel association test), "SKATO"</span></span>
<span><span class="co">#&gt;      (combination of BT and SKAT), "sumchi" (sum of χ2-statistics),</span></span>
<span><span class="co">#&gt;      "ACAT" (aggregated Cauchy association test for combining P</span></span>
<span><span class="co">#&gt;      values), "PCA"(principal component approach), "FLM"( functional</span></span>
<span><span class="co">#&gt;      multiple linear regression model), "simpleM" (Bonferroni</span></span>
<span><span class="co">#&gt;      correction test), "minp" (minimum P-value) leveraging PLINK1.9</span></span>
<span><span class="co">#&gt;      (Purcell et al. 2007) and sumFREGAT (Svishcheva et al. 2019;</span></span>
<span><span class="co">#&gt;      Belonogova et al. 2022) tools.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Though this function implicitly performs X-linked gene-based test,</span></span>
<span><span class="co">#&gt;      it is flexible to perform this analysis genome-wide. For the</span></span>
<span><span class="co">#&gt;      details about the different tests, please follow the associated</span></span>
<span><span class="co">#&gt;      paper.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      TestXGene(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        sumstat,</span></span>
<span><span class="co">#&gt;        gene_file,</span></span>
<span><span class="co">#&gt;        gene_range = 5e+05,</span></span>
<span><span class="co">#&gt;        score_file,</span></span>
<span><span class="co">#&gt;        ref_data = NULL,</span></span>
<span><span class="co">#&gt;        max_gene = NULL,</span></span>
<span><span class="co">#&gt;        sample_size = NULL,</span></span>
<span><span class="co">#&gt;        genebasedTest = c("SKAT", "SKATO", "sumchi", "ACAT", "BT", "PCA", "FLM", "simpleM",</span></span>
<span><span class="co">#&gt;          "minp"),</span></span>
<span><span class="co">#&gt;        gene_approximation = TRUE,</span></span>
<span><span class="co">#&gt;        beta_par,</span></span>
<span><span class="co">#&gt;        weights_function,</span></span>
<span><span class="co">#&gt;        geno_variance_weights,</span></span>
<span><span class="co">#&gt;        kernel_p_method = "kuonen",</span></span>
<span><span class="co">#&gt;        acc_devies = 1e-08,</span></span>
<span><span class="co">#&gt;        lim_devies = 1e+06,</span></span>
<span><span class="co">#&gt;        rho = TRUE,</span></span>
<span><span class="co">#&gt;        skato_p_threshold = 0.8,</span></span>
<span><span class="co">#&gt;        anno_type = "",</span></span>
<span><span class="co">#&gt;        mac_threshold,</span></span>
<span><span class="co">#&gt;        reference_matrix_used,</span></span>
<span><span class="co">#&gt;        regularize_fun,</span></span>
<span><span class="co">#&gt;        pca_var_fraction = 0.85,</span></span>
<span><span class="co">#&gt;        flm_basis_function = "fourier",</span></span>
<span><span class="co">#&gt;        flm_num_basis = 25,</span></span>
<span><span class="co">#&gt;        flm_poly_order = 4,</span></span>
<span><span class="co">#&gt;        flip_genotypes = FALSE,</span></span>
<span><span class="co">#&gt;        omit_linear_variant = FALSE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the all the input</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the genotype data. This file is used to</span></span>
<span><span class="co">#&gt;           compute the correlation between the SNPs. This file needs to</span></span>
<span><span class="co">#&gt;           be in DataDir. If the base genotype data is unavailable, then</span></span>
<span><span class="co">#&gt;           users can use the 1000 Genomes Project samples. Users should</span></span>
<span><span class="co">#&gt;           use the population that most closely represents the base</span></span>
<span><span class="co">#&gt;           sample. For ACAT model, this parameter is not mandatory and</span></span>
<span><span class="co">#&gt;           could be set 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  sumstat: A dataframe object with GWAS summary statistics. When the</span></span>
<span><span class="co">#&gt;           base-genotype data is used to compute genetic correlations,</span></span>
<span><span class="co">#&gt;           the mandatory columns are:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 1: 'CHROM' (i.e., chromosome code),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 2: 'POS' (i.e., base-pair position),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 3: 'ID' (i.e. SNP IDs),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 4: 'P' (i.e., p-values),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 5: 'BETA' (i.e., effect-size),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 6: 'A1' (i.e., effect allele),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 7: 'A2' (i.e., alternative allele) and</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • Column 8: 'EAF' (i.e., the effect allele frequency)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           These are mandatory when base-genotype data is used to</span></span>
<span><span class="co">#&gt;           compute genetic correlations. Otherwise, if the users are</span></span>
<span><span class="co">#&gt;           using reference data, then columns 5 to 8 are optional. Also,</span></span>
<span><span class="co">#&gt;           in that case, columns, such as 'REF' (i.e., reference</span></span>
<span><span class="co">#&gt;           allele), and 'ALT' (i.e., alternative allele) could be</span></span>
<span><span class="co">#&gt;           present to compare alleles with those in the reference file</span></span>
<span><span class="co">#&gt;           and exclude genetic variants if alleles do not match. There</span></span>
<span><span class="co">#&gt;           could be an additional column, 'ANNO' with functional</span></span>
<span><span class="co">#&gt;           annotations (like "intron_variant", "synonymous", "missense"</span></span>
<span><span class="co">#&gt;           etc.)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; gene_file: Character string, specifying the prefix of the name of a</span></span>
<span><span class="co">#&gt;           .txt file listing genes in refFlat format. This file needs to</span></span>
<span><span class="co">#&gt;           be in 'DataDir'. The X-linked gene files,</span></span>
<span><span class="co">#&gt;           "Xlinkedgenes_hg19.txt" and "Xlinkedgenes_hg38.txt" and</span></span>
<span><span class="co">#&gt;           autosomal gene files, “Autosomes_hg19.txt” and</span></span>
<span><span class="co">#&gt;           “Autosomes_hg38.txt” can be specified. The default is</span></span>
<span><span class="co">#&gt;           "Xlinkedgenes_hg19.txt". The genome built should be in</span></span>
<span><span class="co">#&gt;           agreement with the analysis.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; gene_range: Integer value, specifying the up stream and down stream</span></span>
<span><span class="co">#&gt;           range (in kilo base) of a gene for SNPs to be considered. The</span></span>
<span><span class="co">#&gt;           default is 500000.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; score_file: Character string, specifying the prefix of a file which</span></span>
<span><span class="co">#&gt;           will be used to produce score files with Z scores from P</span></span>
<span><span class="co">#&gt;           values and beta input from GWAS summary statistics.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ref_data: Character string, specifying the path to a reference</span></span>
<span><span class="co">#&gt;           dataframe with additional data needed to recode user data</span></span>
<span><span class="co">#&gt;           according to correlation matrices that will be used. It</span></span>
<span><span class="co">#&gt;           contains 'ID' column with names of SNPs, 'REF' and 'ALT'</span></span>
<span><span class="co">#&gt;           columns with alleles that were coded as 0 and 1,</span></span>
<span><span class="co">#&gt;           respectively. Effect sizes from data will be inverted for</span></span>
<span><span class="co">#&gt;           variants with effect alleles different from 'ALT' alleles in</span></span>
<span><span class="co">#&gt;           reference data. If presented, 'REF' and 'ALT' columns from</span></span>
<span><span class="co">#&gt;           the input data will be used to sort out variants with alleles</span></span>
<span><span class="co">#&gt;           different from those in reference data. This dataframe can</span></span>
<span><span class="co">#&gt;           also be a source of map data and allele frequencies if they</span></span>
<span><span class="co">#&gt;           are not present in data. 'AF' column in the reference file</span></span>
<span><span class="co">#&gt;           represents the allele frequency of 'ALT' allele. The default</span></span>
<span><span class="co">#&gt;           is "ref1KG.MAC5.EUR_AF.RData".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; max_gene: Positive integer value, specifying the number of genes for</span></span>
<span><span class="co">#&gt;           which the gene-based test will be performed. The default is</span></span>
<span><span class="co">#&gt;           NULL to consider all the genes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; sample_size: Positive integer value, specifying the sample size of the</span></span>
<span><span class="co">#&gt;           GWAS. Only needed for FLM and PCA models.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; genebasedTest: Character string, specifying the name of the gene-based</span></span>
<span><span class="co">#&gt;           test. Nine different tests can be specified,</span></span>
<span><span class="co">#&gt;           "SKAT","SKATO","sumchi","ACAT","BT","PCA",</span></span>
<span><span class="co">#&gt;           "FLM","simpleM","minp". The default is "SKAT".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; gene_approximation: Boolean value, 'TRUE' or 'FALSE', specifying</span></span>
<span><span class="co">#&gt;           whether approximation for large genes (&gt;= 500 SNPs) should be</span></span>
<span><span class="co">#&gt;           used. Applicable for SKAT, SKATO, sumchi, PCA, FLM. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE' for these methods).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; beta_par: Boolean value, 'TRUE' or 'FALSE', specifying whether</span></span>
<span><span class="co">#&gt;           approximation for large genes (&gt;= 500 SNPs) should be used.</span></span>
<span><span class="co">#&gt;           Applicable for SKAT, SKATO, sumchi, PCA, FLM (default =</span></span>
<span><span class="co">#&gt;           'TRUE' for these methods).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; weights_function: A function of MAF to assign weights for each genetic</span></span>
<span><span class="co">#&gt;           variant. By default is 'NULL'. In this case the weights will</span></span>
<span><span class="co">#&gt;           be calculated using the beta distribution.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; geno_variance_weights: Character string, indicating whether scores</span></span>
<span><span class="co">#&gt;           should be weighted by the variance of genotypes: "none"</span></span>
<span><span class="co">#&gt;           (i.e., no weights applied, resulting in a sum chi-square</span></span>
<span><span class="co">#&gt;           test); "se.beta" (i.e., scores weighted by variance of</span></span>
<span><span class="co">#&gt;           genotypes estimated from P values and effect sizes); "af"</span></span>
<span><span class="co">#&gt;           (i.e., scores weighted by variance of genotypes calculated as</span></span>
<span><span class="co">#&gt;           AF * (1 - AF), where AF is allele frequency.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; kernel_p_method: Character string, specifying the method for computing</span></span>
<span><span class="co">#&gt;           P value in kernel-based tests, such as SKAT, SKATO and</span></span>
<span><span class="co">#&gt;           sumchi. Available methods are "kuonen" (Belonogova et al.</span></span>
<span><span class="co">#&gt;           2022) "davies" (Belonogova et al. 2022) and "hybrid"</span></span>
<span><span class="co">#&gt;           (Belonogova et al. 2022). The default is "kuonen".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; acc_devies: Positive numeric value, specifying the accuracy parameter</span></span>
<span><span class="co">#&gt;           for "davies" method. The default is 1e-8.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; lim_devies: Positive numeric value, specifying the limit parameter for</span></span>
<span><span class="co">#&gt;           "davies" method. The default is 1e+6.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      rho: Logical value, 'TRUE' or 'FALSE' or can be a vector of grid</span></span>
<span><span class="co">#&gt;           values from 0 to 1. If TRUE, the optimal test (SKAT-O) is</span></span>
<span><span class="co">#&gt;           performed (12). The default grid is c(0, 0.1^2, 0.2^2, 0.3^2,</span></span>
<span><span class="co">#&gt;           0.4^2, 0.5^2, 0.5, 1).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; skato_p_threshold: Positive numeric value, specifying the largest P</span></span>
<span><span class="co">#&gt;           value that will be considered as important when performing</span></span>
<span><span class="co">#&gt;           computational optimization in SKAT-O. All P values larger</span></span>
<span><span class="co">#&gt;           than skato_p_threshold will be processed via burden test. The</span></span>
<span><span class="co">#&gt;           default is 0.8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; anno_type: A character (or character vector) indicating annotation</span></span>
<span><span class="co">#&gt;           types to be used. The default is "" (i.e, nothing).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; mac_threshold: Integer value, specifying the threshold of MACs (Minor</span></span>
<span><span class="co">#&gt;           allele content) calculated from MAFs. In ACAT, scores with</span></span>
<span><span class="co">#&gt;           MAC &lt;= 10 will be combined using Burden test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; reference_matrix_used: Boolean value, 'TRUE' or 'FALSE' logical</span></span>
<span><span class="co">#&gt;           indicating whether the correlation matrices were generated</span></span>
<span><span class="co">#&gt;           using the reference matrix. The default is 'FALSE'. If</span></span>
<span><span class="co">#&gt;           'TRUE', regularization algorithms will be applied to ensure</span></span>
<span><span class="co">#&gt;           the invertibility and numerical stability of the matrices.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; regularize_fun: Character string, specifying the one of two</span></span>
<span><span class="co">#&gt;           regularization algorithms if ‘reference_matrix’ is TRUE: 'LH'</span></span>
<span><span class="co">#&gt;           (default) or 'derivLH'. Currently, both give similar results.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pca_var_fraction: P ositive numeric value, specifying the minimal</span></span>
<span><span class="co">#&gt;           proportion of genetic variance within the region that should</span></span>
<span><span class="co">#&gt;           be explained by principal components used in PCA method. This</span></span>
<span><span class="co">#&gt;           is also valid in 'simpleM'. The default is 0.85.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; flm_basis_function: Character string, specifying the name of a basis</span></span>
<span><span class="co">#&gt;           function type for beta-smooth in FLM method. Can be set to</span></span>
<span><span class="co">#&gt;           "bspline" (B-spline basis) or "fourier" (Fourier basis,</span></span>
<span><span class="co">#&gt;           default).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; flm_num_basis: Positive integer value, specifying the number of basis</span></span>
<span><span class="co">#&gt;           functions to be used for beta-smooth in FLM method. The</span></span>
<span><span class="co">#&gt;           default is 25.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; flm_poly_order: Positive integer value, specifying the polynomial order</span></span>
<span><span class="co">#&gt;           to be used in "bspline" for FLM model. The default = 4, which</span></span>
<span><span class="co">#&gt;           corresponds to the cubic B-splines. This has no effect if</span></span>
<span><span class="co">#&gt;           only Fourier bases are used</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; flip_genotypes: L ogical value, 'TRUE' or 'FALSE', indicating whether</span></span>
<span><span class="co">#&gt;           the genotypes of some genetic variants should be flipped</span></span>
<span><span class="co">#&gt;           (relabelled) for their better functional representation (13).</span></span>
<span><span class="co">#&gt;           The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; omit_linear_variant: Logical value, 'TRUE' or 'FALSE', indicating</span></span>
<span><span class="co">#&gt;           whether to omit linearly dependent genetic variants. It was</span></span>
<span><span class="co">#&gt;           done in the FLM test (4). The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A data frame with columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • gene</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • chrom</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • start</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • end</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • markers (i.e., numbers of SNPs),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • filtered.markers (i.e. filtered SNPs)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • pvalue (i.e., p-value).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Additionally, for “BT”, there will be “beta” (i.e., gene-level</span></span>
<span><span class="co">#&gt;      estimates of betas) and “beta.se” (i.e., standard errors of</span></span>
<span><span class="co">#&gt;      betas). For “FLM”, there will be the “model” column with the names</span></span>
<span><span class="co">#&gt;      of the functional models used for each region. Names shortly</span></span>
<span><span class="co">#&gt;      describe the functional basis and the number of basis functions</span></span>
<span><span class="co">#&gt;      used. E.g., "F25" means 25 Fourier basis functions, "B15" means 15</span></span>
<span><span class="co">#&gt;      B-spline basis functions. For “PCA”, there will be the</span></span>
<span><span class="co">#&gt;      “ncomponents” (the number of the principal components used for</span></span>
<span><span class="co">#&gt;      each region) and “explained.variance.fraction” (i.e., the</span></span>
<span><span class="co">#&gt;      proportion of genetic variance they make up) columns.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Belonogova NM, Svishcheva GR, Kirichenko AV, Zorkoltseva IV,</span></span>
<span><span class="co">#&gt;      Tsepilov YA, Axenovich TI (2022). "sumSTAAR: A flexible framework</span></span>
<span><span class="co">#&gt;      for gene-based association studies using GWAS summary statistics."</span></span>
<span><span class="co">#&gt;      _PLoS Computational Biology_, *18*(6), e1010172.</span></span>
<span><span class="co">#&gt;      doi:10.1371/journal.pcbi.1010172</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1371/journal.pcbi.1010172&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/35653402&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Svishcheva GR, Belonogova NM, Zorkoltseva IV, Kirichenko AV,</span></span>
<span><span class="co">#&gt;      Axenovich TI (2019). "Gene-based association tests using GWAS</span></span>
<span><span class="co">#&gt;      summary statistics." _Bioinformatics_, *35*(19), 3701-3708.</span></span>
<span><span class="co">#&gt;      doi:10.1093/bioinformatics/btz172</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1093/bioinformatics/btz172&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/30860568&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      if (!(Sys.getenv("CI") == "true" &amp;&amp; Sys.info()[["sysname"]] == "Darwin")) {</span></span>
<span><span class="co">#&gt;          data("XWAS_Summary_Example", package = "GXwasR")</span></span>
<span><span class="co">#&gt;          DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;          ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;          finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;          sumstat &lt;- XWAS_Summary_Example</span></span>
<span><span class="co">#&gt;          ref_data &lt;- NULL</span></span>
<span><span class="co">#&gt;          gene_file &lt;- "Xlinkedgenes_hg19.txt"</span></span>
<span><span class="co">#&gt;          gene_range &lt;- 500000</span></span>
<span><span class="co">#&gt;          max_gene &lt;- 10</span></span>
<span><span class="co">#&gt;          gene_approximation &lt;- TRUE</span></span>
<span><span class="co">#&gt;          beta_par &lt;- c(1, 25)</span></span>
<span><span class="co">#&gt;          weights_function &lt;- NULL</span></span>
<span><span class="co">#&gt;          geno_variance_weights &lt;- "se.beta"</span></span>
<span><span class="co">#&gt;          method &lt;- "kuonen"</span></span>
<span><span class="co">#&gt;          acc_devies &lt;- 1e-8</span></span>
<span><span class="co">#&gt;          lim_devies &lt;- 1e+6</span></span>
<span><span class="co">#&gt;          rho &lt;- TRUE</span></span>
<span><span class="co">#&gt;          skato_p_threshold &lt;- 0.8</span></span>
<span><span class="co">#&gt;          mac_threshold &lt;- 3</span></span>
<span><span class="co">#&gt;          sample_size &lt;- 4000</span></span>
<span><span class="co">#&gt;          reference_matrix_used &lt;- FALSE</span></span>
<span><span class="co">#&gt;          regularize_fun &lt;- "LH"</span></span>
<span><span class="co">#&gt;          pca_var_fraction &lt;- 0.85</span></span>
<span><span class="co">#&gt;          flm_basis_function &lt;- "fourier"</span></span>
<span><span class="co">#&gt;          flm_num_basis &lt;- 25</span></span>
<span><span class="co">#&gt;          flm_poly_order &lt;- 4</span></span>
<span><span class="co">#&gt;          flip_genotypes &lt;- FALSE</span></span>
<span><span class="co">#&gt;          omit_linear_variant &lt;- FALSE</span></span>
<span><span class="co">#&gt;          kernel_p_method &lt;- "kuonen"</span></span>
<span><span class="co">#&gt;          anno_type &lt;- ""</span></span>
<span><span class="co">#&gt;          GenetestResult &lt;- TestXGene(DataDir, ResultDir, finput, sumstat, gene_file,</span></span>
<span><span class="co">#&gt;              gene_range, score_file, ref_data, max_gene, sample_size,</span></span>
<span><span class="co">#&gt;              genebasedTest = "SKAT",</span></span>
<span><span class="co">#&gt;              gene_approximation, beta_par, weights_function, geno_variance_weights,</span></span>
<span><span class="co">#&gt;              kernel_p_method, acc_devies, lim_devies, rho, skato_p_threshold, anno_type,</span></span>
<span><span class="co">#&gt;              mac_threshold, reference_matrix_used, regularize_fun, pca_var_fraction,</span></span>
<span><span class="co">#&gt;              flm_basis_function, flm_num_basis, flm_poly_order, flip_genotypes,</span></span>
<span><span class="co">#&gt;              omit_linear_variant</span></span>
<span><span class="co">#&gt;          )</span></span>
<span><span class="co">#&gt;      }</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-14">Function 14<a class="anchor" aria-label="anchor" href="#function-14"></a>
</h4>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MetaGWAS</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; MetaGWAS: Combining summary-level results from two or more GWA studies</span></span>
<span><span class="co">#&gt; into a single estimate.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function combine K sets of GWAS association statistics on</span></span>
<span><span class="co">#&gt;      same (or at least similar) phenotype. This function employs</span></span>
<span><span class="co">#&gt;      PLINK's (Purcell et al. 2007) inverse variance-based analysis to</span></span>
<span><span class="co">#&gt;      run a number of models, including a) Fixed-effect model and b)</span></span>
<span><span class="co">#&gt;      Random-effect model, assuming there may be variation between the</span></span>
<span><span class="co">#&gt;      genuine underlying effects, i.e., effect size beta. 'This function</span></span>
<span><span class="co">#&gt;      also calculates weighted Z-score-based p-values after METAL</span></span>
<span><span class="co">#&gt;      (Willer et al. 2010). For more information about the algorithms,</span></span>
<span><span class="co">#&gt;      please see the associated paper.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      MetaGWAS(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        SummData = c(""),</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        SNPfile = NULL,</span></span>
<span><span class="co">#&gt;        useSNPposition = TRUE,</span></span>
<span><span class="co">#&gt;        UseA1 = FALSE,</span></span>
<span><span class="co">#&gt;        GCse = TRUE,</span></span>
<span><span class="co">#&gt;        plotname = "Meta_Analysis.plot",</span></span>
<span><span class="co">#&gt;        pval_filter = "R",</span></span>
<span><span class="co">#&gt;        top_snp_pval = 1e-08,</span></span>
<span><span class="co">#&gt;        max_top_snps = 6,</span></span>
<span><span class="co">#&gt;        chosen_snps_file = NULL,</span></span>
<span><span class="co">#&gt;        byCHR = FALSE,</span></span>
<span><span class="co">#&gt;        pval_threshold_manplot = 1e-05</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input files</span></span>
<span><span class="co">#&gt;           needed for 'SummData' and 'SNPfile' arguments.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; SummData: Vector value containing the name(s) of the .Rda file(s) with</span></span>
<span><span class="co">#&gt;           GWAS summary statistics, with ‘SNP’ (i.e., SNP identifier),</span></span>
<span><span class="co">#&gt;           ‘BETA’ (i.e., effect-size or logarithm of odds ratio), ‘SE’</span></span>
<span><span class="co">#&gt;           (i.e., standard error of BETA), ‘P’ (i.e., p-values), 'NMISS'</span></span>
<span><span class="co">#&gt;           (i.e., effective sample size), 'L95' (i.e., lower limit of</span></span>
<span><span class="co">#&gt;           95% confidence interval) and 'U95' (i.e., upper limit of 95%</span></span>
<span><span class="co">#&gt;           confidence interval) are in mandatory column headers. These</span></span>
<span><span class="co">#&gt;           files needed to be in DataDir. If the numbers of cases and</span></span>
<span><span class="co">#&gt;           controls are unequal, effective sample size should be 4 /</span></span>
<span><span class="co">#&gt;           (1/&lt;# of cases&gt; + 1/&lt;# of controls&gt;). A smaller "effective"</span></span>
<span><span class="co">#&gt;           sample size may be used for samples that include related</span></span>
<span><span class="co">#&gt;           individuals, however simulations indicate that small changes</span></span>
<span><span class="co">#&gt;           in the effective sample size have relatively little effect on</span></span>
<span><span class="co">#&gt;           the final p-value (Willer et al. 2010). Columns, such as,</span></span>
<span><span class="co">#&gt;           'CHR' (Chromosome code), 'BP' (Basepair position), 'A1'</span></span>
<span><span class="co">#&gt;           (First allele code), 'A2' (Second allele code) columns are</span></span>
<span><span class="co">#&gt;           optional. If these are present, setting 'useSNPposition' to</span></span>
<span><span class="co">#&gt;           'FALSE', causes 'CHR', 'BP' and 'A1' to be ignored and</span></span>
<span><span class="co">#&gt;           setting 'UseA1' to be 'FALSE' causes 'A1' to be ignored. If,</span></span>
<span><span class="co">#&gt;           both these arguments are 'TRUE', this function takes care of</span></span>
<span><span class="co">#&gt;           A1/A2 allele flips properly. Otherwise, A1 mismatches are</span></span>
<span><span class="co">#&gt;           thrown out. Values of CHR/BP are allowed to vary.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  SNPfile: Character string specifying the name of the plain-text file</span></span>
<span><span class="co">#&gt;           with a column of SNP names. These could be LD clumped SNPs or</span></span>
<span><span class="co">#&gt;           any other list of chosen SNPs for Meta analysis. This file</span></span>
<span><span class="co">#&gt;           needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; useSNPposition: Boolean value, 'TRUE' or 'FALSE' for using 'CHR', 'BP',</span></span>
<span><span class="co">#&gt;           and 'A1' or not. The default is 'FALSE.' Note: if this is</span></span>
<span><span class="co">#&gt;           'FALSE' then there will be no Manhattan and QQ plot will be</span></span>
<span><span class="co">#&gt;           generated.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    UseA1: Boolean value, 'TRUE' or 'FALSE' for 'A1' to be used or not.</span></span>
<span><span class="co">#&gt;           The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     GCse: Boolean value, 'TRUE' or 'FALSE' for applying study specific</span></span>
<span><span class="co">#&gt;           genomic control to adjust each study for potential population</span></span>
<span><span class="co">#&gt;           structure for all the SNPs. The default is 'TRUE'. If users</span></span>
<span><span class="co">#&gt;           would want to apply genomic control separately for directly</span></span>
<span><span class="co">#&gt;           genotyped and imputed SNPs prior using the function, set this</span></span>
<span><span class="co">#&gt;           parameter as 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotname: Character string, specifying the plot name of the file</span></span>
<span><span class="co">#&gt;           containing forest plots for the SNPs. The default is</span></span>
<span><span class="co">#&gt;           “Meta_Analysis.plot”.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pval_filter: Character value as "R","F" or "W", specifying whether</span></span>
<span><span class="co">#&gt;           p-value threshold should be chosen based on “Random”, “Fixed”</span></span>
<span><span class="co">#&gt;           or “Weighted” effect model for the SNPs to be included in the</span></span>
<span><span class="co">#&gt;           forest plots.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; top_snp_pval: Numeric value, specifying the threshold to be used to</span></span>
<span><span class="co">#&gt;           filter the SNPs for the forest plots. The default is 1e-08.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; max_top_snps: Integer value, specifying the maximum number of top SNPs</span></span>
<span><span class="co">#&gt;           (SNPs with the lowest p-values) to be ploted in the forest</span></span>
<span><span class="co">#&gt;           plot file. The default is 6.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; chosen_snps_file: Character string specifying the name of the</span></span>
<span><span class="co">#&gt;           plain-text file with a column of SNP names for the forest</span></span>
<span><span class="co">#&gt;           plots. The default is NULL.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying whether the meta</span></span>
<span><span class="co">#&gt;           analysis will be performed chromosome wise or not. The</span></span>
<span><span class="co">#&gt;           default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pval_threshold_manplot: Numeric value, specifying the p-value threshold</span></span>
<span><span class="co">#&gt;           for plotting Manhattan plots.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list object containing five dataframes. The first three</span></span>
<span><span class="co">#&gt;      dataframes, such as Mfixed, Mrandom and Mweighted contain results</span></span>
<span><span class="co">#&gt;      for fixed effect, random effect and weighted model. Each of these</span></span>
<span><span class="co">#&gt;      dataframes can have maximum 12 columns, such as:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'CHR' (Chromosome code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'BP' (Basepair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SNP' (SNP identifier)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'A1' (First allele code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'A2' (Second allele code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'Q' (p-value for Cochrane's Q statistic)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'I' (I^2 heterogeneity index (0-100))</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P' (P-value from mata analysis)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'ES' (Effect-size estimate from mata analysis)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SE' (Standard Error from mata analysis)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'CI_L' (Lower limit of confidence interval)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'CI_U' (Uper limit of confidence interval)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The fourth dataframe contains the same columns 'CHR', 'BP', 'SNP',</span></span>
<span><span class="co">#&gt;      'A1', 'A2', 'Q', 'I'", with column 'N'' ( Number of valid studies</span></span>
<span><span class="co">#&gt;      for this SNP), P (Fixed-effects meta-analysis p-value), and other</span></span>
<span><span class="co">#&gt;      columns as 'Fx...' (Study x (0-based input file indices) effect</span></span>
<span><span class="co">#&gt;      estimate, Examples: F0, F1 etc.).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The fifth dataframe, ProblemSNP has three columns, such as</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'File' (file name of input data),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'SNP' (Problematic SNPs that are thrown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'Problem' (Problem code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Problem codes are:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • BAD_CHR (Invalid chromosome code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • BAD_BP (Invalid base-position code), BAD_ES (Invalid</span></span>
<span><span class="co">#&gt;           effect-size)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • BAD_SE (Invalid standard error), MISSING_A1 (Missing allele 1</span></span>
<span><span class="co">#&gt;           label)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • MISSING_A2 (Missing allele 2 label)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • ALLELE_MISMATCH (Mismatching allele codes across files)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A .pdf file comprising the forest plots of the SNPs is produced in</span></span>
<span><span class="co">#&gt;      the ResultDir with Plotname as prefix. If 'useSNPposition' is set</span></span>
<span><span class="co">#&gt;      'TRUE', a .jpeg file with Manhattan Plot and Q-Q plot will be in</span></span>
<span><span class="co">#&gt;      the 'ResultDir' with Plotname as prefix.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Willer CJ, Li Y, Abecasis GR (2010). "METAL: fast and efficient</span></span>
<span><span class="co">#&gt;      meta-analysis of genomewide association scans." _Bioinformatics_,</span></span>
<span><span class="co">#&gt;      *26*(17), 2190-2191. doi:10.1093/bioinformatics/btq340</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1093/bioinformatics/btq340&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/20616382&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex1", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex2", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      SummData &lt;- list(Summary_Stat_Ex1, Summary_Stat_Ex2)</span></span>
<span><span class="co">#&gt;      SNPfile &lt;- "UniqueLoci"</span></span>
<span><span class="co">#&gt;      useSNPposition &lt;- FALSE</span></span>
<span><span class="co">#&gt;      UseA1 &lt;- TRUE</span></span>
<span><span class="co">#&gt;      GCse &lt;- TRUE</span></span>
<span><span class="co">#&gt;      byCHR &lt;- FALSE</span></span>
<span><span class="co">#&gt;      pval_filter &lt;- "R"</span></span>
<span><span class="co">#&gt;      top_snp_pval &lt;- 1e-08</span></span>
<span><span class="co">#&gt;      max_top_snps &lt;- 10</span></span>
<span><span class="co">#&gt;      chosen_snps_file &lt;- NULL</span></span>
<span><span class="co">#&gt;      pval_threshold_manplot &lt;- 1e-05</span></span>
<span><span class="co">#&gt;      plotname &lt;- "Meta_Analysis.plot"</span></span>
<span><span class="co">#&gt;      x &lt;- MetaGWAS(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, SummData = SummData, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          SNPfile = NULL, useSNPposition = TRUE, UseA1 = UseA1, GCse = GCse,</span></span>
<span><span class="co">#&gt;          plotname = "Meta_Analysis.plot", pval_filter, top_snp_pval, max_top_snps,</span></span>
<span><span class="co">#&gt;          chosen_snps_file = NULL, byCHR, pval_threshold_manplot</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-15">Function 15<a class="anchor" aria-label="anchor" href="#function-15"></a>
</h4>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputePRS</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ComputePRS: Computing polygenic risk score (PRS)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function calculates the polygenic risk score, which is the</span></span>
<span><span class="co">#&gt;      total of allele counts (genotypes) weighted by estimated effect</span></span>
<span><span class="co">#&gt;      sizes from genome-wide association studies. It uses C+T filtering</span></span>
<span><span class="co">#&gt;      techniques. The users could perform clumping procedure</span></span>
<span><span class="co">#&gt;      choromosome-wise and genome-wide. Also, the function offers the</span></span>
<span><span class="co">#&gt;      choice of including several genetic principal components along</span></span>
<span><span class="co">#&gt;      with other covariates. Using this function, users have the freedom</span></span>
<span><span class="co">#&gt;      to experiment with various clumping and thresholding arrangements</span></span>
<span><span class="co">#&gt;      to test a wide range of various parameter values.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      ComputePRS(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        summarystat,</span></span>
<span><span class="co">#&gt;        phenofile,</span></span>
<span><span class="co">#&gt;        covarfile = NULL,</span></span>
<span><span class="co">#&gt;        effectsize = c("BETA", "OR"),</span></span>
<span><span class="co">#&gt;        ldclump = FALSE,</span></span>
<span><span class="co">#&gt;        LDreference,</span></span>
<span><span class="co">#&gt;        clump_p1,</span></span>
<span><span class="co">#&gt;        clump_p2,</span></span>
<span><span class="co">#&gt;        clump_r2,</span></span>
<span><span class="co">#&gt;        clump_kb,</span></span>
<span><span class="co">#&gt;        byCHR = TRUE,</span></span>
<span><span class="co">#&gt;        pthreshold = c(0.001, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5),</span></span>
<span><span class="co">#&gt;        highLD_regions,</span></span>
<span><span class="co">#&gt;        ld_prunning = FALSE,</span></span>
<span><span class="co">#&gt;        window_size = 50,</span></span>
<span><span class="co">#&gt;        step_size = 5,</span></span>
<span><span class="co">#&gt;        r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        nPC = 6,</span></span>
<span><span class="co">#&gt;        pheno_type = "binary"</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the all the input</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is tempdir().</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the genotype data i.e., the target data</span></span>
<span><span class="co">#&gt;           based on which clumping procedure will be performed. This</span></span>
<span><span class="co">#&gt;           file needs to be in DataDir. If your target data are small</span></span>
<span><span class="co">#&gt;           (e.g. N &lt; 500) then you can use the 1000 Genomes Project</span></span>
<span><span class="co">#&gt;           samples. Make sure to use the population that most closely</span></span>
<span><span class="co">#&gt;           reflects represents the base sample.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; summarystat: A dataframe object with GWAS summary statistics.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           The mandatory column headers in this dataframe are:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'CHR'(Chromosome code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'BP'(Basepair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'A1' (effect allele)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'SNP' (i.e., SNP identifier)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'BETA' or 'OR' (i.e., effect-size or logarithm of odds</span></span>
<span><span class="co">#&gt;               ratio)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'P' (i.e., p-values).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Special Notes: The first three columns needed to be 'SNP',</span></span>
<span><span class="co">#&gt;           'A1' and 'BETA' or 'OR'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; phenofile: A character string, specifying the name of the mandatory</span></span>
<span><span class="co">#&gt;           phenotype file. This is a plain text file with no header</span></span>
<span><span class="co">#&gt;           line; columns family ID, individual ID and phenotype columns.</span></span>
<span><span class="co">#&gt;           For binary trait, the phenotypic value should be coded as 0</span></span>
<span><span class="co">#&gt;           or 1, then it will be recognized as a case-control study (0</span></span>
<span><span class="co">#&gt;           for controls and 1 for cases). Missing value should be</span></span>
<span><span class="co">#&gt;           represented by "-9" or "NA". The interested phenotype column</span></span>
<span><span class="co">#&gt;           should be labeled as "Pheno1". This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covarfile: A character string, specifying the name of the covariate</span></span>
<span><span class="co">#&gt;           file which is a plain .text file with no header line; columns</span></span>
<span><span class="co">#&gt;           are family ID, individual ID and the covariates. The default</span></span>
<span><span class="co">#&gt;           is 'NULL'. This file needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; effectsize: Boolean value, 'BETA' or 'OR', specifying the type of the</span></span>
<span><span class="co">#&gt;           GWAS effectsize. The default is 'BETA'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  ldclump: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           perform clumping or not.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LDreference: A character string, specifying the prefix of the PLINK</span></span>
<span><span class="co">#&gt;           files of the population reference panel of the same ancestry,</span></span>
<span><span class="co">#&gt;           and ideally the one that was used for imputing your target</span></span>
<span><span class="co">#&gt;           dataset. These files should be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p1: Numeric value, specifying the significance threshold for</span></span>
<span><span class="co">#&gt;           index SNPs if 'ldclump' was set to be 'TRUE'. The default is</span></span>
<span><span class="co">#&gt;           0.0001.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p2: Numeric value, specifying the secondary significance</span></span>
<span><span class="co">#&gt;           threshold for clumped SNPs if 'ldclump' was set to be 'TRUE'.</span></span>
<span><span class="co">#&gt;           The default is 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_r2: Numeric value, specifying the linkage disequilibrium (LD)</span></span>
<span><span class="co">#&gt;           threshold for clumping if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 0.50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_kb: Integer value, specifying the physical distance threshold in</span></span>
<span><span class="co">#&gt;           base-pair for clumping if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 250.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying chromosome-wise</span></span>
<span><span class="co">#&gt;           clumping procedure if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pthreshold: Numeric vector, containing several p value thresholds to</span></span>
<span><span class="co">#&gt;           maximize predictive ability of the derived polygenic scores.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: Character string, specifying the .txt file name with</span></span>
<span><span class="co">#&gt;           known genomic regions with high LD. The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ld_prunning: Boolean value, 'TRUE' or 'FALSE' for LD-based filtering</span></span>
<span><span class="co">#&gt;           for computing genetic PC as covariates.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; window_size: Integer value, specifying a window size in variant count</span></span>
<span><span class="co">#&gt;           or kilobase for LD-based filtering in computing genetic PC.</span></span>
<span><span class="co">#&gt;           The default is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; step_size: Integer value, specifying a variant count to shift the</span></span>
<span><span class="co">#&gt;           window at the end of each step for LD filtering in computing</span></span>
<span><span class="co">#&gt;           genetic PCs. The default is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; r2_threshold: Numeric value between 0 to 1 of pairwise r^2 threshold</span></span>
<span><span class="co">#&gt;           for LD-based filtering in computing genetic PCs. The default</span></span>
<span><span class="co">#&gt;           is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      nPC: Positive integer value, specifying the number of genetic PCs</span></span>
<span><span class="co">#&gt;           to be included as predictor in the PRS model fit. The default</span></span>
<span><span class="co">#&gt;           is 6.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pheno_type: Boolean value, ‘binary’ or ‘quantitative’, specifying the</span></span>
<span><span class="co">#&gt;           type of the trait. The default is ‘binary’.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list object containing a dataframe a numeric value, a GeneticPC</span></span>
<span><span class="co">#&gt;      plot (if requested), and a PRS plot. The dataframe,PRS, contains</span></span>
<span><span class="co">#&gt;      four mandatory columns, such as, IID (i.e., Individual ID), FID</span></span>
<span><span class="co">#&gt;      (i.e., Family ID), Pheno1 (i.e., the trait for PRS) and Score</span></span>
<span><span class="co">#&gt;      (i.e., the best PRS). Other columns of covariates could be there.</span></span>
<span><span class="co">#&gt;      The numeric value, BestP contains the threshold of of the best</span></span>
<span><span class="co">#&gt;      p-value for the best pRS model fit.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Also, the function produces several plots such as p-value</span></span>
<span><span class="co">#&gt;      thresholds vs PRS model fit and PRS distribution among male and</span></span>
<span><span class="co">#&gt;      females. For case-control data, it shows PRS distribution among</span></span>
<span><span class="co">#&gt;      cases and controls and ROC curves as well.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex1", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_phenofile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_covarfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_pthresoldfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("highLD_hg19", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      summarystat &lt;- Summary_Stat_Ex1[, c(2, 4, 7, 1, 3, 12)]</span></span>
<span><span class="co">#&gt;      phenofile &lt;- Example_phenofile # Cannot be NULL</span></span>
<span><span class="co">#&gt;      # The interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="co">#&gt;      covarfile &lt;- Example_covarfile</span></span>
<span><span class="co">#&gt;      clump_p1 &lt;- 0.0001</span></span>
<span><span class="co">#&gt;      clump_p2 &lt;- 0.0001</span></span>
<span><span class="co">#&gt;      clump_kb &lt;- 500</span></span>
<span><span class="co">#&gt;      clump_r2 &lt;- 0.5</span></span>
<span><span class="co">#&gt;      byCHR &lt;- TRUE</span></span>
<span><span class="co">#&gt;      pthreshold &lt;- Example_pthresoldfile$Threshold</span></span>
<span><span class="co">#&gt;      ld_prunning &lt;- TRUE</span></span>
<span><span class="co">#&gt;      highLD_regions &lt;- highLD_hg19</span></span>
<span><span class="co">#&gt;      window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      nPC &lt;- 6 # We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="co">#&gt;      pheno_type &lt;- "binary"</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      PRSresult &lt;- ComputePRS(DataDir, ResultDir, finput, summarystat, phenofile, covarfile,</span></span>
<span><span class="co">#&gt;          effectsize = "BETA", LDreference = "GXwasR_example", ldclump = FALSE, clump_p1, clump_p2,</span></span>
<span><span class="co">#&gt;          clump_r2, clump_kb, byCHR = TRUE, pthreshold = pthreshold, highLD_regions = highLD_regions,</span></span>
<span><span class="co">#&gt;          ld_prunning = TRUE, window_size = 50, step_size = 5, r2_threshold = 0.02, nPC = 6,</span></span>
<span><span class="co">#&gt;          pheno_type = "binary"</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      ## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="co">#&gt;      PRS &lt;- PRSresult$PRS</span></span>
<span><span class="co">#&gt;      PRS[seq_len(10), ]</span></span>
<span><span class="co">#&gt;      ## The best threshold</span></span>
<span><span class="co">#&gt;      BestPvalue &lt;- PRSresult$BestP$Threshold</span></span>
<span><span class="co">#&gt;      BestPvalue</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-16">Function 16<a class="anchor" aria-label="anchor" href="#function-16"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GeneticCorrBT</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GeneticCorrBT: Computing genetic correlation between two traits.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function computes genetic correlation, a quantitative genetic</span></span>
<span><span class="co">#&gt;      measure that describes the genetic link between two traits and has</span></span>
<span><span class="co">#&gt;      been predicted to indicate pleiotropic gene activity or</span></span>
<span><span class="co">#&gt;      correlation between causative loci in two traits. For example, it</span></span>
<span><span class="co">#&gt;      does a bivariate GREML analysis to determine the genetic</span></span>
<span><span class="co">#&gt;      association between two quantitative traits, two binary disease</span></span>
<span><span class="co">#&gt;      traits from case-control studies, and between a quantitative trait</span></span>
<span><span class="co">#&gt;      and a binary disease trait following (Yang et al. 2011; Lee et al.</span></span>
<span><span class="co">#&gt;      2012). If users want, this function gives the flexibility to</span></span>
<span><span class="co">#&gt;      compute the genetic correlation chromosome-wise.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      GeneticCorrBT(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        byCHR = FALSE,</span></span>
<span><span class="co">#&gt;        REMLalgo = c(0, 1, 2),</span></span>
<span><span class="co">#&gt;        nitr = 100,</span></span>
<span><span class="co">#&gt;        phenofile,</span></span>
<span><span class="co">#&gt;        cat_covarfile = NULL,</span></span>
<span><span class="co">#&gt;        quant_covarfile = NULL,</span></span>
<span><span class="co">#&gt;        computeGRM = TRUE,</span></span>
<span><span class="co">#&gt;        grmfile_name = NULL,</span></span>
<span><span class="co">#&gt;        partGRM = FALSE,</span></span>
<span><span class="co">#&gt;        autosome = TRUE,</span></span>
<span><span class="co">#&gt;        Xsome = TRUE,</span></span>
<span><span class="co">#&gt;        nGRM = 3,</span></span>
<span><span class="co">#&gt;        cripticut = 0.025,</span></span>
<span><span class="co">#&gt;        minMAF = NULL,</span></span>
<span><span class="co">#&gt;        maxMAF = NULL,</span></span>
<span><span class="co">#&gt;        excludeResidual = FALSE,</span></span>
<span><span class="co">#&gt;        ncores = 2</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the all the input</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the genotype data. This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying whether the</span></span>
<span><span class="co">#&gt;           analysis will be performed chromosome wise or not. The</span></span>
<span><span class="co">#&gt;           default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; REMLalgo: Integer value of 0, 1 or 2, specifying the algorithm to run</span></span>
<span><span class="co">#&gt;           REML iterations, 0 for average information (AI), 1 for</span></span>
<span><span class="co">#&gt;           Fisher-scoring and 2 for EM. The default option is 0, i.e.</span></span>
<span><span class="co">#&gt;           AI-REML (1).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     nitr: Integer value, specifying the number of iterations for</span></span>
<span><span class="co">#&gt;           performing the REML. The default is 100.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; phenofile: A dataframe for Bivar RELM has four columns family ID,</span></span>
<span><span class="co">#&gt;           individual ID and two trait columns. For binary trait, the</span></span>
<span><span class="co">#&gt;           phenotypic value should be coded as 0 or 1, then it will be</span></span>
<span><span class="co">#&gt;           recognized as a case-control study (0 for controls and 1 for</span></span>
<span><span class="co">#&gt;           cases). Missing value should be represented by "-9" or "NA".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; cat_covarfile: A character string, specifying the name of the</span></span>
<span><span class="co">#&gt;           categorical covariate file which is a plain text file with no</span></span>
<span><span class="co">#&gt;           header line; columns are family ID, individual ID and</span></span>
<span><span class="co">#&gt;           discrete covariates. The default is 'NULL'. This file needs</span></span>
<span><span class="co">#&gt;           to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; quant_covarfile: A character string, specifying the name of the</span></span>
<span><span class="co">#&gt;           quantitative covariate file which is a plain text file with</span></span>
<span><span class="co">#&gt;           no header line; columns are family ID, individual ID and</span></span>
<span><span class="co">#&gt;           continuous covariates. The default is 'NULL'. This file needs</span></span>
<span><span class="co">#&gt;           to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; computeGRM: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           compute GRM matrices or not. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; grmfile_name: A string of characters specifying the prefix of autosomal</span></span>
<span><span class="co">#&gt;           .grm.bin file. Users need to provide separate GRM files for</span></span>
<span><span class="co">#&gt;           autosomes and X chromosome in 'ResultDir'. The X chromosomal</span></span>
<span><span class="co">#&gt;           GRM file should have "x" added in the autosomal prefix as</span></span>
<span><span class="co">#&gt;           file name.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For instance, if autosomal file is "ABC.grm.bin", then X</span></span>
<span><span class="co">#&gt;           chromosomal file should be "xABC.grm.bim". If you are</span></span>
<span><span class="co">#&gt;           providing chromosome-wise GRMs, then the prefix should add</span></span>
<span><span class="co">#&gt;           "ChrNumber_" at the starting of the prefix like,</span></span>
<span><span class="co">#&gt;           "Chr1_ABC.grm.bin". The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  partGRM: Boolean value, 'TRUE' or 'FALSE', specifying whether the GRM</span></span>
<span><span class="co">#&gt;           will be partitioned into n parts (by row) in GREML model. The</span></span>
<span><span class="co">#&gt;           default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; autosome: Boolean value, 'TRUE' or 'FALSE', specifying whether estimate</span></span>
<span><span class="co">#&gt;           of heritability will be done for autosomes or not. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Xsome: Boolean value, 'TRUE' or 'FALSE', specifying whether estimate</span></span>
<span><span class="co">#&gt;           of heritability will be done for X chromosome or not. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     nGRM: Integer value, specifying the number of the partition of the</span></span>
<span><span class="co">#&gt;           GRM in GREML model. The default is 3.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; cripticut: Numeric value, specifying the threshold to create a new GRM</span></span>
<span><span class="co">#&gt;           of "unrelated" individuals in GREML model. The default is</span></span>
<span><span class="co">#&gt;           arbitrary chosen as 0.025 following (Yang et al. 2011).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   minMAF: Positive numeric value (&lt; maxMAF), specifying the minimum</span></span>
<span><span class="co">#&gt;           threshold for the MAF filter of the SNPs in the Bivariate</span></span>
<span><span class="co">#&gt;           GREML model.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   maxMAF: Positive numeric value (minMAF,1), specifying the maximum</span></span>
<span><span class="co">#&gt;           threshold for the MAF filter of the SNPs in the Bivariate</span></span>
<span><span class="co">#&gt;           GREML model.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; excludeResidual: Boolean value, 'TRUE' or 'FALSE', specifying whether</span></span>
<span><span class="co">#&gt;           to drop the residual covariance from the model. Recommended</span></span>
<span><span class="co">#&gt;           to set this 'TRUE' if the traits were measured on different</span></span>
<span><span class="co">#&gt;           individuals. The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   ncores: Integer value, specifying the number of cores to be used.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with minimum three columns:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • Source" (i.e., source of heritability)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • Variance" (i.e. estimated heritability)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • SE" (i.e., standard error of the estimated heritability)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Source column will have rows, such as V(G)_tr1 (genetic variance</span></span>
<span><span class="co">#&gt;      for trait 1), V(G)_tr2 (genetic variance for trait 2), C(G)_tr12</span></span>
<span><span class="co">#&gt;      (genetic covariance between traits 1 and 2),V(e)_tr1 (residual</span></span>
<span><span class="co">#&gt;      variance for trait 1), V(e)_tr2 (residual variance for trait 2),</span></span>
<span><span class="co">#&gt;      C(e)_tr12 (residual covariance between traits 1 and 2), Vp_tr1</span></span>
<span><span class="co">#&gt;      (proportion of variance explained by all SNPs for trait 1), Vp_tr2</span></span>
<span><span class="co">#&gt;      (proportion of variance explained by all SNPs for trait 2),</span></span>
<span><span class="co">#&gt;      V(G)/Vp_tr1 (phenotypic variance for trait 1), V(G)/Vp_tr2</span></span>
<span><span class="co">#&gt;      (phenotypic variance for trait 2), rG (genetic correlation) and n</span></span>
<span><span class="co">#&gt;      (sample size). In case of chromosome-wise analysis, there will be</span></span>
<span><span class="co">#&gt;      'chromosome' column for chromosome code.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Lee SH, Wray NR, Goddard ME, Visscher PM (2012). "Estimation of</span></span>
<span><span class="co">#&gt;      pleiotropy between complex diseases using SNP-derived genomic</span></span>
<span><span class="co">#&gt;      relationships and restricted maximum likelihood."</span></span>
<span><span class="co">#&gt;      _Bioinformatics_, *28*, 2540-2542.</span></span>
<span><span class="co">#&gt;      doi:10.1093/bioinformatics/bts474</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1093/bioinformatics/bts474&gt;,</span></span>
<span><span class="co">#&gt;      &lt;http://www.ncbi.nlm.nih.gov/pubmed/22843982&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Yang J, Lee SH, Goddard ME, Visscher PM (2011). "GCTA: a tool for</span></span>
<span><span class="co">#&gt;      Genome-wide Complex Trait Analysis." _American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *88*(1), 76-82.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Example_phenofile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      byCHR &lt;- TRUE</span></span>
<span><span class="co">#&gt;      REMLalgo &lt;- 0</span></span>
<span><span class="co">#&gt;      nitr &lt;- 3</span></span>
<span><span class="co">#&gt;      ncores &lt;- 3</span></span>
<span><span class="co">#&gt;      phenofile &lt;- Example_phenofile # Cannot be NULL</span></span>
<span><span class="co">#&gt;      cat_covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      quant_covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      partGRM &lt;- FALSE # Partition the GRM into m parts (by row),</span></span>
<span><span class="co">#&gt;      autosome &lt;- TRUE</span></span>
<span><span class="co">#&gt;      Xsome &lt;- TRUE</span></span>
<span><span class="co">#&gt;      cripticut &lt;- 0.025</span></span>
<span><span class="co">#&gt;      minMAF &lt;- 0.01 # if MAF filter apply</span></span>
<span><span class="co">#&gt;      maxMAF &lt;- 0.04</span></span>
<span><span class="co">#&gt;      excludeResidual &lt;- TRUE</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      genetic_correlation &lt;- GeneticCorrBT(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput, byCHR = byCHR,</span></span>
<span><span class="co">#&gt;          REMLalgo = 0, nitr = 10, phenofile = phenofile, cat_covarfile = NULL, quant_covarfile = NULL,</span></span>
<span><span class="co">#&gt;          partGRM = FALSE, autosome = TRUE, Xsome = TRUE, nGRM = 3,</span></span>
<span><span class="co">#&gt;          cripticut = 0.025, minMAF = NULL, maxMAF = NULL, excludeResidual = TRUE, ncores = ncores</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-17">Function 17<a class="anchor" aria-label="anchor" href="#function-17"></a>
</h4>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">EstimateHerit</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; EstimateHerit: Computing SNP heritability i.e., the proportion of</span></span>
<span><span class="co">#&gt; phenotypic variance explained by SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This functions performs two types of heritability estimation,</span></span>
<span><span class="co">#&gt;      (i)GREML:Genomic relatedness matrix (GRM) restricted maximum</span></span>
<span><span class="co">#&gt;      likelihood-based method following GCTA (Yang et al. 2011) and</span></span>
<span><span class="co">#&gt;      (ii)LDSC: LD score regression-based method following</span></span>
<span><span class="co">#&gt;      (Bulik-Sullivan et al. 2014; Privé et al. 2020). For the details,</span></span>
<span><span class="co">#&gt;      please follow the associated paper.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Prior to using this function, it is recommended to apply QCsnp and</span></span>
<span><span class="co">#&gt;      QCsample to ensure data quality control.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      EstimateHerit(</span></span>
<span><span class="co">#&gt;        DataDir = NULL,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput = NULL,</span></span>
<span><span class="co">#&gt;        precomputedLD = NULL,</span></span>
<span><span class="co">#&gt;        indepSNPs = NULL,</span></span>
<span><span class="co">#&gt;        summarystat = NULL,</span></span>
<span><span class="co">#&gt;        ncores = 2,</span></span>
<span><span class="co">#&gt;        model = c("LDSC", "GREML"),</span></span>
<span><span class="co">#&gt;        computeGRM = TRUE,</span></span>
<span><span class="co">#&gt;        grmfile_name = NULL,</span></span>
<span><span class="co">#&gt;        byCHR = FALSE,</span></span>
<span><span class="co">#&gt;        r2_LD = 0,</span></span>
<span><span class="co">#&gt;        LDSC_blocks = 20,</span></span>
<span><span class="co">#&gt;        intercept = NULL,</span></span>
<span><span class="co">#&gt;        chi2_thr1 = 30,</span></span>
<span><span class="co">#&gt;        chi2_thr2 = Inf,</span></span>
<span><span class="co">#&gt;        REMLalgo = c(0, 1, 2),</span></span>
<span><span class="co">#&gt;        nitr = 100,</span></span>
<span><span class="co">#&gt;        cat_covarfile = NULL,</span></span>
<span><span class="co">#&gt;        quant_covarfile = NULL,</span></span>
<span><span class="co">#&gt;        prevalence = NULL,</span></span>
<span><span class="co">#&gt;        partGRM = FALSE,</span></span>
<span><span class="co">#&gt;        autosome = TRUE,</span></span>
<span><span class="co">#&gt;        Xsome = TRUE,</span></span>
<span><span class="co">#&gt;        nGRM = 3,</span></span>
<span><span class="co">#&gt;        cripticut = 0.025,</span></span>
<span><span class="co">#&gt;        minMAF = NULL,</span></span>
<span><span class="co">#&gt;        maxMAF = NULL,</span></span>
<span><span class="co">#&gt;        hg = c("hg19", "hg38"),</span></span>
<span><span class="co">#&gt;        PlotIndepSNP = TRUE,</span></span>
<span><span class="co">#&gt;        IndepSNP_window_size = 50,</span></span>
<span><span class="co">#&gt;        IndepSNP_step_size = 5,</span></span>
<span><span class="co">#&gt;        IndepSNP_r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        highLD_regions = NULL,</span></span>
<span><span class="co">#&gt;        plotjpeg = TRUE,</span></span>
<span><span class="co">#&gt;        plotname = "Heritability_Plots"</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the all the input</span></span>
<span><span class="co">#&gt;           files. The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the genotype data. This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'. For LDSC model, if the original genotype data is</span></span>
<span><span class="co">#&gt;           not available, Hapmap 3 or 1000Genome data can be used. If</span></span>
<span><span class="co">#&gt;           use NULL, then you need to provide 'precomputedLD' argument.</span></span>
<span><span class="co">#&gt;           See below.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; precomputedLD: A dataframe object as LD matrix with columns: 'CHR',</span></span>
<span><span class="co">#&gt;           'SNP', 'BP', 'ld_size', 'MAF', 'ld_score'. . The default is</span></span>
<span><span class="co">#&gt;           'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; indepSNPs: A dataframe with independent SNP ids with column name</span></span>
<span><span class="co">#&gt;           "rsid". The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; summarystat: A dataframe object with GWAS summary statistics. The</span></span>
<span><span class="co">#&gt;           mandatory column headers in this dataframe are</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'chr' (Chromosome code),</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'pos' (Basepair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'a1' (First allele code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'rsid' (i.e., SNP identifier)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'beta' (i.e., effect-size or logarithm of odds ratio)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'beta_se' (i.e., standard error of beta)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'P' (i.e., p-values)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'n_eff' (i.e., effective sample size)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           For case-control study, effective sample size should be 4 /</span></span>
<span><span class="co">#&gt;           (1/&lt;# of cases&gt; + 1/&lt;# of controls&gt;). The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   ncores: Integer value, specifying the number of cores to be used for</span></span>
<span><span class="co">#&gt;           running LDSC model. The default is 2.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    model: Character string, specifying the heritability estimation</span></span>
<span><span class="co">#&gt;           model. There are two options, “GREML” or “LDSC”. The default</span></span>
<span><span class="co">#&gt;           is “GREML”.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: argument For LDSC, DataDir and finput can be 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; computeGRM: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           compute GRM matrices or not. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; grmfile_name: A string of characters specifying the prefix of autosomal</span></span>
<span><span class="co">#&gt;           .grm.bin file. Users need to provide separate GRM files for</span></span>
<span><span class="co">#&gt;           autosomes and X chromosome in 'ResultDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           The X chromosomal GRM file should have "x" added in the</span></span>
<span><span class="co">#&gt;           autosomal prefix as file name. For instance, if autosomal</span></span>
<span><span class="co">#&gt;           file is "ABC.grm.bin", then X chromosomal file should be</span></span>
<span><span class="co">#&gt;           "xABC.grm.bim".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           If you are providing chromosome-wise GRMs, then the prefix</span></span>
<span><span class="co">#&gt;           should add "ChrNumber_" at the start of the prefix like,</span></span>
<span><span class="co">#&gt;           "Chr1_ABC.grm.bin". The default is NULL.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying whether the</span></span>
<span><span class="co">#&gt;           analysis will be performed chromosome wise or not. The</span></span>
<span><span class="co">#&gt;           default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    r2_LD: Numeric value, specifying the LD threshold for clumping in</span></span>
<span><span class="co">#&gt;           LDSC model. The default is 0.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LDSC_blocks: Integer value, specifying the block size for performing</span></span>
<span><span class="co">#&gt;           jackknife variance estimator in LDSC model following (Privé</span></span>
<span><span class="co">#&gt;           et al. 2020). The default is 200.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; intercept: Numeric value to constrain the intercept to some value (e.g.</span></span>
<span><span class="co">#&gt;           1) in LDSC regression. Default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; chi2_thr1: Numeric value for threshold on chi2 in step 1 of LDSC</span></span>
<span><span class="co">#&gt;           regression. Default is 30.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; chi2_thr2: Numeric value for threshold on chi2 in step 2. Default is</span></span>
<span><span class="co">#&gt;           'Inf' (none).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; REMLalgo: Integer value of 0, 1 or 2, specifying the algorithm to run</span></span>
<span><span class="co">#&gt;           REML iterations, 0 for average information (AI), 1 for</span></span>
<span><span class="co">#&gt;           Fisher-scoring and 2 for EM. The default option is 0, i.e.</span></span>
<span><span class="co">#&gt;           AI-REML (Yang et al. 2011).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     nitr: Integer value, specifying the number of iterations for</span></span>
<span><span class="co">#&gt;           performing the REML. The default is 100.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; cat_covarfile: A character string, specifying the name of the</span></span>
<span><span class="co">#&gt;           categorical covariate file which is a plain text file with no</span></span>
<span><span class="co">#&gt;           header line; columns are family ID, individual ID and</span></span>
<span><span class="co">#&gt;           discrete covariates. The default is 'NULL'. This file needs</span></span>
<span><span class="co">#&gt;           to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; quant_covarfile: A character string, specifying the name of the</span></span>
<span><span class="co">#&gt;           quantitative covariate file which is a plain text file with</span></span>
<span><span class="co">#&gt;           no header line; columns are family ID, individual ID and</span></span>
<span><span class="co">#&gt;           continuous covariates. The default is 'NULL'. This file needs</span></span>
<span><span class="co">#&gt;           to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; prevalence: Numeric value, specifying the disease prevalence. The</span></span>
<span><span class="co">#&gt;           default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Note: for the continuous trait value, users should use the</span></span>
<span><span class="co">#&gt;           default.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  partGRM: Boolean value, 'TRUE' or 'FALSE', specifying whether the GRM</span></span>
<span><span class="co">#&gt;           will be partitioned into n parts (by row) in GREML model. The</span></span>
<span><span class="co">#&gt;           default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; autosome: Boolean value, 'TRUE' or 'FALSE', specifying whether estimate</span></span>
<span><span class="co">#&gt;           of heritability will be done for autosomes or not. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    Xsome: Boolean value, 'TRUE' or 'FALSE', specifying whether estimate</span></span>
<span><span class="co">#&gt;           of heritability will be done for X chromosome or not. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     nGRM: Integer value, specifying the number of the partition of the</span></span>
<span><span class="co">#&gt;           GRM in GREML model. The default is 3.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; cripticut: Numeric value, specifying the threshold to create a new GRM</span></span>
<span><span class="co">#&gt;           of "unrelated" individuals in GREML model. The default is</span></span>
<span><span class="co">#&gt;           arbitrary chosen as 0.025 following (Yang et al. 2011).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   minMAF: Positive numeric value (0,1), specifying the minimum</span></span>
<span><span class="co">#&gt;           threshold for the MAF filter of the SNPs in the GREML model.</span></span>
<span><span class="co">#&gt;           This value cannot be greater than 'maxMAF' parameter. The</span></span>
<span><span class="co">#&gt;           default is 'NULL'. For 'NULL', maximum MAF value of the</span></span>
<span><span class="co">#&gt;           genotype data will be computed and printed on the plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   maxMAF: Positive numeric value (0,1), specifying the maximum</span></span>
<span><span class="co">#&gt;           threshold for the MAF filter of the SNPs in the GREML model.</span></span>
<span><span class="co">#&gt;           This value cannot be less than 'minMAF' parameter. The</span></span>
<span><span class="co">#&gt;           default is 'NULL'. For 'NULL', minimum MAF value of the</span></span>
<span><span class="co">#&gt;           genotype data will be computed and printed on the plot.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       hg: Boolean value, specifying the genome built, “hg19” or “hg38”</span></span>
<span><span class="co">#&gt;           to use chromosome length from UCSC genome browser and getting</span></span>
<span><span class="co">#&gt;           genes and proteins according to this built. The default is</span></span>
<span><span class="co">#&gt;           “hg19”.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; PlotIndepSNP: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           use independent SNPs i.e., chromosome-wise LD pruned SNPs in</span></span>
<span><span class="co">#&gt;           the plots or not. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; IndepSNP_window_size: Integer value, specifying a window size in</span></span>
<span><span class="co">#&gt;           variant count or kilobase for LD-based filtering. The default</span></span>
<span><span class="co">#&gt;           is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; IndepSNP_step_size: Integer value, specifying a variant count to shift</span></span>
<span><span class="co">#&gt;           the window at the end of each step for LD filtering for</span></span>
<span><span class="co">#&gt;           pruned SNPs in the plots. The default is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; IndepSNP_r2_threshold: Numeric value between 0 to 1 of pairwise r^2</span></span>
<span><span class="co">#&gt;           threshold for LD-based filtering for pruned SNPs in the</span></span>
<span><span class="co">#&gt;           plots. The default is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: Character string, specifying the .txt file name with</span></span>
<span><span class="co">#&gt;           genomic regions with high LD for using in finding pruned SNPs</span></span>
<span><span class="co">#&gt;           in the plots. This file needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotjpeg: Boolean value, 'TRUE' or 'FALSE', specifying whether to save</span></span>
<span><span class="co">#&gt;           the plots in jpeg file in 'ResultDir'. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; plotname: String of character value specifying the name of the jpeg</span></span>
<span><span class="co">#&gt;           file with the plots. The default is "Heritability_Plots".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with maximum eight columns for GREML (here, three</span></span>
<span><span class="co">#&gt;      columns if running genome-wide) and ten columns for LDSC model if</span></span>
<span><span class="co">#&gt;      byCHR is 'TRUE'. The columns, such as, "chromosome"(i.e.,</span></span>
<span><span class="co">#&gt;      chromosome code),"snp_proportion" (i.e.,chromosome-wise SNP</span></span>
<span><span class="co">#&gt;      proportion)", "no.of.genes" (i.e., number of genes per</span></span>
<span><span class="co">#&gt;      chromosome), "no.of.proteins" (i.e., number of genes per</span></span>
<span><span class="co">#&gt;      chromosome),"size_mb" (i.e., chromosome length), "Source" (i.e.,</span></span>
<span><span class="co">#&gt;      source of heritability), "Variance" (i.e., estimated</span></span>
<span><span class="co">#&gt;      heritability), and "SE" (i.e., standard error of the estimated</span></span>
<span><span class="co">#&gt;      heritability) are common for both GREML and LDSC model. The</span></span>
<span><span class="co">#&gt;      column, "Intercept" (i.e., LDSC regression intercept) and "Int_SE"</span></span>
<span><span class="co">#&gt;      (i.e., standard error of the intercept) will be two extra columns</span></span>
<span><span class="co">#&gt;      for LDSC models. Source column will have rows, such as 'V(1)'</span></span>
<span><span class="co">#&gt;      (i.e., name of genetic variance), 'V(e)' (i.e., residual</span></span>
<span><span class="co">#&gt;      variance), 'V(p)' (i.e., phenotypic variance), 'V(1)/Vp' (i.e.,</span></span>
<span><span class="co">#&gt;      ratio of genetic variance to phenotypic variance), and 'V(1)/Vp_L'</span></span>
<span><span class="co">#&gt;      (i.e., ratio of genetic variance to phenotypic variance in</span></span>
<span><span class="co">#&gt;      liability scale for binary phenotypes). If 'byCHR' is 'FALSE',</span></span>
<span><span class="co">#&gt;      then the first five columns will not be reported in the dataframe.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Bulik-Sullivan B, Finucane HK, Anttila V, Gusev A, Day FR, Loh P,</span></span>
<span><span class="co">#&gt;      Consortium R, Consortium PG, of the Wellcome Trust Case Control</span></span>
<span><span class="co">#&gt;      Consortium GCfAN, Perry JRB, Patterson N, Robinson EB, Daly MJ,</span></span>
<span><span class="co">#&gt;      Price AL, Neale BM (2014). "LD score regression distinguishes</span></span>
<span><span class="co">#&gt;      confounding from polygenicity in genome-wide association studies."</span></span>
<span><span class="co">#&gt;      _Nature Genetics_, *47*, 291-295.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Privé F, Arbel J, Vilhjálmsson BJ (2020). "LDpred2: better,</span></span>
<span><span class="co">#&gt;      faster, stronger." _Bioinformatics_, *36*(22-23), 5424-5431.</span></span>
<span><span class="co">#&gt;      doi:10.1093/bioinformatics/btaa1029</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1093/bioinformatics/btaa1029&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Yang J, Lee SH, Goddard ME, Visscher PM (2011). "GCTA: a tool for</span></span>
<span><span class="co">#&gt;      Genome-wide Complex Trait Analysis." _American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *88*(1), 76-82.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex1", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("highLD_hg19", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      precomputedLD &lt;- NULL</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      test.sumstats &lt;- na.omit(Summary_Stat_Ex1[Summary_Stat_Ex1$TEST == "ADD", c(seq_len(4), 6:8)])</span></span>
<span><span class="co">#&gt;      colnames(test.sumstats) &lt;- c("chr", "rsid", "pos", "a1", "n_eff", "beta", "beta_se")</span></span>
<span><span class="co">#&gt;      summarystat &lt;- test.sumstats</span></span>
<span><span class="co">#&gt;      ncores &lt;- 3</span></span>
<span><span class="co">#&gt;      model &lt;- "GREML"</span></span>
<span><span class="co">#&gt;      byCHR &lt;- FALSE</span></span>
<span><span class="co">#&gt;      r2_LD &lt;- 0</span></span>
<span><span class="co">#&gt;      LDSC_blocks &lt;- 20</span></span>
<span><span class="co">#&gt;      REMLalgo &lt;- 0</span></span>
<span><span class="co">#&gt;      nitr &lt;- 3</span></span>
<span><span class="co">#&gt;      cat_covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      quant_covarfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      prevalence &lt;- 0.01</span></span>
<span><span class="co">#&gt;      partGRM &lt;- FALSE</span></span>
<span><span class="co">#&gt;      autosome &lt;- TRUE</span></span>
<span><span class="co">#&gt;      Xsome &lt;- TRUE</span></span>
<span><span class="co">#&gt;      nGRM &lt;- 3</span></span>
<span><span class="co">#&gt;      cripticut &lt;- 0.025</span></span>
<span><span class="co">#&gt;      minMAF &lt;- NULL</span></span>
<span><span class="co">#&gt;      maxMAF &lt;- NULL</span></span>
<span><span class="co">#&gt;      hg &lt;- "hg19"</span></span>
<span><span class="co">#&gt;      PlotIndepSNP &lt;- TRUE</span></span>
<span><span class="co">#&gt;      IndepSNP_window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      IndepSNP_step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      IndepSNP_r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      highLD_regions &lt;- highLD_hg19</span></span>
<span><span class="co">#&gt;      H2 &lt;- EstimateHerit(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir, finput = finput,</span></span>
<span><span class="co">#&gt;          summarystat = NULL, ncores, model = "GREML", byCHR = TRUE, r2_LD = 0,</span></span>
<span><span class="co">#&gt;          LDSC_blocks = 20, REMLalgo = 0, nitr = 100, cat_covarfile = NULL, quant_covarfile = NULL,</span></span>
<span><span class="co">#&gt;          prevalence = 0.01, partGRM = FALSE, autosome = TRUE, Xsome = TRUE, nGRM = 3,</span></span>
<span><span class="co">#&gt;          cripticut = 0.025, minMAF = NULL, maxMAF = NULL, hg = "hg19", PlotIndepSNP = TRUE,</span></span>
<span><span class="co">#&gt;          IndepSNP_window_size = 50, IndepSNP_step_size = 5, IndepSNP_r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;          highLD_regions = highLD_hg19</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-18">Function 18<a class="anchor" aria-label="anchor" href="#function-18"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexRegress</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexRegress: Performing linear regression analysis with quantitative</span></span>
<span><span class="co">#&gt; response variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function could be used to check association of two variables.</span></span>
<span><span class="co">#&gt;      For instance, PRS with sex.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexRegress(fdata, regressor_index, response_index)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    fdata: R dataframe object. The column with header 'response' should</span></span>
<span><span class="co">#&gt;           contain the response variable. All other column are the</span></span>
<span><span class="co">#&gt;           regressor.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; regressor_index: Integer value, specifying the column number of the</span></span>
<span><span class="co">#&gt;           main regressor variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; response_index: Integer value, specifying the column number of the</span></span>
<span><span class="co">#&gt;           response variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Numeric value containing the regression estimate ("Estimate"),</span></span>
<span><span class="co">#&gt;      standard error ("Std. Error"), statistics ("t value") and p-value</span></span>
<span><span class="co">#&gt;      (Pr(&gt;|t|))</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Regression_Ex", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      fdata &lt;- Regression_Ex</span></span>
<span><span class="co">#&gt;      fdata$SEX &lt;- as.factor(as.character(fdata$SEX))</span></span>
<span><span class="co">#&gt;      response_index &lt;- 1</span></span>
<span><span class="co">#&gt;      regressor_index &lt;- 2</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      x &lt;- SexRegress(fdata, regressor_index, response_index)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-19">Function 19<a class="anchor" aria-label="anchor" href="#function-19"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterPlinkSample</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; FilterPlinkSample: Making PLINK files with desired samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function prepares PLINK binary files with the desired</span></span>
<span><span class="co">#&gt;      samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      FilterPlinkSample(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput = NULL,</span></span>
<span><span class="co">#&gt;        filter_sample = "cases",</span></span>
<span><span class="co">#&gt;        keep_remove_sample_file = NULL,</span></span>
<span><span class="co">#&gt;        keep = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the all input files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: character string for the file path where the output PLINK</span></span>
<span><span class="co">#&gt;           files will be stored.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; filter_sample: Character string, specifying the sample type to be</span></span>
<span><span class="co">#&gt;           retained. The choices are, "cases", "controls", "males" and</span></span>
<span><span class="co">#&gt;           "females". The default is "cases".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; keep_remove_sample_file: Character string, specifying the prefix of a</span></span>
<span><span class="co">#&gt;           space/tab-delimited text file with no header. For the samples</span></span>
<span><span class="co">#&gt;           that we want to keep or remove, the family IDs should be in</span></span>
<span><span class="co">#&gt;           the first column and within-family IDs in the second column.</span></span>
<span><span class="co">#&gt;           This file needs to be in the 'DataDir'. The default is</span></span>
<span><span class="co">#&gt;           'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     keep: Boolean value, 'TRUE' or 'FALSE' for specifying desired</span></span>
<span><span class="co">#&gt;           samples to keep or remove. The default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'NULL'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The output PLINK files with passed samples will be saved in</span></span>
<span><span class="co">#&gt;      ResultDir.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "casesPlink"</span></span>
<span><span class="co">#&gt;      filter_sample &lt;- "cases"</span></span>
<span><span class="co">#&gt;      keep_remove_sample_file &lt;- "samples_example"</span></span>
<span><span class="co">#&gt;      keep &lt;- FALSE</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      FilterPlinkSample(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, foutput = foutput, keep_remove_sample_file = keep_remove_sample_file,</span></span>
<span><span class="co">#&gt;          keep = keep</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-20">Function 20<a class="anchor" aria-label="anchor" href="#function-20"></a>
</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputeGeneticPC</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ComputeGeneticPC: Computing principal components from genetic</span></span>
<span><span class="co">#&gt; relationship matrix</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function performs principal components analysis (PCA) based</span></span>
<span><span class="co">#&gt;      on the variance-standardized relationship matrix (Purcell et al.</span></span>
<span><span class="co">#&gt;      2007).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Top principal components are generally used as covariates in</span></span>
<span><span class="co">#&gt;      association analysis regressions to help correct for population</span></span>
<span><span class="co">#&gt;      stratification</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      ComputeGeneticPC(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        countPC = 10,</span></span>
<span><span class="co">#&gt;        plotPC = TRUE,</span></span>
<span><span class="co">#&gt;        highLD_regions = NULL,</span></span>
<span><span class="co">#&gt;        ld_prunning = TRUE,</span></span>
<span><span class="co">#&gt;        window_size = 50,</span></span>
<span><span class="co">#&gt;        step_size = 5,</span></span>
<span><span class="co">#&gt;        r2_threshold = 0.02</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files. This file needs to be in 'DataDir.'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  countPC: Integer value, specifying the number of principal components.</span></span>
<span><span class="co">#&gt;           The default is 10.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   plotPC: Boolean value, 'TRUE' or 'FALSE', specifying whether to plot</span></span>
<span><span class="co">#&gt;           the first two PCs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: A R dataframe with genomic regions with high LD for</span></span>
<span><span class="co">#&gt;           using in finding pruned SNPs in the plots. The default is</span></span>
<span><span class="co">#&gt;           'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ld_prunning: Numeric value between 0 to 1 of pairwise r^2 threshold for</span></span>
<span><span class="co">#&gt;           LD-based filtering for pruned SNPs in the plots. The default</span></span>
<span><span class="co">#&gt;           is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; window_size: Integer value, specifying a window size in variant count</span></span>
<span><span class="co">#&gt;           or kilobase for LD-based filtering. The default is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; step_size: Integer value, specifying a variant count to shift the</span></span>
<span><span class="co">#&gt;           window at the end of each step for LD filtering for pruned</span></span>
<span><span class="co">#&gt;           SNPs in the plots. The default is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; r2_threshold: Numeric value between 0 to 1 of pairwise r^2 threshold</span></span>
<span><span class="co">#&gt;           for LD-based filtering for pruned SNPs in the plots. The</span></span>
<span><span class="co">#&gt;           default is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A dataframe with genetic principal components. The first two</span></span>
<span><span class="co">#&gt;      columns are IID (i.e., Individual Id) and FID (i.e., Family ID).</span></span>
<span><span class="co">#&gt;      The other columns are PCs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("highLD_hg19", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      highLD_regions &lt;- highLD_hg19</span></span>
<span><span class="co">#&gt;      ld_prunning &lt;- "TRUE"</span></span>
<span><span class="co">#&gt;      window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      countPC &lt;- 20</span></span>
<span><span class="co">#&gt;      ## Genetic PC</span></span>
<span><span class="co">#&gt;      GP &lt;- ComputeGeneticPC(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, highLD_regions = highLD_hg19, countPC = 20</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-21">Function 21<a class="anchor" aria-label="anchor" href="#function-21"></a>
</h4>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ClumpLD</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ClumpLD: Clumping SNPs using linkage disequilibrium between SNPs</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function, which is based on empirical estimations of linkage</span></span>
<span><span class="co">#&gt;      disequilibrium between SNPs, groups the SNP-based results across</span></span>
<span><span class="co">#&gt;      one or more datasets or analysis. This approach can be used in two</span></span>
<span><span class="co">#&gt;      basic scenarios: (i) To summarize the top X single SNP findings</span></span>
<span><span class="co">#&gt;      from a genome-wide scan as fewer clusters of connected SNPs (i.e.,</span></span>
<span><span class="co">#&gt;      to assess how many independent loci are associated). (ii) To give</span></span>
<span><span class="co">#&gt;      researchers a simple approach to merge sets of data from multiple</span></span>
<span><span class="co">#&gt;      studies when those studies may have used various marker sets for</span></span>
<span><span class="co">#&gt;      genotyping.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The clumping process begins with the index SNPs that are</span></span>
<span><span class="co">#&gt;      significant at threshold p1 and have not yet been clumped. It then</span></span>
<span><span class="co">#&gt;      creates clumps of all additional SNPs that are within a specified</span></span>
<span><span class="co">#&gt;      kb of the index SNP and that are in linkage disequilibrium with</span></span>
<span><span class="co">#&gt;      the index SNP based on an r-squared threshold. Following that,</span></span>
<span><span class="co">#&gt;      these SNPs are filtered based on the outcome for that SNP. As this</span></span>
<span><span class="co">#&gt;      method is greedy (Purcell et al. 2007), each SNP will, at most,</span></span>
<span><span class="co">#&gt;      only appear in one clump. The P value and ALLELES would always, at</span></span>
<span><span class="co">#&gt;      random, be chosen from the first input file if the same SNP</span></span>
<span><span class="co">#&gt;      appeared in several input files in SNPdata argument. Instead of</span></span>
<span><span class="co">#&gt;      the best p-value, the function refer to the SNP that has the</span></span>
<span><span class="co">#&gt;      strongest LD to the index as the best proxy. Based on the genotype</span></span>
<span><span class="co">#&gt;      data, the SNP with the highest LD will be the same for all input</span></span>
<span><span class="co">#&gt;      files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      ClumpLD(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        SNPdata,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        clump_p1,</span></span>
<span><span class="co">#&gt;        clump_p2,</span></span>
<span><span class="co">#&gt;        clump_r2,</span></span>
<span><span class="co">#&gt;        clump_kb,</span></span>
<span><span class="co">#&gt;        byCHR = TRUE,</span></span>
<span><span class="co">#&gt;        clump_best = TRUE,</span></span>
<span><span class="co">#&gt;        clump_index_first = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files which will be used to calculate linkage</span></span>
<span><span class="co">#&gt;           disequilibrium between the SNPs. This actual genotype data</span></span>
<span><span class="co">#&gt;           may or may not be the same dataset that was used to generate</span></span>
<span><span class="co">#&gt;           the summary statistics. This file needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  SNPdata: A list of R dataframes containing a single or multiple</span></span>
<span><span class="co">#&gt;           summary statistics with SNP and P (i.e., p-values) in</span></span>
<span><span class="co">#&gt;           mandatory column headers. Other columns could be present.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p1: Numeric value, specifying the significance threshold for</span></span>
<span><span class="co">#&gt;           index SNPs. The default is 0.0001.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p2: Numeric value, specifying the secondary significance</span></span>
<span><span class="co">#&gt;           threshold for clumped SNPs. The default is 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_r2: Numeric value, specifying the LD threshold for clumping. The</span></span>
<span><span class="co">#&gt;           default is 0.50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_kb: Integer value, specifying the physical distance threshold in</span></span>
<span><span class="co">#&gt;           base-pair for clumping. The default is 250.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           perform the clumping chromosome-wise.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_best: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           select and output the best SNP from each clump. Default is</span></span>
<span><span class="co">#&gt;           'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_index_first: Boolean value, 'TRUE' or 'FALSE', specifying whether</span></span>
<span><span class="co">#&gt;           to force the index SNP to appear first in each clump. This</span></span>
<span><span class="co">#&gt;           option should typically be 'TRUE' if clump_best is 'TRUE.'</span></span>
<span><span class="co">#&gt;           Default is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list with two dataframes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      BestClump: a dataframe with eight columns showing the single best</span></span>
<span><span class="co">#&gt;      proxy SNP for each index SNP with columns "INDEX"(Index SNP</span></span>
<span><span class="co">#&gt;      identifier), "PSNP"(Best proxy SNP), "RSQ LD"(r-squared) between</span></span>
<span><span class="co">#&gt;      index and proxy, "KB"(Physical distance between index and proxy),</span></span>
<span><span class="co">#&gt;      P(p-value for proxy SNP), "ALLELES"(The associated haplotypes for</span></span>
<span><span class="co">#&gt;      the index and proxy SNP), and "F"(Which file used for clumping</span></span>
<span><span class="co">#&gt;      from which this result came from).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      AllClump: a dataframe with eight columns providing a detailed</span></span>
<span><span class="co">#&gt;      summary of each clump identified by PLINK. It includes "INDEX_SNP"</span></span>
<span><span class="co">#&gt;      (the identifier for the index SNP that represents the clump),</span></span>
<span><span class="co">#&gt;      "SNP" (the SNP being reported, which for the index SNP is the same</span></span>
<span><span class="co">#&gt;      as INDEX_SNP), "DISTANCE" (the physical distance in base pairs</span></span>
<span><span class="co">#&gt;      between the index SNP and the reported SNP, with 0.0 indicating</span></span>
<span><span class="co">#&gt;      the index itself), "RSQ" (the r-squared value showing the degree</span></span>
<span><span class="co">#&gt;      of linkage disequilibrium between the index SNP and the SNP in the</span></span>
<span><span class="co">#&gt;      clump), "ALLELES" (the allele information, which in some cases may</span></span>
<span><span class="co">#&gt;      appear misaligned if the data isn’t formatted as expected), "F" (a</span></span>
<span><span class="co">#&gt;      statistic or indicator related to the association test, which may</span></span>
<span><span class="co">#&gt;      be NA when not applicable), "P" (the p-value for the association</span></span>
<span><span class="co">#&gt;      test of the SNP), and "CHR" (the chromosome on which the SNP is</span></span>
<span><span class="co">#&gt;      located).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex1", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex2", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      SNPdata &lt;- list(Summary_Stat_Ex1, Summary_Stat_Ex2)</span></span>
<span><span class="co">#&gt;      clump_p1 &lt;- 0.0001</span></span>
<span><span class="co">#&gt;      clump_p2 &lt;- 0.001</span></span>
<span><span class="co">#&gt;      clump_r2 &lt;- 0.5</span></span>
<span><span class="co">#&gt;      clump_kb &lt;- 250</span></span>
<span><span class="co">#&gt;      byCHR &lt;- TRUE</span></span>
<span><span class="co">#&gt;      clumpedResult &lt;- ClumpLD(</span></span>
<span><span class="co">#&gt;          DataDir, finput, SNPdata, ResultDir, clump_p1,</span></span>
<span><span class="co">#&gt;          clump_p2, clump_r2, clump_kb, byCHR</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-22">Function 22<a class="anchor" aria-label="anchor" href="#function-22"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GetMFPlink</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GetMFPlink: Getting male and female PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function prepares separate male and female PLINK binary files</span></span>
<span><span class="co">#&gt;      from combined PLINK files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      GetMFPlink(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput,</span></span>
<span><span class="co">#&gt;        sex,</span></span>
<span><span class="co">#&gt;        xplink = FALSE,</span></span>
<span><span class="co">#&gt;        autoplink = FALSE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string for the file path of the input PLINK binary</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      sex: Boolean value, 'males' or 'females', specifying output PLINK</span></span>
<span><span class="co">#&gt;           binary files with male or female samples.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   xplink: Boolean value, 'TRUE' or 'FALSE', specifying output PLINK</span></span>
<span><span class="co">#&gt;           binary files with only X chromosome or not. Default is</span></span>
<span><span class="co">#&gt;           'FALSE.'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; autoplink: Boolean value, 'TRUE' or 'FALSE', specifying output PLINK</span></span>
<span><span class="co">#&gt;           binary files with only autosome or not. Default is 'FALSE.'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      None</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      sex &lt;- "females"</span></span>
<span><span class="co">#&gt;      x &lt;- GetMFPlink(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          finput = finput, foutput = foutput, sex = sex,</span></span>
<span><span class="co">#&gt;          xplink = FALSE, autoplink = FALSE</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-23">Function 23<a class="anchor" aria-label="anchor" href="#function-23"></a>
</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">plinkVCF</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; plinkVCF: Converting VCF files to PLINK binary files and vice-versa.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function performs the conversion between VCF files to PLINK</span></span>
<span><span class="co">#&gt;      binary formats.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      For VCF to PLINK files conversion, if you do not specify any FAM</span></span>
<span><span class="co">#&gt;      file when you are converting from VCF to PLINK format, then PLINK</span></span>
<span><span class="co">#&gt;      will just create a 'dummy' FAM file with the same name as your</span></span>
<span><span class="co">#&gt;      dataset with missing phenotypes and missing sex.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      plinkVCF(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        foutput,</span></span>
<span><span class="co">#&gt;        VtoP = FALSE,</span></span>
<span><span class="co">#&gt;        PtoV = TRUE,</span></span>
<span><span class="co">#&gt;        Famfile = NULL,</span></span>
<span><span class="co">#&gt;        PVbyCHR = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files and all other input files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files or vcf files. This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if filtering option for the SNPs is chosen. The</span></span>
<span><span class="co">#&gt;           default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     VtoP: Boolean value, 'TRUE' or 'FALSE', specifying the conversion</span></span>
<span><span class="co">#&gt;           of VCF files to PLINK binary files or not. The default is</span></span>
<span><span class="co">#&gt;           'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     PtoV: Boolean value, 'TRUE' or 'FALSE', specifying the conversion</span></span>
<span><span class="co">#&gt;           of PLINK binary files to VCF files or not. The default is</span></span>
<span><span class="co">#&gt;           'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  Famfile: Character string, specifying the name of the original .fam</span></span>
<span><span class="co">#&gt;           file if VtoP was set to be 'TRUE'. This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'. The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  PVbyCHR: Boolean value, 'TRUE' or 'FALSE' specifying to do the PLINK</span></span>
<span><span class="co">#&gt;           to vcf conversion chromosome-wise or not. The default is</span></span>
<span><span class="co">#&gt;           'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'NULL'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The output files will be saved in 'ResultDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example" # Plink file</span></span>
<span><span class="co">#&gt;      foutput &lt;- "GXwasR_example1"</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      PtoV &lt;- TRUE</span></span>
<span><span class="co">#&gt;      VtoP &lt;- FALSE</span></span>
<span><span class="co">#&gt;      Famfile &lt;- NULL</span></span>
<span><span class="co">#&gt;      PVbyCHR &lt;- FALSE</span></span>
<span><span class="co">#&gt;      plinkVCF(DataDir, ResultDir, finput, foutput, VtoP, PtoV, Famfile, PVbyCHR)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-24">Function 24<a class="anchor" aria-label="anchor" href="#function-24"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MergeRegion</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; MergeRegion: Merging two sets of PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function combines the two genotype datasets based on either</span></span>
<span><span class="co">#&gt;      common SNPs or all the SNPs between them.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      MergeRegion(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir,</span></span>
<span><span class="co">#&gt;        finput1,</span></span>
<span><span class="co">#&gt;        finput2,</span></span>
<span><span class="co">#&gt;        foutput,</span></span>
<span><span class="co">#&gt;        use_common_snps = TRUE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is tempdir().</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  finput1: Character string, specifying the prefix of the first input</span></span>
<span><span class="co">#&gt;           PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  finput2: Character string, specifying the prefix of the first input</span></span>
<span><span class="co">#&gt;           PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if filtering option for the SNPs is chosen. The</span></span>
<span><span class="co">#&gt;           default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; use_common_snps: Boolean value, TRUE or FALSE, specifying to use common</span></span>
<span><span class="co">#&gt;           SNPs for merging or to use all the SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      The output PLINK files will be saved in ResultDir.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput1 &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      finput2 &lt;- "GXwasR_example_imputed"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Test_output"</span></span>
<span><span class="co">#&gt;      y &lt;- MergeRegion(DataDir, ResultDir, finput1, finput2, foutput, use_common_snps = TRUE)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-25">Function 25<a class="anchor" aria-label="anchor" href="#function-25"></a>
</h4>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterAllele</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; FilterAllele: Filtering out the multi-allelic variants</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function filters out the multi-allelic SNPs from the input</span></span>
<span><span class="co">#&gt;      dataset.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      FilterAllele(DataDir, ResultDir, finput, foutput)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files. If multi-allelic variants are present, this</span></span>
<span><span class="co">#&gt;           file will be produced after filtering out these variants.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'NULL'. After multi-allelic variant filtering, the filtered PLINK</span></span>
<span><span class="co">#&gt;      files with only biallelic SNPs will be saved in 'ResultDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Filter_Test"</span></span>
<span><span class="co">#&gt;      x &lt;- FilterAllele(DataDir, ResultDir, finput, foutput)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-26">Function 26<a class="anchor" aria-label="anchor" href="#function-26"></a>
</h4>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">PlinkSummary</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; PlinkSummary: Summary of PLINK format genotype dataset</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      PlinkSummary: Summary of PLINK format genotype dataset</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      PlinkSummary(DataDir, ResultDir = tempdir(), finput)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path of the PLINK program to</span></span>
<span><span class="co">#&gt;           be set up.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files. This file needs to be in DataDir.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function is called for its side effect: printing summary</span></span>
<span><span class="co">#&gt;      statistics to the console. It returns 'NULL' invisibly.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      x &lt;- PlinkSummary(DataDir, ResultDir, finput)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-27">Function 27<a class="anchor" aria-label="anchor" href="#function-27"></a>
</h4>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterSNP</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; FilterSNP: Filter out SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      FilterSNP: Filter out SNPs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      FilterSNP(DataDir, ResultDir, finput, foutput, SNPvec, extract = FALSE)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  foutput: Character string, specifying the prefix of the output PLINK</span></span>
<span><span class="co">#&gt;           binary files if the filtering option for the SNPs is chosen.</span></span>
<span><span class="co">#&gt;           The default is "FALSE".</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   SNPvec: R dataframe with SNP names to be excluded.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  extract: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           extract the snps or discard the snps. The default is 'FALSE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      'NULL'. The filtered file will be saved in 'ResultDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      SNPvec &lt;- c("rs6529954", "rs12858640", "rs5962098")</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      foutput &lt;- "Filter_Test"</span></span>
<span><span class="co">#&gt;      FilterSNP(DataDir, ResultDir, finput, foutput, SNPvec = SNPvec, extract = TRUE)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-28">Function 28<a class="anchor" aria-label="anchor" href="#function-28"></a>
</h4>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">DummyCovar</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; DummyCovar: Recode a categorical variable to a set of binary dummy</span></span>
<span><span class="co">#&gt; variables.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      When dealing with categorical variables in genetic analysis using,</span></span>
<span><span class="co">#&gt;      a common approach is to convert these into dummy variables for</span></span>
<span><span class="co">#&gt;      proper analysis (Purcell et al. 2007). This function creates K-1</span></span>
<span><span class="co">#&gt;      new dummy variables for a variable with K categories. One level is</span></span>
<span><span class="co">#&gt;      automatically excluded from the dummy variables which serves as</span></span>
<span><span class="co">#&gt;      the reference category for subsequent analyses. This setup</span></span>
<span><span class="co">#&gt;      implicitly sets the excluded that category as the baseline against</span></span>
<span><span class="co">#&gt;      which other categories are compared.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DummyCovar(DataDir, ResultDir = DataDir, bfile, incovar, outcovar)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is 'tempdir()'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    bfile: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for which covariate file will be generated.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  incovar: Character string, specifying the prefix of the input</span></span>
<span><span class="co">#&gt;           covariate file. First two columns will be, FID (i.e., Family</span></span>
<span><span class="co">#&gt;           ID) and IID (i.e., Sample ID) and rest of the columns are</span></span>
<span><span class="co">#&gt;           covariates.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; outcovar: Character string, specifying the prefix of the Output</span></span>
<span><span class="co">#&gt;           covariate file</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      R dataframe object with covariates</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Purcell S, Neale B, Todd-Brown K, Thomas L, Ferreira MAR, Bender</span></span>
<span><span class="co">#&gt;      D, Maller J, Sklar P, de Bakker PIW, Daly MJ, others (2007).</span></span>
<span><span class="co">#&gt;      "PLINK: A Tool Set for Whole-Genome Association and</span></span>
<span><span class="co">#&gt;      Population-Based Linkage Analyses." _The American Journal of Human</span></span>
<span><span class="co">#&gt;      Genetics_, *81*(3), 559-575. doi:10.1086/519795</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1086/519795&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      bfile &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      incovar &lt;- "covarfile_w_pc_age.txt"</span></span>
<span><span class="co">#&gt;      outcovar &lt;- "dummycovarfile"</span></span>
<span><span class="co">#&gt;      dummy_covars &lt;- DummyCovar(</span></span>
<span><span class="co">#&gt;          DataDir = DataDir, ResultDir = ResultDir,</span></span>
<span><span class="co">#&gt;          bfile = bfile, incovar = incovar,</span></span>
<span><span class="co">#&gt;          outcovar = outcovar</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-29">Function 29<a class="anchor" aria-label="anchor" href="#function-29"></a>
</h4>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXWASmiami</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Miami plot</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function generates Miami plots for GWAS and XWAS.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      GXWASmiami(</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        FemaleWAS,</span></span>
<span><span class="co">#&gt;        MaleWAS,</span></span>
<span><span class="co">#&gt;        snp_pval = 1e-08,</span></span>
<span><span class="co">#&gt;        Xchr = FALSE</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the folder path where the outputs will</span></span>
<span><span class="co">#&gt;           be saved.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; FemaleWAS: R dataframe of summary statistics of GWAS or XWAS of female</span></span>
<span><span class="co">#&gt;           samples with four columns, SNP(Variant), CHR(Chromosome</span></span>
<span><span class="co">#&gt;           number), POS(Base pair position) and pvalue(P-value of the</span></span>
<span><span class="co">#&gt;           test). This can be generated by running FM01comb or FM02comb</span></span>
<span><span class="co">#&gt;           model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  MaleWAS: R dataframe of summary statistics of GWAS or XWAS of male</span></span>
<span><span class="co">#&gt;           samples with four columns, SNP(Variant), CHR(Chromosome</span></span>
<span><span class="co">#&gt;           number), POS(Base pair position) and pvalue(P-value of the</span></span>
<span><span class="co">#&gt;           test). This can be generated by running FM01comb or FM02comb</span></span>
<span><span class="co">#&gt;           model with GXWAS function.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; snp_pval: Numeric value as p-value threshold for annotation. SNPs below</span></span>
<span><span class="co">#&gt;           this p-value will be annotated on the plot. The default is</span></span>
<span><span class="co">#&gt;           1e-08.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Xchr: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           generate Miami plot for stratified XWAS or not. The default</span></span>
<span><span class="co">#&gt;           is 'TRUE'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Invisibly returns 'NULL'. Generates and saves Miami plots as a</span></span>
<span><span class="co">#&gt;      side effect.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Ffile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Mfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      FemaleWAS &lt;- na.omit(Ffile[, c("SNP", "CHR", "BP", "P")])</span></span>
<span><span class="co">#&gt;      colnames(FemaleWAS) &lt;- c("SNP", "CHR", "POS", "pvalue")</span></span>
<span><span class="co">#&gt;      MaleWAS &lt;- na.omit(Mfile[, c("SNP", "CHR", "BP", "P")])</span></span>
<span><span class="co">#&gt;      colnames(MaleWAS) &lt;- c("SNP", "CHR", "POS", "pvalue")</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      GXWASmiami(FemaleWAS = FemaleWAS, MaleWAS = MaleWAS, snp_pval = 0.05)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-30">Function 30<a class="anchor" aria-label="anchor" href="#function-30"></a>
</h4>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SumstatGenCorr</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; SumstatGenCorr: Genetic Correlation Calculation from GWAS Summary</span></span>
<span><span class="co">#&gt; Statistics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function calculates the genetic correlation between two</span></span>
<span><span class="co">#&gt;      summary statistics using a specified reference linkage</span></span>
<span><span class="co">#&gt;      disequilibrium (LD) matrix from the UK Biobank.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SumstatGenCorr(</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        referenceLD,</span></span>
<span><span class="co">#&gt;        sumstat1,</span></span>
<span><span class="co">#&gt;        sumstat2,</span></span>
<span><span class="co">#&gt;        Nref = 335265,</span></span>
<span><span class="co">#&gt;        N0 = min(sumstat1$N),</span></span>
<span><span class="co">#&gt;        eigen.cut = "automatic",</span></span>
<span><span class="co">#&gt;        lim = exp(-18),</span></span>
<span><span class="co">#&gt;        parallel = FALSE,</span></span>
<span><span class="co">#&gt;        numCores = 2</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Directory where results should be saved.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; referenceLD: Reference LD matrix identifier. These are the LD matrices</span></span>
<span><span class="co">#&gt;           and their eigen-decomposition from 335,265 genomic British UK</span></span>
<span><span class="co">#&gt;           Biobank individuals. Two sets of reference panel are</span></span>
<span><span class="co">#&gt;           provided:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            1. 307,519 QCed UK Biobank Axiom Array SNPs. The size is</span></span>
<span><span class="co">#&gt;               about 7.5 GB after unzipping.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;            2. 1,029,876 QCed UK Biobank imputed SNPs. The size is about</span></span>
<span><span class="co">#&gt;               31 GB after unzipping. Although it takes more time, using</span></span>
<span><span class="co">#&gt;               the imputed panel provides more accurate estimates of</span></span>
<span><span class="co">#&gt;               genetic correlations. Therefore if the GWAS includes most</span></span>
<span><span class="co">#&gt;               of the HapMap3 SNPs, then it is recommend using the</span></span>
<span><span class="co">#&gt;               imputed reference panel.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; sumstat1: Data frame for the first set of summary statistics. The input</span></span>
<span><span class="co">#&gt;           data frame should include following columns: SNP, SNP ID; A1,</span></span>
<span><span class="co">#&gt;           effect allele; A2, reference allele; N, sample size; Z,</span></span>
<span><span class="co">#&gt;           z-score; If Z is not given, alternatively, you may provide:</span></span>
<span><span class="co">#&gt;           b, estimate of marginal effect in GWAS; se, standard error of</span></span>
<span><span class="co">#&gt;           the estimates of marginal effects in GWAS.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; sumstat2: Data frame for the second set of summary statistics. The</span></span>
<span><span class="co">#&gt;           input data frame should include following columns: SNP, SNP</span></span>
<span><span class="co">#&gt;           ID; A1, effect allele; A2, reference allele; N, sample size;</span></span>
<span><span class="co">#&gt;           Z, z-score; If Z is not given, alternatively, you may</span></span>
<span><span class="co">#&gt;           provide: b, estimate of marginal effect in GWAS; se, standard</span></span>
<span><span class="co">#&gt;           error of the estimates of marginal effects in GWAS.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Nref: Sample size of the reference sample where LD is computed. If</span></span>
<span><span class="co">#&gt;           the default UK Biobank reference sample is used, Nref =</span></span>
<span><span class="co">#&gt;           335265</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;       N0: Number of individuals included in both cohorts. The estimated</span></span>
<span><span class="co">#&gt;           genetic correlation is usually robust against misspecified</span></span>
<span><span class="co">#&gt;           N0. If not given, the default value is set to the minimum</span></span>
<span><span class="co">#&gt;           sample size across all SNPs in cohort 1 and cohort 2.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; eigen.cut: Which eigenvalues and eigenvectors in each LD score matrix</span></span>
<span><span class="co">#&gt;           should be used for HDL. Users are allowed to specify a</span></span>
<span><span class="co">#&gt;           numeric value between 0 and 1 for eigen.cut. For example,</span></span>
<span><span class="co">#&gt;           eigen.cut = 0.99 means using the leading eigenvalues</span></span>
<span><span class="co">#&gt;           explaining 99% of the variance and their correspondent</span></span>
<span><span class="co">#&gt;           eigenvectors. If the default 'automatic' is used, the</span></span>
<span><span class="co">#&gt;           eigen.cut gives the most stable heritability estimates will</span></span>
<span><span class="co">#&gt;           be used.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      lim: Tolerance limitation, default lim = exp(-18).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; parallel: Boolean value, TRUE or FALSE for whether to perform parallel</span></span>
<span><span class="co">#&gt;           computation. The default is FALSE</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; numCores: The number of cores to be used. The default is 2.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Details:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function requires access to the reference LD data via an</span></span>
<span><span class="co">#&gt;      environment variable. You must set one of the following</span></span>
<span><span class="co">#&gt;      environment variables to the appropriate directory:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'UKB_ARRAY_PATH' for the Axiom Array reference</span></span>
<span><span class="co">#&gt;           ('UKB_array_SVD_eigen90_extraction')</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'UKB_IMPUTED_PATH' for the full imputed reference</span></span>
<span><span class="co">#&gt;           ('UKB_imputed_SVD_eigen99_extraction')</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'UKB_IMPUTED_HAPMAP2_PATH' for the imputed HapMap2 subset</span></span>
<span><span class="co">#&gt;           ('UKB_imputed_hapmap2_SVD_eigen99_extraction')</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list is returned with:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'rg': The estimated genetic correlation.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'rg.se': The standard error of the estimated genetic</span></span>
<span><span class="co">#&gt;           correlation.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'P': P-value based on Wald test.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'estimates.df': A detailed matrix includes the estimates and</span></span>
<span><span class="co">#&gt;           standard errors of heritabilities, genetic covariance and</span></span>
<span><span class="co">#&gt;           genetic correlation.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         • 'eigen.use': The eigen.cut used in computation.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; References:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Ning Z, Pawitan Y, Shen X (2020). "High-definition likelihood</span></span>
<span><span class="co">#&gt;      inference of genetic correlations across human complex traits."</span></span>
<span><span class="co">#&gt;      _Nature Genetics_, *52*, 859-864. doi:10.1038/s41588-020-0676-4</span></span>
<span><span class="co">#&gt;      &lt;https://doi.org/10.1038/s41588-020-0676-4&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      sumstat1 &lt;- GXwasR:::simulateSumstats()</span></span>
<span><span class="co">#&gt;      sumstat2 &lt;- GXwasR:::simulateSumstats()</span></span>
<span><span class="co">#&gt;      if (nzchar(Sys.getenv("UKB_IMPUTED_HAPMAP2_PATH"))) {</span></span>
<span><span class="co">#&gt;          res &lt;- SumstatGenCorr(</span></span>
<span><span class="co">#&gt;              ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;              referenceLD = "UKB_imputed_hapmap2_SVD_eigen99_extraction",</span></span>
<span><span class="co">#&gt;              sumstat1 = sumstat1,</span></span>
<span><span class="co">#&gt;              sumstat2 = sumstat2,</span></span>
<span><span class="co">#&gt;              parallel = TRUE</span></span>
<span><span class="co">#&gt;          )</span></span>
<span><span class="co">#&gt;      }</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-31">Function 31<a class="anchor" aria-label="anchor" href="#function-31"></a>
</h4>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">LDPrune</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; LDPrune: Performs LD pruning on SNP data using PLINK</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function utilizes PLINK to perform LD pruning on genetic</span></span>
<span><span class="co">#&gt;      data. It identifies and removes SNPs that are in high linkage</span></span>
<span><span class="co">#&gt;      disequilibrium with each other within specified windows.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      LDPrune(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        window_size = 50,</span></span>
<span><span class="co">#&gt;        step_size = 5,</span></span>
<span><span class="co">#&gt;        r2_threshold = 0.2</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string representing the file path of the input</span></span>
<span><span class="co">#&gt;           PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored, defaulting to a temporary directory.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; window_size: Integer, specifying the number of SNPs to include in the</span></span>
<span><span class="co">#&gt;           sliding window.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; step_size: Integer, specifying the number of SNPs the window moves over</span></span>
<span><span class="co">#&gt;           in each step.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; r2_threshold: Numeric, specifying the R^2 threshold for LD pruning.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Returns a character vector of SNP identifiers that remain after LD</span></span>
<span><span class="co">#&gt;      pruning or NULL if an error occurs.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      prunedSNPs &lt;- LDPrune(DataDir, finput, ResultDir, 50, 5, 0.2)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-32">Function 32<a class="anchor" aria-label="anchor" href="#function-32"></a>
</h4>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">executePlinkMAF</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; executePlinkMAF: Execute PLINK to Calculate Minor Allele Frequencies</span></span>
<span><span class="co">#&gt; (MAF)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function executes PLINK to calculate minor allele frequencies</span></span>
<span><span class="co">#&gt;      (MAF) for a given dataset. It sets up the necessary PLINK</span></span>
<span><span class="co">#&gt;      environment, runs the PLINK command, and returns the MAF results</span></span>
<span><span class="co">#&gt;      as a DataFrame. Intermediate files generated by PLINK are cleaned</span></span>
<span><span class="co">#&gt;      up after execution.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      executePlinkMAF(DataDir, ResultDir, finput)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character. Directory containing the input PLINK files (.bed,</span></span>
<span><span class="co">#&gt;           .bim, .fam).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character. Directory to store the output files generated by</span></span>
<span><span class="co">#&gt;           PLINK.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character. Base name of the PLINK input files (without</span></span>
<span><span class="co">#&gt;           extensions).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataFrame containing the minor allele frequency (MAF) results.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      maf_data &lt;- executePlinkMAF(DataDir, ResultDir, finput)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-33">Function 33<a class="anchor" aria-label="anchor" href="#function-33"></a>
</h4>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputeLD</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ComputeLD: Compute Linkage Disequilibrium (LD) for SNP Data</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function computes linkage disequilibrium (LD) statistics for</span></span>
<span><span class="co">#&gt;      SNP data using PLINK. It allows for computation across all SNPs or</span></span>
<span><span class="co">#&gt;      within specific chromosomes.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      ComputeLD(DataDir, ResultDir, finput, ByCHR = FALSE, CHRnum = NULL, r2_LD)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: Character string representing the file path of the input</span></span>
<span><span class="co">#&gt;           PLINK binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: Character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored, defaulting to a temporary directory.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    ByCHR: Logical indicating whether to perform the computation by</span></span>
<span><span class="co">#&gt;           chromosome. The default is FALSE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   CHRnum: If ByCHR is TRUE, specifies the chromosome number for which</span></span>
<span><span class="co">#&gt;           LD should be computed. The default is NULL.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    r2_LD: The threshold for r-squared LD values to report in the</span></span>
<span><span class="co">#&gt;           output.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Returns a data frame containing the computed LD values among SNPs,</span></span>
<span><span class="co">#&gt;      read from the output file generated by PLINK.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      snpld &lt;- ComputeLD(</span></span>
<span><span class="co">#&gt;          DataDir = GXwasR:::GXwasR_data(), ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;          finput = "GXwasR_example", ByCHR = TRUE, CHRnum = 1, r2_LD = 0.2</span></span>
<span><span class="co">#&gt;      )</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<p>Please follow these tutorials to learn more about the functionality
of the package GXwasR.</p>
<p>Tutorial for performing post-imputation QC followed by sex-aware
association tests: (vignette:Use of GXwasR)</p>
<ul>
<li>Tutorial for performing pre-imputation QC using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/preimputationQC.html" class="external-link">https://boseb.github.io/GXwasR/articles/preimputationQC.html</a>
</li>
<li>Tutorial for performing post-imputation QC using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/postimputationQC.html" class="external-link">https://boseb.github.io/GXwasR/articles/postimputationQC.html</a>
</li>
<li>Tutorial for running GWAS, XWAS and Sex-differential Tests using
GXwasR: <a href="https://boseb.github.io/GXwasR/articles/gwas_models.html" class="external-link">https://boseb.github.io/GXwasR/articles/gwas_models.html</a>
</li>
<li>Tutorial for computing Polygenic Risk Score using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/GXwasR_PRS.html" class="external-link">https://boseb.github.io/GXwasR/articles/GXwasR_PRS.html</a>
</li>
<li>Tutorial for ancestry estimation: <a href="https://boseb.github.io/GXwasR/articles/decoding_ancestry.html" class="external-link">https://boseb.github.io/GXwasR/articles/decoding_ancestry.html</a>
</li>
<li>Tutorial for meta analysis using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/meta_analysis.html" class="external-link">https://boseb.github.io/GXwasR/articles/meta_analysis.html</a>
</li>
<li>Tutorial for genetic correlation using GWAS Summary Statistics using
GXwasR: <a href="https://boseb.github.io/GXwasR/articles/genetic_correlation_sumstat.html" class="external-link">https://boseb.github.io/GXwasR/articles/genetic_correlation_sumstat.html</a>
</li>
</ul>
<div class="section level3">
<h3 id="running-examples">Running Examples<a class="anchor" aria-label="anchor" href="#running-examples"></a>
</h3>
<div class="section level4">
<h4 id="qcsnp">
<code>QCsnp()</code><a class="anchor" aria-label="anchor" href="#qcsnp"></a>
</h4>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>,</span>
<span>    geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>,</span>
<span>    ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>,</span>
<span>    caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using PLINK v1.9.0-b.7.7 64-bit (22 Oct 2024)</span></span>
<span><span class="co">#&gt; ℹ 4214 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; ℹ 5467 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; ℹ No filter based on differential missingness will be applied.</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files prefixed as ,Test_output, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 26527 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 16846 SNPs after filtering.</span></span>
<span><span class="co">#&gt; This message is displayed once every 8 hours.</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; $MonomorSNPs</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $DiffMissSNPs</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="qcsample">
<code>QCsample()</code><a class="anchor" aria-label="anchor" href="#qcsample"></a>
</h4>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">imiss</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span><span class="va">het</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="va">small_sample_mod</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">IBDmatrix</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">ambi_out</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="co">#'</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsample.html">QCsample</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,</span>
<span>    foutput <span class="op">=</span> <span class="va">foutput</span>, imiss <span class="op">=</span> <span class="va">imiss</span>, het <span class="op">=</span> <span class="va">het</span>, IBD <span class="op">=</span> <span class="va">IBD</span>,</span>
<span>    ambi_out <span class="op">=</span> <span class="va">ambi_out</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; • Plots are initiated.</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for heterozygosity threshold: 3</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness and heterozygosity: 3</span></span>
<span><span class="co">#&gt; ℹ No. of samples marked to be filtered out for IDB after missingness and heterozygosity filter: 2</span></span>
<span><span class="co">#&gt; ℹ No. of samples in input PLINK files: 276</span></span>
<span><span class="co">#&gt; ℹ No. of samples in output PLINK files: 271</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, Test_output with final samples are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX.</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">HM</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">FID</th>
<th align="left">IID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">78</td>
<td align="left">EUR_GBR</td>
<td align="left">HG00240</td>
</tr>
<tr class="even">
<td align="left">166</td>
<td align="left">EUR_FIN</td>
<td align="left">HG00367</td>
</tr>
<tr class="odd">
<td align="left">220</td>
<td align="left">EUR_TSI</td>
<td align="left">NA20585</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">Failed_Missingness</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
</tr></thead>
<tbody></tbody>
</table>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">Failed_heterozygosity</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">FID</th>
<th align="left">IID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">82</td>
<td align="left">EUR_FIN</td>
<td align="left">HG00367</td>
</tr>
<tr class="even">
<td align="left">155</td>
<td align="left">EUR_GBR</td>
<td align="left">HG00240</td>
</tr>
<tr class="odd">
<td align="left">220</td>
<td align="left">EUR_TSI</td>
<td align="left">NA20585</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">Failed_IBD</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">V1</th>
<th align="left">V2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">112</td>
<td align="left">EUR_GBR</td>
<td align="left">HG00119</td>
</tr>
<tr class="even">
<td align="left">117</td>
<td align="left">EUR_GBR</td>
<td align="left">HG00124</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">Missingness_results</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">V1</th>
<th align="left">V2</th>
<th align="left">V3</th>
<th align="left">V4</th>
<th align="left">V5</th>
<th align="left">V6</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">FID</td>
<td align="left">IID</td>
<td align="left">MISS_PHENO</td>
<td align="left">N_MISS</td>
<td align="left">N_GENO</td>
<td align="left">F_MISS</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="left">N</td>
<td align="left">0</td>
<td align="left">26515</td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="left">N</td>
<td align="left">0</td>
<td align="left">26515</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="left">N</td>
<td align="left">0</td>
<td align="left">26515</td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="left">N</td>
<td align="left">0</td>
<td align="left">26515</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="left">N</td>
<td align="left">0</td>
<td align="left">26515</td>
<td align="left">0</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="ancestrycheck">
<code>AncestryCheck()</code><a class="anchor" aria-label="anchor" href="#ancestrycheck"></a>
</h4>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_data_study_sample_ancestry"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="st">"HapMapIII_NCBI36"</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">study_pop</span> <span class="op">&lt;-</span> <span class="va">example_data_study_sample_ancestry</span> <span class="co"># PreimputeEX</span></span>
<span><span class="va">studyLD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">studyLD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">studyLD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">filterSNP</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">studyLD</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">referLD</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">referLD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">referLD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">referLD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">outlier_threshold</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AncestryCheck.html">AncestryCheck</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,</span>
<span>    reference <span class="op">=</span> <span class="va">reference</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>,</span>
<span>    study_pop <span class="op">=</span> <span class="va">study_pop</span>, studyLD <span class="op">=</span> <span class="va">studyLD</span>, referLD <span class="op">=</span> <span class="va">referLD</span>,</span>
<span>    outlierOf <span class="op">=</span> <span class="st">"EUR"</span>, outlier <span class="op">=</span> <span class="va">outlier</span>, outlier_threshold <span class="op">=</span> <span class="va">outlier_threshold</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 'HapMapIII_NCBI36' reference data found at /Users/mayerdav/Downloads/HapMapIII_NCBI36.</span></span>
<span><span class="co">#&gt; 4214 ambiguous SNPs removed from study data.</span></span>
<span><span class="co">#&gt; 111854 ambiguous SNPs removed from reference data.</span></span>
<span><span class="co">#&gt; ! LD pruning is recommended for reference dataset. Set referLD = TRUE.</span></span>
<span><span class="co">#&gt; ! LD pruning is recommended for study dataset. Set studyLD = TRUE.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Number of overlapping SNPs between study and reference data using rsID: 3722</span></span>
<span><span class="co">#&gt; This message is displayed once every 8 hours.</span></span></code></pre></div>
<div class="float">
<img src="overview-AncestryCheck-1.png" alt="plot of chunk overview-AncestryCheck"><div class="figcaption">plot of chunk overview-AncestryCheck</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ No allele flips between study and reference data.</span></span>
<span><span class="co">#&gt; ℹ 20 samples are outliers of selected reference population.</span></span>
<span><span class="co">#&gt; ℹ 168 samples are NOT outliers of selected reference population.</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">Outlier_samples</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00179</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00180</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00185</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00189</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00269</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">Samples_with_predicted_ancestry</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="left">Predicted Ancestry</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00096</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">EUR_GBR</td>
<td align="left">HG00097</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00099</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">EUR_GBR</td>
<td align="left">HG00100</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00101</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">EUR_GBR</td>
<td align="left">HG00102</td>
<td align="left">EUR</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">NonOutlier_samples</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">sample</th>
<th align="left">Assigned_Pop_ZScore</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">410</td>
<td align="left">HG00171</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">411</td>
<td align="left">HG00173</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">412</td>
<td align="left">HG00174</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">413</td>
<td align="left">HG00176</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">415</td>
<td align="left">HG00178</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">418</td>
<td align="left">HG00182</td>
<td align="left">EUR</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="sexcheck">
<code>SexCheck()</code><a class="anchor" aria-label="anchor" href="#sexcheck"></a>
</h4>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">LD</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">LD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">LD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">LD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">fmax_F</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">mmin_F</span> <span class="op">&lt;-</span> <span class="fl">0.8</span></span>
<span><span class="va">impute_sex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">compute_freq</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexCheck.html">SexCheck</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, impute_sex <span class="op">=</span> <span class="va">impute_sex</span>,</span>
<span>    compute_freq <span class="op">=</span> <span class="va">compute_freq</span>, LD_window_size <span class="op">=</span> <span class="va">LD_window_size</span>, LD_step_size <span class="op">=</span> <span class="va">LD_step_size</span>,</span>
<span>    LD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, fmax_F <span class="op">=</span> <span class="fl">0.2</span>, mmin_F <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Checking if there is any wrong sex assignment</span></span>
<span><span class="va">problematic_sex</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[</span><span class="va">x</span><span class="op">$</span><span class="va">STATUS</span> <span class="op">!=</span> <span class="st">"OK"</span>, <span class="op">]</span></span>
<span><span class="va">problematic_sex</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">PEDSEX</th>
<th align="right">SNPSEX</th>
<th align="left">STATUS</th>
<th align="right">F</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">78</td>
<td align="left">EUR_FIN</td>
<td align="left">HG00361</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.4890</td>
</tr>
<tr class="even">
<td align="left">183</td>
<td align="left">EUR_TSI</td>
<td align="left">NA20506</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.7969</td>
</tr>
<tr class="odd">
<td align="left">203</td>
<td align="left">EUR_TSI</td>
<td align="left">NA20530</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.7478</td>
</tr>
<tr class="even">
<td align="left">206</td>
<td align="left">EUR_TSI</td>
<td align="left">NA20533</td>
<td align="right">2</td>
<td align="right">0</td>
<td align="left">PROBLEM</td>
<td align="right">0.3285</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="xhwe">
<code>Xhwe()</code><a class="anchor" aria-label="anchor" href="#xhwe"></a>
</h4>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Xhwe.html">Xhwe</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, filterSNP <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, prefixed as female, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="co">#&gt; This test is running on a case-control dataset with female samples.</span></span>
<span><span class="co">#&gt; ℹ Failed SNPs are excluded from the output PLINK files prefixed as Test_output is in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; [1] "rs56053951" "rs12353847" "rs5940058"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="mafdiffsexcontrol">
<code>MAFdiffSexControl()</code><a class="anchor" aria-label="anchor" href="#mafdiffsexcontrol"></a>
</h4>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAFdiffSexControl.html">MAFdiffSexControl</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, filterSNP <span class="op">=</span> <span class="cn">TRUE</span>, foutput <span class="op">=</span> <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ No SNP to be flagged or excluded.</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filterregion">
<code>FilterRegion()</code><a class="anchor" aria-label="anchor" href="#filterregion"></a>
</h4>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterRegion.html">FilterRegion</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">TRUE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    filterPAR <span class="op">=</span> <span class="cn">TRUE</span>, filterXTR <span class="op">=</span> <span class="cn">TRUE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="cn">NULL</span>, Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ chrX</span></span>
<span><span class="co">#&gt; ✖ There is no PAR region in the input data. Argument filterPAR cannot set to be TRUE.</span></span>
<span><span class="co">#&gt; ℹ Changing filterPAR to FALSE</span></span>
<span><span class="co">#&gt; ✖ There is no XTR region in the input data. Argument filterXTR cannot set to be TRUE.</span></span>
<span><span class="co">#&gt; ℹ Changing filterXTR to FALSE</span></span>
<span><span class="co">#&gt; ℹ Ampliconic SNPs:9</span></span>
<span><span class="co">#&gt; ℹ 9 SNPs are discarded.</span></span>
<span><span class="co">#&gt; ✔ PLINK files with passed SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX prefixed as PostimputeEX_QC1</span></span>
<span><span class="co">#&gt; ✔ PLINK files with discarded SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX prefixed as PostimputeEX_QC1_snps_extracted</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; $PAR</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $XTR</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Ampliconic</span></span>
<span><span class="co">#&gt;   CHR         SNP START       END A1 A2</span></span>
<span><span class="co">#&gt; 1  23  rs62602496     0  48349540  G  A</span></span>
<span><span class="co">#&gt; 2  23   rs6612314     0  55442087  A  C</span></span>
<span><span class="co">#&gt; 3  23 rs151231489     0  55445831  C  A</span></span>
<span><span class="co">#&gt; 4  23  rs73498395     0  71740842  T  C</span></span>
<span><span class="co">#&gt; 5  23      rs6527     0  73066891  A  C</span></span>
<span><span class="co">#&gt; 6  23   rs4907822     0 102264099  A  G</span></span>
<span><span class="co">#&gt; 7  23 rs142219143     0 102354248  G  A</span></span>
<span><span class="co">#&gt; 8  23   rs5987512     0 102483549  G  C</span></span>
<span><span class="co">#&gt; 9  23 rs139353379     0 103970756  G  A</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="gxwas">
<code>GXwas()</code><a class="anchor" aria-label="anchor" href="#gxwas"></a>
</h4>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">standard_beta</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">xsex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">Inphenocov</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">covartest</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">interaction</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">MF.na.rm</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">B</span> <span class="op">&lt;-</span> <span class="fl">10000</span></span>
<span><span class="va">MF.zero.sub</span> <span class="op">&lt;-</span> <span class="fl">0.00001</span></span>
<span><span class="va">trait</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">xmodel</span> <span class="op">&lt;-</span> <span class="st">"FMcombx02"</span></span>
<span><span class="va">combtest</span> <span class="op">&lt;-</span> <span class="st">"fisher.method"</span></span>
<span><span class="va">snp_pval</span> <span class="op">&lt;-</span> <span class="fl">1e-08</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">MF.mc.cores</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">ResultGXwas</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GXwas.html">GXwas</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, xmodel <span class="op">=</span> <span class="va">xmodel</span>, trait <span class="op">=</span> <span class="va">trait</span>, covarfile <span class="op">=</span> <span class="va">covarfile</span>,</span>
<span>    sex <span class="op">=</span> <span class="va">sex</span>, xsex <span class="op">=</span> <span class="va">xsex</span>, combtest <span class="op">=</span> <span class="va">combtest</span>, MF.p.corr <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>    snp_pval <span class="op">=</span> <span class="va">snp_pval</span>, plot.jpeg <span class="op">=</span> <span class="cn">TRUE</span>, suggestiveline <span class="op">=</span> <span class="fl">5</span>, genomewideline <span class="op">=</span> <span class="fl">7.3</span>,</span>
<span>    MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, ncores <span class="op">=</span> <span class="va">ncores</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; • Running FMcombx02 model</span></span>
<span><span class="co">#&gt; ℹ If you want parallel computation, please provide non-zero value for argument ncores.</span></span>
<span><span class="va">ResultGXwas</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="4%">
<col width="14%">
<col width="8%">
<col width="3%">
<col width="6%">
<col width="7%">
<col width="10%">
<col width="8%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">276</td>
<td align="right">0.11850</td>
<td align="right">0.1923</td>
<td align="right">-0.2584</td>
<td align="right">0.4953</td>
<td align="right">0.6161</td>
<td align="right">0.5378</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">SEX</td>
<td align="right">276</td>
<td align="right">0.24120</td>
<td align="right">0.2483</td>
<td align="right">-0.2456</td>
<td align="right">0.7279</td>
<td align="right">0.9711</td>
<td align="right">0.3315</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs75530702</td>
<td align="right">720797</td>
<td align="left">G</td>
<td align="left">ADD</td>
<td align="right">276</td>
<td align="right">0.43500</td>
<td align="right">1.4220</td>
<td align="right">-2.3530</td>
<td align="right">3.2230</td>
<td align="right">0.3058</td>
<td align="right">0.7598</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs75530702</td>
<td align="right">720797</td>
<td align="left">G</td>
<td align="left">SEX</td>
<td align="right">276</td>
<td align="right">0.25010</td>
<td align="right">0.2477</td>
<td align="right">-0.2355</td>
<td align="right">0.7357</td>
<td align="right">1.0100</td>
<td align="right">0.3127</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">276</td>
<td align="right">-0.05108</td>
<td align="right">0.7425</td>
<td align="right">-1.5060</td>
<td align="right">1.4040</td>
<td align="right">-0.0688</td>
<td align="right">0.9452</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">SEX</td>
<td align="right">276</td>
<td align="right">0.25020</td>
<td align="right">0.2478</td>
<td align="right">-0.2355</td>
<td align="right">0.7358</td>
<td align="right">1.0100</td>
<td align="right">0.3127</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="pvalcomb">
<code>PvalComb()</code><a class="anchor" aria-label="anchor" href="#pvalcomb"></a>
</h4>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Mfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Ffile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">SumstatMale</span> <span class="op">&lt;-</span> <span class="va">Mfile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SumstatMale</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"POS"</span></span>
<span><span class="va">SumstatFemale</span> <span class="op">&lt;-</span> <span class="va">Ffile</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">SumstatFemale</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"POS"</span></span>
<span><span class="va">PvalComb_Result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PvalComb.html">PvalComb</a></span><span class="op">(</span></span>
<span>    SumstatMale <span class="op">=</span> <span class="va">SumstatMale</span>, SumstatFemale <span class="op">=</span> <span class="va">SumstatFemale</span>,</span>
<span>    combtest <span class="op">=</span> <span class="st">"fisher.method"</span>, MF.mc.cores <span class="op">=</span> <span class="fl">1</span>, snp_pval <span class="op">=</span> <span class="fl">0.001</span>, plot.jpeg <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    suggestiveline <span class="op">=</span> <span class="fl">3</span>, genomewideline <span class="op">=</span> <span class="fl">5.69897</span>, ncores <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/Stratified_GWAS.png</span></span></code></pre></div>
<div class="float">
<img src="overview-PvalComb-1.png" alt="plot of chunk overview-PvalComb"><div class="figcaption">plot of chunk overview-PvalComb</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/Stratified_XWAS.png</span></span></code></pre>
<p><img src="overview-PvalComb-2.png" alt="plot of chunk overview-PvalComb"><img src="overview-PvalComb-3.png" alt="plot of chunk overview-PvalComb"><img src="overview-PvalComb-4.png" alt="plot of chunk overview-PvalComb"><img src="overview-PvalComb-5.png" alt="plot of chunk overview-PvalComb"><img src="overview-PvalComb-6.png" alt="plot of chunk overview-PvalComb"></p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PvalComb_Result</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">POS</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs1015493</td>
<td align="right">5</td>
<td align="right">11359627</td>
<td align="right">0.1305102</td>
</tr>
<tr class="even">
<td align="left">rs1028717</td>
<td align="right">14</td>
<td align="right">85392037</td>
<td align="right">0.9297428</td>
</tr>
<tr class="odd">
<td align="left">rs1030166</td>
<td align="right">5</td>
<td align="right">140165657</td>
<td align="right">0.9436807</td>
</tr>
<tr class="even">
<td align="left">rs10416111</td>
<td align="right">19</td>
<td align="right">8970692</td>
<td align="right">0.8329715</td>
</tr>
<tr class="odd">
<td align="left">rs10494251</td>
<td align="right">1</td>
<td align="right">145490518</td>
<td align="right">0.0501473</td>
</tr>
<tr class="even">
<td align="left">rs10518607</td>
<td align="right">4</td>
<td align="right">133249707</td>
<td align="right">0.1032385</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="sexdiff">
<code>SexDiff()</code><a class="anchor" aria-label="anchor" href="#sexdiff"></a>
</h4>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Mfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Ffile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">Difftest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexDiff.html">SexDiff</a></span><span class="op">(</span><span class="va">Mfile</span>, <span class="va">Ffile</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="overview-SexDiff-1.png" alt="plot of chunk overview-SexDiff"><div class="figcaption">plot of chunk overview-SexDiff</div>
</div>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">significant_snps</span> <span class="op">&lt;-</span> <span class="va">Difftest</span><span class="op">[</span><span class="va">Difftest</span><span class="op">$</span><span class="va">adjP</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span></span>
<span><span class="va">significant_snps</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">SNP</th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="right">tstat</th>
<th align="right">P</th>
<th align="right">adjP</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">rs11073280</td>
<td align="right">15</td>
<td align="right">35967595</td>
<td align="left">C</td>
<td align="right">6.156697</td>
<td align="right">0.00e+00</td>
<td align="right">0.0000002</td>
</tr>
<tr class="even">
<td align="left">rs12694063</td>
<td align="right">2</td>
<td align="right">207445996</td>
<td align="left">T</td>
<td align="right">5.365174</td>
<td align="right">1.00e-07</td>
<td align="right">0.0000115</td>
</tr>
<tr class="odd">
<td align="left">rs10894121</td>
<td align="right">11</td>
<td align="right">129107457</td>
<td align="left">A</td>
<td align="right">5.300321</td>
<td align="right">1.00e-07</td>
<td align="right">0.0000158</td>
</tr>
<tr class="even">
<td align="left">rs11717371</td>
<td align="right">3</td>
<td align="right">15077542</td>
<td align="left">T</td>
<td align="right">4.045074</td>
<td align="right">3.54e-05</td>
<td align="right">0.0042428</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="sexdiffzscore">
<code>SexDiffZscore()</code><a class="anchor" aria-label="anchor" href="#sexdiffzscore"></a>
</h4>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_h2data"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">inputdata</span> <span class="op">&lt;-</span> <span class="va">Example_h2data</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexDiffZscore.html">SexDiffZscore</a></span><span class="op">(</span><span class="va">inputdata</span><span class="op">)</span></span>
<span><span class="va">x</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">ID</th>
<th align="right">Mstat</th>
<th align="right">Mse</th>
<th align="right">Fstat</th>
<th align="right">Fse</th>
<th align="right">Zscore</th>
<th align="right">p</th>
<th align="right">adjP</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ADHD</td>
<td align="right">0.249</td>
<td align="right">0.021</td>
<td align="right">0.135</td>
<td align="right">0.028</td>
<td align="right">3.2571429</td>
<td align="right">0.0011254</td>
<td align="right">0.0090032</td>
</tr>
<tr class="even">
<td align="left">AFB</td>
<td align="right">0.113</td>
<td align="right">0.010</td>
<td align="right">0.052</td>
<td align="right">0.004</td>
<td align="right">5.6637078</td>
<td align="right">0.0000000</td>
<td align="right">0.0000001</td>
</tr>
<tr class="odd">
<td align="left">ALCC</td>
<td align="right">0.086</td>
<td align="right">0.010</td>
<td align="right">0.082</td>
<td align="right">0.011</td>
<td align="right">0.2690691</td>
<td align="right">0.7878765</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">ALCD</td>
<td align="right">0.113</td>
<td align="right">0.031</td>
<td align="right">0.014</td>
<td align="right">0.052</td>
<td align="right">1.6353029</td>
<td align="right">0.1019856</td>
<td align="right">0.8158845</td>
</tr>
<tr class="odd">
<td align="left">ANX</td>
<td align="right">0.009</td>
<td align="right">0.003</td>
<td align="right">0.005</td>
<td align="right">0.002</td>
<td align="right">1.1094004</td>
<td align="right">0.2672575</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">ASD</td>
<td align="right">0.227</td>
<td align="right">0.019</td>
<td align="right">-0.075</td>
<td align="right">0.034</td>
<td align="right">7.7537921</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">BD</td>
<td align="right">0.364</td>
<td align="right">0.032</td>
<td align="right">0.339</td>
<td align="right">0.027</td>
<td align="right">0.5971027</td>
<td align="right">0.5504388</td>
<td align="right">1.0000000</td>
</tr>
<tr class="even">
<td align="left">CUE</td>
<td align="right">0.076</td>
<td align="right">0.010</td>
<td align="right">0.067</td>
<td align="right">0.008</td>
<td align="right">0.7027819</td>
<td align="right">0.4821917</td>
<td align="right">1.0000000</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="diffzeroone">
<code>DiffZeroOne()</code><a class="anchor" aria-label="anchor" href="#diffzeroone"></a>
</h4>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_rgdata"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">inputdata</span> <span class="op">&lt;-</span> <span class="va">Example_rgdata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">inputdata</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Trait"</span>, <span class="st">"Stat"</span>, <span class="st">"SE"</span><span class="op">)</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DiffZeroOne.html">DiffZeroOne</a></span><span class="op">(</span><span class="va">inputdata</span>, <span class="cn">FALSE</span>, <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">x</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Trait</th>
<th align="right">Stat</th>
<th align="right">SE</th>
<th align="right">p1</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">RTB</td>
<td align="right">0.811</td>
<td align="right">0.041</td>
<td align="right">0.0000020</td>
</tr>
<tr class="even">
<td align="left">EA</td>
<td align="right">0.919</td>
<td align="right">0.022</td>
<td align="right">0.0001158</td>
</tr>
<tr class="odd">
<td align="left">BD</td>
<td align="right">0.862</td>
<td align="right">0.057</td>
<td align="right">0.0077378</td>
</tr>
<tr class="even">
<td align="left">CUE</td>
<td align="right">0.769</td>
<td align="right">0.091</td>
<td align="right">0.0055671</td>
</tr>
<tr class="odd">
<td align="left">NEU</td>
<td align="right">0.928</td>
<td align="right">0.029</td>
<td align="right">0.0065185</td>
</tr>
<tr class="even">
<td align="left">SCZ</td>
<td align="right">0.923</td>
<td align="right">0.033</td>
<td align="right">0.0098153</td>
</tr>
<tr class="odd">
<td align="left">AFB</td>
<td align="right">0.886</td>
<td align="right">0.057</td>
<td align="right">0.0227501</td>
</tr>
<tr class="even">
<td align="left">SMKP</td>
<td align="right">0.915</td>
<td align="right">0.044</td>
<td align="right">0.0266910</td>
</tr>
<tr class="odd">
<td align="left">INS</td>
<td align="right">0.786</td>
<td align="right">0.125</td>
<td align="right">0.0434483</td>
</tr>
<tr class="even">
<td align="left">ANX</td>
<td align="right">0.678</td>
<td align="right">0.275</td>
<td align="right">0.1208177</td>
</tr>
<tr class="odd">
<td align="left">ALCC</td>
<td align="right">0.911</td>
<td align="right">0.090</td>
<td align="right">0.1613588</td>
</tr>
<tr class="even">
<td align="left">NEB</td>
<td align="right">0.903</td>
<td align="right">0.097</td>
<td align="right">0.1586553</td>
</tr>
<tr class="odd">
<td align="left">MDD</td>
<td align="right">0.996</td>
<td align="right">0.305</td>
<td align="right">0.4947681</td>
</tr>
<tr class="even">
<td align="left">OCD</td>
<td align="right">1.034</td>
<td align="right">0.564</td>
<td align="right">0.5240352</td>
</tr>
<tr class="odd">
<td align="left">SMKC</td>
<td align="right">1.022</td>
<td align="right">0.050</td>
<td align="right">0.6700314</td>
</tr>
<tr class="even">
<td align="left">ADHD</td>
<td align="right">1.209</td>
<td align="right">0.127</td>
<td align="right">0.9500841</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="testxgene">
<code>TestXGene()</code><a class="anchor" aria-label="anchor" href="#testxgene"></a>
</h4>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"XWAS_Summary_Example"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">sumstat</span> <span class="op">&lt;-</span> <span class="va">XWAS_Summary_Example</span></span>
<span><span class="va">ref_data</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">gene_file</span> <span class="op">&lt;-</span> <span class="st">"Xlinkedgenes_hg19.txt"</span></span>
<span><span class="va">gene_range</span> <span class="op">&lt;-</span> <span class="fl">500000</span></span>
<span><span class="va">max_gene</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">gene_approximation</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">beta_par</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">25</span><span class="op">)</span></span>
<span><span class="va">weights_function</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">geno_variance_weights</span> <span class="op">&lt;-</span> <span class="st">"se.beta"</span></span>
<span><span class="va">method</span> <span class="op">&lt;-</span> <span class="st">"kuonen"</span></span>
<span><span class="va">acc_devies</span> <span class="op">&lt;-</span> <span class="fl">1e-8</span></span>
<span><span class="va">lim_devies</span> <span class="op">&lt;-</span> <span class="fl">1e+6</span></span>
<span><span class="va">rho</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">skato_p_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.8</span></span>
<span><span class="va">mac_threshold</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">sample_size</span> <span class="op">&lt;-</span> <span class="fl">4000</span></span>
<span><span class="va">reference_matrix_used</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">regularize_fun</span> <span class="op">&lt;-</span> <span class="st">"LH"</span></span>
<span><span class="va">pca_var_fraction</span> <span class="op">&lt;-</span> <span class="fl">0.85</span></span>
<span><span class="va">flm_basis_function</span> <span class="op">&lt;-</span> <span class="st">"fourier"</span></span>
<span><span class="va">flm_num_basis</span> <span class="op">&lt;-</span> <span class="fl">25</span></span>
<span><span class="va">flm_poly_order</span> <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">flip_genotypes</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">omit_linear_variant</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">kernel_p_method</span> <span class="op">&lt;-</span> <span class="st">"kuonen"</span></span>
<span><span class="va">anno_type</span> <span class="op">&lt;-</span> <span class="st">""</span></span>
<span><span class="va">GenetestResult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TestXGene.html">TestXGene</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">sumstat</span>, <span class="va">gene_file</span>,</span>
<span>    <span class="va">gene_range</span>, <span class="va">score_file</span>, <span class="va">ref_data</span>, <span class="va">max_gene</span>, <span class="va">sample_size</span>,</span>
<span>    genebasedTest <span class="op">=</span> <span class="st">"SKAT"</span>,</span>
<span>    <span class="va">gene_approximation</span>, <span class="va">beta_par</span>, <span class="va">weights_function</span>, <span class="va">geno_variance_weights</span>,</span>
<span>    <span class="va">kernel_p_method</span>, <span class="va">acc_devies</span>, <span class="va">lim_devies</span>, <span class="va">rho</span>, <span class="va">skato_p_threshold</span>, <span class="va">anno_type</span>,</span>
<span>    <span class="va">mac_threshold</span>, <span class="va">reference_matrix_used</span>, <span class="va">regularize_fun</span>, <span class="va">pca_var_fraction</span>,</span>
<span>    <span class="va">flm_basis_function</span>, <span class="va">flm_num_basis</span>, <span class="va">flm_poly_order</span>, <span class="va">flip_genotypes</span>,</span>
<span>    <span class="va">omit_linear_variant</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 80 of 80 variants found in reference</span></span>
<span><span class="co">#&gt; ℹ Effect sizes recoded for 0 variant(s)</span></span>
<span><span class="co">#&gt; ✔ Allele frequencies found and linked</span></span>
<span><span class="co">#&gt; ✔ File /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/gene.test.score.file.vcf.gz has been created and indexed</span></span>
<span><span class="co">#&gt; ℹ 157 genes are having 321 SNPs</span></span>
<span><span class="co">#&gt; • SNP-SNP correlation matrices are being created...</span></span>
<span><span class="co">#&gt; ✔ SNP-SNP correlation matrices are done.</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004cqy.3, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004csf.3, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004csr.3, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004cst.2, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004csu.1, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004csx.4, skipped</span></span>
<span><span class="co">#&gt; ! Warning: No variants to analyze in gene uc004cuw.3, skipped</span></span>
<span><span class="va">GenetestResult</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="4%">
<col width="15%">
<col width="8%">
<col width="12%">
<col width="12%">
<col width="11%">
<col width="24%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">gene</th>
<th align="right">chrom</th>
<th align="right">start</th>
<th align="right">end</th>
<th align="right">markers</th>
<th align="right">filtered.markers</th>
<th align="right">pvalue</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">24</td>
<td align="left">uc004cqy.3</td>
<td align="right">23</td>
<td align="right">2822010</td>
<td align="right">2847416</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">38</td>
<td align="left">uc004crr.3</td>
<td align="right">23</td>
<td align="right">5808082</td>
<td align="right">6146706</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.9365</td>
</tr>
<tr class="odd">
<td align="left">49</td>
<td align="left">uc004csf.3</td>
<td align="right">23</td>
<td align="right">8496914</td>
<td align="right">8700227</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">70</td>
<td align="left">uc004csr.3</td>
<td align="right">23</td>
<td align="right">9433200</td>
<td align="right">9687780</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">71</td>
<td align="left">uc004cst.2</td>
<td align="right">23</td>
<td align="right">9693452</td>
<td align="right">9734005</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">72</td>
<td align="left">uc004csu.1</td>
<td align="right">23</td>
<td align="right">9754495</td>
<td align="right">9917481</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">74</td>
<td align="left">uc004csx.4</td>
<td align="right">23</td>
<td align="right">9983794</td>
<td align="right">10112518</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">76</td>
<td align="left">uc004cti.4</td>
<td align="right">23</td>
<td align="right">10413349</td>
<td align="right">10851809</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.1846</td>
</tr>
<tr class="odd">
<td align="left">82</td>
<td align="left">uc004cup.1</td>
<td align="right">23</td>
<td align="right">11155662</td>
<td align="right">11683821</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0.9751</td>
</tr>
<tr class="even">
<td align="left">85</td>
<td align="left">uc004cuw.3</td>
<td align="right">23</td>
<td align="right">11776277</td>
<td align="right">11793872</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="metagwas">
<code>MetaGWAS()</code><a class="anchor" aria-label="anchor" href="#metagwas"></a>
</h4>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex1"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex2"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">SummData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span>, <span class="va">Summary_Stat_Ex2</span><span class="op">)</span></span>
<span><span class="va">SNPfile</span> <span class="op">&lt;-</span> <span class="st">"UniqueLoci"</span></span>
<span><span class="va">useSNPposition</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">UseA1</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">GCse</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">pval_filter</span> <span class="op">&lt;-</span> <span class="st">"R"</span></span>
<span><span class="va">top_snp_pval</span> <span class="op">&lt;-</span> <span class="fl">1e-08</span></span>
<span><span class="va">max_top_snps</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">chosen_snps_file</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">pval_threshold_manplot</span> <span class="op">&lt;-</span> <span class="fl">1e-05</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"Meta_Analysis.plot"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetaGWAS.html">MetaGWAS</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, SummData <span class="op">=</span> <span class="va">SummData</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    SNPfile <span class="op">=</span> <span class="cn">NULL</span>, useSNPposition <span class="op">=</span> <span class="cn">TRUE</span>, UseA1 <span class="op">=</span> <span class="va">UseA1</span>, GCse <span class="op">=</span> <span class="va">GCse</span>,</span>
<span>    plotname <span class="op">=</span> <span class="st">"Meta_Analysis.plot"</span>, <span class="va">pval_filter</span>, <span class="va">top_snp_pval</span>, <span class="va">max_top_snps</span>,</span>
<span>    chosen_snps_file <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">byCHR</span>, <span class="va">pval_threshold_manplot</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Processing file number 1</span></span>
<span><span class="co">#&gt; ℹ Processing file number 2</span></span>
<span><span class="co">#&gt; ℹ Applying study-specific genomic control.</span></span>
<span><span class="co">#&gt; ℹ Applying study-specific genomic control.</span></span>
<span><span class="co">#&gt; Processing chromosome</span></span></code></pre></div>
<p><img src="overview-MetaGWAS-1.png" alt="plot of chunk overview-MetaGWAS"><img src="overview-MetaGWAS-2.png" alt="plot of chunk overview-MetaGWAS"><img src="overview-MetaGWAS-3.png" alt="plot of chunk overview-MetaGWAS"></p>
<pre><code><span><span class="co">#&gt; ✔ Forest plot files for Meta_Analysis.plot SNPs have been created.</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="co">#&gt; ✔ Forest plot files for Meta_Analysis.plot SNPs have been created.</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="va">x</span><span class="op">$</span><span class="va">Resultfixed</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre>
<table class="table">
<colgroup>
<col width="4%">
<col width="8%">
<col width="12%">
<col width="3%">
<col width="3%">
<col width="7%">
<col width="6%">
<col width="10%">
<col width="8%">
<col width="10%">
<col width="11%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">Q</th>
<th align="right">I</th>
<th align="right">P</th>
<th align="right">ES</th>
<th align="right">SE</th>
<th align="right">CI_L</th>
<th align="right">CI_U</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">73841</td>
<td align="left">rs143773730</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">0.2588</td>
<td align="right">21.57</td>
<td align="right">0.5535000</td>
<td align="right">0.1108</td>
<td align="right">0.1869968</td>
<td align="right">-0.2557137</td>
<td align="right">0.4773137</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">775125</td>
<td align="left">rs147281566</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">0.7118</td>
<td align="right">0.00</td>
<td align="right">0.9414000</td>
<td align="right">-0.0534</td>
<td align="right">0.7264282</td>
<td align="right">-1.4771993</td>
<td align="right">1.3703993</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">863863</td>
<td align="left">rs35854196</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.2820</td>
<td align="right">13.60</td>
<td align="right">0.5093000</td>
<td align="right">0.3621</td>
<td align="right">0.5486962</td>
<td align="right">-0.7133445</td>
<td align="right">1.4375445</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">928969</td>
<td align="left">rs115490086</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">0.6896000</td>
<td align="right">0.5649</td>
<td align="right">1.4143788</td>
<td align="right">-2.2072824</td>
<td align="right">3.3370824</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">1109154</td>
<td align="left">rs12041521</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.6258</td>
<td align="right">0.00</td>
<td align="right">0.0159200</td>
<td align="right">-0.5550</td>
<td align="right">0.2302193</td>
<td align="right">-1.0062299</td>
<td align="right">-0.1037701</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">1127860</td>
<td align="left">rs148527527</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.5778</td>
<td align="right">0.00</td>
<td align="right">0.0008237</td>
<td align="right">1.3978</td>
<td align="right">0.4179142</td>
<td align="right">0.5786881</td>
<td align="right">2.2169119</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span><span class="op">$</span><span class="va">ProblemSNP</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="74%">
<col width="10%">
<col width="15%">
</colgroup>
<thead><tr class="header">
<th align="left">File</th>
<th align="left">SNP</th>
<th align="left">Problem</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs2803333</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs672606</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs7519955</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs1538466</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs61777960</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/SNPdata_2</td>
<td align="left">rs2865211</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="computeprs">
<code>ComputePRS()</code><a class="anchor" aria-label="anchor" href="#computeprs"></a>
</h4>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex1"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_phenofile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_covarfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_pthresoldfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL</span></span>
<span><span class="co"># The interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="va">Example_covarfile</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span></span>
<span><span class="va">PRSresult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputePRS.html">ComputePRS</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>,</span>
<span>    effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">clump_p1</span>, <span class="va">clump_p2</span>,</span>
<span>    <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>,</span>
<span>    ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>,</span>
<span>    pheno_type <span class="op">=</span> <span class="st">"binary"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/temp_libpatheab742dff3cd', reason 'Directory not empty'</span></span>
<span><span class="co">#&gt; ℹ 0.001</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.001</span></span>
<span><span class="co">#&gt; ℹ 0.05</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.05</span></span>
<span><span class="co">#&gt; ℹ 0.1</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.1</span></span>
<span><span class="co">#&gt; ℹ 0.2</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.2</span></span>
<span><span class="co">#&gt; ℹ 0.3</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.3</span></span>
<span><span class="co">#&gt; ℹ 0.4</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.4</span></span>
<span><span class="co">#&gt; ℹ 0.5</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.5</span></span>
<span><span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="7%">
<col width="6%">
<col width="3%">
<col width="8%">
<col width="8%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Pheno1</th>
<th align="right">AGE</th>
<th align="right">testcovar</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">SCORE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">1</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0964618</td>
<td align="right">-0.0382514</td>
<td align="right">0.0090380</td>
<td align="right">-0.0895293</td>
<td align="right">-0.0252291</td>
<td align="right">0.0146543</td>
<td align="right">0.0484212</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">1</td>
<td align="right">81</td>
<td align="right">1</td>
<td align="right">0.0743932</td>
<td align="right">-0.0567271</td>
<td align="right">0.0396994</td>
<td align="right">0.0205343</td>
<td align="right">0.0230410</td>
<td align="right">-0.0360380</td>
<td align="right">0.0293742</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">1</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">0.0645349</td>
<td align="right">-0.0561035</td>
<td align="right">-0.0145786</td>
<td align="right">-0.0264343</td>
<td align="right">-0.0018095</td>
<td align="right">0.0144521</td>
<td align="right">0.0099674</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">2</td>
<td align="right">75</td>
<td align="right">0</td>
<td align="right">0.0832472</td>
<td align="right">-0.0560208</td>
<td align="right">0.0302654</td>
<td align="right">-0.0011077</td>
<td align="right">-0.0530371</td>
<td align="right">-0.0335803</td>
<td align="right">0.1042060</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">1</td>
<td align="right">88</td>
<td align="right">1</td>
<td align="right">0.0775687</td>
<td align="right">-0.0094644</td>
<td align="right">0.0451785</td>
<td align="right">-0.0373215</td>
<td align="right">-0.0617431</td>
<td align="right">-0.0719951</td>
<td align="right">0.0314303</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">1</td>
<td align="right">24</td>
<td align="right">1</td>
<td align="right">0.0661714</td>
<td align="right">-0.0504906</td>
<td align="right">0.0248539</td>
<td align="right">-0.0828066</td>
<td align="right">0.0409418</td>
<td align="right">0.0055953</td>
<td align="right">0.0722689</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PRSresult</span><span class="op">$</span><span class="va">BestP</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">Threshold</th>
</tr></thead>
<tbody><tr class="odd">
<td align="right">0.05</td>
</tr></tbody>
</table>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS_plot</span><span class="op">)</span></span></code></pre></div>
<div class="float">
<img src="overview-ComputePRS-1.png" alt="plot of chunk compute_prs"><div class="figcaption">plot of chunk compute_prs</div>
</div>
</div>
<div class="section level4">
<h4 id="geneticcorrbt">
<code>GeneticCorrBT()</code><a class="anchor" aria-label="anchor" href="#geneticcorrbt"></a>
</h4>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Example_phenofile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">REMLalgo</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">nitr</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL</span></span>
<span><span class="va">cat_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">quant_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">partGRM</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Partition the GRM into m parts (by row),</span></span>
<span><span class="va">autosome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">Xsome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">cripticut</span> <span class="op">&lt;-</span> <span class="fl">0.025</span></span>
<span><span class="va">minMAF</span> <span class="op">&lt;-</span> <span class="fl">0.01</span> <span class="co"># if MAF filter apply</span></span>
<span><span class="va">maxMAF</span> <span class="op">&lt;-</span> <span class="fl">0.04</span></span>
<span><span class="va">excludeResidual</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span></span>
<span><span class="va">genetic_correlation</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GeneticCorrBT.html">GeneticCorrBT</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, byCHR <span class="op">=</span> <span class="va">byCHR</span>,</span>
<span>    REMLalgo <span class="op">=</span> <span class="fl">0</span>, nitr <span class="op">=</span> <span class="fl">10</span>, phenofile <span class="op">=</span> <span class="va">phenofile</span>, cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>, quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    partGRM <span class="op">=</span> <span class="cn">FALSE</span>, autosome <span class="op">=</span> <span class="cn">TRUE</span>, Xsome <span class="op">=</span> <span class="cn">TRUE</span>, nGRM <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    cripticut <span class="op">=</span> <span class="fl">0.025</span>, minMAF <span class="op">=</span> <span class="cn">NULL</span>, maxMAF <span class="op">=</span> <span class="cn">NULL</span>, excludeResidual <span class="op">=</span> <span class="cn">TRUE</span>, ncores <span class="op">=</span> <span class="va">ncores</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Processing chromosome 1</span></span>
<span><span class="co">#&gt; Using GCTA version v1.94.4 Mac</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 2</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 3</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 4</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 5</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 6</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 7</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 8</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 9</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 10</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 23</span></span>
<span><span class="co">#&gt; ✔ GRM has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/xGXwasR.grm.bin]</span></span>
<span><span class="co">#&gt; ℹ Number of SNPs in each pair of individuals has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/xGXwasR.grm.N.bin]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 24</span></span>
<span><span class="co">#&gt; ✖ Error: Log-likelihood not converged (stop after 10 iteractions). </span></span>
<span><span class="co">#&gt; ℹ Note: to constrain the correlation being from -1 to 1, a genetic (or residual) variance-covariance matrix is bended to be positive definite. In this case, the SE is unreliable.</span></span>
<span><span class="co">#&gt; ✖ Convergence issue occurred, please:</span></span>
<span><span class="co">#&gt;   - set byCHR = TRUE</span></span>
<span><span class="co">#&gt;   - set different options</span></span>
<span><span class="co">#&gt;   - verify SNP partitioning or quality of the data</span></span>
<span><span class="co">#&gt; ℹ The result will be provided for the last iteration.</span></span>
<span><span class="va">genetic_correlation</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">chromosome</th>
<th align="left">Source</th>
<th align="right">Variance</th>
<th align="right">NA</th>
<th align="right">NA</th>
<th align="right">NA</th>
<th align="right">SE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">343.88</td>
<td align="right">350.43000</td>
<td align="right">351.09000</td>
<td align="right">351.29000</td>
<td align="right">334.23000</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">0.05138</td>
<td align="right">0.01988</td>
<td align="right">0.01092</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">0.00213</td>
<td align="right">0.00014</td>
<td align="right">0.00075</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">0.00469</td>
<td align="right">-0.00166</td>
<td align="right">-0.00286</td>
<td align="right">0.00000</td>
<td align="right">0.00000</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">0.18285</td>
<td align="right">0.21472</td>
<td align="right">0.22398</td>
<td align="right">0.23846</td>
<td align="right">0.23783</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">0.03878</td>
<td align="right">0.04109</td>
<td align="right">0.04135</td>
<td align="right">0.03513</td>
<td align="right">0.02517</td>
<td align="right">NA</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="estimateherit">
<code>EstimateHerit()</code><a class="anchor" aria-label="anchor" href="#estimateherit"></a>
</h4>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex1"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">precomputedLD</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">test.sumstats</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span><span class="op">[</span><span class="va">Summary_Stat_Ex1</span><span class="op">$</span><span class="va">TEST</span> <span class="op">==</span> <span class="st">"ADD"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">4</span><span class="op">)</span>, <span class="fl">6</span><span class="op">:</span><span class="fl">8</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">test.sumstats</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="st">"rsid"</span>, <span class="st">"pos"</span>, <span class="st">"a1"</span>, <span class="st">"n_eff"</span>, <span class="st">"beta"</span>, <span class="st">"beta_se"</span><span class="op">)</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">test.sumstats</span></span>
<span><span class="va">ncores</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">model</span> <span class="op">&lt;-</span> <span class="st">"GREML"</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">r2_LD</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">LDSC_blocks</span> <span class="op">&lt;-</span> <span class="fl">20</span></span>
<span><span class="va">REMLalgo</span> <span class="op">&lt;-</span> <span class="fl">0</span></span>
<span><span class="va">nitr</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">cat_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">quant_covarfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">prevalence</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span><span class="va">partGRM</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">autosome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">Xsome</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">nGRM</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">cripticut</span> <span class="op">&lt;-</span> <span class="fl">0.025</span></span>
<span><span class="va">minMAF</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">maxMAF</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hg</span> <span class="op">&lt;-</span> <span class="st">"hg19"</span></span>
<span><span class="va">PlotIndepSNP</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">IndepSNP_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">IndepSNP_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">IndepSNP_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">H2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/EstimateHerit.html">EstimateHerit</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,</span>
<span>    summarystat <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">ncores</span>, model <span class="op">=</span> <span class="st">"GREML"</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, r2_LD <span class="op">=</span> <span class="fl">0</span>,</span>
<span>    LDSC_blocks <span class="op">=</span> <span class="fl">20</span>, REMLalgo <span class="op">=</span> <span class="fl">0</span>, nitr <span class="op">=</span> <span class="fl">100</span>, cat_covarfile <span class="op">=</span> <span class="cn">NULL</span>, quant_covarfile <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    prevalence <span class="op">=</span> <span class="fl">0.01</span>, partGRM <span class="op">=</span> <span class="cn">FALSE</span>, autosome <span class="op">=</span> <span class="cn">TRUE</span>, Xsome <span class="op">=</span> <span class="cn">TRUE</span>, nGRM <span class="op">=</span> <span class="fl">3</span>,</span>
<span>    cripticut <span class="op">=</span> <span class="fl">0.025</span>, minMAF <span class="op">=</span> <span class="cn">NULL</span>, maxMAF <span class="op">=</span> <span class="cn">NULL</span>, hg <span class="op">=</span> <span class="st">"hg19"</span>, PlotIndepSNP <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    IndepSNP_window_size <span class="op">=</span> <span class="fl">50</span>, IndepSNP_step_size <span class="op">=</span> <span class="fl">5</span>, IndepSNP_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    highLD_regions <span class="op">=</span> <span class="va">highLD_hg19</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Processing chromosome 1</span></span>
<span><span class="co">#&gt; Processing chromosome 2</span></span>
<span><span class="co">#&gt; Processing chromosome 3</span></span>
<span><span class="co">#&gt; Processing chromosome 4</span></span>
<span><span class="co">#&gt; Processing chromosome 5</span></span>
<span><span class="co">#&gt; Processing chromosome 6</span></span>
<span><span class="co">#&gt; Processing chromosome 7</span></span>
<span><span class="co">#&gt; Processing chromosome 8</span></span>
<span><span class="co">#&gt; Processing chromosome 9</span></span>
<span><span class="co">#&gt; Processing chromosome 10</span></span>
<span><span class="co">#&gt; Processing chromosome 23</span></span>
<span><span class="co">#&gt; ✔ GRM has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/xGXwasR.grm.bin]</span></span>
<span><span class="co">#&gt; ℹ Number of SNPs in each pair of individuals has been saved in the file [/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/xGXwasR.grm.N.bin]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Processing chromosome 24</span></span>
<span><span class="co">#&gt; ✔ Plots are saved in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX with name Heritability_Plots.jpeg</span></span>
<span><span class="co">#&gt; ✔ All GRM related files are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="va">H2</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="12%">
<col width="16%">
<col width="13%">
<col width="16%">
<col width="10%">
<col width="11%">
<col width="10%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="right">Chromosome</th>
<th align="right">SNP_proportion</th>
<th align="right">No.of.genes</th>
<th align="right">No.of.proteins</th>
<th align="right">Size_mb</th>
<th align="left">Source</th>
<th align="right">Variance</th>
<th align="right">SE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">0.0277453</td>
<td align="right">2177</td>
<td align="right">1936</td>
<td align="right">249.2506</td>
<td align="left">V(G)</td>
<td align="right">0.000000</td>
<td align="right">0.054312</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">0.0277453</td>
<td align="right">2177</td>
<td align="right">1936</td>
<td align="right">249.2506</td>
<td align="left">V(e)</td>
<td align="right">0.239076</td>
<td align="right">0.058127</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0.0277453</td>
<td align="right">2177</td>
<td align="right">1936</td>
<td align="right">249.2506</td>
<td align="left">Vp</td>
<td align="right">0.239076</td>
<td align="right">0.020390</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">0.0277453</td>
<td align="right">2177</td>
<td align="right">1936</td>
<td align="right">249.2506</td>
<td align="left">V(G)/Vp</td>
<td align="right">0.000001</td>
<td align="right">0.227173</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">0.0277453</td>
<td align="right">2177</td>
<td align="right">1936</td>
<td align="right">249.2506</td>
<td align="left">V(G)/Vp_L</td>
<td align="right">0.000001</td>
<td align="right">0.131597</td>
</tr>
<tr class="even">
<td align="right">2</td>
<td align="right">0.0275568</td>
<td align="right">1377</td>
<td align="right">1188</td>
<td align="right">243.1994</td>
<td align="left">V(G)</td>
<td align="right">0.000000</td>
<td align="right">0.050757</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="sexregress">
<code>SexRegress()</code><a class="anchor" aria-label="anchor" href="#sexregress"></a>
</h4>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Regression_Ex"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">fdata</span> <span class="op">&lt;-</span> <span class="va">Regression_Ex</span></span>
<span><span class="va">fdata</span><span class="op">$</span><span class="va">SEX</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">fdata</span><span class="op">$</span><span class="va">SEX</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">response_index</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">regressor_index</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexRegress.html">SexRegress</a></span><span class="op">(</span><span class="va">fdata</span>, <span class="va">regressor_index</span>, <span class="va">response_index</span><span class="op">)</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt;    Estimate  Std. Error     t value    Pr(&gt;|t|) </span></span>
<span><span class="co">#&gt; 0.001305843 0.004337332 0.301070570 0.763594133</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filterplinksample">
<code>FilterPlinkSample()</code><a class="anchor" aria-label="anchor" href="#filterplinksample"></a>
</h4>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"casesPlink"</span></span>
<span><span class="va">filter_sample</span> <span class="op">&lt;-</span> <span class="st">"cases"</span></span>
<span><span class="va">keep_remove_sample_file</span> <span class="op">&lt;-</span> <span class="st">"samples_example"</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span></span>
<span><span class="fu"><a href="../reference/FilterPlinkSample.html">FilterPlinkSample</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, keep_remove_sample_file <span class="op">=</span> <span class="va">keep_remove_sample_file</span>,</span>
<span>    keep <span class="op">=</span> <span class="va">keep</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ casesPlink PLINK files with desired samples are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="computegeneticpc">
<code>ComputeGeneticPC()</code><a class="anchor" aria-label="anchor" href="#computegeneticpc"></a>
</h4>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="st">"TRUE"</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">countPC</span> <span class="op">&lt;-</span> <span class="fl">20</span></span>
<span><span class="co">## Genetic PC</span></span>
<span><span class="va">GP</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputeGeneticPC.html">ComputeGeneticPC</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_hg19</span>, countPC <span class="op">=</span> <span class="fl">20</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/temp_libpatheab742dff3cd', reason 'Directory not empty'</span></span>
<span><span class="va">GP</span><span class="op">$</span><span class="va">PCs1</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">V1</th>
<th align="left">V2</th>
<th align="right">V3</th>
<th align="right">V4</th>
<th align="right">V5</th>
<th align="right">V6</th>
<th align="right">V7</th>
<th align="right">V8</th>
<th align="right">V9</th>
<th align="right">V10</th>
<th align="right">V11</th>
<th align="right">V12</th>
<th align="right">V13</th>
<th align="right">V14</th>
<th align="right">V15</th>
<th align="right">V16</th>
<th align="right">V17</th>
<th align="right">V18</th>
<th align="right">V19</th>
<th align="right">V20</th>
<th align="right">V21</th>
<th align="right">V22</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">0.0964618</td>
<td align="right">-0.0382514</td>
<td align="right">0.0090380</td>
<td align="right">-0.0895293</td>
<td align="right">-0.0252291</td>
<td align="right">0.0146543</td>
<td align="right">0.0153706</td>
<td align="right">-0.0076534</td>
<td align="right">-0.0055936</td>
<td align="right">0.0111945</td>
<td align="right">-0.0292772</td>
<td align="right">0.0001423</td>
<td align="right">-0.0344247</td>
<td align="right">-0.0076093</td>
<td align="right">0.0160821</td>
<td align="right">-0.1051830</td>
<td align="right">0.1415490</td>
<td align="right">-0.0112478</td>
<td align="right">0.0171623</td>
<td align="right">-0.0243035</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">0.0743932</td>
<td align="right">-0.0567271</td>
<td align="right">0.0396994</td>
<td align="right">0.0205343</td>
<td align="right">0.0230410</td>
<td align="right">-0.0360380</td>
<td align="right">-0.0422706</td>
<td align="right">-0.0036093</td>
<td align="right">-0.0133624</td>
<td align="right">0.0075615</td>
<td align="right">-0.0808966</td>
<td align="right">-0.0576625</td>
<td align="right">-0.0426657</td>
<td align="right">-0.0859803</td>
<td align="right">0.0511761</td>
<td align="right">-0.0529104</td>
<td align="right">0.0896545</td>
<td align="right">0.0756350</td>
<td align="right">-0.0112599</td>
<td align="right">-0.0034034</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">0.0645349</td>
<td align="right">-0.0561035</td>
<td align="right">-0.0145786</td>
<td align="right">-0.0264343</td>
<td align="right">-0.0018095</td>
<td align="right">0.0144521</td>
<td align="right">-0.0223747</td>
<td align="right">0.0297751</td>
<td align="right">0.0460854</td>
<td align="right">0.0145766</td>
<td align="right">0.0413334</td>
<td align="right">-0.0359086</td>
<td align="right">0.0016855</td>
<td align="right">0.1225740</td>
<td align="right">-0.0197903</td>
<td align="right">-0.0435513</td>
<td align="right">-0.0530760</td>
<td align="right">-0.0081402</td>
<td align="right">-0.0208054</td>
<td align="right">-0.0405354</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">0.0832472</td>
<td align="right">-0.0560208</td>
<td align="right">0.0302654</td>
<td align="right">-0.0011077</td>
<td align="right">-0.0530371</td>
<td align="right">-0.0335803</td>
<td align="right">-0.0772808</td>
<td align="right">0.0554254</td>
<td align="right">-0.0124211</td>
<td align="right">-0.0232910</td>
<td align="right">-0.0896787</td>
<td align="right">-0.0065826</td>
<td align="right">0.0044798</td>
<td align="right">0.0008519</td>
<td align="right">0.0037424</td>
<td align="right">0.0693772</td>
<td align="right">-0.0460331</td>
<td align="right">-0.0548226</td>
<td align="right">0.0251519</td>
<td align="right">0.0227838</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">0.0775687</td>
<td align="right">-0.0094644</td>
<td align="right">0.0451785</td>
<td align="right">-0.0373215</td>
<td align="right">-0.0617431</td>
<td align="right">-0.0719951</td>
<td align="right">-0.0718680</td>
<td align="right">-0.0358363</td>
<td align="right">0.0750992</td>
<td align="right">0.0004759</td>
<td align="right">-0.0146171</td>
<td align="right">-0.0595273</td>
<td align="right">-0.0020504</td>
<td align="right">-0.0301174</td>
<td align="right">0.0271127</td>
<td align="right">0.0367063</td>
<td align="right">0.0220704</td>
<td align="right">0.0080734</td>
<td align="right">0.0021652</td>
<td align="right">-0.0367468</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">0.0661714</td>
<td align="right">-0.0504906</td>
<td align="right">0.0248539</td>
<td align="right">-0.0828066</td>
<td align="right">0.0409418</td>
<td align="right">0.0055953</td>
<td align="right">-0.0220680</td>
<td align="right">-0.0612912</td>
<td align="right">-0.0328064</td>
<td align="right">-0.0164471</td>
<td align="right">0.0347364</td>
<td align="right">-0.0506891</td>
<td align="right">-0.0068565</td>
<td align="right">0.1102030</td>
<td align="right">0.0692138</td>
<td align="right">0.0478395</td>
<td align="right">0.0104342</td>
<td align="right">0.0588439</td>
<td align="right">0.0424195</td>
<td align="right">-0.0275438</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GP</span><span class="op">$</span><span class="va">plot</span></span></code></pre></div>
<div class="float">
<img src="overview-ComputeGeneticPC-1.png" alt="plot of chunk overview-ComputeGeneticPC"><div class="figcaption">plot of chunk overview-ComputeGeneticPC</div>
</div>
</div>
<div class="section level4">
<h4 id="clumpld">
<code>ClumpLD()</code><a class="anchor" aria-label="anchor" href="#clumpld"></a>
</h4>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex1"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Summary_Stat_Ex2"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">SNPdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span>, <span class="va">Summary_Stat_Ex2</span><span class="op">)</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.001</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">250</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">clumpedResult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ClumpLD.html">ClumpLD</a></span><span class="op">(</span></span>
<span>    <span class="va">DataDir</span>, <span class="va">finput</span>, <span class="va">SNPdata</span>, <span class="va">ResultDir</span>, <span class="va">clump_p1</span>,</span>
<span>    <span class="va">clump_p2</span>, <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, <span class="va">byCHR</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; • Processing summary statistics 1</span></span>
<span><span class="co">#&gt; • Processing summary statistics 2</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 1</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 1</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 2</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 2</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 3</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 3</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 4</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 4</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 5</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 5</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 6</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 6</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 7</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 7</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 8</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 8</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 9</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 9</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 10</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 10</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 23</span></span>
<span><span class="co">#&gt; • Running LD clumping for chromosome 24</span></span>
<span><span class="co">#&gt; ℹ No significant clump results for chromosome 24</span></span>
<span><span class="va">clumpedResult</span></span>
<span><span class="co">#&gt; $BestClump</span></span>
<span><span class="co">#&gt;         INDEX       PSNP    RSQ     KB        P ALLELES      F    CHR</span></span>
<span><span class="co">#&gt;        &lt;char&gt;     &lt;char&gt; &lt;char&gt; &lt;char&gt;   &lt;char&gt;  &lt;char&gt; &lt;char&gt; &lt;char&gt;</span></span>
<span><span class="co">#&gt; 1:  rs6529954  rs6529954      *      0 4.41e-09   AA/GG      2     23</span></span>
<span><span class="co">#&gt; 2: rs12858640 rs12858640      *      0 3.89e-09   CC/TT      2     23</span></span>
<span><span class="co">#&gt; 3:  rs5962098  rs5962098      *      0 1.71e-06   AA/GG      2     23</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AllClump</span></span>
<span><span class="co">#&gt;      INDEX_SNP        SNP DISTANCE   RSQ  ALLELES     F        P   CHR</span></span>
<span><span class="co">#&gt;         &lt;char&gt;     &lt;char&gt;    &lt;num&gt; &lt;num&gt;   &lt;char&gt; &lt;num&gt;    &lt;num&gt; &lt;int&gt;</span></span>
<span><span class="co">#&gt;  1:  rs6529954  rs6529954      0.0 1.000  2.8e-09    NA       NA    23</span></span>
<span><span class="co">#&gt;  2:  rs6529954  rs5962098    -47.2 0.702    AG/GA     1 1.96e-06    23</span></span>
<span><span class="co">#&gt;  3:  rs6529954  rs5962098    -47.2 0.702    AG/GA     2 1.71e-06    23</span></span>
<span><span class="co">#&gt;  4:  rs6529954  rs6529954      0.0 1.000    AA/GG     2 4.41e-09    23</span></span>
<span><span class="co">#&gt;  5:  rs6529954 rs12858640     30.5 0.981    AC/GT     1 4.37e-09    23</span></span>
<span><span class="co">#&gt;  6:  rs6529954 rs12858640     30.5 0.981    AC/GT     2 3.89e-09    23</span></span>
<span><span class="co">#&gt;  7: rs12858640 rs12858640      0.0 1.000 4.37e-09    NA       NA    23</span></span>
<span><span class="co">#&gt;  8: rs12858640  rs5962098    -77.7 0.716    CG/TA     1 1.96e-06    23</span></span>
<span><span class="co">#&gt;  9: rs12858640  rs5962098    -77.7 0.716    CG/TA     2 1.71e-06    23</span></span>
<span><span class="co">#&gt; 10: rs12858640  rs6529954    -30.5 0.981    CA/TG     1 2.80e-09    23</span></span>
<span><span class="co">#&gt; 11: rs12858640  rs6529954    -30.5 0.981    CA/TG     2 4.41e-09    23</span></span>
<span><span class="co">#&gt; 12: rs12858640 rs12858640      0.0 1.000    CC/TT     2 3.89e-09    23</span></span>
<span><span class="co">#&gt; 13:  rs5962098  rs5962098      0.0 1.000 1.96e-06    NA       NA    23</span></span>
<span><span class="co">#&gt; 14:  rs5962098  rs5962098      0.0 1.000    AA/GG     2 1.71e-06    23</span></span>
<span><span class="co">#&gt; 15:  rs5962098  rs6529954     47.2 0.702    AG/GA     1 2.80e-09    23</span></span>
<span><span class="co">#&gt; 16:  rs5962098  rs6529954     47.2 0.702    AG/GA     2 4.41e-09    23</span></span>
<span><span class="co">#&gt; 17:  rs5962098 rs12858640     77.7 0.716    AT/GC     1 4.37e-09    23</span></span>
<span><span class="co">#&gt; 18:  rs5962098 rs12858640     77.7 0.716    AT/GC     2 3.89e-09    23</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="getmfplink">
<code>GetMFPlink()</code><a class="anchor" aria-label="anchor" href="#getmfplink"></a>
</h4>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"females"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>,</span>
<span>    xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, prefixed as Test_output, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plinkvcf">
<code>plinkVCF()</code><a class="anchor" aria-label="anchor" href="#plinkvcf"></a>
</h4>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span> <span class="co"># Plink file</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example1"</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">PtoV</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">VtoP</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">Famfile</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">PVbyCHR</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="fu"><a href="../reference/plinkVCF.html">plinkVCF</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span>, <span class="va">VtoP</span>, <span class="va">PtoV</span>, <span class="va">Famfile</span>, <span class="va">PVbyCHR</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output files created in ResultDir: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="mergeregion">
<code>MergeRegion()</code><a class="anchor" aria-label="anchor" href="#mergeregion"></a>
</h4>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example_imputed"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX prefixed as Test_output</span></span>
<span><span class="va">y</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filterallele">
<code>FilterAllele()</code><a class="anchor" aria-label="anchor" href="#filterallele"></a>
</h4>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Filter_Test"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterAllele.html">FilterAllele</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ There is no multi-allelic SNP present in the input dataset.</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="plinksummary">
<code>PlinkSummary()</code><a class="anchor" aria-label="anchor" href="#plinksummary"></a>
</h4>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="filtersnp">FilterSNP<a class="anchor" aria-label="anchor" href="#filtersnp"></a>
</h4>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">SNPvec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs6529954"</span>, <span class="st">"rs12858640"</span>, <span class="st">"rs5962098"</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Filter_Test"</span></span>
<span><span class="fu"><a href="../reference/FilterSNP.html">FilterSNP</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span>, SNPvec <span class="op">=</span> <span class="va">SNPvec</span>, extract <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 3 SNPs are extracted</span></span>
<span><span class="co">#&gt; ✔ Plink files with extracted SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX prefixed as Filter_Test</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="dummycovar">
<code>DummyCovar()</code><a class="anchor" aria-label="anchor" href="#dummycovar"></a>
</h4>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">bfile</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">incovar</span> <span class="op">&lt;-</span> <span class="st">"covarfile_w_pc_age.txt"</span></span>
<span><span class="va">outcovar</span> <span class="op">&lt;-</span> <span class="st">"dummycovarfile"</span></span>
<span><span class="va">dummy_covars</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/DummyCovar.html">DummyCovar</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    bfile <span class="op">=</span> <span class="va">bfile</span>, incovar <span class="op">=</span> <span class="va">incovar</span>,</span>
<span>    outcovar <span class="op">=</span> <span class="va">outcovar</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Covariate file: dummycovarfile.cov is in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span>
<span><span class="va">dummy_covars</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Categorical</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">-9</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">-9</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">-9</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">-9</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">-9</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">-9</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="gxwasmiami">
<code>GXWASmiami()</code><a class="anchor" aria-label="anchor" href="#gxwasmiami"></a>
</h4>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Ffile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Mfile"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">FemaleWAS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">Ffile</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"BP"</span>, <span class="st">"P"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">FemaleWAS</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"POS"</span>, <span class="st">"pvalue"</span><span class="op">)</span></span>
<span><span class="va">MaleWAS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">Mfile</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"BP"</span>, <span class="st">"P"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">MaleWAS</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SNP"</span>, <span class="st">"CHR"</span>, <span class="st">"POS"</span>, <span class="st">"pvalue"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/GXWASmiami.html">GXWASmiami</a></span><span class="op">(</span>FemaleWAS <span class="op">=</span> <span class="va">FemaleWAS</span>, MaleWAS <span class="op">=</span> <span class="va">MaleWAS</span>, snp_pval <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Generating Miami plots for stratified test.</span></span>
<span><span class="co">#&gt; ℹ Saving plot to /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX/Stratified_GWAS.png</span></span></code></pre></div>
<div class="float">
<img src="overview-GXWASmiami-1.png" alt="plot of chunk overview-GXWASmiami"><div class="figcaption">plot of chunk overview-GXWASmiami</div>
</div>
<pre><code><span><span class="co">#&gt; ✔ Miami plot of stratified GWAS is saved in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp7SvuuX</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="sumstatgencorr">
<code>SumstatGenCorr()</code><a class="anchor" aria-label="anchor" href="#sumstatgencorr"></a>
</h4>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" tabindex="-1"></a>sumstat1 <span class="ot">&lt;-</span> GXwasR<span class="sc">:::</span><span class="fu">simulateSumstats</span>()</span>
<span id="cb81-2"><a href="#cb81-2" tabindex="-1"></a>sumstat2 <span class="ot">&lt;-</span> GXwasR<span class="sc">:::</span><span class="fu">simulateSumstats</span>()</span>
<span id="cb81-3"><a href="#cb81-3" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">nzchar</span>(<span class="fu">Sys.getenv</span>(<span class="st">"UKB_IMPUTED_HAPMAP2_PATH"</span>))) {</span>
<span id="cb81-4"><a href="#cb81-4" tabindex="-1"></a>    res <span class="ot">&lt;-</span> <span class="fu">SumstatGenCorr</span>(</span>
<span id="cb81-5"><a href="#cb81-5" tabindex="-1"></a>        <span class="at">ResultDir =</span> <span class="fu">tempdir</span>(),</span>
<span id="cb81-6"><a href="#cb81-6" tabindex="-1"></a>        <span class="at">referenceLD =</span> <span class="st">"UKB_imputed_hapmap2_SVD_eigen99_extraction"</span>,</span>
<span id="cb81-7"><a href="#cb81-7" tabindex="-1"></a>        <span class="at">sumstat1 =</span> sumstat1,</span>
<span id="cb81-8"><a href="#cb81-8" tabindex="-1"></a>        <span class="at">sumstat2 =</span> sumstat2,</span>
<span id="cb81-9"><a href="#cb81-9" tabindex="-1"></a>        <span class="at">parallel =</span> <span class="cn">TRUE</span></span>
<span id="cb81-10"><a href="#cb81-10" tabindex="-1"></a>    )</span>
<span id="cb81-11"><a href="#cb81-11" tabindex="-1"></a>    res</span>
<span id="cb81-12"><a href="#cb81-12" tabindex="-1"></a>}</span>
<span id="cb81-13"><a href="#cb81-13" tabindex="-1"></a><span class="co">#&gt; Analysis starts on Wed Aug  6 11:42:35 2025</span></span>
<span id="cb81-14"><a href="#cb81-14" tabindex="-1"></a><span class="co">#&gt; ℹ 9436 out of 769306 (1.23%) SNPs in reference panel are available in GWAS 1.</span></span>
<span id="cb81-15"><a href="#cb81-15" tabindex="-1"></a><span class="co">#&gt; ℹ 9409 out of 769306 (1.22%) SNPs in reference panel are available in GWAS 2.</span></span>
<span id="cb81-16"><a href="#cb81-16" tabindex="-1"></a><span class="co">#&gt; ! Warning: More than 1% SNPs in reference panel are missed in GWAS 1. This may generate bias in estimation. Please make sure that you are using the correct reference panel.</span></span>
<span id="cb81-17"><a href="#cb81-17" tabindex="-1"></a><span class="co">#&gt; ! Warning: More than 1% SNPs in reference panel are missed in GWAS 2. This may generate bias in estimation. Please make sure that you are using the correct reference panel.</span></span>
<span id="cb81-18"><a href="#cb81-18" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-19"><a href="#cb81-19" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-20"><a href="#cb81-20" tabindex="-1"></a>  <span class="er">|</span>                                                                                                                                                                                                                                                                              <span class="er">|</span>   <span class="dv">0</span>%</span>
<span id="cb81-21"><a href="#cb81-21" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-22"><a href="#cb81-22" tabindex="-1"></a>  <span class="er">|====</span>                                                                                                                                                                                                                                                                          <span class="er">|</span>   <span class="dv">2</span>%</span>
<span id="cb81-23"><a href="#cb81-23" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-24"><a href="#cb81-24" tabindex="-1"></a>  <span class="er">|=========</span>                                                                                                                                                                                                                                                                     <span class="er">|</span>   <span class="dv">3</span>%</span>
<span id="cb81-25"><a href="#cb81-25" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-26"><a href="#cb81-26" tabindex="-1"></a>  <span class="er">|=============</span>                                                                                                                                                                                                                                                                 <span class="er">|</span>   <span class="dv">5</span>%</span>
<span id="cb81-27"><a href="#cb81-27" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-28"><a href="#cb81-28" tabindex="-1"></a>  <span class="er">|=================</span>                                                                                                                                                                                                                                                             <span class="er">|</span>   <span class="dv">6</span>%</span>
<span id="cb81-29"><a href="#cb81-29" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-30"><a href="#cb81-30" tabindex="-1"></a>  <span class="er">|======================</span>                                                                                                                                                                                                                                                        <span class="er">|</span>   <span class="dv">8</span>%</span>
<span id="cb81-31"><a href="#cb81-31" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-32"><a href="#cb81-32" tabindex="-1"></a>  <span class="er">|==========================</span>                                                                                                                                                                                                                                                    <span class="er">|</span>  <span class="dv">10</span>%</span>
<span id="cb81-33"><a href="#cb81-33" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-34"><a href="#cb81-34" tabindex="-1"></a>  <span class="er">|==============================</span>                                                                                                                                                                                                                                                <span class="er">|</span>  <span class="dv">11</span>%</span>
<span id="cb81-35"><a href="#cb81-35" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-36"><a href="#cb81-36" tabindex="-1"></a>  <span class="er">|===================================</span>                                                                                                                                                                                                                                           <span class="er">|</span>  <span class="dv">13</span>%</span>
<span id="cb81-37"><a href="#cb81-37" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-38"><a href="#cb81-38" tabindex="-1"></a>  <span class="er">|=======================================</span>                                                                                                                                                                                                                                       <span class="er">|</span>  <span class="dv">15</span>%</span>
<span id="cb81-39"><a href="#cb81-39" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-40"><a href="#cb81-40" tabindex="-1"></a>  <span class="er">|============================================</span>                                                                                                                                                                                                                                  <span class="er">|</span>  <span class="dv">16</span>%</span>
<span id="cb81-41"><a href="#cb81-41" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-42"><a href="#cb81-42" tabindex="-1"></a>  <span class="er">|================================================</span>                                                                                                                                                                                                                              <span class="er">|</span>  <span class="dv">18</span>%</span>
<span id="cb81-43"><a href="#cb81-43" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-44"><a href="#cb81-44" tabindex="-1"></a>  <span class="er">|====================================================</span>                                                                                                                                                                                                                          <span class="er">|</span>  <span class="dv">19</span>%</span>
<span id="cb81-45"><a href="#cb81-45" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-46"><a href="#cb81-46" tabindex="-1"></a>  <span class="er">|=========================================================</span>                                                                                                                                                                                                                     <span class="er">|</span>  <span class="dv">21</span>%</span>
<span id="cb81-47"><a href="#cb81-47" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-48"><a href="#cb81-48" tabindex="-1"></a>  <span class="er">|=============================================================</span>                                                                                                                                                                                                                 <span class="er">|</span>  <span class="dv">23</span>%</span>
<span id="cb81-49"><a href="#cb81-49" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-50"><a href="#cb81-50" tabindex="-1"></a>  <span class="er">|=================================================================</span>                                                                                                                                                                                                             <span class="er">|</span>  <span class="dv">24</span>%</span>
<span id="cb81-51"><a href="#cb81-51" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-52"><a href="#cb81-52" tabindex="-1"></a>  <span class="er">|======================================================================</span>                                                                                                                                                                                                        <span class="er">|</span>  <span class="dv">26</span>%</span>
<span id="cb81-53"><a href="#cb81-53" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-54"><a href="#cb81-54" tabindex="-1"></a>  <span class="er">|==========================================================================</span>                                                                                                                                                                                                    <span class="er">|</span>  <span class="dv">27</span>%</span>
<span id="cb81-55"><a href="#cb81-55" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-56"><a href="#cb81-56" tabindex="-1"></a>  <span class="er">|==============================================================================</span>                                                                                                                                                                                                <span class="er">|</span>  <span class="dv">29</span>%</span>
<span id="cb81-57"><a href="#cb81-57" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-58"><a href="#cb81-58" tabindex="-1"></a>  <span class="er">|===================================================================================</span>                                                                                                                                                                                           <span class="er">|</span>  <span class="dv">31</span>%</span>
<span id="cb81-59"><a href="#cb81-59" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-60"><a href="#cb81-60" tabindex="-1"></a>  <span class="er">|=======================================================================================</span>                                                                                                                                                                                       <span class="er">|</span>  <span class="dv">32</span>%</span>
<span id="cb81-61"><a href="#cb81-61" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-62"><a href="#cb81-62" tabindex="-1"></a>  <span class="er">|===========================================================================================</span>                                                                                                                                                                                   <span class="er">|</span>  <span class="dv">34</span>%</span>
<span id="cb81-63"><a href="#cb81-63" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-64"><a href="#cb81-64" tabindex="-1"></a>  <span class="er">|================================================================================================</span>                                                                                                                                                                              <span class="er">|</span>  <span class="dv">35</span>%</span>
<span id="cb81-65"><a href="#cb81-65" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-66"><a href="#cb81-66" tabindex="-1"></a>  <span class="er">|====================================================================================================</span>                                                                                                                                                                          <span class="er">|</span>  <span class="dv">37</span>%</span>
<span id="cb81-67"><a href="#cb81-67" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-68"><a href="#cb81-68" tabindex="-1"></a>  <span class="er">|=========================================================================================================</span>                                                                                                                                                                     <span class="er">|</span>  <span class="dv">39</span>%</span>
<span id="cb81-69"><a href="#cb81-69" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-70"><a href="#cb81-70" tabindex="-1"></a>  <span class="er">|=============================================================================================================</span>                                                                                                                                                                 <span class="er">|</span>  <span class="dv">40</span>%</span>
<span id="cb81-71"><a href="#cb81-71" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-72"><a href="#cb81-72" tabindex="-1"></a>  <span class="er">|=================================================================================================================</span>                                                                                                                                                             <span class="er">|</span>  <span class="dv">42</span>%</span>
<span id="cb81-73"><a href="#cb81-73" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-74"><a href="#cb81-74" tabindex="-1"></a>  <span class="er">|======================================================================================================================</span>                                                                                                                                                        <span class="er">|</span>  <span class="dv">44</span>%</span>
<span id="cb81-75"><a href="#cb81-75" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-76"><a href="#cb81-76" tabindex="-1"></a>  <span class="er">|==========================================================================================================================</span>                                                                                                                                                    <span class="er">|</span>  <span class="dv">45</span>%</span>
<span id="cb81-77"><a href="#cb81-77" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-78"><a href="#cb81-78" tabindex="-1"></a>  <span class="er">|==============================================================================================================================</span>                                                                                                                                                <span class="er">|</span>  <span class="dv">47</span>%</span>
<span id="cb81-79"><a href="#cb81-79" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-80"><a href="#cb81-80" tabindex="-1"></a>  <span class="er">|===================================================================================================================================</span>                                                                                                                                           <span class="er">|</span>  <span class="dv">48</span>%</span>
<span id="cb81-81"><a href="#cb81-81" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-82"><a href="#cb81-82" tabindex="-1"></a>  <span class="er">|=======================================================================================================================================</span>                                                                                                                                       <span class="er">|</span>  <span class="dv">50</span>%</span>
<span id="cb81-83"><a href="#cb81-83" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-84"><a href="#cb81-84" tabindex="-1"></a>  <span class="er">|===========================================================================================================================================</span>                                                                                                                                   <span class="er">|</span>  <span class="dv">52</span>%</span>
<span id="cb81-85"><a href="#cb81-85" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-86"><a href="#cb81-86" tabindex="-1"></a>  <span class="er">|================================================================================================================================================</span>                                                                                                                              <span class="er">|</span>  <span class="dv">53</span>%</span>
<span id="cb81-87"><a href="#cb81-87" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-88"><a href="#cb81-88" tabindex="-1"></a>  <span class="er">|====================================================================================================================================================</span>                                                                                                                          <span class="er">|</span>  <span class="dv">55</span>%</span>
<span id="cb81-89"><a href="#cb81-89" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-90"><a href="#cb81-90" tabindex="-1"></a>  <span class="er">|========================================================================================================================================================</span>                                                                                                                      <span class="er">|</span>  <span class="dv">56</span>%</span>
<span id="cb81-91"><a href="#cb81-91" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-92"><a href="#cb81-92" tabindex="-1"></a>  <span class="er">|=============================================================================================================================================================</span>                                                                                                                 <span class="er">|</span>  <span class="dv">58</span>%</span>
<span id="cb81-93"><a href="#cb81-93" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-94"><a href="#cb81-94" tabindex="-1"></a>  <span class="er">|=================================================================================================================================================================</span>                                                                                                             <span class="er">|</span>  <span class="dv">60</span>%</span>
<span id="cb81-95"><a href="#cb81-95" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-96"><a href="#cb81-96" tabindex="-1"></a>  <span class="er">|=====================================================================================================================================================================</span>                                                                                                         <span class="er">|</span>  <span class="dv">61</span>%</span>
<span id="cb81-97"><a href="#cb81-97" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-98"><a href="#cb81-98" tabindex="-1"></a>  <span class="er">|==========================================================================================================================================================================</span>                                                                                                    <span class="er">|</span>  <span class="dv">63</span>%</span>
<span id="cb81-99"><a href="#cb81-99" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-100"><a href="#cb81-100" tabindex="-1"></a>  <span class="er">|==============================================================================================================================================================================</span>                                                                                                <span class="er">|</span>  <span class="dv">65</span>%</span>
<span id="cb81-101"><a href="#cb81-101" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-102"><a href="#cb81-102" tabindex="-1"></a>  <span class="er">|===================================================================================================================================================================================</span>                                                                                           <span class="er">|</span>  <span class="dv">66</span>%</span>
<span id="cb81-103"><a href="#cb81-103" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-104"><a href="#cb81-104" tabindex="-1"></a>  <span class="er">|=======================================================================================================================================================================================</span>                                                                                       <span class="er">|</span>  <span class="dv">68</span>%</span>
<span id="cb81-105"><a href="#cb81-105" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-106"><a href="#cb81-106" tabindex="-1"></a>  <span class="er">|===========================================================================================================================================================================================</span>                                                                                   <span class="er">|</span>  <span class="dv">69</span>%</span>
<span id="cb81-107"><a href="#cb81-107" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-108"><a href="#cb81-108" tabindex="-1"></a>  <span class="er">|================================================================================================================================================================================================</span>                                                                              <span class="er">|</span>  <span class="dv">71</span>%</span>
<span id="cb81-109"><a href="#cb81-109" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-110"><a href="#cb81-110" tabindex="-1"></a>  <span class="er">|====================================================================================================================================================================================================</span>                                                                          <span class="er">|</span>  <span class="dv">73</span>%</span>
<span id="cb81-111"><a href="#cb81-111" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-112"><a href="#cb81-112" tabindex="-1"></a>  <span class="er">|========================================================================================================================================================================================================</span>                                                                      <span class="er">|</span>  <span class="dv">74</span>%</span>
<span id="cb81-113"><a href="#cb81-113" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-114"><a href="#cb81-114" tabindex="-1"></a>  <span class="er">|=============================================================================================================================================================================================================</span>                                                                 <span class="er">|</span>  <span class="dv">76</span>%</span>
<span id="cb81-115"><a href="#cb81-115" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-116"><a href="#cb81-116" tabindex="-1"></a>  <span class="er">|=================================================================================================================================================================================================================</span>                                                             <span class="er">|</span>  <span class="dv">77</span>%</span>
<span id="cb81-117"><a href="#cb81-117" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-118"><a href="#cb81-118" tabindex="-1"></a>  <span class="er">|=====================================================================================================================================================================================================================</span>                                                         <span class="er">|</span>  <span class="dv">79</span>%</span>
<span id="cb81-119"><a href="#cb81-119" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-120"><a href="#cb81-120" tabindex="-1"></a>  <span class="er">|==========================================================================================================================================================================================================================</span>                                                    <span class="er">|</span>  <span class="dv">81</span>%</span>
<span id="cb81-121"><a href="#cb81-121" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-122"><a href="#cb81-122" tabindex="-1"></a>  <span class="er">|==============================================================================================================================================================================================================================</span>                                                <span class="er">|</span>  <span class="dv">82</span>%</span>
<span id="cb81-123"><a href="#cb81-123" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-124"><a href="#cb81-124" tabindex="-1"></a>  <span class="er">|==================================================================================================================================================================================================================================</span>                                            <span class="er">|</span>  <span class="dv">84</span>%</span>
<span id="cb81-125"><a href="#cb81-125" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-126"><a href="#cb81-126" tabindex="-1"></a>  <span class="er">|=======================================================================================================================================================================================================================================</span>                                       <span class="er">|</span>  <span class="dv">85</span>%</span>
<span id="cb81-127"><a href="#cb81-127" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-128"><a href="#cb81-128" tabindex="-1"></a>  <span class="er">|===========================================================================================================================================================================================================================================</span>                                   <span class="er">|</span>  <span class="dv">87</span>%</span>
<span id="cb81-129"><a href="#cb81-129" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-130"><a href="#cb81-130" tabindex="-1"></a>  <span class="er">|================================================================================================================================================================================================================================================</span>                              <span class="er">|</span>  <span class="dv">89</span>%</span>
<span id="cb81-131"><a href="#cb81-131" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-132"><a href="#cb81-132" tabindex="-1"></a>  <span class="er">|====================================================================================================================================================================================================================================================</span>                          <span class="er">|</span>  <span class="dv">90</span>%</span>
<span id="cb81-133"><a href="#cb81-133" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-134"><a href="#cb81-134" tabindex="-1"></a>  <span class="er">|========================================================================================================================================================================================================================================================</span>                      <span class="er">|</span>  <span class="dv">92</span>%</span>
<span id="cb81-135"><a href="#cb81-135" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-136"><a href="#cb81-136" tabindex="-1"></a>  <span class="er">|=============================================================================================================================================================================================================================================================</span>                 <span class="er">|</span>  <span class="dv">94</span>%</span>
<span id="cb81-137"><a href="#cb81-137" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-138"><a href="#cb81-138" tabindex="-1"></a>  <span class="er">|=================================================================================================================================================================================================================================================================</span>             <span class="er">|</span>  <span class="dv">95</span>%</span>
<span id="cb81-139"><a href="#cb81-139" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-140"><a href="#cb81-140" tabindex="-1"></a>  <span class="er">|=====================================================================================================================================================================================================================================================================</span>         <span class="er">|</span>  <span class="dv">97</span>%</span>
<span id="cb81-141"><a href="#cb81-141" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-142"><a href="#cb81-142" tabindex="-1"></a>  <span class="er">|==========================================================================================================================================================================================================================================================================</span>    <span class="er">|</span>  <span class="dv">98</span>%</span>
<span id="cb81-143"><a href="#cb81-143" tabindex="-1"></a>  <span class="sc">|</span>                                                                                                                                                                                                                                                                                    </span>
<span id="cb81-144"><a href="#cb81-144" tabindex="-1"></a>  <span class="er">|==============================================================================================================================================================================================================================================================================|</span> <span class="dv">100</span>%</span>
<span id="cb81-145"><a href="#cb81-145" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-146"><a href="#cb81-146" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-147"><a href="#cb81-147" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-148"><a href="#cb81-148" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb81-149"><a href="#cb81-149" tabindex="-1"></a><span class="co">#&gt; Integrating piecewise resultsError in estimating Genetic Correlation: Algorithm failed to converge after trying different initial values.</span></span>
<span id="cb81-150"><a href="#cb81-150" tabindex="-1"></a><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="ldprune">
<code>LDPrune()</code><a class="anchor" aria-label="anchor" href="#ldprune"></a>
</h4>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">prunedSNPs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LDPrune.html">LDPrune</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">finput</span>, <span class="va">ResultDir</span>, <span class="fl">50</span>, <span class="fl">5</span>, <span class="fl">0.2</span><span class="op">)</span></span>
<span><span class="va">prunedSNPs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "rs143773730" "rs75530702"  "rs147281566" "rs35854196"  "rs115490086" "rs12041521"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="executeplinkmaf">
<code>executePlinkMAF()</code><a class="anchor" aria-label="anchor" href="#executeplinkmaf"></a>
</h4>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">maf_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/executePlinkMAF.html">executePlinkMAF</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="va">maf_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">MAF</th>
<th align="right">NCHROBS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.233700</td>
<td align="right">552</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs75530702</td>
<td align="left">G</td>
<td align="left">A</td>
<td align="right">0.003623</td>
<td align="right">552</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.014490</td>
<td align="right">552</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">0.027170</td>
<td align="right">552</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="left">rs115490086</td>
<td align="left">T</td>
<td align="left">C</td>
<td align="right">0.007246</td>
<td align="right">552</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="left">rs12041521</td>
<td align="left">A</td>
<td align="left">G</td>
<td align="right">0.202900</td>
<td align="right">552</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="computeld">
<code>ComputeLD()</code><a class="anchor" aria-label="anchor" href="#computeld"></a>
</h4>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">snpld</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputeLD.html">ComputeLD</a></span><span class="op">(</span></span>
<span>    DataDir <span class="op">=</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span>, ResultDir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    finput <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ByCHR <span class="op">=</span> <span class="cn">TRUE</span>, CHRnum <span class="op">=</span> <span class="fl">1</span>, r2_LD <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="op">)</span></span>
<span><span class="va">snpld</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">CHR_A</th>
<th align="right">BP_A</th>
<th align="left">SNP_A</th>
<th align="right">CHR_B</th>
<th align="right">BP_B</th>
<th align="left">SNP_B</th>
<th align="right">R2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">1</td>
<td align="right">1617443</td>
<td align="left">rs112033089</td>
<td align="right">1</td>
<td align="right">1711414</td>
<td align="left">rs867207</td>
<td align="right">0.247273</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">1821625</td>
<td align="left">rs6603805</td>
<td align="right">1</td>
<td align="right">1875530</td>
<td align="left">rs2803333</td>
<td align="right">0.220796</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">2793754</td>
<td align="left">rs113572417</td>
<td align="right">1</td>
<td align="right">2806390</td>
<td align="left">rs112163476</td>
<td align="right">1.000000</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">2838513</td>
<td align="left">rs78083445</td>
<td align="right">1</td>
<td align="right">2845752</td>
<td align="left">rs10797367</td>
<td align="right">0.476771</td>
</tr>
<tr class="odd">
<td align="right">1</td>
<td align="right">4520285</td>
<td align="left">rs6426412</td>
<td align="right">1</td>
<td align="right">4549482</td>
<td align="left">rs667500</td>
<td align="right">0.228998</td>
</tr>
<tr class="even">
<td align="right">1</td>
<td align="right">4558972</td>
<td align="left">rs13374152</td>
<td align="right">1</td>
<td align="right">4580135</td>
<td align="left">rs13375764</td>
<td align="right">0.249483</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits." _medRxiv 2025.06.10.25329327_.</span></span>
<span><span class="co">#&gt;   doi:10.1101/2025.06.10.25329327 &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim, Winters, Actkins, Mayer, Congivaram,
Niarchou, Edwards, Davis, and Stranger, 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-06</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-06 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mgcv                   1.9-3      2025-04-04 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  nlme                   3.1-168    2025-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  polynom                1.4-1      2022-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  seqminer               9.7        2024-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp7SvuuX/temp_libpatheab742dff3cd</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
