<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>GXwasR Overview • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GXwasR Overview">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>GXwasR Overview</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">5 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/GXwasR_overview.Rmd" class="external-link"><code>vignettes/GXwasR_overview.Rmd</code></a></small>
      <div class="d-none name"><code>GXwasR_overview.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="gxwasr-description">
<a id="GXwasR Description"></a> GXwasR Description:<a class="anchor" aria-label="anchor" href="#gxwasr-description"></a>
</h2>
<p>This package implements various statistical genetics models for
Genome-Wide Association (GWA) and X-Chromosome Wide Association (XWA)
analyses in a sex-combined or sex-stratified way considering
X-Chromosome Inactivation (XCI) pattern. In addition to association
analysis, the package also enables testing for sex differences in
genetic effects, including the implementation of specific models and
applying best practices for additional quality control (QC) of genetic
data required for these tests. The package includes thirty different
functions in six different categories (A-F) which enable a comprehensive
pipeline for sex-aware genetic association analysis of common variants
with unrelated individuals.</p>
</div>
<div class="section level2">
<h2 id="basics">Basics<a class="anchor" aria-label="anchor" href="#basics"></a>
</h2>
<div class="section level3">
<h3 id="install-gxwasr">Install <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#install-gxwasr"></a>
</h3>
<p><code>R</code> is an open-source statistical environment which can be
easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em> is
a <code>R</code> package available via the <a href="http://bioconductor.org" class="external-link">Bioconductor</a> repository for packages.
<code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a> after which you can install
<em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em> by
using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Check that you have a valid Bioconductor installation</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="functions-overview">Functions Overview<a class="anchor" aria-label="anchor" href="#functions-overview"></a>
</h2>
<p>This document lists all the functions in GXwasR package.It offers
thirty-three distinct functions, which are organized into six main
categories: A) Pre-imputation QC, B) Post-imputation QC, C) Sex-combined
and sex-stratified GWAS with specialized analysis for XWAS, D)
Sex-differential test, E) High level analysis and a range of F) Utility
Functions. These categories and their respective functions are detailed
in the below Table.</p>
<div class="section level3">
<h3 id="table-of-functions">Table of Functions<a class="anchor" aria-label="anchor" href="#table-of-functions"></a>
</h3>
<table class="table">
<colgroup>
<col width="10%">
<col width="66%">
<col width="22%">
</colgroup>
<thead><tr class="header">
<th>Function</th>
<th>Description</th>
<th>Category</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>QCsnp()</td>
<td>Performs quality control for SNPs in PLINK binary files.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="even">
<td>QCsample()</td>
<td>Identifies outlier individuals based on heterozygosity and missing
genotype rates.</td>
<td>Pre-imputation QC, Post-imputation QC</td>
</tr>
<tr class="odd">
<td>AncestryCheck()</td>
<td>Evaluates samples’ ancestry and flags outliers using PCA.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="even">
<td>SexCheck()</td>
<td>Compares sex assignments with predictions from X chromosome
inbreeding coefficients.</td>
<td>Pre-imputation QC</td>
</tr>
<tr class="odd">
<td>Xhwe()</td>
<td>Filters X-chromosome variants violating Hardy-Weinberg Equilibrium
in females.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="even">
<td>MAFdiffSexControl()</td>
<td>Tests for significant MAF differences between sexes in control
samples.</td>
<td>Post-imputation QC</td>
</tr>
<tr class="odd">
<td>FilterRegion()</td>
<td>Filters out specific chromosomal regions from input PLINK
files.</td>
<td>Post-imputation QC, Utility Functions</td>
</tr>
<tr class="even">
<td>GXwas()</td>
<td>Runs GWAS models in autosomes with XWAS models like “FMcomb01”,
“FMcomb02”, “FMstratified” for binary and quantitative traits, and
“GWAScxci” for binary traits. Focuses on additive SNP effects,
multi-collinearity issues, and includes multiple covariates and their
interactions.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="odd">
<td>PvalComb()</td>
<td>Combines p-values from separate GWAS using various statistical
methods and corrects summary p-values.</td>
<td>Sex-combined and sex-stratified GWAS with XWAS</td>
</tr>
<tr class="even">
<td>SexDiff()</td>
<td>Evaluates sex differences in genetic effect size for each SNP.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>SexDiffZscore()</td>
<td>Presumably analyzes sex differences using Z-score methodology,
comparing genetic effect sizes between males and females.</td>
<td>Sex-differential test</td>
</tr>
<tr class="even">
<td>DiffZeroOne()</td>
<td>Assesses Z-scores for deviation from one and zero for statistics
like genetic correlation.</td>
<td>Sex-differential test</td>
</tr>
<tr class="odd">
<td>TestXGene()</td>
<td>Performs gene-based association tests using GWAS/XWAS summary
statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>MetaGWAS()</td>
<td>Combines summary-level GWAS results using fixed-effect and
random-effect models.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>ComputePRS()</td>
<td>Calculates polygenic risk scores from GWAS summary statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>GeneticCorrBT()</td>
<td>Computes genetic correlation between to traits.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>EstimateHerit()</td>
<td>Computes SNP heritability using GREML or LDSC models.</td>
<td>High level analysis</td>
</tr>
<tr class="even">
<td>SexRegress()</td>
<td>Not previously described. Presumably involves regression analyses
specific to sex-stratified data.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>FilterPlinkSample()</td>
<td>Prepares PLINK binary files with desired samples based on specified
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>ComputeGeneticPC()</td>
<td>Computes principal components from a genetic relationship matrix for
population stratification correction.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>ClumpLD()</td>
<td>Performs linkage disequilibrium clumping of SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>GetMFPlink()</td>
<td>Prepares separate male and female PLINK binary files from combined
files.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>plinkVCF()</td>
<td>Converts VCF files to plink binary formats and vice versa, including
creation of dummy FAM files.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>MergeRegion()</td>
<td>Combines two genotype datasets based on common or all SNPs.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterAllele()</td>
<td>Filters out multi-allelic variants from the genetic dataset,
essential for maintaining dataset integrity and simplifying genetic
analyses.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>PlinkSummary()</td>
<td>Provides a summary of genotype datasets in plink format.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>FilterSNP()</td>
<td>Filters out specific SNPs from the dataset based on user-defined
criteria.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>DummyCovar()</td>
<td>Recodes a categorical covariate into binary dummy variables for
statistical analysis in GXwasR.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>GXWASmiami()</td>
<td>Generates Miami plots for GWAS and XWAS.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>SumstatGenCorr()</td>
<td>Genetic Correlation Calculation from GWAS Summary Statistics.</td>
<td>High level analysis</td>
</tr>
<tr class="odd">
<td>LDPrune()</td>
<td>Performs LD pruning on SNP data.</td>
<td>Utility Functions</td>
</tr>
<tr class="even">
<td>executePlinkMAF()</td>
<td>Calculate minor allele frequencies.</td>
<td>Utility Functions</td>
</tr>
<tr class="odd">
<td>ComputeLD()</td>
<td>Calculate LD matrix.</td>
<td>Utility Functions</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="function-1">Function 1<a class="anchor" aria-label="anchor" href="#function-1"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">QCsnp</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "QCsnp"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-2">Function 2<a class="anchor" aria-label="anchor" href="#function-2"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">QCsample</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "QCsample"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-3">Function 3<a class="anchor" aria-label="anchor" href="#function-3"></a>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">AncestryCheck</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "AncestryCheck"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-4">Function 4<a class="anchor" aria-label="anchor" href="#function-4"></a>
</h4>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexCheck</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SexCheck"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-5">Function 5<a class="anchor" aria-label="anchor" href="#function-5"></a>
</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">Xhwe</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "Xhwe"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-6">Function 6<a class="anchor" aria-label="anchor" href="#function-6"></a>
</h4>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MAFdiffSexControl</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "MAFdiffSexControl"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-7">Function 7<a class="anchor" aria-label="anchor" href="#function-7"></a>
</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterRegion</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "FilterRegion"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-8">Function 8<a class="anchor" aria-label="anchor" href="#function-8"></a>
</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXwas</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "GXwas"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-9">Function 9<a class="anchor" aria-label="anchor" href="#function-9"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">PvalComb</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "PvalComb"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-10">Function 10<a class="anchor" aria-label="anchor" href="#function-10"></a>
</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiff</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SexDiff"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-11">Function 11<a class="anchor" aria-label="anchor" href="#function-11"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexDiffZscore</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SexDiffZscore"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-12">Function 12<a class="anchor" aria-label="anchor" href="#function-12"></a>
</h4>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">DiffZeroOne</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "DiffZeroOne"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-13">Function 13<a class="anchor" aria-label="anchor" href="#function-13"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">TestXGene</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "TestXGene"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-14">Function 14<a class="anchor" aria-label="anchor" href="#function-14"></a>
</h4>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MetaGXwas</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Error in `dev_help()`:</span></span>
<span><span class="co">#&gt; ! Can't find development topic `MetaGXwas`.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-15">Function 15<a class="anchor" aria-label="anchor" href="#function-15"></a>
</h4>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputePRS</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "ComputePRS"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-16">Function 16<a class="anchor" aria-label="anchor" href="#function-16"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GeneticCorrBT</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "GeneticCorrBT"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-17">Function 17<a class="anchor" aria-label="anchor" href="#function-17"></a>
</h4>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">EstimateHerit</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "EstimateHerit"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-18">Function 18<a class="anchor" aria-label="anchor" href="#function-18"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexRegress</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SexRegress"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-19">Function 19<a class="anchor" aria-label="anchor" href="#function-19"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterPlinkSample</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "FilterPlinkSample"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-20">Function 20<a class="anchor" aria-label="anchor" href="#function-20"></a>
</h4>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputeGeneticPC</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "ComputeGeneticPC"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-21">Function 21<a class="anchor" aria-label="anchor" href="#function-21"></a>
</h4>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ClumpLD</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "ClumpLD"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-22">Function 22<a class="anchor" aria-label="anchor" href="#function-22"></a>
</h4>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GetMFPlink</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "GetMFPlink"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-23">Function 23<a class="anchor" aria-label="anchor" href="#function-23"></a>
</h4>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">plinkVCF</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "plinkVCF"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-24">Function 24<a class="anchor" aria-label="anchor" href="#function-24"></a>
</h4>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MergeRegion</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "MergeRegion"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-25">Function 25<a class="anchor" aria-label="anchor" href="#function-25"></a>
</h4>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterAllele</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "FilterAllele"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-26">Function 26<a class="anchor" aria-label="anchor" href="#function-26"></a>
</h4>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">PlinkSummary</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "PlinkSummary"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-27">Function 27<a class="anchor" aria-label="anchor" href="#function-27"></a>
</h4>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">FilterSNP</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "FilterSNP"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-28">Function 28<a class="anchor" aria-label="anchor" href="#function-28"></a>
</h4>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">DummyCovar</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "DummyCovar"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-29">Function 29<a class="anchor" aria-label="anchor" href="#function-29"></a>
</h4>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">GXWASmiami</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "GXWASmiami"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-30">Function 30<a class="anchor" aria-label="anchor" href="#function-30"></a>
</h4>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SumstatGenCorr</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "SumstatGenCorr"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-31">Function 31<a class="anchor" aria-label="anchor" href="#function-31"></a>
</h4>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">LDPrune</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "LDPrune"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-32">Function 32<a class="anchor" aria-label="anchor" href="#function-32"></a>
</h4>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">executePlinkMAF</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "executePlinkMAF"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="function-33">Function 33<a class="anchor" aria-label="anchor" href="#function-33"></a>
</h4>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputeLD</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "ComputeLD"</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<p>Please follow these tutorials to know more about the functionality of
the package GXwasR.</p>
<p>Tutorial for performing post-imputation QC followed by sex-aware
association tests: (vignette:Use of GXwasR)</p>
<ul>
<li>Tutorial for performing pre-imputation QC using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/preimputationQC.html" class="external-link">https://boseb.github.io/GXwasR/articles/preimputationQC.html</a>
</li>
<li>Tutorial for performing post-imputation QC using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/postimputationQC.html" class="external-link">https://boseb.github.io/GXwasR/articles/postimputationQC.html</a>
</li>
<li>Tutorial for running GWAS, XWAS and Sex-differential Tests using
GXwasR: <a href="https://boseb.github.io/GXwasR/articles/gwas_models.html" class="external-link">https://boseb.github.io/GXwasR/articles/gwas_models.html</a>
</li>
<li>Tutorial for computing Polygenic Risk Score using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/GXwasR_PRS.html" class="external-link">https://boseb.github.io/GXwasR/articles/GXwasR_PRS.html</a>
</li>
<li>Tutorial for ancestry estimation: <a href="https://boseb.github.io/GXwasR/articles/decoding_ancestry.html" class="external-link">https://boseb.github.io/GXwasR/articles/decoding_ancestry.html</a>
</li>
<li>Tutorial for meta analysis using GXwasR: <a href="https://boseb.github.io/GXwasR/articles/meta_analysis.html" class="external-link">https://boseb.github.io/GXwasR/articles/meta_analysis.html</a>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-05</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  ! package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;    abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    commonmark             2.0.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    desc                   1.4.3      2023-12-10 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;    GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  P GXwasR               * 0.99.0     2025-08-05 [?] Bioconductor</span></span>
<span><span class="co">#&gt;    hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    hunspell               3.0.6      2025-03-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;    pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rcmdcheck              1.4.0      2021-09-27 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    roxygen2               7.3.2      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rstudioapi             0.17.1     2024-10-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    spelling               2.3.1      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    xopen                  1.0.1      2024-04-25 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpdyNNez/temp_libpath548171db3e3e</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt;  P ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
