<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Computing Polygenic Risk Score using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Computing Polygenic Risk Score using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Computing Polygenic Risk Score using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">8 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/vignettes/GXwasR_PRS.Rmd" class="external-link"><code>vignettes/GXwasR_PRS.Rmd</code></a></small>
      <div class="d-none name"><code>GXwasR_PRS.Rmd</code></div>
    </div>

    
    
<p><strong>This document provides a tutorial for computing
state-of-the-art polygenic risk score and sex-aware ploygenic risk score
utilizing GXwasR R package.</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some libraries (Users can ignore this.)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="call-gxwasr-library-and-load-example-datasets">Call GXwasR library and load example datasets:<a class="anchor" aria-label="anchor" href="#call-gxwasr-library-and-load-example-datasets"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call GXwasR library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="computing-standard-prs-using-computeprs">Computing standard PRS using ComputePRS():<a class="anchor" aria-label="anchor" href="#computing-standard-prs-using-computeprs"></a>
</h2>
<div class="section level3">
<h3 id="about-computeprs">About ComputePRS()<a class="anchor" aria-label="anchor" href="#about-computeprs"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">ComputePRS</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; ComputePRS: Computing polygenic risk score (PRS)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function calculates the polygenic risk score, which is the</span></span>
<span><span class="co">#&gt;      total of allele counts (genotypes) weighted by estimated effect</span></span>
<span><span class="co">#&gt;      sizes from genome-wide association studies. It uses C+T filtering</span></span>
<span><span class="co">#&gt;      techniques. The users could perform clumping procedure</span></span>
<span><span class="co">#&gt;      choromosome-wise and genome-wide. Also, the function offers the</span></span>
<span><span class="co">#&gt;      choice of including several genetic principal components along</span></span>
<span><span class="co">#&gt;      with other covariates. Using this function, users have the freedom</span></span>
<span><span class="co">#&gt;      to experiment with various clumping and thresholding arrangements</span></span>
<span><span class="co">#&gt;      to test a wide range of various parameter values.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      ComputePRS(</span></span>
<span><span class="co">#&gt;        DataDir,</span></span>
<span><span class="co">#&gt;        ResultDir = tempdir(),</span></span>
<span><span class="co">#&gt;        finput,</span></span>
<span><span class="co">#&gt;        summarystat,</span></span>
<span><span class="co">#&gt;        phenofile,</span></span>
<span><span class="co">#&gt;        covarfile = NULL,</span></span>
<span><span class="co">#&gt;        effectsize = c("BETA", "OR"),</span></span>
<span><span class="co">#&gt;        ldclump = FALSE,</span></span>
<span><span class="co">#&gt;        LDreference,</span></span>
<span><span class="co">#&gt;        clump_p1,</span></span>
<span><span class="co">#&gt;        clump_p2,</span></span>
<span><span class="co">#&gt;        clump_r2,</span></span>
<span><span class="co">#&gt;        clump_kb,</span></span>
<span><span class="co">#&gt;        byCHR = TRUE,</span></span>
<span><span class="co">#&gt;        pthreshold = c(0.001, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5),</span></span>
<span><span class="co">#&gt;        highLD_regions,</span></span>
<span><span class="co">#&gt;        ld_prunning = FALSE,</span></span>
<span><span class="co">#&gt;        window_size = 50,</span></span>
<span><span class="co">#&gt;        step_size = 5,</span></span>
<span><span class="co">#&gt;        r2_threshold = 0.02,</span></span>
<span><span class="co">#&gt;        nPC = 6,</span></span>
<span><span class="co">#&gt;        pheno_type = "binary"</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  DataDir: A character string for the file path of the all the input</span></span>
<span><span class="co">#&gt;           files.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ResultDir: A character string for the file path where all output files</span></span>
<span><span class="co">#&gt;           will be stored. The default is tempdir().</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   finput: Character string, specifying the prefix of the input PLINK</span></span>
<span><span class="co">#&gt;           binary files for the genotype data i.e., the target data</span></span>
<span><span class="co">#&gt;           based on which clumping procedure will be performed. This</span></span>
<span><span class="co">#&gt;           file needs to be in DataDir. If your target data are small</span></span>
<span><span class="co">#&gt;           (e.g. N &lt; 500) then you can use the 1000 Genomes Project</span></span>
<span><span class="co">#&gt;           samples. Make sure to use the population that most closely</span></span>
<span><span class="co">#&gt;           reflects represents the base sample.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; summarystat: A dataframe object with GWAS summary statistics.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           The mandatory column headers in this dataframe are:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'CHR'(Chromosome code)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'BP'(Basepair position)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'A1' (effect allele)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'SNP' (i.e., SNP identifier)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'BETA' or 'OR' (i.e., effect-size or logarithm of odds</span></span>
<span><span class="co">#&gt;               ratio)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;             • 'P' (i.e., p-values).</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;           Special Notes: The first three columns needed to be 'SNP',</span></span>
<span><span class="co">#&gt;           'A1' and 'BETA' or 'OR'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; phenofile: A character string, specifying the name of the mandatory</span></span>
<span><span class="co">#&gt;           phenotype file. This is a plain text file with no header</span></span>
<span><span class="co">#&gt;           line; columns family ID, individual ID and phenotype columns.</span></span>
<span><span class="co">#&gt;           For binary trait, the phenotypic value should be coded as 0</span></span>
<span><span class="co">#&gt;           or 1, then it will be recognized as a case-control study (0</span></span>
<span><span class="co">#&gt;           for controls and 1 for cases). Missing value should be</span></span>
<span><span class="co">#&gt;           represented by "-9" or "NA". The interested phenotype column</span></span>
<span><span class="co">#&gt;           should be labeled as "Pheno1". This file needs to be in</span></span>
<span><span class="co">#&gt;           'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; covarfile: A character string, specifying the name of the covariate</span></span>
<span><span class="co">#&gt;           file which is a plain .text file with no header line; columns</span></span>
<span><span class="co">#&gt;           are family ID, individual ID and the covariates. The default</span></span>
<span><span class="co">#&gt;           is 'NULL'. This file needs to be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; effectsize: Boolean value, 'BETA' or 'OR', specifying the type of the</span></span>
<span><span class="co">#&gt;           GWAS effectsize. The default is 'BETA'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  ldclump: Boolean value, 'TRUE' or 'FALSE', specifying whether to</span></span>
<span><span class="co">#&gt;           perform clumping or not.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; LDreference: A character string, specifying the prefix of the PLINK</span></span>
<span><span class="co">#&gt;           files of the population reference panel of the same ancestry,</span></span>
<span><span class="co">#&gt;           and ideally the one that was used for imputing your target</span></span>
<span><span class="co">#&gt;           dataset. These files should be in 'DataDir'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p1: Numeric value, specifying the significance threshold for</span></span>
<span><span class="co">#&gt;           index SNPs if 'ldclump' was set to be 'TRUE'. The default is</span></span>
<span><span class="co">#&gt;           0.0001.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_p2: Numeric value, specifying the secondary significance</span></span>
<span><span class="co">#&gt;           threshold for clumped SNPs if 'ldclump' was set to be 'TRUE'.</span></span>
<span><span class="co">#&gt;           The default is 0.01</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_r2: Numeric value, specifying the linkage disequilibrium (LD)</span></span>
<span><span class="co">#&gt;           threshold for clumping if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 0.50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; clump_kb: Integer value, specifying the physical distance threshold in</span></span>
<span><span class="co">#&gt;           base-pair for clumping if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 250.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    byCHR: Boolean value, 'TRUE' or 'FALSE', specifying chromosome-wise</span></span>
<span><span class="co">#&gt;           clumping procedure if 'ldclump' was set to be 'TRUE'. The</span></span>
<span><span class="co">#&gt;           default is 'TRUE'</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pthreshold: Numeric vector, containing several p value thresholds to</span></span>
<span><span class="co">#&gt;           maximize predictive ability of the derived polygenic scores.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; highLD_regions: Character string, specifying the .txt file name with</span></span>
<span><span class="co">#&gt;           known genomic regions with high LD. The default is 'NULL'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ld_prunning: Boolean value, 'TRUE' or 'FALSE' for LD-based filtering</span></span>
<span><span class="co">#&gt;           for computing genetic PC as covariates.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; window_size: Integer value, specifying a window size in variant count</span></span>
<span><span class="co">#&gt;           or kilobase for LD-based filtering in computing genetic PC.</span></span>
<span><span class="co">#&gt;           The default is 50.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; step_size: Integer value, specifying a variant count to shift the</span></span>
<span><span class="co">#&gt;           window at the end of each step for LD filtering in computing</span></span>
<span><span class="co">#&gt;           genetic PCs. The default is 5.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; r2_threshold: Numeric value between 0 to 1 of pairwise r^2 threshold</span></span>
<span><span class="co">#&gt;           for LD-based filtering in computing genetic PCs. The default</span></span>
<span><span class="co">#&gt;           is 0.02.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      nPC: Positive integer value, specifying the number of genetic PCs</span></span>
<span><span class="co">#&gt;           to be included as predictor in the PRS model fit. The default</span></span>
<span><span class="co">#&gt;           is 6.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; pheno_type: Boolean value, ‘binary’ or ‘quantitative’, specifying the</span></span>
<span><span class="co">#&gt;           type of the trait. The default is ‘binary’.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      A list object containing a dataframe a numeric value, a GeneticPC</span></span>
<span><span class="co">#&gt;      plot (if requested), and a PRS plot. The dataframe,PRS, contains</span></span>
<span><span class="co">#&gt;      four mandatory columns, such as, IID (i.e., Individual ID), FID</span></span>
<span><span class="co">#&gt;      (i.e., Family ID), Pheno1 (i.e., the trait for PRS) and Score</span></span>
<span><span class="co">#&gt;      (i.e., the best PRS). Other columns of covariates could be there.</span></span>
<span><span class="co">#&gt;      The numeric value, BestP contains the threshold of of the best</span></span>
<span><span class="co">#&gt;      p-value for the best pRS model fit.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Also, the function produces several plots such as p-value</span></span>
<span><span class="co">#&gt;      thresholds vs PRS model fit and PRS distribution among male and</span></span>
<span><span class="co">#&gt;      females. For case-control data, it shows PRS distribution among</span></span>
<span><span class="co">#&gt;      cases and controls and ROC curves as well.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Author(s):</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Banabithi Bose</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Summary_Stat_Ex1", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_phenofile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_covarfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("Example_pthresoldfile", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      data("highLD_hg19", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="co">#&gt;      ResultDir &lt;- tempdir()</span></span>
<span><span class="co">#&gt;      finput &lt;- "GXwasR_example"</span></span>
<span><span class="co">#&gt;      summarystat &lt;- Summary_Stat_Ex1[, c(2, 4, 7, 1, 3, 12)]</span></span>
<span><span class="co">#&gt;      phenofile &lt;- Example_phenofile # Cannot be NULL</span></span>
<span><span class="co">#&gt;      # The interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="co">#&gt;      covarfile &lt;- Example_covarfile</span></span>
<span><span class="co">#&gt;      clump_p1 &lt;- 0.0001</span></span>
<span><span class="co">#&gt;      clump_p2 &lt;- 0.0001</span></span>
<span><span class="co">#&gt;      clump_kb &lt;- 500</span></span>
<span><span class="co">#&gt;      clump_r2 &lt;- 0.5</span></span>
<span><span class="co">#&gt;      byCHR &lt;- TRUE</span></span>
<span><span class="co">#&gt;      pthreshold &lt;- Example_pthresoldfile$Threshold</span></span>
<span><span class="co">#&gt;      ld_prunning &lt;- TRUE</span></span>
<span><span class="co">#&gt;      highLD_regions &lt;- highLD_hg19</span></span>
<span><span class="co">#&gt;      window_size &lt;- 50</span></span>
<span><span class="co">#&gt;      step_size &lt;- 5</span></span>
<span><span class="co">#&gt;      r2_threshold &lt;- 0.02</span></span>
<span><span class="co">#&gt;      nPC &lt;- 6 # We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="co">#&gt;      pheno_type &lt;- "binary"</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      PRSresult &lt;- ComputePRS(DataDir, ResultDir, finput, summarystat, phenofile, covarfile,</span></span>
<span><span class="co">#&gt;          effectsize = "BETA", LDreference = "GXwasR_example", ldclump = FALSE, clump_p1, clump_p2,</span></span>
<span><span class="co">#&gt;          clump_r2, clump_kb, byCHR = TRUE, pthreshold = pthreshold, highLD_regions = highLD_regions,</span></span>
<span><span class="co">#&gt;          ld_prunning = TRUE, window_size = 50, step_size = 5, r2_threshold = 0.02, nPC = 6,</span></span>
<span><span class="co">#&gt;          pheno_type = "binary"</span></span>
<span><span class="co">#&gt;      )</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      ## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="co">#&gt;      PRS &lt;- PRSresult$PRS</span></span>
<span><span class="co">#&gt;      PRS[seq_len(10), ]</span></span>
<span><span class="co">#&gt;      ## The best threshold</span></span>
<span><span class="co">#&gt;      BestPvalue &lt;- PRSresult$BestP$Threshold</span></span>
<span><span class="co">#&gt;      BestPvalue</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="running-computeprs">Running ComputePRS()<a class="anchor" aria-label="anchor" href="#running-computeprs"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Example_phenofile</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Example_covarfile</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Example_pthresoldfile</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">highLD_hg19</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL, the interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="va">Example_covarfile</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">ldclump</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">PRSresult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputePRS.html">ComputePRS</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>, effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">clump_p1</span>, <span class="va">clump_p2</span>, <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>, ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>, pheno_type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f', reason 'Directory not empty'</span></span>
<span><span class="co">#&gt; ℹ 0.001</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.001</span></span>
<span><span class="co">#&gt; ℹ 0.05</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.05</span></span>
<span><span class="co">#&gt; ℹ 0.1</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.1</span></span>
<span><span class="co">#&gt; ℹ 0.2</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.2</span></span>
<span><span class="co">#&gt; ℹ 0.3</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.3</span></span>
<span><span class="co">#&gt; ℹ 0.4</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.4</span></span>
<span><span class="co">#&gt; ℹ 0.5</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.5</span></span>
<span><span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS_plot</span></span>
<span><span class="co">#&gt; Error in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, : Metric information not available for this family/device</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="va">PRS</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS</span></span>
<span><span class="va">PRS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="7%">
<col width="6%">
<col width="3%">
<col width="8%">
<col width="8%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Pheno1</th>
<th align="right">AGE</th>
<th align="right">testcovar</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">SCORE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">1</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0964618</td>
<td align="right">-0.0382514</td>
<td align="right">0.0090380</td>
<td align="right">-0.0895293</td>
<td align="right">-0.0252291</td>
<td align="right">0.0146543</td>
<td align="right">0.0484212</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">1</td>
<td align="right">81</td>
<td align="right">1</td>
<td align="right">0.0743932</td>
<td align="right">-0.0567271</td>
<td align="right">0.0396994</td>
<td align="right">0.0205343</td>
<td align="right">0.0230410</td>
<td align="right">-0.0360380</td>
<td align="right">0.0293742</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">1</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">0.0645349</td>
<td align="right">-0.0561035</td>
<td align="right">-0.0145786</td>
<td align="right">-0.0264343</td>
<td align="right">-0.0018095</td>
<td align="right">0.0144521</td>
<td align="right">0.0099674</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">2</td>
<td align="right">75</td>
<td align="right">0</td>
<td align="right">0.0832472</td>
<td align="right">-0.0560208</td>
<td align="right">0.0302654</td>
<td align="right">-0.0011077</td>
<td align="right">-0.0530371</td>
<td align="right">-0.0335803</td>
<td align="right">0.1042060</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">1</td>
<td align="right">88</td>
<td align="right">1</td>
<td align="right">0.0775687</td>
<td align="right">-0.0094644</td>
<td align="right">0.0451785</td>
<td align="right">-0.0373215</td>
<td align="right">-0.0617431</td>
<td align="right">-0.0719951</td>
<td align="right">0.0314303</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">1</td>
<td align="right">24</td>
<td align="right">1</td>
<td align="right">0.0661714</td>
<td align="right">-0.0504906</td>
<td align="right">0.0248539</td>
<td align="right">-0.0828066</td>
<td align="right">0.0409418</td>
<td align="right">0.0055953</td>
<td align="right">0.0722689</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00179</td>
<td align="right">2</td>
<td align="right">78</td>
<td align="right">1</td>
<td align="right">0.0806110</td>
<td align="right">-0.0115409</td>
<td align="right">-0.0075072</td>
<td align="right">-0.0328823</td>
<td align="right">0.0355973</td>
<td align="right">0.0424512</td>
<td align="right">0.1165240</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00180</td>
<td align="right">1</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">0.0840638</td>
<td align="right">0.0011783</td>
<td align="right">0.0128646</td>
<td align="right">0.0591540</td>
<td align="right">0.0547667</td>
<td align="right">-0.0101595</td>
<td align="right">0.0617159</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00182</td>
<td align="right">1</td>
<td align="right">50</td>
<td align="right">1</td>
<td align="right">0.1029760</td>
<td align="right">-0.0430046</td>
<td align="right">0.0199971</td>
<td align="right">-0.0302365</td>
<td align="right">-0.0901911</td>
<td align="right">-0.0378515</td>
<td align="right">0.0283703</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00183</td>
<td align="right">1</td>
<td align="right">58</td>
<td align="right">0</td>
<td align="right">0.0827103</td>
<td align="right">-0.0386530</td>
<td align="right">0.0509366</td>
<td align="right">-0.0015378</td>
<td align="right">-0.1045610</td>
<td align="right">0.0623634</td>
<td align="right">0.0386297</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The best threshold</span></span>
<span><span class="va">BestPvalue</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">BestP</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">BestPvalue</span></span>
<span><span class="co">#&gt; [1] 0.05</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="computing-sex-aware-prs">Computing sex-aware PRS<a class="anchor" aria-label="anchor" href="#computing-sex-aware-prs"></a>
</h2>
<div class="section level3">
<h3 id="datasets-for-computing-prs">Datasets for computing PRS<a class="anchor" aria-label="anchor" href="#datasets-for-computing-prs"></a>
</h3>
<p><strong>Discovery Data</strong> i.e., GWAS summary statistics with
mandatory columns such as “SNP”(SNP names), “A1”(effect allele),
“BETA”(effect size in beta value) and “P”(p-value).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example discovery data is included in GXwasR</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span><span class="op">)</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p><strong>Target Data</strong> i.e., genotype dataset in plink .bed,
.bim and .fam format.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example target data is included in GXwasR</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quality-control-of-the-datasets-before-computing-sex-aware-prs-">Quality control of the datasets before computing sex-aware PRS.<a class="anchor" aria-label="anchor" href="#quality-control-of-the-datasets-before-computing-sex-aware-prs-"></a>
</h3>
<p>Users must ensure that both datasets are having the <strong>same
genome built</strong>. Here’s the quality control procedure:</p>
<p><strong>(A) Discovery Data: Remove multi-allelic, indels and
ambiguous (A/T or C/G) SNPs. Then remove SNPs with minor allele
frequency (MAF) &lt; 0.05 and quality info score &lt; 0.1. For this
filtering, R packages like data.table, dplyr, tidyverse can be
used.</strong></p>
<p><strong>(B) Target Data: Remove multi-allelic, indels and ambiguous
(A/T or C/G) SNPs. Then remove SNPs with MAF &lt; 0.05 and Hardy
Weinberg Equilibrium (hwe) &gt; e-10. For this filtering steps, users
should use <code><a href="../reference/FilterAllele.html">FilterAllele()</a></code> and <code><a href="../reference/QCsnp.html">QCsnp()</a></code>
functions in R.</strong></p>
<p>Let’s see how to use these functions. For the details of these
utility functions, please follow <a href="https://boseb.github.io/GXwasR/articles/GXwasR_overview.html" class="external-link">https://boseb.github.io/GXwasR/articles/GXwasR_overview.html</a>.</p>
<p>We will use <code><a href="../reference/FilterAllele.html">FilterAllele()</a></code> on target data to filter out
the multi-allelic SNPs. Also, note that GXwasR doesn’t work with
multi-allelic variants. So this filtering step is important.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Target data</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"filtered_multiallelic"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterAllele.html">FilterAllele</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ There is no multi-allelic SNP present in the input dataset.</span></span></code></pre></div>
<p>Now, we will use <code><a href="../reference/QCsnp.html">QCsnp()</a></code> to remove ambiguous (A/T or
C/G) SNPs and SNPs with MAF &lt; 0.05 and Hardy Weinberg Equilibrium
(hwe) &gt; e-10.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Since there was no multiallelic SNPs, we will continue with original input data.</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"filtered_step1"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">diffmissFilter</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">dmissX</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">dmissAutoY</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co">## Since the filtering doesn't require us to run on cases and controls separately, we will make this parameter FALSE.</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hwe</span> <span class="op">&lt;-</span> <span class="fl">1e-10</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, hwe <span class="op">=</span> <span class="va">hwe</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 4214 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; --hwe: 3 variants removed due to Hardy-Weinberg exact test.</span></span>
<span><span class="co">#&gt; 5467 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; ℹ No filter based on differential missingness will be applied.</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files prefixed as ,filtered_step1, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 26527 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 16843 SNPs after filtering.</span></span></code></pre></div>
<p><strong>(C) Check whether SNPs present in discovery data are in
target data.</strong></p>
<p>Let’s gather the common SNPs between these.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">SNP1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">summarystat</span><span class="op">$</span><span class="va">SNP</span><span class="op">)</span></span>
<span><span class="va">targetbim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/filtered_step1.bim"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">SNP2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">targetbim</span><span class="op">$</span><span class="va">V2</span><span class="op">)</span></span>
<span><span class="va">commonSNP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">SNP1</span>, <span class="va">SNP2</span><span class="op">)</span> <span class="co">## 991 SNPs are common between our discovery data and target data</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="filtering-discovery-data-and-target-data-with-common-snps">Filtering discovery data and target data with common SNPs<a class="anchor" aria-label="anchor" href="#filtering-discovery-data-and-target-data-with-common-snps"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">commonSNP</span> <span class="op">&lt;-</span> <span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/as.data.table.html" class="external-link">as.data.table</a></span><span class="op">(</span><span class="va">commonSNP</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">commonSNP</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"SNP"</span></span>
<span><span class="va">NewDiscoveryData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">commonSNP</span>, <span class="va">summarystat</span>, by <span class="op">=</span> <span class="st">"SNP"</span><span class="op">)</span></span>
<span><span class="co">## For making target data with common SNPs, we will use FilterSNP().</span></span>
<span><span class="va">SNPvec</span> <span class="op">&lt;-</span> <span class="va">commonSNP</span></span>
<span><span class="co"># Need to copy filtered_step1 file from ResultDir to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"filtered_step1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"filtered_step1"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"NewtargetData"</span></span>
<span><span class="fu"><a href="../reference/FilterSNP.html">FilterSNP</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span>, <span class="va">SNPvec</span>, extract <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 991 SNPs are extracted</span></span>
<span><span class="co">#&gt; ✔ Plink files with extracted SNPs are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ prefixed as NewtargetData</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>Now, our discovery data and target data are ready. We will proceed
towards computing first sex-combined PRS and then sex-stratified
PRS.</p>
</div>
</div>
<div class="section level2">
<h2 id="sex-combined-prs-computation">Sex-combined PRS computation<a class="anchor" aria-label="anchor" href="#sex-combined-prs-computation"></a>
</h2>
<p><strong>(A) Perform LD clumping for discovery data using target data
as reference.</strong></p>
<p><strong>(B) Compute PRS at a variety of p-value
thresholds.</strong></p>
<p><strong>(C) Select best threshold based on R-square (for quantitative
trait) or MacFadden R-square (for binary trait) using generalized linear
model with covariates, such as genetic PCs.</strong></p>
<p>We will utilize the <code><a href="../reference/ComputePRS.html">ComputePRS()</a></code> to perform the steps
(A), (B) and (C).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="co"># Filtered target data needs to be copied from ResultDir to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"NewtargetData"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"NewtargetData"</span></span>
<span></span>
<span><span class="co"># Filtered discovery data.</span></span>
<span><span class="co"># Need to maintain the first three coulmn of this dataset as SNP ID, Effect Allele and Effect Size</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">NewDiscoveryData</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="va">Example_phenofile</span> <span class="co"># Cannot be NULL, the interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="co">## Added "AGE" and "testcovar" as covariates.</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="va">Example_covarfile</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">effectsize</span> <span class="op">&lt;-</span> <span class="st">"BETA"</span></span>
<span><span class="va">ldclump</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">PRSresult</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputePRS.html">ComputePRS</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>, effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">clump_p1</span>, <span class="va">clump_p2</span>, <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>, ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>, pheno_type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f', reason 'Directory not empty'</span></span>
<span><span class="co">#&gt; ℹ 0.001</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.001</span></span>
<span><span class="co">#&gt; ℹ 0.05</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.05</span></span>
<span><span class="co">#&gt; ℹ 0.1</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.1</span></span>
<span><span class="co">#&gt; ℹ 0.2</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.2</span></span>
<span><span class="co">#&gt; ℹ 0.3</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.3</span></span>
<span><span class="co">#&gt; ℹ 0.4</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.4</span></span>
<span><span class="co">#&gt; ℹ 0.5</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.5</span></span>
<span><span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS_plot</span></span>
<span><span class="co">#&gt; Error in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, : Metric information not available for this family/device</span></span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="va">PRS</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">PRS</span></span>
<span><span class="va">PRS</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="7%">
<col width="7%">
<col width="6%">
<col width="3%">
<col width="8%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Pheno1</th>
<th align="right">AGE</th>
<th align="right">testcovar</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">SCORE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">1</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0816192</td>
<td align="right">0.0088405</td>
<td align="right">0.0457488</td>
<td align="right">-0.0458072</td>
<td align="right">0.0361127</td>
<td align="right">-0.0444544</td>
<td align="right">0.0446702</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">1</td>
<td align="right">81</td>
<td align="right">1</td>
<td align="right">0.0783950</td>
<td align="right">0.0793499</td>
<td align="right">-0.0203422</td>
<td align="right">0.0507417</td>
<td align="right">0.0060022</td>
<td align="right">0.0682457</td>
<td align="right">0.0227245</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">1</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">0.0417082</td>
<td align="right">-0.0182760</td>
<td align="right">-0.0141222</td>
<td align="right">-0.0796694</td>
<td align="right">-0.1079240</td>
<td align="right">0.0417553</td>
<td align="right">-0.0021202</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">2</td>
<td align="right">75</td>
<td align="right">0</td>
<td align="right">0.0330363</td>
<td align="right">-0.0687181</td>
<td align="right">0.0420013</td>
<td align="right">0.0035531</td>
<td align="right">-0.0326060</td>
<td align="right">0.0546183</td>
<td align="right">0.1291310</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">1</td>
<td align="right">88</td>
<td align="right">1</td>
<td align="right">0.1078350</td>
<td align="right">0.0402309</td>
<td align="right">-0.0575659</td>
<td align="right">0.0204330</td>
<td align="right">-0.0190359</td>
<td align="right">-0.1153630</td>
<td align="right">0.0456979</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">1</td>
<td align="right">24</td>
<td align="right">1</td>
<td align="right">0.0446610</td>
<td align="right">-0.0178283</td>
<td align="right">-0.0541228</td>
<td align="right">-0.1293990</td>
<td align="right">0.0191170</td>
<td align="right">0.0419359</td>
<td align="right">0.0974915</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00179</td>
<td align="right">2</td>
<td align="right">78</td>
<td align="right">1</td>
<td align="right">-0.0079884</td>
<td align="right">0.0495156</td>
<td align="right">-0.0417300</td>
<td align="right">-0.0908318</td>
<td align="right">-0.1013690</td>
<td align="right">-0.0286235</td>
<td align="right">0.1043860</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00180</td>
<td align="right">1</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">-0.0643263</td>
<td align="right">-0.0789866</td>
<td align="right">-0.0961588</td>
<td align="right">-0.0117982</td>
<td align="right">0.0919479</td>
<td align="right">-0.0083767</td>
<td align="right">0.0943266</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00182</td>
<td align="right">1</td>
<td align="right">50</td>
<td align="right">1</td>
<td align="right">0.0477393</td>
<td align="right">0.0258317</td>
<td align="right">0.1084700</td>
<td align="right">-0.0102109</td>
<td align="right">-0.0949523</td>
<td align="right">0.0224777</td>
<td align="right">0.0248974</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00183</td>
<td align="right">1</td>
<td align="right">58</td>
<td align="right">0</td>
<td align="right">0.0181373</td>
<td align="right">-0.0707456</td>
<td align="right">0.0252158</td>
<td align="right">-0.0319165</td>
<td align="right">-0.0351717</td>
<td align="right">-0.0521393</td>
<td align="right">0.0521662</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The best threshold</span></span>
<span><span class="va">BestPvalue</span> <span class="op">&lt;-</span> <span class="va">PRSresult</span><span class="op">$</span><span class="va">BestP</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">BestPvalue</span></span>
<span><span class="co">#&gt; [1] 0.05</span></span></code></pre></div>
<p><strong>(D) Perform regression to test the association of PRS and
sex.</strong></p>
<p><strong>(E) Perform the regression using different thresholds of PRS
to check for consistent evidence of sex difference.</strong></p>
<p>In this regression, we could include genetic PCs and other covariates
as well.</p>
<p><strong>PRS ~ Sex + PCs + Other Covariates.</strong></p>
<p>For this test, we will use <code><a href="../reference/SexRegress.html">SexRegress()</a></code> function.</p>
</div>
<div class="section level2">
<h2 id="about-sexregress">About SexRegress()<a class="anchor" aria-label="anchor" href="#about-sexregress"></a>
</h2>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">SexRegress</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; SexRegress: Performing linear regression analysis with quantitative</span></span>
<span><span class="co">#&gt; response variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Description:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      This function could be used to check association of two variables.</span></span>
<span><span class="co">#&gt;      For instance, PRS with sex.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Usage:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      SexRegress(fdata, regressor_index, response_index)</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt; Arguments:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    fdata: R dataframe object. The column with header 'response' should</span></span>
<span><span class="co">#&gt;           contain the response variable. All other column are the</span></span>
<span><span class="co">#&gt;           regressor.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; regressor_index: Integer value, specifying the column number of the</span></span>
<span><span class="co">#&gt;           main regressor variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; response_index: Integer value, specifying the column number of the</span></span>
<span><span class="co">#&gt;           response variable.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Value:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      Numeric value containing the regression estimate ("Estimate"),</span></span>
<span><span class="co">#&gt;      standard error ("Std. Error"), statistics ("t value") and p-value</span></span>
<span><span class="co">#&gt;      (Pr(&gt;|t|))</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Examples:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;      data("Regression_Ex", package = "GXwasR")</span></span>
<span><span class="co">#&gt;      fdata &lt;- Regression_Ex</span></span>
<span><span class="co">#&gt;      fdata$SEX &lt;- as.factor(as.character(fdata$SEX))</span></span>
<span><span class="co">#&gt;      response_index &lt;- 1</span></span>
<span><span class="co">#&gt;      regressor_index &lt;- 2</span></span>
<span><span class="co">#&gt;      </span></span>
<span><span class="co">#&gt;      x &lt;- SexRegress(fdata, regressor_index, response_index)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-sexregress">Running SexRegress()<a class="anchor" aria-label="anchor" href="#running-sexregress"></a>
</h2>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="co"># First, we need to make fdata object.</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="va">famfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/NewtargetData.fam"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">famfile</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span>, <span class="st">"Sex"</span><span class="op">)</span></span>
<span><span class="va">prefdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfile</span>, <span class="va">PRS</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IID"</span>, <span class="st">"FID"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fdata</span> <span class="op">&lt;-</span> <span class="va">prefdata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">13</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">fdata</span><span class="op">$</span><span class="va">Sex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">fdata</span><span class="op">$</span><span class="va">Sex</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">response_index</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">regressor_index</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexRegress.html">SexRegress</a></span><span class="op">(</span><span class="va">fdata</span>, <span class="va">regressor_index</span>, <span class="va">response_index</span><span class="op">)</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt;     Estimate   Std. Error      t value     Pr(&gt;|t|) </span></span>
<span><span class="co">#&gt; -0.005453123  0.005495108 -0.992359622  0.321927372</span></span></code></pre></div>
<p>In this case, the association between PRS and sex seems
insignificant. Users can check with different thresholds of PRS as
mentioned in step (D).</p>
<div class="section level3">
<h3 id="sex-stratified-prs">Sex-stratified PRS<a class="anchor" aria-label="anchor" href="#sex-stratified-prs"></a>
</h3>
<p>The steps are:</p>
<p><strong>(A) Generate separate male and female discovery data by
stratified GWAS.</strong></p>
<p>We will use SumstatMale and SumstatFemale for these.</p>
<p><strong>(B) Prepare separate male and female target
datasets.</strong></p>
<p>For this, we will utilize <code><a href="../reference/GetMFPlink.html">GetMFPlink()</a></code>.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="co">## We will use NewtargetData as input for this function.</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"NewtargetData"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"maletarget"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"males"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, prefixed as maletarget, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ</span></span>
<span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"femaletarget"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"females"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, prefixed as femaletarget, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ</span></span></code></pre></div>
<p><strong>(C) Perform LD clumping separately in both discovery
data.</strong></p>
<p><strong>(D) Compute PRS at a variety of p-value thresholds, using
male-only discovery data to compute scores for female-only target data
and vice versa.</strong></p>
<p><strong>(E) Select best threshold based on R-square (for quantitative
trait) or MacFadden R-square (for binary trait) using generalized linear
model with covariates, such as genetic PCs.</strong></p>
<p>We will utilize the <code><a href="../reference/ComputePRS.html">ComputePRS()</a></code> to perform the steps
(C), (D) and (E).</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="co"># Male and female target datasets need to be copied from ResultDir to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"target"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="computing-prs-for-female-only-target-data-using-male-only-discovery-data">Computing PRS for female-only target data using male-only discovery
data<a class="anchor" aria-label="anchor" href="#computing-prs-for-female-only-target-data-using-male-only-discovery-data"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># DataDir &lt;- GXwasR:::GXwasR_data()</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"femaletarget"</span></span>
<span><span class="co"># Filtered discovery data.</span></span>
<span><span class="co"># Need to maintain the first three coulmn of this dataset as SNP ID, Effect Allele and Effect Size</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex1</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">## Making phenofile only with females</span></span>
<span><span class="va">famfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/femaletarget.fam"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">famfile</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span>, <span class="st">"Sex"</span><span class="op">)</span></span>
<span><span class="va">famfileF</span> <span class="op">&lt;-</span> <span class="va">famfile</span><span class="op">[</span><span class="va">famfile</span><span class="op">$</span><span class="va">Sex</span> <span class="op">==</span> <span class="fl">2</span>, <span class="op">]</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfileF</span>, <span class="va">Example_phenofile</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span><span class="op">)</span><span class="op">)</span> <span class="co"># Cannot be NULL, the interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="co">## Added "AGE" and "testcovar" as covariates.</span></span>
<span><span class="co">## Making cpvarfile with females</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfileF</span>, <span class="va">Example_covarfile</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">ldclump</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">effectsize</span> <span class="op">&lt;-</span> <span class="st">"BETA"</span></span>
<span><span class="va">PRSresultFemale</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputePRS.html">ComputePRS</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>, effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, <span class="va">clump_p1</span>, <span class="va">clump_p2</span>, <span class="va">clump_r2</span>, <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>, ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>, pheno_type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f', reason 'Directory not empty'</span></span>
<span><span class="co">#&gt; ℹ 0.001</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.001</span></span>
<span><span class="co">#&gt; ℹ 0.05</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.05</span></span>
<span><span class="co">#&gt; ℹ 0.1</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.1</span></span>
<span><span class="co">#&gt; ℹ 0.2</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.2</span></span>
<span><span class="co">#&gt; ℹ 0.3</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.3</span></span>
<span><span class="co">#&gt; ℹ 0.4</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.4</span></span>
<span><span class="co">#&gt; ℹ 0.5</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.5</span></span>
<span><span class="va">PRSresultFemale</span><span class="op">$</span><span class="va">PRS_plot</span></span>
<span><span class="co">#&gt; Error in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, : Metric information not available for this family/device</span></span></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="va">PRSfemale</span> <span class="op">&lt;-</span> <span class="va">PRSresultFemale</span><span class="op">$</span><span class="va">PRS</span></span>
<span><span class="va">PRSfemale</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="6%">
<col width="6%">
<col width="5%">
<col width="3%">
<col width="3%">
<col width="8%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Pheno1</th>
<th align="right">Sex</th>
<th align="right">AGE</th>
<th align="right">testcovar</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">SCORE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0358056</td>
<td align="right">-0.0103993</td>
<td align="right">0.0304770</td>
<td align="right">0.0804331</td>
<td align="right">-0.0316828</td>
<td align="right">0.2308030</td>
<td align="right">-0.975333</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">81</td>
<td align="right">1</td>
<td align="right">-0.0259195</td>
<td align="right">0.0522046</td>
<td align="right">0.0338950</td>
<td align="right">-0.0783466</td>
<td align="right">-0.0301168</td>
<td align="right">0.0481648</td>
<td align="right">-0.327300</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">83</td>
<td align="right">1</td>
<td align="right">0.0690632</td>
<td align="right">0.1000850</td>
<td align="right">0.0177107</td>
<td align="right">-0.0062435</td>
<td align="right">0.2347970</td>
<td align="right">-0.1600570</td>
<td align="right">-0.975333</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">75</td>
<td align="right">0</td>
<td align="right">0.0464837</td>
<td align="right">-0.0611293</td>
<td align="right">-0.0300357</td>
<td align="right">-0.1261310</td>
<td align="right">0.0221893</td>
<td align="right">-0.0430133</td>
<td align="right">0.320733</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">88</td>
<td align="right">1</td>
<td align="right">0.0175615</td>
<td align="right">-0.0407396</td>
<td align="right">-0.1278730</td>
<td align="right">0.1634300</td>
<td align="right">0.1062830</td>
<td align="right">0.0622563</td>
<td align="right">-0.975333</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00178</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">24</td>
<td align="right">1</td>
<td align="right">0.1526140</td>
<td align="right">0.1886470</td>
<td align="right">-0.0341148</td>
<td align="right">-0.0824660</td>
<td align="right">-0.0736962</td>
<td align="right">-0.0412650</td>
<td align="right">0.320733</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00179</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">78</td>
<td align="right">1</td>
<td align="right">0.0881766</td>
<td align="right">0.0494024</td>
<td align="right">0.1288620</td>
<td align="right">0.0523562</td>
<td align="right">-0.0191810</td>
<td align="right">0.0078241</td>
<td align="right">-0.327300</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00180</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">39</td>
<td align="right">1</td>
<td align="right">0.0459778</td>
<td align="right">0.0170708</td>
<td align="right">0.0761808</td>
<td align="right">-0.0344214</td>
<td align="right">-0.0240829</td>
<td align="right">0.0729217</td>
<td align="right">-0.327300</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00266</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">64</td>
<td align="right">1</td>
<td align="right">0.0238295</td>
<td align="right">-0.0127790</td>
<td align="right">-0.1954810</td>
<td align="right">-0.0758269</td>
<td align="right">-0.0017235</td>
<td align="right">0.0638968</td>
<td align="right">0.320733</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00268</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">66</td>
<td align="right">1</td>
<td align="right">-0.1570560</td>
<td align="right">-0.0401398</td>
<td align="right">0.1022290</td>
<td align="right">0.0203453</td>
<td align="right">-0.0103913</td>
<td align="right">0.0914439</td>
<td align="right">-0.327300</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The best threshold</span></span>
<span><span class="va">BestPvalue</span> <span class="op">&lt;-</span> <span class="va">PRSresultFemale</span><span class="op">$</span><span class="va">BestP</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">BestPvalue</span></span>
<span><span class="co">#&gt; [1] 0.001</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="computing-prs-for-male-only-target-data-using-female-only-discovery-data">Computing PRS for male-only target data using female-only discovery
data<a class="anchor" aria-label="anchor" href="#computing-prs-for-male-only-target-data-using-female-only-discovery-data"></a>
</h2>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex2</span><span class="op">)</span></span>
<span><span class="co"># DataDir &lt;- system.file("extdata", package = "GXwasR")</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"maletarget"</span></span>
<span><span class="co"># Filtered discovery data.</span></span>
<span><span class="co"># Need to maintain the first three coulmn of this dataset as SNP ID, Effect Allele and Effect Size</span></span>
<span><span class="va">summarystat</span> <span class="op">&lt;-</span> <span class="va">Summary_Stat_Ex2</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">1</span>, <span class="fl">3</span>, <span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co">## Making phenofile only with females</span></span>
<span><span class="va">famfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/maletarget.fam"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">5</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">famfile</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span>, <span class="st">"Sex"</span><span class="op">)</span></span>
<span><span class="va">famfileM</span> <span class="op">&lt;-</span> <span class="va">famfile</span><span class="op">[</span><span class="va">famfile</span><span class="op">$</span><span class="va">Sex</span> <span class="op">==</span> <span class="fl">1</span>, <span class="op">]</span></span>
<span><span class="va">phenofile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfileM</span>, <span class="va">Example_phenofile</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span><span class="op">)</span><span class="op">)</span> <span class="co"># Cannot be NULL, the interested phenotype column should be labeled as "Pheno1".</span></span>
<span><span class="co">## Added "AGE" and "testcovar" as covariates.</span></span>
<span><span class="co">## Making cpvarfile with females</span></span>
<span><span class="va">covarfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfileM</span>, <span class="va">Example_covarfile</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">clump_p1</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_p2</span> <span class="op">&lt;-</span> <span class="fl">0.0001</span></span>
<span><span class="va">clump_kb</span> <span class="op">&lt;-</span> <span class="fl">500</span></span>
<span><span class="va">clump_r2</span> <span class="op">&lt;-</span> <span class="fl">0.5</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">pthreshold</span> <span class="op">&lt;-</span> <span class="va">Example_pthresoldfile</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">nPC</span> <span class="op">&lt;-</span> <span class="fl">6</span> <span class="co"># We can incorporate PCs into our PRS analysis to account for population stratification.</span></span>
<span><span class="va">pheno_type</span> <span class="op">&lt;-</span> <span class="st">"binary"</span></span>
<span><span class="va">ldclump</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">effectsize</span> <span class="op">&lt;-</span> <span class="st">"BETA"</span></span>
<span><span class="va">PRSresultmale</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ComputePRS.html">ComputePRS</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">summarystat</span>, <span class="va">phenofile</span>, <span class="va">covarfile</span>, effectsize <span class="op">=</span> <span class="st">"BETA"</span>, LDreference <span class="op">=</span> <span class="st">"GXwasR_example"</span>, ldclump <span class="op">=</span> <span class="cn">FALSE</span>, clump_p1 <span class="op">=</span> <span class="va">clump_p1</span>, clump_p2 <span class="op">=</span> <span class="va">clump_p2</span>, clump_r2 <span class="op">=</span> <span class="va">clump_r2</span>, clump_kb <span class="op">=</span> <span class="va">clump_kb</span>, byCHR <span class="op">=</span> <span class="cn">TRUE</span>, pthreshold <span class="op">=</span> <span class="va">pthreshold</span>, highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>, ld_prunning <span class="op">=</span> <span class="cn">TRUE</span>, window_size <span class="op">=</span> <span class="fl">50</span>, step_size <span class="op">=</span> <span class="fl">5</span>, r2_threshold <span class="op">=</span> <span class="fl">0.02</span>, nPC <span class="op">=</span> <span class="fl">6</span>, pheno_type <span class="op">=</span> <span class="st">"binary"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: cannot remove file '/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f', reason 'Directory not empty'</span></span>
<span><span class="co">#&gt; ℹ 0.001</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.001</span></span>
<span><span class="co">#&gt; ℹ 0.05</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.05</span></span>
<span><span class="co">#&gt; ℹ 0.1</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.1</span></span>
<span><span class="co">#&gt; ℹ 0.2</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.2</span></span>
<span><span class="co">#&gt; ℹ 0.3</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.3</span></span>
<span><span class="co">#&gt; ℹ 0.4</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.4</span></span>
<span><span class="co">#&gt; ℹ 0.5</span></span>
<span><span class="co">#&gt; • Computing PRS for threshold 0.5</span></span>
<span><span class="va">PRSresultmale</span><span class="op">$</span><span class="va">PRS_plot</span></span>
<span><span class="co">#&gt; Error in grid.Call.graphics(C_text, as.graphicsAnnot(x$label), x$x, x$y, : Metric information not available for this family/device</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## This table shows 10 samples with phenotype, covariates and a PRS column.</span></span>
<span><span class="va">PRSmale</span> <span class="op">&lt;-</span> <span class="va">PRSresultmale</span><span class="op">$</span><span class="va">PRS</span></span>
<span><span class="va">PRSmale</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="6%">
<col width="6%">
<col width="5%">
<col width="3%">
<col width="3%">
<col width="8%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">Pheno1</th>
<th align="right">Sex</th>
<th align="right">AGE</th>
<th align="right">testcovar</th>
<th align="right">PC1</th>
<th align="right">PC2</th>
<th align="right">PC3</th>
<th align="right">PC4</th>
<th align="right">PC5</th>
<th align="right">PC6</th>
<th align="right">SCORE</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00182</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">50</td>
<td align="right">1</td>
<td align="right">0.0647406</td>
<td align="right">-0.0928744</td>
<td align="right">-0.0671189</td>
<td align="right">0.1612960</td>
<td align="right">0.0577821</td>
<td align="right">-0.0080738</td>
<td align="right">-0.7348330</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00183</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">58</td>
<td align="right">0</td>
<td align="right">-0.1385060</td>
<td align="right">-0.0695656</td>
<td align="right">-0.0741905</td>
<td align="right">-0.0772780</td>
<td align="right">0.0448000</td>
<td align="right">0.0203577</td>
<td align="right">-0.0018333</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00185</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">22</td>
<td align="right">1</td>
<td align="right">-0.0149102</td>
<td align="right">-0.1042650</td>
<td align="right">-0.0490031</td>
<td align="right">0.0909954</td>
<td align="right">-0.1272400</td>
<td align="right">0.0736424</td>
<td align="right">-0.0018333</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00186</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">52</td>
<td align="right">0</td>
<td align="right">0.0936989</td>
<td align="right">-0.0658094</td>
<td align="right">-0.1384690</td>
<td align="right">0.0711780</td>
<td align="right">-0.1307370</td>
<td align="right">-0.2262310</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00187</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">49</td>
<td align="right">1</td>
<td align="right">-0.0495608</td>
<td align="right">-0.0536840</td>
<td align="right">-0.0351695</td>
<td align="right">0.0991132</td>
<td align="right">0.0429508</td>
<td align="right">0.1034500</td>
<td align="right">0.4873330</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00188</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">47</td>
<td align="right">1</td>
<td align="right">-0.0168334</td>
<td align="right">-0.0884520</td>
<td align="right">-0.0100132</td>
<td align="right">0.0212827</td>
<td align="right">0.0980303</td>
<td align="right">0.0024415</td>
<td align="right">-0.4883330</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00189</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">27</td>
<td align="right">1</td>
<td align="right">0.0926677</td>
<td align="right">-0.0486674</td>
<td align="right">-0.0237400</td>
<td align="right">0.0180525</td>
<td align="right">-0.0864605</td>
<td align="right">-0.0121730</td>
<td align="right">-0.7348330</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00190</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">42</td>
<td align="right">0</td>
<td align="right">0.1214390</td>
<td align="right">0.1955690</td>
<td align="right">0.0718894</td>
<td align="right">0.0852533</td>
<td align="right">0.0009986</td>
<td align="right">-0.0935724</td>
<td align="right">-0.7348330</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00267</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">36</td>
<td align="right">1</td>
<td align="right">0.0838363</td>
<td align="right">-0.1059630</td>
<td align="right">-0.0289262</td>
<td align="right">-0.0295922</td>
<td align="right">0.0224057</td>
<td align="right">-0.1815780</td>
<td align="right">-0.7348330</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00271</td>
<td align="right">2</td>
<td align="right">1</td>
<td align="right">51</td>
<td align="right">1</td>
<td align="right">0.0015542</td>
<td align="right">-0.0608128</td>
<td align="right">0.0592212</td>
<td align="right">-0.0144662</td>
<td align="right">-0.0084652</td>
<td align="right">-0.2149010</td>
<td align="right">0.2408330</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The best threshold</span></span>
<span><span class="va">BestPvalue</span> <span class="op">&lt;-</span> <span class="va">PRSresultmale</span><span class="op">$</span><span class="va">BestP</span><span class="op">$</span><span class="va">Threshold</span></span>
<span><span class="va">BestPvalue</span></span>
<span><span class="co">#&gt; [1] 0.001</span></span></code></pre></div>
<p><strong>(F) Perform regression to test the association of PRS for a
trait in one sex and case status in other sex.</strong></p>
<p><strong>PRS ~ trait + PCs + other covariates</strong></p>
<p>Let’s check the association between PRS and trait in female.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="co"># First, we need to make fdata object.</span></span>
<span><span class="va">famfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/femaletarget.fam"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">6</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">famfile</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"FID"</span>, <span class="st">"IID"</span>, <span class="st">"Trait"</span><span class="op">)</span></span>
<span><span class="va">prefdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">famfile</span>, <span class="va">PRS</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IID"</span>, <span class="st">"FID"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fdata</span> <span class="op">&lt;-</span> <span class="va">prefdata</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">13</span>, <span class="fl">3</span><span class="op">:</span><span class="fl">12</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">fdata</span><span class="op">$</span><span class="va">Sex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">fdata</span><span class="op">$</span><span class="va">Trait</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">response_index</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span><span class="va">regressor_index</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/SexRegress.html">SexRegress</a></span><span class="op">(</span><span class="va">fdata</span>, <span class="va">regressor_index</span>, <span class="va">response_index</span><span class="op">)</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt;     Estimate   Std. Error      t value     Pr(&gt;|t|) </span></span>
<span><span class="co">#&gt; 4.942281e-02 6.939184e-03 7.122280e+00 4.988950e-11</span></span>
<span><span class="co"># In this case, there is a significant association between PRS and trait value.</span></span></code></pre></div>
<div class="section level3">
<h3 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h3>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.6</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-08</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.33 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-08 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mgcv                   1.9-3      2025-04-04 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  nlme                   3.1-168    2025-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  polynom                1.4-1      2022-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  seqminer               9.7        2024-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
