<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>GWAS Meta analysis Using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GWAS Meta analysis Using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>GWAS Meta analysis Using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">4 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/meta_analysis.Rmd" class="external-link"><code>vignettes/meta_analysis.Rmd</code></a></small>
      <div class="d-none name"><code>meta_analysis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="meta-analysis-using-gxwasr">Meta analysis using GXwasR<a class="anchor" aria-label="anchor" href="#meta-analysis-using-gxwasr"></a>
</h2>
<p>Here we will present a tutorial showing how to perform meta analysis
utilizing the R package GXwasR.</p>
<p>The example data sets for performing the tutorial can be accessed
from “data” folder of the package.</p>
</div>
<div class="section level2">
<h2 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a>
</h2>
<p>Example datasets are provided with the package and can be accessed by
calling the <code><a href="https://rdrr.io/r/utils/data.html" class="external-link">data()</a></code> function.</p>
<p>Load the example datasets to perform this tutorial.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Two sets of summary GWAS summary statistics in .Rda files will be
required to perform this tutorial.</li>
</ol>
<ul>
<li><strong>Summary_Stat_Ex1.Rda</strong></li>
<li><strong>Summary_Stat_Ex2.Rda</strong></li>
</ul>
<p>Check one of the summary statistics data</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span><span class="op">)</span></span>
<span><span class="co">## Visualize three rows and all the columns</span></span>
<span><span class="va">Summary_Stat_Ex1</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>, <span class="op">]</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="3%">
<col width="4%">
<col width="14%">
<col width="8%">
<col width="3%">
<col width="5%">
<col width="7%">
<col width="9%">
<col width="8%">
<col width="9%">
<col width="7%">
<col width="9%">
<col width="8%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">BP</th>
<th align="left">A1</th>
<th align="left">TEST</th>
<th align="right">NMISS</th>
<th align="right">BETA</th>
<th align="right">SE</th>
<th align="right">L95</th>
<th align="right">U95</th>
<th align="right">STAT</th>
<th align="right">P</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="right">1</td>
<td align="left">rs143773730</td>
<td align="right">73841</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.0789</td>
<td align="right">0.2643</td>
<td align="right">-0.5968</td>
<td align="right">0.439</td>
<td align="right">-0.2986</td>
<td align="right">0.7653</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="right">1</td>
<td align="left">rs147281566</td>
<td align="right">775125</td>
<td align="left">T</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">-0.3959</td>
<td align="right">1.2380</td>
<td align="right">-2.8230</td>
<td align="right">2.031</td>
<td align="right">-0.3197</td>
<td align="right">0.7492</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="right">1</td>
<td align="left">rs35854196</td>
<td align="right">863863</td>
<td align="left">A</td>
<td align="left">ADD</td>
<td align="right">125</td>
<td align="right">1.0500</td>
<td align="right">0.8858</td>
<td align="right">-0.6864</td>
<td align="right">2.786</td>
<td align="right">1.1850</td>
<td align="right">0.2360</td>
</tr>
</tbody>
</table>
<p>Among these 12 columns, some columns are mandatory for this tutorial,
such as: ‘SNP’ (i.e., SNP identifier), ‘BETA’ (i.e., effect-size or
logarithm of odds ratio), ‘SE’ (i.e., standard error of BETA), ‘P’
(i.e., p-values) and ‘NMISS’ (i.e., effective sample size). The other
columns are, ‘CHR’ (i.e., chromosome code), ‘BP’ (i.e., base pair
position), A1 (i.e., disease allele), TEST (i.e., association test
type), L95 (i.e., lower limit of 95 percentile confidence interval), U95
(i.e., upper limit of 95 percentile confidence interval) and STAT (i.e.,
test statistic).</p>
<ol start="2" style="list-style-type: decimal">
<li><p><strong>UniqueLoci.Rda:</strong> .Rda file with a single column
containing SNP names. These could be LD clumped SNPs or any other list
of chosen SNPs for Meta analysis.</p></li>
<li><p><strong>SNPsPlot.Rda:</strong> .Rda file with a single column
containing SNP names for the forest plots.</p></li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some other libraries for this vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-function-for-meta-analysis">The function for meta analysis:<a class="anchor" aria-label="anchor" href="#the-function-for-meta-analysis"></a>
</h2>
<p><strong>MetaGWAS()</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/help.html" class="external-link">help</a></span><span class="op">(</span><span class="va">MetaGWAS</span>, package <span class="op">=</span> <span class="va">GXwasR</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Rendering development documentation for "MetaGWAS"</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="running-metagwas">Running MetaGWAS<a class="anchor" aria-label="anchor" href="#running-metagwas"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex2</span><span class="op">)</span></span>
<span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">SummData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">Summary_Stat_Ex1</span>, <span class="va">Summary_Stat_Ex2</span><span class="op">)</span></span>
<span><span class="va">SNPfile</span> <span class="op">&lt;-</span> <span class="st">"UniqueLoci"</span></span>
<span><span class="va">useSNPposition</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">UseA1</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">GCse</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">byCHR</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">pval_filter</span> <span class="op">&lt;-</span> <span class="st">"R"</span></span>
<span><span class="va">top_snp_pval</span> <span class="op">&lt;-</span> <span class="fl">1e-08</span></span>
<span><span class="va">max_top_snps</span> <span class="op">&lt;-</span> <span class="fl">10</span></span>
<span><span class="va">chosen_snps_file</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">pval_threshold_manplot</span> <span class="op">&lt;-</span> <span class="fl">1e-05</span></span>
<span><span class="va">plotname</span> <span class="op">&lt;-</span> <span class="st">"Meta_Analysis.plot"</span></span>
<span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MetaGWAS.html">MetaGWAS</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, SummData <span class="op">=</span> <span class="va">SummData</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, SNPfile <span class="op">=</span> <span class="cn">NULL</span>, useSNPposition <span class="op">=</span> <span class="cn">TRUE</span>, UseA1 <span class="op">=</span> <span class="va">UseA1</span>, GCse <span class="op">=</span> <span class="va">GCse</span>, plotname <span class="op">=</span> <span class="st">"Meta_Analysis.plot"</span>, <span class="va">pval_filter</span>, <span class="va">top_snp_pval</span>, <span class="va">max_top_snps</span>, chosen_snps_file <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">byCHR</span>, <span class="va">pval_threshold_manplot</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Processing file number 1</span></span>
<span><span class="co">#&gt; ℹ Processing file number 2</span></span>
<span><span class="co">#&gt; ℹ Applying study-specific genomic control.</span></span>
<span><span class="co">#&gt; ℹ Applying study-specific genomic control.</span></span>
<span><span class="co">#&gt; Processing chromosome</span></span></code></pre></div>
<p><img src="unnamed-chunk-5-1.png" alt="plot of chunk unnamed-chunk-5"><img src="unnamed-chunk-5-2.png" alt="plot of chunk unnamed-chunk-5"></p>
<pre><code><span><span class="co">#&gt; ✔ Forest plot files for Meta_Analysis.plot SNPs have been created.</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp0uzDD3</span></span>
<span><span class="co">#&gt; ✔ Forest plot files for Meta_Analysis.plot SNPs have been created.</span></span>
<span><span class="co">#&gt; ℹ You can find them in the directory: /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//Rtmp0uzDD3</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="random-effect-result">Random effect result<a class="anchor" aria-label="anchor" href="#random-effect-result"></a>
</h2>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataframe with the fixed effect result</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Resultrandom</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Top ten associations from random effect model."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Top ten associations from random effect model.</caption>
<colgroup>
<col width="5%">
<col width="4%">
<col width="9%">
<col width="12%">
<col width="3%">
<col width="3%">
<col width="7%">
<col width="6%">
<col width="10%">
<col width="8%">
<col width="10%">
<col width="11%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">Q</th>
<th align="right">I</th>
<th align="right">P</th>
<th align="right">ES</th>
<th align="right">SE</th>
<th align="right">CI_L</th>
<th align="right">CI_U</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1054</td>
<td align="right">23</td>
<td align="right">4184349</td>
<td align="left">rs6529954</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1162</td>
<td align="right">59.48</td>
<td align="right">0.0000002</td>
<td align="right">-1.7980</td>
<td align="right">0.3441459</td>
<td align="right">-2.4725260</td>
<td align="right">-1.1234740</td>
</tr>
<tr class="even">
<td align="left">1053</td>
<td align="right">23</td>
<td align="right">4137114</td>
<td align="left">rs5962098</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1753</td>
<td align="right">45.56</td>
<td align="right">0.0000010</td>
<td align="right">1.1488</td>
<td align="right">0.2351018</td>
<td align="right">0.6880004</td>
<td align="right">1.6095996</td>
</tr>
<tr class="odd">
<td align="left">1043</td>
<td align="right">23</td>
<td align="right">3376304</td>
<td align="left">rs6420571</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.3068</td>
<td align="right">4.26</td>
<td align="right">0.0000015</td>
<td align="right">-1.1667</td>
<td align="right">0.2426946</td>
<td align="right">-1.6423814</td>
<td align="right">-0.6910186</td>
</tr>
<tr class="even">
<td align="left">1055</td>
<td align="right">23</td>
<td align="right">4214861</td>
<td align="left">rs12858640</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.0725</td>
<td align="right">69.00</td>
<td align="right">0.0000062</td>
<td align="right">-1.7572</td>
<td align="right">0.3887229</td>
<td align="right">-2.5190968</td>
<td align="right">-0.9953032</td>
</tr>
<tr class="odd">
<td align="left">559</td>
<td align="right">2</td>
<td align="right">2579014</td>
<td align="left">rs10186455</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.4490</td>
<td align="right">0.00</td>
<td align="right">0.0000476</td>
<td align="right">1.2694</td>
<td align="right">0.3121061</td>
<td align="right">0.6576721</td>
<td align="right">1.8811279</td>
</tr>
<tr class="even">
<td align="left">1052</td>
<td align="right">23</td>
<td align="right">4119808</td>
<td align="left">rs10521557</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.8021</td>
<td align="right">0.00</td>
<td align="right">0.0000631</td>
<td align="right">1.4884</td>
<td align="right">0.3720121</td>
<td align="right">0.7592563</td>
<td align="right">2.2175437</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="right">1</td>
<td align="right">1127860</td>
<td align="left">rs148527527</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.5778</td>
<td align="right">0.00</td>
<td align="right">0.0008237</td>
<td align="right">1.3978</td>
<td align="right">0.4179142</td>
<td align="right">0.5786881</td>
<td align="right">2.2169119</td>
</tr>
<tr class="even">
<td align="left">558</td>
<td align="right">2</td>
<td align="right">2535670</td>
<td align="left">rs13430614</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.6681</td>
<td align="right">0.00</td>
<td align="right">0.0010230</td>
<td align="right">0.6358</td>
<td align="right">0.1935981</td>
<td align="right">0.2563478</td>
<td align="right">1.0152522</td>
</tr>
<tr class="odd">
<td align="left">1382</td>
<td align="right">23</td>
<td align="right">45563626</td>
<td align="left">rs1207312</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">0.5388</td>
<td align="right">0.00</td>
<td align="right">0.0046670</td>
<td align="right">0.6138</td>
<td align="right">0.2169547</td>
<td align="right">0.1885689</td>
<td align="right">1.0390311</td>
</tr>
<tr class="even">
<td align="left">1383</td>
<td align="right">23</td>
<td align="right">45565805</td>
<td align="left">rs1780835</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.5206</td>
<td align="right">0.00</td>
<td align="right">0.0064620</td>
<td align="right">0.5404</td>
<td align="right">0.1984308</td>
<td align="right">0.1514757</td>
<td align="right">0.9293243</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="fixed-effect-result">Fixed effect result<a class="anchor" aria-label="anchor" href="#fixed-effect-result"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataframe with the fixed effect result</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Resultfixed</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Top ten associations from fixed effect model."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Top ten associations from fixed effect model.</caption>
<colgroup>
<col width="5%">
<col width="4%">
<col width="9%">
<col width="12%">
<col width="3%">
<col width="3%">
<col width="7%">
<col width="6%">
<col width="10%">
<col width="8%">
<col width="10%">
<col width="11%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">Q</th>
<th align="right">I</th>
<th align="right">P</th>
<th align="right">ES</th>
<th align="right">SE</th>
<th align="right">CI_L</th>
<th align="right">CI_U</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1054</td>
<td align="right">23</td>
<td align="right">4184349</td>
<td align="left">rs6529954</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1162</td>
<td align="right">59.48</td>
<td align="right">0.0000000</td>
<td align="right">-1.7127</td>
<td align="right">0.2025697</td>
<td align="right">-2.1097365</td>
<td align="right">-1.3156635</td>
</tr>
<tr class="even">
<td align="left">1055</td>
<td align="right">23</td>
<td align="right">4214861</td>
<td align="left">rs12858640</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.0725</td>
<td align="right">69.00</td>
<td align="right">0.0000000</td>
<td align="right">-1.6364</td>
<td align="right">0.1955735</td>
<td align="right">-2.0197240</td>
<td align="right">-1.2530760</td>
</tr>
<tr class="odd">
<td align="left">1053</td>
<td align="right">23</td>
<td align="right">4137114</td>
<td align="left">rs5962098</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1753</td>
<td align="right">45.56</td>
<td align="right">0.0000000</td>
<td align="right">1.1030</td>
<td align="right">0.1618941</td>
<td align="right">0.7856876</td>
<td align="right">1.4203124</td>
</tr>
<tr class="even">
<td align="left">1043</td>
<td align="right">23</td>
<td align="right">3376304</td>
<td align="left">rs6420571</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.3068</td>
<td align="right">4.26</td>
<td align="right">0.0000009</td>
<td align="right">-1.1647</td>
<td align="right">0.2371161</td>
<td align="right">-1.6294476</td>
<td align="right">-0.6999524</td>
</tr>
<tr class="odd">
<td align="left">559</td>
<td align="right">2</td>
<td align="right">2579014</td>
<td align="left">rs10186455</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.4490</td>
<td align="right">0.00</td>
<td align="right">0.0000476</td>
<td align="right">1.2694</td>
<td align="right">0.3121061</td>
<td align="right">0.6576721</td>
<td align="right">1.8811279</td>
</tr>
<tr class="even">
<td align="left">1052</td>
<td align="right">23</td>
<td align="right">4119808</td>
<td align="left">rs10521557</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.8021</td>
<td align="right">0.00</td>
<td align="right">0.0000631</td>
<td align="right">1.4884</td>
<td align="right">0.3720121</td>
<td align="right">0.7592563</td>
<td align="right">2.2175437</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="right">1</td>
<td align="right">1127860</td>
<td align="left">rs148527527</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.5778</td>
<td align="right">0.00</td>
<td align="right">0.0008237</td>
<td align="right">1.3978</td>
<td align="right">0.4179142</td>
<td align="right">0.5786881</td>
<td align="right">2.2169119</td>
</tr>
<tr class="even">
<td align="left">558</td>
<td align="right">2</td>
<td align="right">2535670</td>
<td align="left">rs13430614</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.6681</td>
<td align="right">0.00</td>
<td align="right">0.0010230</td>
<td align="right">0.6358</td>
<td align="right">0.1935981</td>
<td align="right">0.2563478</td>
<td align="right">1.0152522</td>
</tr>
<tr class="odd">
<td align="left">638</td>
<td align="right">2</td>
<td align="right">8181194</td>
<td align="left">rs7370955</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.2403</td>
<td align="right">27.48</td>
<td align="right">0.0024130</td>
<td align="right">0.7752</td>
<td align="right">0.2555007</td>
<td align="right">0.2744187</td>
<td align="right">1.2759813</td>
</tr>
<tr class="even">
<td align="left">1382</td>
<td align="right">23</td>
<td align="right">45563626</td>
<td align="left">rs1207312</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">0.5388</td>
<td align="right">0.00</td>
<td align="right">0.0046670</td>
<td align="right">0.6138</td>
<td align="right">0.2169547</td>
<td align="right">0.1885689</td>
<td align="right">1.0390311</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="weighted-effect-result">Weighted effect result<a class="anchor" aria-label="anchor" href="#weighted-effect-result"></a>
</h2>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataframe with the fixed effect result</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Resultweighted</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">P</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Top ten associations from weighted effect model."</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>Top ten associations from weighted effect model.</caption>
<colgroup>
<col width="5%">
<col width="4%">
<col width="9%">
<col width="12%">
<col width="3%">
<col width="3%">
<col width="7%">
<col width="6%">
<col width="10%">
<col width="7%">
<col width="10%">
<col width="11%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">Q</th>
<th align="right">I</th>
<th align="right">P</th>
<th align="right">ES</th>
<th align="right">SE</th>
<th align="right">CI_L</th>
<th align="right">CI_U</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1054</td>
<td align="right">23</td>
<td align="right">4184349</td>
<td align="left">rs6529954</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1162</td>
<td align="right">59.48</td>
<td align="right">0.0000000</td>
<td align="right">-8.340</td>
<td align="right">0.0000000</td>
<td align="right">-8.3400000</td>
<td align="right">-8.340000</td>
</tr>
<tr class="even">
<td align="left">1055</td>
<td align="right">23</td>
<td align="right">4214861</td>
<td align="left">rs12858640</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.0725</td>
<td align="right">69.00</td>
<td align="right">0.0000000</td>
<td align="right">-8.306</td>
<td align="right">0.0000000</td>
<td align="right">-8.3060000</td>
<td align="right">-8.306000</td>
</tr>
<tr class="odd">
<td align="left">1053</td>
<td align="right">23</td>
<td align="right">4137114</td>
<td align="left">rs5962098</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.1753</td>
<td align="right">45.56</td>
<td align="right">0.0000000</td>
<td align="right">6.741</td>
<td align="right">1.0000820</td>
<td align="right">4.7808392</td>
<td align="right">8.701161</td>
</tr>
<tr class="even">
<td align="left">1043</td>
<td align="right">23</td>
<td align="right">3376304</td>
<td align="left">rs6420571</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.3068</td>
<td align="right">4.26</td>
<td align="right">0.0000010</td>
<td align="right">-4.883</td>
<td align="right">1.0000062</td>
<td align="right">-6.8430122</td>
<td align="right">-2.922988</td>
</tr>
<tr class="odd">
<td align="left">559</td>
<td align="right">2</td>
<td align="right">2579014</td>
<td align="left">rs10186455</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.4490</td>
<td align="right">0.00</td>
<td align="right">0.0000723</td>
<td align="right">3.969</td>
<td align="right">1.0000874</td>
<td align="right">2.0088287</td>
<td align="right">5.929171</td>
</tr>
<tr class="even">
<td align="left">1052</td>
<td align="right">23</td>
<td align="right">4119808</td>
<td align="left">rs10521557</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.8021</td>
<td align="right">0.00</td>
<td align="right">0.0000955</td>
<td align="right">3.902</td>
<td align="right">1.0000837</td>
<td align="right">1.9418359</td>
<td align="right">5.862164</td>
</tr>
<tr class="odd">
<td align="left">6</td>
<td align="right">1</td>
<td align="right">1127860</td>
<td align="left">rs148527527</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">0.5778</td>
<td align="right">0.00</td>
<td align="right">0.0010750</td>
<td align="right">3.270</td>
<td align="right">0.9999618</td>
<td align="right">1.3100749</td>
<td align="right">5.229925</td>
</tr>
<tr class="even">
<td align="left">558</td>
<td align="right">2</td>
<td align="right">2535670</td>
<td align="left">rs13430614</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">0.6681</td>
<td align="right">0.00</td>
<td align="right">0.0014260</td>
<td align="right">3.189</td>
<td align="right">0.9998946</td>
<td align="right">1.2292065</td>
<td align="right">5.148794</td>
</tr>
<tr class="odd">
<td align="left">638</td>
<td align="right">2</td>
<td align="right">8181194</td>
<td align="left">rs7370955</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">0.2403</td>
<td align="right">27.48</td>
<td align="right">0.0032420</td>
<td align="right">2.944</td>
<td align="right">1.0000649</td>
<td align="right">0.9838728</td>
<td align="right">4.904127</td>
</tr>
<tr class="even">
<td align="left">1382</td>
<td align="right">23</td>
<td align="right">45563626</td>
<td align="left">rs1207312</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">0.5388</td>
<td align="right">0.00</td>
<td align="right">0.0048350</td>
<td align="right">2.818</td>
<td align="right">1.0000617</td>
<td align="right">0.8578790</td>
<td align="right">4.778121</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="metadata-of-the-meta-analysis">Metadata of the meta analysis<a class="anchor" aria-label="anchor" href="#metadata-of-the-meta-analysis"></a>
</h2>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataframe with the metadata</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Metadata</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="va">x1</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">x1</span><span class="op">$</span><span class="va">Q</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Metadata of the top ten associations based on Cochrane’s Q statistics."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Metadata of the top ten associations based on Cochrane’s Q
statistics.</caption>
<colgroup>
<col width="7%">
<col width="5%">
<col width="13%">
<col width="16%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="10%">
<col width="8%">
<col width="11%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">CHR</th>
<th align="right">BP</th>
<th align="left">SNP</th>
<th align="left">A1</th>
<th align="left">A2</th>
<th align="right">N</th>
<th align="right">Q</th>
<th align="right">I</th>
<th align="right">F0</th>
<th align="right">F1</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">812</td>
<td align="right">2</td>
<td align="right">21385538</td>
<td align="left">rs575905</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0010</td>
<td align="right">90.73</td>
<td align="right">-0.5218</td>
<td align="right">0.6472</td>
</tr>
<tr class="even">
<td align="left">1110</td>
<td align="right">23</td>
<td align="right">9277424</td>
<td align="left">rs2214279</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0015</td>
<td align="right">90.09</td>
<td align="right">0.5001</td>
<td align="right">-0.5365</td>
</tr>
<tr class="odd">
<td align="left">655</td>
<td align="right">2</td>
<td align="right">9913645</td>
<td align="left">rs1106144</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0020</td>
<td align="right">89.48</td>
<td align="right">0.5909</td>
<td align="right">-0.8972</td>
</tr>
<tr class="even">
<td align="left">663</td>
<td align="right">2</td>
<td align="right">10210922</td>
<td align="left">rs11678624</td>
<td align="left">T</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0021</td>
<td align="right">89.45</td>
<td align="right">-0.3659</td>
<td align="right">0.7374</td>
</tr>
<tr class="odd">
<td align="left">1207</td>
<td align="right">23</td>
<td align="right">23909933</td>
<td align="left">rs2428144</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0031</td>
<td align="right">88.55</td>
<td align="right">0.5344</td>
<td align="right">-0.4081</td>
</tr>
<tr class="even">
<td align="left">1253</td>
<td align="right">23</td>
<td align="right">29189630</td>
<td align="left">rs225456</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0031</td>
<td align="right">88.56</td>
<td align="right">0.3208</td>
<td align="right">-1.1170</td>
</tr>
<tr class="odd">
<td align="left">1248</td>
<td align="right">23</td>
<td align="right">28763567</td>
<td align="left">rs12008039</td>
<td align="left">G</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0032</td>
<td align="right">88.48</td>
<td align="right">-0.3187</td>
<td align="right">0.5565</td>
</tr>
<tr class="even">
<td align="left">810</td>
<td align="right">2</td>
<td align="right">21275825</td>
<td align="left">rs12468735</td>
<td align="left">A</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0036</td>
<td align="right">88.19</td>
<td align="right">-0.5095</td>
<td align="right">0.5175</td>
</tr>
<tr class="odd">
<td align="left">664</td>
<td align="right">2</td>
<td align="right">10343419</td>
<td align="left">rs759347</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0041</td>
<td align="right">87.88</td>
<td align="right">0.9790</td>
<td align="right">-0.3894</td>
</tr>
<tr class="even">
<td align="left">1251</td>
<td align="right">23</td>
<td align="right">29104124</td>
<td align="left">rs16988439</td>
<td align="left">C</td>
<td align="left">?</td>
<td align="right">2</td>
<td align="right">0.0042</td>
<td align="right">87.81</td>
<td align="right">0.3208</td>
<td align="right">-1.0730</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="problematic-snps">Problematic SNPs<a class="anchor" aria-label="anchor" href="#problematic-snps"></a>
</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Dataframe with the problematic SNPs</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">ProblemSNP</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x1</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>, caption <span class="op">=</span> <span class="st">"Top ten problematic SNPs."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Top ten problematic SNPs.</caption>
<colgroup>
<col width="74%">
<col width="10%">
<col width="15%">
</colgroup>
<thead><tr class="header">
<th align="left">File</th>
<th align="left">SNP</th>
<th align="left">Problem</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs2803333</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs672606</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs7519955</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs1538466</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs61777960</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs2865211</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs10917217</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs196402</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="odd">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs6662038</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
<tr class="even">
<td align="left">/private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/SNPdata_2</td>
<td align="left">rs3845484</td>
<td align="left">ALLELE_MISMATCH</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="including-plots">Including Plots<a class="anchor" aria-label="anchor" href="#including-plots"></a>
</h2>
<div class="section level3">
<h3 id="manhattan-plots-and-qq-plots">Manhattan Plots and QQ plots:<a class="anchor" aria-label="anchor" href="#manhattan-plots-and-qq-plots"></a>
</h3>
<div class="figure">
<img src="Meta_Analysis.plot.jpeg" alt="plot of chunk unnamed-chunk-11" width="1500px" height="500px"><p class="caption">
plot of chunk unnamed-chunk-11
</p>
</div>
</div>
<div class="section level3">
<h3 id="all-forest-plots">All Forest plots<a class="anchor" aria-label="anchor" href="#all-forest-plots"></a>
</h3>
<p>Now, let’s see all the Forest plots.</p>
<div class="figure">
<img src="Meta_Analysis.forest.plot.1.jpg" alt="plot of chunk unnamed-chunk-12" width="1000px" height="500px"><p class="caption">
plot of chunk unnamed-chunk-12
</p>
</div>
<div class="figure">
<img src="Meta_Analysis.forest.plot.2.jpg" alt="plot of chunk unnamed-chunk-13" width="1000px" height="500px"><p class="caption">
plot of chunk unnamed-chunk-13
</p>
</div>
<div class="figure">
<img src="Meta_Analysis.forest.plot.3.jpg" alt="plot of chunk unnamed-chunk-14" width="1000px" height="500px"><p class="caption">
plot of chunk unnamed-chunk-14
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-04</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  ! package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;    abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    askpass                1.2.1      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    biocthis               1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    commonmark             2.0.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    credentials            2.0.2      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    desc                   1.4.3      2023-12-10 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;    GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    gert                   2.1.5      2025-03-25 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gh                     1.5.0      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gitcreds               0.1.2      2022-09-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  P GXwasR               * 0.99.0     2025-08-05 [?] Bioconductor</span></span>
<span><span class="co">#&gt;    hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    httr2                  1.2.1      2025-07-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    hunspell               3.0.6      2025-03-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    openssl                2.3.3      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;    pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.cache                0.17.0     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rappdirs               0.3.3      2021-01-31 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rcmdcheck              1.4.0      2021-09-27 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    roxygen2               7.3.2      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rstudioapi             0.17.1     2024-10-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    spelling               2.3.1      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    styler                 1.10.3     2024-04-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    whisker                0.4.1      2022-12-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    xopen                  1.0.1      2024-04-25 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/Rtmp0uzDD3/temp_libpathfc413d99b05c</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt;  P ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
