<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Sex-aware Post-impute Quality Control Pipeline Using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Sex-aware Post-impute Quality Control Pipeline Using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Sex-aware Post-impute Quality Control Pipeline Using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">5 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/main/vignettes/postimputationQC.Rmd" class="external-link"><code>vignettes/postimputationQC.Rmd</code></a></small>
      <div class="d-none name"><code>postimputationQC.Rmd</code></div>
    </div>

    
    
<p>This document outlines the sex-aware post-imputation quality control
(QC) steps, which represent our recommended best practices. Users have
the flexibility to create their own pipelines using the GXwasR functions
based on their specific needs.</p>
<div class="section level2">
<h2 id="sex-aware-post-imputation-qc-pipeline">Sex-aware Post-imputation QC pipeline:<a class="anchor" aria-label="anchor" href="#sex-aware-post-imputation-qc-pipeline"></a>
</h2>
<p>We will run the post-imputation QC pipeline in this vignette
utilizing the functions from GXwasR package.</p>
<div class="section level3">
<h3 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a>
</h3>
<p>The PLINK bed, bim, and fam files are the three mandatory files
representing a genotype dataset to run this pipeline. To know about
these file extensions, please check <a href="https://www.cog-genomics.org/plink/1.9/formats" class="external-link uri">https://www.cog-genomics.org/plink/1.9/formats</a>.</p>
<p><strong>GXwasR_example:</strong></p>
<ul>
<li><p>GXwasR_example.bed</p></li>
<li><p>GXwasR_example.bim</p></li>
<li><p>GXwasR_example.fam</p></li>
</ul>
<p>These plink files contain genotypes for 276 individuals (males and
females) simulated from 1000Genome from European decent with 26515
variants across twelve chromosomes (1-10,23,24). This dataset contains
125 males, 151 females, 108 cases and 168 controls. We will utilize this
set of plink files as proxy of pre-imputated genotype data.</p>
</div>
<div class="section level3">
<h3 id="post-imputation-qc-steps">Post-imputation QC steps<a class="anchor" aria-label="anchor" href="#post-imputation-qc-steps"></a>
</h3>
<p>-<strong>(A) Sex-combined QC:</strong> 1.Remove SNPs with IMPUTE2
info score &lt; 0.6 and certainty &lt; 0.8 This part should be done in
the imputation server with the VCF files.</p>
<p>-<strong>(B) Sex-specific autosomal QC at the variant level:</strong>
1.Filter PAR, XTR, Ampliconic regions from the X-chromosome and combine
with autosomes for autosomal QC. 2.Any SNPs in the controls with a sex
difference in MAF should be carefully examined and flagged for further
examination, since the differences may be due to technical confounding
or sampling biases for the study cohorts 3.Prepare separate male and
female subsets of the genotypes 4.Remove SNPs with MAF &lt; 0.05 from
each sex</p>
<p>-<strong>(C) Sex-specific X-chromosomal QC at the variant
level:</strong> 1.Separate PAR, XTR, Ampliconic regions from the
X-chromosome 2.Prepare separate male and female subsets of the genotypes
3.Remove SNPs with a missingness difference in cases vs controls,
separately in each sex 4.In each sex, filter out SNPs based on 1% MAF,
10% missingness per individual, and 5% missingness per genotype. We
recommend setting the threshold filters to be the same in males and
females for analyses of common variants. 5.Identify the set of SNPs
found in both of the post-QC male and female datasets and create a
combined dataset with only these SNPs 6.Testing for HWE across the X in
females (cases and controls combined), and later removing these regions
from analysis in all samples. If assuming equal allele frequency between
males and females, HWE for chrX can be considered in females only as
described in Khramatsova et. al., 2023. 7.Test for significantly
different MAF (p &lt; 0.05/#SNPs) between sexes in control samples only
for binary traits. ChrX variants with MAF sex differences in controls
may be biologically meaningful, one could consider flagging these
variants rather than removing them (Khramatsova et. al., 2023.)</p>
<p>-<strong>(D) Sex-specific QC at sample level: Filter samples
if:</strong> 1.Remove individual with missing genotype rate &gt; 0.1
2.Remove individual with the absolute value of the heterozygosity F
statistic &gt; 0.20 3.Remove pair of individuals with IBD statistics
pihat &gt; 0.2.</p>
</div>
<div class="section level3">
<h3 id="loading-the-gxwasr-library">Loading the GXwasR library<a class="anchor" aria-label="anchor" href="#loading-the-gxwasr-library"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="learn-about-all-the-gxwasr-functions">Learn about all the GXwasR functions<a class="anchor" aria-label="anchor" href="#learn-about-all-the-gxwasr-functions"></a>
</h3>
<p><strong>Before beginning the pre-imputation quality control (QC), it
is recommended that users familiarize themselves with all the functions
in the GXwasR package. This knowledge will help them understand the
required input arguments, how to execute the functions, and what outputs
to expect.</strong></p>
<p>To explore all available functions, users should first load the
GXwasR library and then use the <code><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes()</a></code>
function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="post-imputation-qc-pipeline">Post-imputation QC pipeline<a class="anchor" aria-label="anchor" href="#post-imputation-qc-pipeline"></a>
</h3>
<p>We will use plink binary files from DataDir, prefixed as
“GXwasR_example” for the input of the post-imputation QC functions and
considered that the SNPs are passed IMPUTE2 info score &lt; 0.6 and
certainty &lt; 0.8.</p>
</div>
</div>
<div class="section level2">
<h2 id="example-dataset-summary">Example Dataset Summary<a class="anchor" aria-label="anchor" href="#example-dataset-summary"></a>
</h2>
<p><strong>Dataset: GXwasR_example</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
<div class="section level3">
<h3 id="step1-filtering-multi-allelic-variants">Step1: Filtering multi-allelic variants<a class="anchor" aria-label="anchor" href="#step1-filtering-multi-allelic-variants"></a>
</h3>
<p>GXwasR cannot work with multi-allelic variants</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterAllele.html">FilterAllele</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ There is no multi-allelic SNP present in the input dataset.</span></span></code></pre></div>
<p>Since there is no multi-allelic SNPs, proceed with the same input
dataset.</p>
</div>
<div class="section level3">
<h3 id="step-2-filtering-ambiguous-snps-indels-and-snps-failed-for-hwe">Step 2: Filtering Ambiguous SNPs, indels and SNPs failed for
HWE<a class="anchor" aria-label="anchor" href="#step-2-filtering-ambiguous-snps-indels-and-snps-failed-for-hwe"></a>
</h3>
<p>For this we will use <code><a href="../reference/QCsnp.html">QCsnp()</a></code> blocking all other
filtering. The function will implicitly perform filtering based on
ambiguous SNPs.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="fl">1e-10</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="fl">1e-06</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 4214 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; ℹ There are no monomorphic SNPs.</span></span>
<span><span class="co">#&gt; ℹ No filter based on differential missingness will be applied.</span></span>
<span><span class="co">#&gt; ✔ Output plink files prefixed as ,PostimputeEX_QC1, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 26527 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 22313 SNPs after filtering.</span></span></code></pre></div>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE TRUE TRUE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-flag-par-xtr-and-ampliconic-regions">Step 3: Flag PAR, XTR and Ampliconic Regions<a class="anchor" aria-label="anchor" href="#step-3-flag-par-xtr-and-ampliconic-regions"></a>
</h3>
<p>Now, we will filter Pseudo-Autosomal Region (PAR), X-transposed
region (XTR) and Ampliconic regions from the X-chromosome and will
combine these regions with autosomes for autosomal QC.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PostimputeEX_QC1</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 22313</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC2_Dummy"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterRegion.html">FilterRegion</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">TRUE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>, filterPAR <span class="op">=</span> <span class="cn">TRUE</span>, filterXTR <span class="op">=</span> <span class="cn">TRUE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">TRUE</span>, regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="cn">NULL</span>, Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ chrX</span></span>
<span><span class="co">#&gt; ✖ There is no PAR region in the input data. Argument filterPAR cannot set to be TRUE.</span></span>
<span><span class="co">#&gt; ℹ Changing filterPAR to FALSE</span></span>
<span><span class="co">#&gt; ✖ There is no XTR region in the input data. Argument filterXTR cannot set to be TRUE.</span></span>
<span><span class="co">#&gt; ℹ Changing filterXTR to FALSE</span></span>
<span><span class="co">#&gt; ℹ Ampliconic SNPs:8</span></span>
<span><span class="co">#&gt; ℹ SNPs are only flagged for the desired region.</span></span>
<span></span>
<span><span class="va">XTR_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">XTR</span></span>
<span><span class="va">Ampli_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Ampliconic</span></span>
<span><span class="va">PAR_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">PAR</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">XTR_SNPs</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/Postimpute_XTR_SNPs.Rda"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">Ampli_SNPs</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/Postimpute_Ampli_SNPs.Rda"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">PAR_SNPs</span>, file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/Postimpute_PAR_SNPs.Rda"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Output</span></span>
<span><span class="co">## SNPs in PAR</span></span>
<span><span class="va">PAR_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">PAR</span> <span class="co">## No PAR SNP</span></span>
<span><span class="co">## SNPs in XTR</span></span>
<span><span class="va">XTR_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">XTR</span> <span class="co">## 42 SNPs</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">XTR_SNPs</span>, caption <span class="op">=</span> <span class="st">"SNPs in XTR."</span><span class="op">)</span></span></code></pre></div>
<p>Table: SNPs in XTR.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## SNPs in Ampliconic region</span></span>
<span><span class="va">Ampliconic_SNPs</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">$</span><span class="va">Ampliconic</span> <span class="co"># 7 SNPs</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">Ampliconic_SNPs</span>, caption <span class="op">=</span> <span class="st">"SNPs in Ampliconic region."</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>SNPs in Ampliconic region.</caption>
<thead><tr class="header">
<th align="right">CHR</th>
<th align="left">SNP</th>
<th align="right">START</th>
<th align="right">END</th>
<th align="left">A1</th>
<th align="left">A2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">23</td>
<td align="left">rs62602496</td>
<td align="right">0</td>
<td align="right">48349540</td>
<td align="left">G</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="left">rs6612314</td>
<td align="right">0</td>
<td align="right">55442087</td>
<td align="left">A</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="right">23</td>
<td align="left">rs151231489</td>
<td align="right">0</td>
<td align="right">55445831</td>
<td align="left">C</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="left">rs73498395</td>
<td align="right">0</td>
<td align="right">71740842</td>
<td align="left">T</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="right">23</td>
<td align="left">rs6527</td>
<td align="right">0</td>
<td align="right">73066891</td>
<td align="left">A</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="left">rs4907822</td>
<td align="right">0</td>
<td align="right">102264099</td>
<td align="left">A</td>
<td align="left">G</td>
</tr>
<tr class="odd">
<td align="right">23</td>
<td align="left">rs142219143</td>
<td align="right">0</td>
<td align="right">102354248</td>
<td align="left">G</td>
<td align="left">A</td>
</tr>
<tr class="even">
<td align="right">23</td>
<td align="left">rs139353379</td>
<td align="right">0</td>
<td align="right">103970756</td>
<td align="left">G</td>
<td align="left">A</td>
</tr>
</tbody>
</table>
<p><strong>Note: We will not perform step 4, since we do not have any
PAR regions. The steps are provided for the users in the case if they
have PAR regions in their dataset.</strong></p>
</div>
<div class="section level3">
<h3 id="step-4-filter-only-par-regions-from-x-chromosome">Step 4: Filter only PAR Regions from X-chromosome<a class="anchor" aria-label="anchor" href="#step-4-filter-only-par-regions-from-x-chromosome"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC2"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterRegion.html">FilterRegion</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">TRUE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>, filterPAR <span class="op">=</span> <span class="cn">TRUE</span>, filterXTR <span class="op">=</span> <span class="cn">FALSE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">FALSE</span>, regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="cn">NULL</span>, Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Now, users need to move the QC-ed files (i.e., “PostimputeEX_QC2” and
“PostimputeEX_QC2_snps_extracted”) to DataDir to be used as input
datasets for the next step of the pipeline.</p>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] FALSE</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-5-preparing-autosomal-and-x-chromosomal-plink-binary-files">Step 5: Preparing autosomal and X-chromosomal plink binary
files<a class="anchor" aria-label="anchor" href="#step-5-preparing-autosomal-and-x-chromosomal-plink-binary-files"></a>
</h3>
<p>Now, we will prepare two sets of plink binary files from
“PostimputeEX_QC1” containing autosomes and X-chromosomes, respectively.
For this purpose, we will again use <code><a href="../reference/FilterRegion.html">FilterRegion()</a></code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC1"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3"</span></span>
<span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterRegion.html">FilterRegion</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">FALSE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>, filterPAR <span class="op">=</span> <span class="cn">FALSE</span>, filterXTR <span class="op">=</span> <span class="cn">FALSE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">FALSE</span>, regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="fl">23</span>, Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>“PostimputeEX_QC3_snps_extracted” contains X-chromosome, while
“PostimputeEX_QC3” contains autosomal SNPs.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Now, copying these files to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>Note: We will not perform step 6, since we do not have any
PAR regions. The steps are provided for the users in the case if they
have PAR regions in their dataset.</strong></p>
</div>
<div class="section level3">
<h3 id="step-6-merging-the-autosomal-regions-with-par-regions">Step 6: Merging the autosomal regions with PAR regions<a class="anchor" aria-label="anchor" href="#step-6-merging-the-autosomal-regions-with-par-regions"></a>
</h3>
<p>We will now merge the autosomal files with PAR regions using
<code><a href="../reference/MergeRegion.html">MergeRegion()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC2_snps_extracted"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_autopar"</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">## Copying the files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC3_autopar"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>With “PostimputeEX_QC3”, we will flag the SNPs in the controls with a
sex difference in MAF, since the differences may be due to technical
confounding or sampling biases for the study cohorts.</p>
</div>
<div class="section level3">
<h3 id="step-7-flag-the-snps-in-the-controls-with-a-sex-difference-in-maf">Step 7: Flag the SNPs in the controls with a sex difference in
MAF<a class="anchor" aria-label="anchor" href="#step-7-flag-the-snps-in-the-controls-with-a-sex-difference-in-maf"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAFdiffSexControl.html">MAFdiffSexControl</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span>, filterSNP <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ No SNP to be flagged or excluded.</span></span></code></pre></div>
<p>Since no SNP is having MAF difference in sexes in controls, we will
proceed to the next steps of the pipeline.</p>
</div>
<div class="section level3">
<h3 id="step-8-prepare-separate-male-and-female-subsets-of-the-genotypes-">Step 8: Prepare separate male and female subsets of the
genotypes.<a class="anchor" aria-label="anchor" href="#step-8-prepare-separate-male-and-female-subsets-of-the-genotypes-"></a>
</h3>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_Female"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"females"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as PostimputeEX_QC4_Female, are in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span>
<span><span class="co">## Making male plink files</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_Male"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"males"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as PostimputeEX_QC4_Male, are in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span></code></pre></div>
<p>Removing the previous QC-ed file from DataDir and copying the new
QC-ed file to DataDir.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Copying new plink files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-9-remove-snps-with-maf-0-05-from-each-sex-from-the-sex-specific-autosomal-plink-files">Step 9: Remove SNPs with MAF &lt; 0.05 from each sex from the
sex-specific autosomal plink files<a class="anchor" aria-label="anchor" href="#step-9-remove-snps-with-maf-0-05-from-each-sex-from-the-sex-specific-autosomal-plink-files"></a>
</h3>
<p>Here, we will perform QC of SNPs using MAF, call rate, monomorphism
and differential missingness in cases vs controls from each sex from the
sex-specific autosomal datasets.</p>
<p>We recommend setting the threshold filters to be the same in males
and females for analyses of common variants.</p>
<p>For this, we will use <code><a href="../reference/QCsnp.html">QCsnp()</a></code> function.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Applying filter to female-specific plink files</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_Female"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Female"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 11 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 5363 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases, </span></span>
<span><span class="co">#&gt; • In controls, </span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ There are no monomorphic SNPs.</span></span>
<span><span class="co">#&gt; ℹ Filtering for differential missingness between cases and controls is turned off.</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output plink files prefixed as ,PostimputeEX_QC5_Female, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 21330 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 15956 SNPs after filtering.</span></span>
<span></span>
<span><span class="co">## Applying filter to male-specific plink files</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_Male"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Male"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 11 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 5309 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases, </span></span>
<span><span class="co">#&gt; • In controls, </span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ There are no monomorphic SNPs.</span></span>
<span><span class="co">#&gt; ℹ Filtering for differential missingness between cases and controls is turned off.</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output plink files prefixed as ,PostimputeEX_QC5_Male, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 21330 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 16010 SNPs after filtering.</span></span></code></pre></div>
<p>Removing the previous QC-ed file from DataDir and copying the new
QC-ed file to DataDir.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-10-combine-sex-specific-autosomal-qc-ed-files-using-common-snps">Step 10: Combine sex-specific autosomal QC-ed files using common
SNPs:<a class="anchor" aria-label="anchor" href="#step-10-combine-sex-specific-autosomal-qc-ed-files-using-common-snps"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Female"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Male"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_Auto"</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as PostimputeEX_Auto</span></span></code></pre></div>
<p>Removing the previous QC-ed file from DataDir and copying the new
QC-ed file to DataDir.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_Auto"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC5"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_Auto"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-11-get-sex-specific-x-chromosomal-plink-files">Step 11: Get sex-specific X-chromosomal plink files<a class="anchor" aria-label="anchor" href="#step-11-get-sex-specific-x-chromosomal-plink-files"></a>
</h3>
<p>For this, we will use plink files prefixed as
“PostimputeEX_QC3_snps_extracted”. As a reminder, these plink files
contain X chromosome.</p>
<p>We will utilize <code><a href="../reference/GetMFPlink.html">GetMFPlink()</a></code> function to get male and
female plink binary files with X chromosomes. We will move these files
from ResultDir to DataDir and will run the sex-specific QC steps.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_snps_extracted"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_XFemale"</span></span>
<span><span class="co">## Making female-specific plink files X chromosome</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"females"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as PostimputeEX_QC3_XFemale, are in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span>
<span><span class="co">## Making male-specific plink files X chromosome</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_XMale"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"males"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as PostimputeEX_QC3_XMale, are in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span>
<span><span class="co">## Copying sex-specific plink files</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC3_X"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-12-qc-of-sex-specific-x-chromosomal-snps">Step 12: QC of sex-specific X-chromosomal SNPs<a class="anchor" aria-label="anchor" href="#step-12-qc-of-sex-specific-x-chromosomal-snps"></a>
</h3>
<p>Here, we will remove SNPs with a missingness difference in cases vs
controls, separately in each sex, and will filter out SNPs based on 1%
MAF, 5% missingness per genotype, and monomorphic SNPs. We recommend
setting the threshold filters to be the same in males and females for
analyses of common variants.</p>
<p>For this, we will use <code><a href="../reference/QCsnp.html">QCsnp()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Female-specific QC</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_XFemale"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_XFemale"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fl">0.05</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">diffmissFilter</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">dmissX</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span> <span class="co"># Since our input has X chromosome, setting this as TRUE wouldn’t change the result.</span></span>
<span><span class="va">dmissAutoY</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span> <span class="co"># Since our input has X chromosome, setting this as TRUE wouldn’t change the result.</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 0 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 110 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases, </span></span>
<span><span class="co">#&gt; • In controls, </span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output plink files prefixed as ,PostimputeEX_QC4_XFemale, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 983 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 873 SNPs after filtering.</span></span>
<span></span>
<span><span class="co">## For male-specific QC:</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC3_XMale"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_XMale"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>, hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 0 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 131 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases, </span></span>
<span><span class="co">#&gt; • In controls, </span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output plink files prefixed as ,PostimputeEX_QC4_XMale, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 983 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 852 SNPs after filtering.</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Copying sex-specific plink files</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4_X"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-13-combine-sex-specific-qc-ed-x-chromosomal-dataset-using-common-snps">Step 13: Combine sex-specific QC-ed X chromosomal dataset using
common SNPs<a class="anchor" aria-label="anchor" href="#step-13-combine-sex-specific-qc-ed-x-chromosomal-dataset-using-common-snps"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_XFemale"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC4_XMale"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Xchr"</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as PostimputeEX_QC5_Xchr</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Copying merged plink files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC5_Xchr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## Removing other QC-ed files from DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-14-test-for-hwe-across-the-x-chromosomes-in-females">Step 14: Test for HWE across the X chromosomes in females<a class="anchor" aria-label="anchor" href="#step-14-test-for-hwe-across-the-x-chromosomes-in-females"></a>
</h3>
<p>Test for HWE across the X in females (cases and controls combined)
and remove these regions from analysis for all samples.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_QC5_Xchr"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_Xchr"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Xhwe.html">Xhwe</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, filterSNP <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as female, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span>
<span><span class="co">#&gt; This test is running on a case-control dataset with female samples.</span></span>
<span><span class="co">#&gt; ℹ Failed SNPs are excluded from the output plink files prefixed as PostimputeEX_Xchr is in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## No. of the SNPs failed the test.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="co"># 2 SNPs in X chr in females failed HWE test.</span></span>
<span><span class="co">#&gt; [1] 3</span></span>
<span><span class="co">## The failed SNPs</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; [1] "rs56053951" "rs12353847" "rs5940058"</span></span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_QC5_Xchr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Copying new QC-ed plink files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX_Xchr"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-15-checking-for-x-chromosomal-snps-having-maf-difference-in-sexes-in-controls">Step 15: Checking for X-chromosomal SNPs having MAF difference in
sexes in controls<a class="anchor" aria-label="anchor" href="#step-15-checking-for-x-chromosomal-snps-having-maf-difference-in-sexes-in-controls"></a>
</h3>
<p>This test should be performed only for binary traits. We will flag
the failed SNPs if any.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_Xchr"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Test_output"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAFdiffSexControl.html">MAFdiffSexControl</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, filterSNP <span class="op">=</span> <span class="cn">FALSE</span>, foutput <span class="op">=</span> <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ No SNP to be flagged or excluded.</span></span>
<span><span class="va">x</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="step-16-combining-autosomal-and-x-chromosomal-qc-ed-datasets">Step 16: Combining autosomal and X-chromosomal QC-ed datasets<a class="anchor" aria-label="anchor" href="#step-16-combining-autosomal-and-x-chromosomal-qc-ed-datasets"></a>
</h2>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_Auto"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"PostimputeEX_Xchr"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeQC"</span></span>
<span><span class="va">use_common_snps</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Merging is done with all the SNPs i.e., union of the SNPs.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as PostimputeQC</span></span></code></pre></div>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Removing the previous QC-ed file from DataDir and copying the new QC-ed file to DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeEX"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"PostimputeQC"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="step-17-preparing-separate-male-and-female-genotype-dataset-for-sample-level-qc">Step 17: Preparing separate male and female genotype dataset for
sample level QC<a class="anchor" aria-label="anchor" href="#step-17-preparing-separate-male-and-female-genotype-dataset-for-sample-level-qc"></a>
</h3>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Making female-specific plink files</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeQC"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Female"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"females"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as Postimpute_Female, are in</span></span>
<span><span class="co">#&gt; /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span>
<span></span>
<span><span class="co">## Making male-specific plink files</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeQC"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Male"</span></span>
<span><span class="va">sex</span> <span class="op">&lt;-</span> <span class="st">"males"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GetMFPlink.html">GetMFPlink</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, sex <span class="op">=</span> <span class="va">sex</span>, xplink <span class="op">=</span> <span class="cn">FALSE</span>, autoplink <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Output plink files, prefixed as Postimpute_Male, are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Copying sex-specific plink files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"Postimpute_Female"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"Postimpute_Male"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-18-sex-specific-sample-level-qc">Step 18: Sex-specific sample level QC<a class="anchor" aria-label="anchor" href="#step-18-sex-specific-sample-level-qc"></a>
</h3>
<p>From each sex-specific plink files, remove individuals with missing
genotype rate &gt; 0.1, with absolute value of the heterozygosity F
statistic &gt; 0.20 and IBD statistics pihat &gt; 0.2.</p>
<p>For this, we will use <code><a href="../reference/QCsample.html">QCsample()</a></code>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running the function female-specific QC</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Female"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Female1"</span></span>
<span><span class="va">imiss</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="co">## Since het = 3 was removing a lot of samples (185), I decided to increase the threshold after looking at the plot.</span></span>
<span><span class="va">het</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">small_sample_mod</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsample.html">QCsample</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, imiss <span class="op">=</span> <span class="va">imiss</span>, het <span class="op">=</span> <span class="va">het</span>, small_sample_mod <span class="op">=</span> <span class="va">small_sample_mod</span>, IBD <span class="op">=</span> <span class="va">IBD</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Plots are initiated.</span></span></code></pre></div>
<div class="float">
<img src="female-missingness-1.png" alt="plot of chunk female-missingness"><div class="figcaption">plot of chunk female-missingness</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for heterozygosity threshold: 2</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness and heterozygosity: 2</span></span>
<span><span class="co">#&gt; ℹ No sample is filtered out for IDB after missingness and heterozygosity filter.</span></span>
<span><span class="co">#&gt; ℹ No. of samples in input plink files: 151</span></span>
<span><span class="co">#&gt; ℹ No. of samples in output plink files: 149</span></span>
<span><span class="co">#&gt; ✔ Output plink files, Postimpute_Female1 with final samples are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez.</span></span></code></pre>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Running the function male-specific QC</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Male"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Male1"</span></span>
<span><span class="va">imiss</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="co">## Since het = 3 was removing a lot of samples (185), I decided to increase the threshold after looking at the plot.</span></span>
<span><span class="va">het</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">small_sample_mod</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">IBD</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsample.html">QCsample</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, small_sample_mod <span class="op">=</span> <span class="va">small_sample_mod</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, imiss <span class="op">=</span> <span class="va">imiss</span>, het <span class="op">=</span> <span class="va">het</span>, IBD <span class="op">=</span> <span class="va">IBD</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Plots are initiated.</span></span></code></pre></div>
<div class="float">
<img src="male-missingness-1.png" alt="plot of chunk male-missingness"><div class="figcaption">plot of chunk male-missingness</div>
</div>
<pre><code><span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for heterozygosity: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered for missingness and heterozygosity: 0</span></span>
<span><span class="co">#&gt; ℹ No sample is filtered out for IDB after missingness and heterozygosity filter.</span></span>
<span><span class="co">#&gt; ℹ No. of samples in input plink files: 125</span></span>
<span><span class="co">#&gt; ℹ No. of samples in output plink files: 125</span></span>
<span><span class="co">#&gt; ✔ Output plink files, Postimpute_Male1 with final samples are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez.</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Copying filtered plink files to DataDir</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"Postimpute_"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span>, <span class="va">DataDir</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-19-combine-male-and-female-specific-qc-ed-genotype-datasets">Step 19: Combine male and female-specific Qc-ed genotype
datasets<a class="anchor" aria-label="anchor" href="#step-19-combine-male-and-female-specific-qc-ed-genotype-datasets"></a>
</h3>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Combine male and female-specific files</span></span>
<span><span class="co">## Running the function</span></span>
<span><span class="va">finput1</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Female1"</span></span>
<span><span class="va">finput2</span> <span class="op">&lt;-</span> <span class="st">"Postimpute_Male1"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeFinal"</span></span>
<span><span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MergeRegion.html">MergeRegion</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput1</span>, <span class="va">finput2</span>, <span class="va">foutput</span>, use_common_snps <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpdyNNez prefixed as PostimputeFinal</span></span></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Removing the not needed files from DataDir.</span></span>
<span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span><span class="op">)</span>, pattern <span class="op">=</span> <span class="st">"Postimpute_"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/invisible.html" class="external-link">invisible</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="st">"/"</span>, <span class="va">ftemp</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="final-summary-of-the-final-qc-ed-genotype-dataset">Final Summary of the final QC-ed genotype dataset<a class="anchor" aria-label="anchor" href="#final-summary-of-the-final-qc-ed-genotype-dataset"></a>
</h2>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PostimputeFinal"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">ResultDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PostimputeFinal</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 149</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 107</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 167</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 54</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 95</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 16208</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 274</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citing-gxwasr">Citing <code>GXwasR</code><a class="anchor" aria-label="anchor" href="#citing-gxwasr"></a>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
will be useful for your research. Please use the following information
to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.5</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-05</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.32 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  ! package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;    abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    commonmark             2.0.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    desc                   1.4.3      2023-12-10 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;    GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  P GXwasR               * 0.99.0     2025-08-05 [?] Bioconductor</span></span>
<span><span class="co">#&gt;    hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    hunspell               3.0.6      2025-03-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    magrittr               2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;    pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rcmdcheck              1.4.0      2021-09-27 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    roxygen2               7.3.2      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rstudioapi             0.17.1     2024-10-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    spelling               2.3.1      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;    sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;    xopen                  1.0.1      2024-04-25 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;    XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;    yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpdyNNez/temp_libpath548171db3e3e</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt;  P ── Loaded and on-disk path mismatch.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
