<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pre-impute Quality Control Pipeline Using GXwasR • GXwasR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="96x96" href="../favicon-96x96.png">
<link rel="icon" type="”image/svg+xml”" href="../favicon.svg">
<link rel="apple-touch-icon" sizes="180x180" href="../apple-touch-icon.png">
<link rel="icon" sizes="any" href="../favicon.ico">
<link rel="manifest" href="../site.webmanifest">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pre-impute Quality Control Pipeline Using GXwasR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">GXwasR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/GXwasR_overview.html">GXwasR Overview</a></li>
    <li><a class="dropdown-item" href="../articles/preimputationQC.html">Pre-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/postimputationQC.html">Sex-aware Post-impute Quality Control Pipeline Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/gwas_models.html">Running GWAS, XWAS and Sex-differential tests Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/GXwasR_PRS.html">Computing Polygenic Risk Score using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/decoding_ancestry.html">Decoding Ancestry: A Guide to Using GXwasR for Genetic Ancestry Estimation</a></li>
    <li><a class="dropdown-item" href="../articles/meta_analysis.html">GWAS Meta analysis Using GXwasR</a></li>
    <li><a class="dropdown-item" href="../articles/genetic_correlation_sumstat.html">Genetic Correlation using GWAS Summary Statistics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/boseb/GXwasR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Pre-impute Quality Control Pipeline Using GXwasR</h1>
                        <h4 data-toc-skip class="author">Banabithi
Bose</h4>
            <address class="author_afil">
      Northwestern UniversityUniversity of Colorado Anschutz Medical
Campus<br><a class="author_email" href="mailto:#"></a><a href="mailto:banabithi.bose@gmail.com" class="email">banabithi.bose@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">8 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/boseb/GXwasR/blob/HEAD/vignettes/preimputationQC.Rmd" class="external-link"><code>vignettes/preimputationQC.Rmd</code></a></small>
      <div class="d-none name"><code>preimputationQC.Rmd</code></div>
    </div>

    
    
<p>This document outlines the pre-imputation quality control (QC) steps,
which represent our recommended best practices. Users have the
flexibility to create their own pipelines using the GXwasR functions
based on their specific needs.</p>
<div class="section level3">
<h3 id="pre-imputation-qc-pipeline">Pre-imputation QC pipeline:<a class="anchor" aria-label="anchor" href="#pre-imputation-qc-pipeline"></a>
</h3>
<p>We will run the pre-imputation QC pipeline in this vignette utilizing
the functions from GXwasR package.</p>
</div>
<div class="section level3">
<h3 id="example-datasets">Example Datasets<a class="anchor" aria-label="anchor" href="#example-datasets"></a>
</h3>
<p>The PLINK bed, bim, and fam files are the three mandatory files
representing a genotype dataset to run this pipeline. To know about
these file extensions, please check <a href="https://www.cog-genomics.org/plink/1.9/formats" class="external-link uri">https://www.cog-genomics.org/plink/1.9/formats</a>.</p>
<p><strong>GXwasR_example:</strong></p>
<ul>
<li>GXwasR_example.bed</li>
<li>GXwasR_example.bim</li>
<li>GXwasR_example.fam</li>
</ul>
<p>These plink files contain genotypes for 276 individuals (males and
females) simulated from 1000Genome from European decent with 26515
variants across twelve chromosomes (1-10,23,24). This dataset contains
125 males, 151 females, 108 cases and 168 controls. We will utilize this
set of plink files as proxy of pre-imputated genotype data.</p>
</div>
<div class="section level3">
<h3 id="pre-imputation-qc-steps">Pre-imputation QC steps<a class="anchor" aria-label="anchor" href="#pre-imputation-qc-steps"></a>
</h3>
<ul>
<li><p><strong>Filtering Multi-Allelic Variants:</strong> This step is
essential for simplifying the genetic data analysis by removing variants
that have more than two alleles. Multi-allelic variants can complicate
analyses and interpretations, making this filtering crucial for a
streamlined dataset.</p></li>
<li><p><strong>Removal of Ambiguous SNPs and Indels:</strong> We
specifically target SNPs with ambiguous mappings (like AT and GC
pairings) and indels (insertions and deletions), which can lead to
inaccuracies in genotype calling. Removing these elements helps in
maintaining the integrity of our genetic data.</p></li>
<li><p><strong>QC of SNPs with Relaxed Genotype Call Rate:</strong> In
this phase, we filter SNPs based on a less stringent genotype call rate.
This step aims to balance between retaining enough informative SNPs and
excluding those with too much missing data.</p></li>
<li><p><strong>Filtering Samples with Relaxed Missing Rate
Threshold:</strong> Similar to SNPs, we apply a relaxed threshold for
sample missingness. This approach ensures that we don’t prematurely
exclude samples that might be informative despite having a slightly
higher rate of missing data.</p></li>
<li><p><strong>Application of Sex Check:</strong> We verify the
consistency between reported and genetic sex. This check helps identify
potential sample mix-ups or issues with data integrity.</p></li>
<li><p><strong>Filter for Needed Chromosomes:</strong> Focus is narrowed
to autosomes and sex chromosomes, as these are typically the most
relevant for genetic studies. This step streamlines the dataset to the
chromosomes of interest.</p></li>
<li><p><strong>Second Round of SNP Filtering:</strong> This more
stringent phase includes filters for minor allele frequency (MAF) &lt;
0.01, call rate &lt; 0.02, case-control differential missingness,
Hardy-Weinberg equilibrium (cases &lt; e-10; controls &lt; e-06), and
the removal of monomorphic SNPs. This rigorous filtering ensures that
only the most reliable SNPs are retained for analysis.</p></li>
<li><p><strong>Second Round of Sample Filtering:</strong> Here, we apply
stringent thresholds for call rate (&lt; 0.02), heterozygosity, and
relatedness (IBD F-stat &lt; 0.2). This step further refines the sample
pool, ensuring that only high-quality samples are included.</p></li>
<li><p><strong>Applying Ancestry Check:</strong> We assess the ancestry
of the samples to identify and exclude outliers from different
ancestries, which is vital for controlling population stratification in
genetic association studies.</p></li>
<li><p><strong>Third Round of SNP Filtering:</strong> This final
filtering step ensures that all previously applied SNP thresholds are
maintained even after the rigorous sample filtering. It’s a crucial step
to guarantee the overall quality and consistency of the SNP data
set.</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="loading-the-gxwasr-library">Loading the GXwasR library<a class="anchor" aria-label="anchor" href="#loading-the-gxwasr-library"></a>
</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Call some libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yihui.org/printr/" class="external-link">printr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rstudio/rmarkdown" class="external-link">rmarkdown</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="learn-about-all-the-gxwasr-functions">Learn about all the GXwasR functions<a class="anchor" aria-label="anchor" href="#learn-about-all-the-gxwasr-functions"></a>
</h3>
<p><strong>Before beginning the pre-imputation quality control (QC), it
is recommended that users familiarize themselves with all the functions
in the GXwasR package. This knowledge will help them understand the
required input arguments, how to execute the functions, and what outputs
to expect.</strong></p>
<p>To explore all available functions, users should first load the
GXwasR library and then use the browseVignettes() function:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example-dataset-summary">Example Dataset Summary<a class="anchor" aria-label="anchor" href="#example-dataset-summary"></a>
</h3>
<p><strong>Dataset: GXwasR_example</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">DataDir</span> <span class="op">&lt;-</span> <span class="fu">GXwasR</span><span class="fu">:::</span><span class="fu"><a href="../reference/GXwasR_data.html">GXwasR_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ResultDir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"GXwasR_example"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: GXwasR_example</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 12</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt;   - Chr: 24</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 26527</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-1-filtering-multi-allelic-variants">Step 1: Filtering multi-allelic variants<a class="anchor" aria-label="anchor" href="#step-1-filtering-multi-allelic-variants"></a>
</h3>
<p>Firstly, users need to ensure that their input plink files are
bi-allelic. To ensure that, it is advised to run the FilterAllele()
function with the input dataset.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC1"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterAllele.html">FilterAllele</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, <span class="va">finput</span>, <span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ There is no multi-allelic SNP present in the input dataset.</span></span></code></pre></div>
<p>Since there was no multi-allelic SNPs, we will proceed with the same
input data.</p>
</div>
<div class="section level3">
<h3 id="step-2-qc-of-snps-for-genotype-call-rate">Step 2: QC of SNPs for genotype call rate<a class="anchor" aria-label="anchor" href="#step-2-qc-of-snps-for-genotype-call-rate"></a>
</h3>
<p>It removes variants that are missing genotype data for more than 20%
of the individuals in your dataset. This function implicitly remove
ambiguous SNPs (i.e., AT&lt;&gt;GC), indels etc. Users can set their own
threshold.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC1"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fl">0.2</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">diffmissFilter</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">dmissX</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">dmissAutoY</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>,hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 4214 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; ℹ 11 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; ℹ No filter based on differential missingness will be applied.</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files prefixed as ,PreimputeEX_QC1, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 26527 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 22302 SNPs after filtering.</span></span></code></pre></div>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span><span class="op">)</span>,pattern <span class="op">=</span> <span class="st">"PreimputeEX_QC1"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span>,<span class="va">ftemp</span><span class="op">)</span>,<span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE TRUE TRUE</span></span>
<span><span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="st">"PreimputeEX_QC1"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PreimputeEX_QC1</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 22302</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-3-filtering-samples-with-high-missing-rate-threshold-">Step 3: Filtering samples with high missing rate threshold.<a class="anchor" aria-label="anchor" href="#step-3-filtering-samples-with-high-missing-rate-threshold-"></a>
</h3>
<p>It excludes individuals who have missing genotype data for more than
20% of the genetic variants in the dataset.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Running</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC1"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC2"</span></span>
<span><span class="va">imiss</span> <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="va">het</span> <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="va">IBD</span> <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="va">filterSample</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ambi_out</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/QCsample.html">QCsample</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>,ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,foutput <span class="op">=</span> <span class="va">foutput</span>, imiss <span class="op">=</span> <span class="va">imiss</span>,het <span class="op">=</span> <span class="va">het</span>, IBD <span class="op">=</span> <span class="cn">NULL</span>, filterSample <span class="op">=</span> <span class="va">filterSample</span>, ambi_out <span class="op">=</span> <span class="va">ambi_out</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Missingness and heterogygosity thresholds are NULL.</span></span>
<span><span class="co">#&gt; ℹ There will be no sample to be filtered for IBD with 'IBD' threshold.</span></span>
<span><span class="co">#&gt; Warning: cannot open file '/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ/PreimputeEX_QC2.fam': No such file or directory</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-4-apply-sex-check">Step 4: Apply sex check<a class="anchor" aria-label="anchor" href="#step-4-apply-sex-check"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC1"</span> <span class="co"># Using the same input file</span></span>
<span><span class="va">LD</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">LD_window_size</span> <span class="op">&lt;-</span> <span class="fl">50</span></span>
<span><span class="va">LD_step_size</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">LD_r2_threshold</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">fmax_F</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span><span class="va">mmin_F</span> <span class="op">&lt;-</span> <span class="fl">0.7</span></span>
<span><span class="co"># We will not impute sex</span></span>
<span><span class="va">impute_sex</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">compute_freq</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">SexCheckResult</span> <span class="op">&lt;-</span><span class="fu"><a href="../reference/SexCheck.html">SexCheck</a></span><span class="op">(</span>DataDir<span class="op">=</span><span class="va">DataDir</span>,ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput<span class="op">=</span><span class="va">finput</span>,impute_sex<span class="op">=</span><span class="va">impute_sex</span>,compute_freq <span class="op">=</span><span class="va">compute_freq</span>,LD_window_size<span class="op">=</span><span class="va">LD_window_size</span>,LD_step_size<span class="op">=</span><span class="va">LD_step_size</span>,LD_r2_threshold<span class="op">=</span><span class="fl">0.02</span>,fmax_F <span class="op">=</span> <span class="va">fmax_F</span>,mmin_F<span class="op">=</span><span class="va">mmin_F</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ There are no Y chromosomes in the input PLINK files. Estimates will be based solely on the X chromosome.</span></span>
<span><span class="va">problematic_sex</span> <span class="op">&lt;-</span> <span class="va">SexCheckResult</span><span class="op">[</span><span class="va">SexCheckResult</span><span class="op">$</span><span class="va">STATUS</span> <span class="op">!=</span> <span class="st">"OK"</span>,<span class="op">]</span></span>
<span><span class="va">problematic_sex</span> <span class="op">&lt;-</span> <span class="va">problematic_sex</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></code></pre></div>
<p>Number of samples with problematic sex assignment: 4</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="va">problematic_sex</span>,file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>,<span class="st">"/problematic_sex_samples"</span><span class="op">)</span>,quote <span class="op">=</span> <span class="cn">F</span>, row.names <span class="op">=</span> <span class="cn">F</span>,col.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span><span class="va">SexCheckResult</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="right">PEDSEX</th>
<th align="right">SNPSEX</th>
<th align="left">STATUS</th>
<th align="right">F</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00171</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">OK</td>
<td align="right">0.081780</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00173</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">OK</td>
<td align="right">-0.031060</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00174</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">OK</td>
<td align="right">0.058430</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00176</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">OK</td>
<td align="right">-0.023280</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00177</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="left">OK</td>
<td align="right">-0.003822</td>
</tr>
</tbody>
</table>
<p>Note, there is no filtering of the samples based on sex
prediction.</p>
<p>This function uses X chromosome data to determine sex (i.e. based on
heterozygosity rates) and flags individuals for whom the reported sex in
the PED file does not match the estimated sex (given genomic data).</p>
<p>A PROBLEM arises if the two sexes do not match, or if the SNP data or
pedigree data are ambiguous with regard to sex. A male call is made if F
is more than 0.8; a female call is made if F is less than 0.2.</p>
<p>SNPSEX Sex as determined by X chromosome</p>
<p>F The actual X chromosome inbreeding (homozygosity) estimate</p>
<p>Users have the ability to impute the sex for their dataset.</p>
</div>
<div class="section level3">
<h3 id="step-5-filter-for-needed-chromosomes-">Step 5: Filter for needed chromosomes.<a class="anchor" aria-label="anchor" href="#step-5-filter-for-needed-chromosomes-"></a>
</h3>
<p>We prefer to keep only chromosome 1 to 23.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">test1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">DataDir</span>,<span class="st">"/"</span>,<span class="va">finput</span>,<span class="st">".bim"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">test1</span><span class="op">$</span><span class="va">V1</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10 23</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC2"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/FilterRegion.html">FilterRegion</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>, foutput <span class="op">=</span> <span class="va">foutput</span>, CHRX <span class="op">=</span> <span class="cn">FALSE</span>, CHRY <span class="op">=</span> <span class="cn">FALSE</span>, filterPAR <span class="op">=</span> <span class="cn">FALSE</span>, filterXTR <span class="op">=</span> <span class="cn">FALSE</span>, filterAmpliconic <span class="op">=</span> <span class="cn">FALSE</span>, regionfile <span class="op">=</span> <span class="cn">FALSE</span>, filterCHR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">24</span>, <span class="fl">25</span>, <span class="fl">26</span><span class="op">)</span>,Hg <span class="op">=</span> <span class="st">"38"</span>, exclude <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span><span class="op">)</span>,pattern <span class="op">=</span> <span class="st">"PreimputeEX_QC2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span>,<span class="va">ftemp</span><span class="op">)</span>,<span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE TRUE TRUE TRUE</span></span>
<span><span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="st">"PreimputeEX_QC2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PreimputeEX_QC2</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 22302</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-6-second-round-of-snp-filtering-">Step 6: Second round of SNP filtering.<a class="anchor" aria-label="anchor" href="#step-6-second-round-of-snp-filtering-"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC2"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC3"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">diffmissFilter</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">dmissX</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">dmissAutoY</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span><span class="va">hwe</span> <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="fl">1e-10</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="fl">1e-06</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>,hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 0 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 1930 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases,  1 variant removed due to Hardy-Weinberg exact test.</span></span>
<span><span class="co">#&gt; • In controls,  19 variants removed due to Hardy-Weinberg exact test.</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ There are no monomorphic SNPs.</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files prefixed as ,PreimputeEX_QC3, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 22302 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 20353 SNPs after filtering.</span></span></code></pre></div>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span><span class="op">)</span>,pattern <span class="op">=</span> <span class="st">"PreimputeEX_QC3"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span>,<span class="va">ftemp</span><span class="op">)</span>,<span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE TRUE TRUE</span></span>
<span><span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="st">"PreimputeEX_QC3"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PreimputeEX_QC3</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 125</span></span>
<span><span class="co">#&gt; ℹ Number of females: 151</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 108</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 53</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 55</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 20353</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 276</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-7-second-round-of-sample-filtering">Step 7: Second round of sample filtering<a class="anchor" aria-label="anchor" href="#step-7-second-round-of-sample-filtering"></a>
</h3>
<p>Here we are using stringent thresholds for sample filter.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC3"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC4"</span></span>
<span><span class="va">imiss</span> <span class="op">=</span> <span class="fl">0.02</span></span>
<span><span class="va">het</span> <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="va">IBD</span> <span class="op">=</span> <span class="fl">0.2</span></span>
<span><span class="va">IBDmatrix</span> <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="va">small_sample_mod</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">filterSample</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">ambi_out</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="../reference/QCsample.html">QCsample</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>,ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,foutput <span class="op">=</span> <span class="va">foutput</span>, imiss <span class="op">=</span> <span class="va">imiss</span>,het <span class="op">=</span> <span class="va">het</span>, IBD <span class="op">=</span> <span class="va">IBD</span>, IBDmatrix <span class="op">=</span> <span class="va">IBDmatrix</span>, filterSample <span class="op">=</span> <span class="va">filterSample</span>, ambi_out <span class="op">=</span> <span class="va">ambi_out</span><span class="op">)</span></span>
<span><span class="co">#&gt; • Plots are initiated.</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for missingness: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered/flagged for heterozygosity: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples filtered for missingness and heterozygosity: 0</span></span>
<span><span class="co">#&gt; ℹ No. of samples marked to be filtered out for IDB after missingness and heterozygosity filter: 2</span></span>
<span><span class="co">#&gt; ℹ No. of samples in input PLINK files: 276</span></span>
<span><span class="co">#&gt; ℹ No. of samples in output PLINK files: 274</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files, PreimputeEX_QC4 with final samples are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ.</span></span></code></pre></div>
<p>Copying the plink files from ResultDir to DataDir.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ftemp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html" class="external-link">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span><span class="op">)</span>,pattern <span class="op">=</span> <span class="st">"PreimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="st">"/"</span>,<span class="va">ftemp</span><span class="op">)</span>,<span class="va">DataDir</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE TRUE TRUE TRUE</span></span>
<span><span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">DataDir</span>, <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="st">"PreimputeEX_QC4"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: PreimputeEX_QC4</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 124</span></span>
<span><span class="co">#&gt; ℹ Number of females: 150</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 106</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 52</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 54</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 20353</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 274</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="step-8-ancestry-check">Step 8: Ancestry check<a class="anchor" aria-label="anchor" href="#step-8-ancestry-check"></a>
</h3>
<p>Prior to use this function, users are highly encouraged to read our
tutorial ““Decoding Ancestry: A Guide to Using GXwasR for Genetic
Ancestry Estimation”.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"highLD_hg19"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"example_data_study_sample_ancestry"</span>, package <span class="op">=</span> <span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC4"</span></span>
<span><span class="va">reference</span> <span class="op">&lt;-</span> <span class="st">"HapMapIII_NCBI36"</span></span>
<span><span class="va">highLD_regions</span> <span class="op">&lt;-</span> <span class="va">highLD_hg19</span></span>
<span><span class="va">study_pop</span> <span class="op">&lt;-</span> <span class="va">example_data_study_sample_ancestry</span></span>
<span><span class="va">studyLD_window_size</span> <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="va">studyLD_step_size</span> <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="va">studyLD_r2_threshold</span> <span class="op">=</span> <span class="fl">0.02</span></span>
<span><span class="va">filterSNP</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">studyLD</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">referLD</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">referLD_window_size</span> <span class="op">=</span> <span class="fl">50</span></span>
<span><span class="va">referLD_step_size</span> <span class="op">=</span> <span class="fl">5</span></span>
<span><span class="va">referLD_r2_threshold</span> <span class="op">=</span> <span class="fl">0.02</span></span>
<span><span class="va">outlier</span> <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="va">outlier_threshold</span> <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="va">outlierOf</span> <span class="op">=</span> <span class="st">"EUR"</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/AncestryCheck.html">AncestryCheck</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">DataDir</span>,</span>
<span>    ResultDir <span class="op">=</span> <span class="va">ResultDir</span>,</span>
<span>    finput <span class="op">=</span> <span class="va">finput</span>,</span>
<span>    reference <span class="op">=</span> <span class="va">reference</span>,</span>
<span>    filterSNP <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    studyLD <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    studyLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    studyLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    studyLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    referLD <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    referLD_window_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>    referLD_step_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>    referLD_r2_threshold <span class="op">=</span> <span class="fl">0.02</span>,</span>
<span>    highLD_regions <span class="op">=</span> <span class="va">highLD_regions</span>,</span>
<span>    study_pop <span class="op">=</span> <span class="va">study_pop</span>,</span>
<span>    outlier <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    outlierOf <span class="op">=</span> <span class="st">"EUR"</span>,</span>
<span>    outlier_threshold <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">#&gt; No A-T or G-C SNPs found in study data.</span></span>
<span><span class="co">#&gt; 111854 ambiguous SNPs removed from reference data.</span></span>
<span><span class="co">#&gt; ! LD pruning is recommended for reference dataset. Set referLD = TRUE.</span></span>
<span><span class="co">#&gt; ✔ LD pruning was performed for study dataset.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ℹ Number of overlapping SNPs between study and reference data using rsID: 1032</span></span>
<span><span class="co">#&gt; ✔ PCA done.</span></span>
<span><span class="co">#&gt; ℹ No allele flips between study and reference data.</span></span>
<span><span class="co">#&gt; ℹ 7 samples are outliers of selected reference population.</span></span>
<span><span class="co">#&gt; ℹ 179 samples are NOT outliers of selected reference population.</span></span>
<span></span>
<span><span class="va">Samples_with_predicted_ancestry</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[[</span><span class="st">"Samples_with_predicted_ancestry"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">outlier</span> <span class="op">&lt;-</span> <span class="va">x</span><span class="op">[[</span><span class="st">"Outlier_samples"</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="va">outlier</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00280</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00315</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00321</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00350</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00369</td>
</tr>
<tr class="even">
<td align="left">EUR_FIN</td>
<td align="left">HG00377</td>
</tr>
<tr class="odd">
<td align="left">EUR_FIN</td>
<td align="left">HG00383</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Samples_with_predicted_ancestry</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="op">]</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">FID</th>
<th align="left">IID</th>
<th align="left">Predicted Ancestry</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00096</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">EUR_GBR</td>
<td align="left">HG00097</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00099</td>
<td align="left">EUR</td>
</tr>
<tr class="even">
<td align="left">EUR_GBR</td>
<td align="left">HG00100</td>
<td align="left">EUR</td>
</tr>
<tr class="odd">
<td align="left">EUR_GBR</td>
<td align="left">HG00101</td>
<td align="left">EUR</td>
</tr>
</tbody>
</table>
<p>There was no outlier samples.</p>
</div>
<div class="section level3">
<h3 id="step-9-third-round-of-snp-filtering">Step 9: Third round of SNP filtering<a class="anchor" aria-label="anchor" href="#step-9-third-round-of-snp-filtering"></a>
</h3>
<p>To make sure all the filtering thresholds for SNPs are maintained
after sample filtering.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/boseb/GXwasR/" class="external-link">GXwasR</a></span><span class="op">)</span></span>
<span><span class="va">finput</span> <span class="op">&lt;-</span> <span class="st">"PreimputeEX_QC4"</span></span>
<span><span class="va">foutput</span> <span class="op">&lt;-</span> <span class="st">"Preimpute_Final"</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fl">0.02</span></span>
<span><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fl">0.01</span></span>
<span><span class="va">casecontrol</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">caldiffmiss</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">diffmissFilter</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">dmissX</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">dmissAutoY</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">monomorphicSNPs</span> <span class="op">&lt;-</span> <span class="cn">TRUE</span></span>
<span><span class="va">ld_prunning</span> <span class="op">&lt;-</span> <span class="cn">FALSE</span></span>
<span> <span class="va">hwe</span> <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="va">hweCase</span> <span class="op">&lt;-</span> <span class="fl">1e-10</span></span>
<span><span class="va">hweControl</span> <span class="op">&lt;-</span> <span class="fl">1e-06</span></span>
<span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/QCsnp.html">QCsnp</a></span><span class="op">(</span>DataDir <span class="op">=</span> <span class="va">ResultDir</span>, ResultDir <span class="op">=</span> <span class="va">ResultDir</span>, finput <span class="op">=</span> <span class="va">finput</span>,foutput <span class="op">=</span> <span class="va">foutput</span>, geno <span class="op">=</span> <span class="va">geno</span>, maf <span class="op">=</span> <span class="va">maf</span>,hweCase <span class="op">=</span> <span class="va">hweCase</span>, hweControl <span class="op">=</span> <span class="va">hweControl</span>, ld_prunning <span class="op">=</span> <span class="va">ld_prunning</span>, casecontrol <span class="op">=</span> <span class="va">casecontrol</span>, monomorphicSNPs <span class="op">=</span> <span class="va">monomorphicSNPs</span>, caldiffmiss <span class="op">=</span> <span class="va">caldiffmiss</span>, dmissX <span class="op">=</span> <span class="va">dmissX</span>, dmissAutoY <span class="op">=</span> <span class="va">dmissAutoY</span>, diffmissFilter <span class="op">=</span> <span class="va">diffmissFilter</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ 0 Ambiguous SNPs (A-T/G-C), indels etc. were removed.</span></span>
<span><span class="co">#&gt; ✔ Thresholds for maf, geno and hwe worked.</span></span>
<span><span class="co">#&gt; 0 variants removed due to missing genotype data (--geno).</span></span>
<span><span class="co">#&gt; 3 variants removed due to minor allele threshold(s)</span></span>
<span><span class="co">#&gt; • In cases,  0 variants removed due to Hardy-Weinberg exact test.</span></span>
<span><span class="co">#&gt; • In controls,  0 variants removed due to Hardy-Weinberg exact test.</span></span>
<span><span class="co">#&gt; ✔ Merging is done using the common SNPs between the input genotype files.</span></span>
<span><span class="co">#&gt; ✔ Plink files with merged regions are in /var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T//RtmpcRh9ZZ prefixed as filtered_temp2</span></span>
<span><span class="co">#&gt; ℹ There are no monomorphic SNPs.</span></span>
<span><span class="co">#&gt; ℹ No SNP with differential missingness between cases and controls.</span></span>
<span><span class="co">#&gt; ✔ Output PLINK files prefixed as ,Preimpute_Final, with passed SNPs are saved in ResultDir.</span></span>
<span><span class="co">#&gt; ℹ Input file has 20353 SNPs.</span></span>
<span><span class="co">#&gt; ℹ Output file has 20350 SNPs after filtering.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="final-summary-of-final-qc-ed-dataset-">Final Summary of final Qc-ed dataset.<a class="anchor" aria-label="anchor" href="#final-summary-of-final-qc-ed-dataset-"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/PlinkSummary.html">PlinkSummary</a></span><span class="op">(</span><span class="va">ResultDir</span>,<span class="va">ResultDir</span>,<span class="va">foutput</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Dataset: Preimpute_Final</span></span>
<span><span class="co">#&gt; ℹ Number of missing phenotypes: 0</span></span>
<span><span class="co">#&gt; ℹ Number of males: 124</span></span>
<span><span class="co">#&gt; ℹ Number of females: 150</span></span>
<span><span class="co">#&gt; ℹ This is case-control data</span></span>
<span><span class="co">#&gt; ℹ Number of cases: 106</span></span>
<span><span class="co">#&gt; ℹ Number of controls: 168</span></span>
<span><span class="co">#&gt; ℹ Number of cases in males: 52</span></span>
<span><span class="co">#&gt; ℹ Number of controls in males: 72</span></span>
<span><span class="co">#&gt; ℹ Number of cases in females: 54</span></span>
<span><span class="co">#&gt; ℹ Number of controls in females: 96</span></span>
<span><span class="co">#&gt; ℹ Number of chromosomes: 11</span></span>
<span><span class="co">#&gt;   - Chr: 1</span></span>
<span><span class="co">#&gt;   - Chr: 2</span></span>
<span><span class="co">#&gt;   - Chr: 3</span></span>
<span><span class="co">#&gt;   - Chr: 4</span></span>
<span><span class="co">#&gt;   - Chr: 5</span></span>
<span><span class="co">#&gt;   - Chr: 6</span></span>
<span><span class="co">#&gt;   - Chr: 7</span></span>
<span><span class="co">#&gt;   - Chr: 8</span></span>
<span><span class="co">#&gt;   - Chr: 9</span></span>
<span><span class="co">#&gt;   - Chr: 10</span></span>
<span><span class="co">#&gt;   - Chr: 23</span></span>
<span><span class="co">#&gt; ℹ Total number of SNPs: 20350</span></span>
<span><span class="co">#&gt; ℹ Total number of samples: 274</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>If you are using GXwasR package and/or following this pipeline in
your study, please cite it as:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"GXwasR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'GXwasR' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Bose B, Blostein F, Kim J, Winters J, Actkins KV, Mayer D, Congivaram H, Niarchou M, Edwards DV, Davis</span></span>
<span><span class="co">#&gt;   LK, Stranger BE (2025). "GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on</span></span>
<span><span class="co">#&gt;   Complex Traits." _medRxiv 2025.06.10.25329327_. doi:10.1101/2025.06.10.25329327</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/2025.06.10.25329327&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {GXwasR: A Toolkit for Investigating Sex-Differentiated Genetic Effects on Complex Traits},</span></span>
<span><span class="co">#&gt;     author = {Banabithi Bose and Freida Blostein and Jeewoo Kim and Jessica Winters and Ky’Era V. Actkins and David Mayer and Harrsha Congivaram and Maria Niarchou and Digna Velez Edwards and Lea K. Davis and Barbara E. Stranger},</span></span>
<span><span class="co">#&gt;     journal = {medRxiv 2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;     year = {2025},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/2025.06.10.25329327},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.21/GXwasR" class="external-link">GXwasR</a></em>
package (Bose, Blostein, Kim et al., 2025) was made possible thanks
to:</p>
<ul>
<li>R (R Core Team, 2025)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight et al., 2025)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.21/biocthis" class="external-link">biocthis</a></em>.</p>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  setting  value</span></span>
<span><span class="co">#&gt;  version  R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">#&gt;  os       macOS Sequoia 15.6</span></span>
<span><span class="co">#&gt;  system   aarch64, darwin24.4.0</span></span>
<span><span class="co">#&gt;  ui       unknown</span></span>
<span><span class="co">#&gt;  language (EN)</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       America/New_York</span></span>
<span><span class="co">#&gt;  date     2025-08-08</span></span>
<span><span class="co">#&gt;  pandoc   3.6.3 @ /Applications/Positron.app/Contents/Resources/app/quarto/bin/tools/aarch64/ (via rmarkdown)</span></span>
<span><span class="co">#&gt;  quarto   1.7.33 @ /usr/local/bin/quarto</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span><span class="co">#&gt;  package              * version    date (UTC) lib source</span></span>
<span><span class="co">#&gt;  abind                  1.4-8      2024-09-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  backports              1.5.0      2024-05-23 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1      2023-01-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigassertr             0.1.7      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigparallelr           0.3.2      2021-10-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  bigsnpr                1.12.18    2024-11-26 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigsparser             0.7.3      2024-09-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bigstatsr              1.6.2      2025-07-29 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Biobase                2.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocGenerics           0.54.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocIO                 1.18.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  BiocManager            1.30.26    2025-06-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocParallel           1.42.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  BiocStyle              2.36.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  Biostrings             2.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  bit                    4.6.0      2025-03-06 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bit64                  4.6.0-1    2025-01-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  bitops                 1.0-9      2024-10-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  brio                   1.1.5      2024-04-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  broom                  1.0.9      2025-07-28 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  BSgenome               1.76.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  cachem                 1.1.0      2024-05-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  calibrate              1.7.7      2020-06-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  callr                  3.7.6      2024-03-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  car                    3.1-3      2024-09-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  carData                3.0-5      2022-01-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  cli                    3.6.5      2025-04-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  codetools              0.2-20     2024-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  cowplot                1.2.0      2025-07-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  crayon                 1.5.3      2024-06-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  curl                   6.4.0      2025-06-22 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  data.table             1.17.8     2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  DelayedArray           0.34.1     2025-04-17 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  devtools             * 2.4.5      2022-10-11 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  digest                 0.6.37     2024-08-19 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doParallel             1.0.17     2022-02-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doRNG                  1.8.6.2    2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  doSNOW                 1.0.20     2022-02-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4      2023-11-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ellipsis               0.3.2      2021-04-29 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  evaluate               1.0.4      2025-06-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  farver                 2.1.2      2024-05-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fastmap                1.2.0      2024-05-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  flock                  0.7        2016-11-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  foreach                1.5.2      2022-02-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Formula                1.2-5      2023-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  fs                     1.6.6      2025-04-12 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gdsfmt                 1.44.1     2025-07-09 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  generics               0.1.4      2025-05-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GenomeInfoDb           1.44.1     2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.14     2025-04-21 [2] Bioconductor</span></span>
<span><span class="co">#&gt;  GenomicAlignments      1.44.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  GenomicRanges          1.60.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  ggplot2                3.5.2      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ggpubr                 0.6.1      2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6      2024-09-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  ggsignif               0.6.4      2022-10-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  glue                   1.8.0      2024-09-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gridExtra              2.3        2017-09-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  gtable                 0.3.6      2024-10-25 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  GXwasR               * 0.99.0     2025-08-08 [1] Bioconductor</span></span>
<span><span class="co">#&gt;  hms                    1.1.3      2023-03-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1    2024-04-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  httpuv                 1.6.16     2025-04-16 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  httr                   1.4.7      2023-08-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  IRanges                2.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  iterators              1.0.14     2022-02-05 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  jsonlite               2.0.0      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  knitr                  1.50       2025-03-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  labeling               0.4.3      2023-08-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  later                  1.4.2      2025-04-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lattice                0.22-7     2025-04-02 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4      2023-11-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  lubridate              1.9.4      2024-12-08 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  magrittr             * 2.0.3      2022-03-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  MASS                   7.3-65     2025-02-28 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mathjaxr               1.8-0      2025-04-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Matrix                 1.7-3      2025-03-11 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  MatrixGenerics         1.20.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  matrixStats            1.5.0      2025-01-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  memoise                2.0.1      2021-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  mgcv                   1.9-3      2025-04-04 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  mime                   0.13       2025-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  miniUI                 0.1.2      2025-04-17 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  nlme                   3.1-168    2025-03-31 [4] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pillar                 1.11.0     2025-07-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  pkgbuild               1.4.8      2025-05-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3      2019-09-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  pkgdev                 0.1.0.9060 2025-08-04 [2] Github (dieghernan/pkgdev@e56f2a8)</span></span>
<span><span class="co">#&gt;  pkgload                1.4.0      2024-06-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  plyr                   1.8.9      2023-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  plyranges              1.28.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  polynom                1.4-1      2022-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  poolr                  1.2-0      2025-05-07 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  prettyunits            1.2.0      2023-09-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  printr               * 0.3        2023-03-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  processx               3.8.6      2025-02-21 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  profvis                0.4.0      2024-09-20 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  progress               1.2.3      2023-12-06 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  promises               1.3.3      2025-05-29 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ps                     1.9.1      2025-04-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  purrr                  1.1.0      2025-07-10 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  qqman                  0.1.9      2023-08-23 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.methodsS3            1.8.2      2022-06-13 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.oo                   1.27.1     2025-05-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R.utils                2.13.0     2025-02-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  R6                     2.6.1      2025-02-15 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  ragg                   1.4.0      2025-04-10 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rbibutils              2.3        2024-10-04 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3      2022-04-03 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rcpp                   1.1.0      2025-07-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  RCurl                  1.98-1.17  2025-03-22 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  Rdpack                 2.6.4      2025-04-09 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0      2022-09-30 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  regioneR               1.40.1     2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  remotes                2.5.0      2024-03-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  restfulr               0.0.16     2025-06-27 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  rjson                  0.2.23     2024-09-16 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rlang                  1.1.6      2025-04-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmarkdown            * 2.29       2024-11-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rmio                   0.4.0      2022-02-17 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rngtools               1.5.2      2021-09-20 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rprojroot              2.1.0      2025-07-12 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  Rsamtools              2.24.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  rstatix                0.7.2      2023-02-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  rtracklayer            1.68.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  S4Arrays               1.8.1      2025-06-01 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  S4Vectors              0.46.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  scales                 1.4.0      2025-04-24 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  seqminer               9.7        2024-10-02 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.3      2025-02-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  shiny                  1.11.1     2025-07-03 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  snow                   0.4-4      2021-10-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  SNPRelate              1.42.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  SparseArray            1.8.1      2025-07-23 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  stringi                1.8.7      2025-03-27 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  stringr                1.5.1      2023-11-14 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sumFREGAT              1.2.5      2022-06-07 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  SummarizedExperiment   1.38.1     2025-04-30 [2] Bioconductor 3.21 (R 4.5.0)</span></span>
<span><span class="co">#&gt;  sys                    3.4.3      2024-10-04 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  systemfonts            1.2.3      2025-04-30 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  testthat             * 3.2.3      2025-01-13 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  textshaping            1.0.1      2025-05-01 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tibble                 3.3.0      2025-06-08 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  tidyr                  1.3.1      2024-01-24 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1      2024-03-11 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  timechange             0.3.0      2024-01-18 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  tzdb                   0.5.0      2025-03-15 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  UCSC.utils             1.4.0      2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  urlchecker             1.0.1      2021-11-30 [3] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  usethis              * 3.1.0      2024-11-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5      2023-12-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  vroom                  1.6.5      2023-12-05 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  withr                  3.0.2      2024-10-28 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xfun                   0.52       2025-04-02 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XML                    3.99-0.18  2025-01-01 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  xml2                   1.3.8      2025-03-14 [2] CRAN (R 4.5.1)</span></span>
<span><span class="co">#&gt;  xtable                 1.8-4      2019-04-21 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt;  XVector                0.48.0     2025-04-15 [2] Bioconduc~</span></span>
<span><span class="co">#&gt;  yaml                   2.3.10     2024-07-26 [2] CRAN (R 4.5.0)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  [1] /private/var/folders/d6/gtwl3_017sj4pp14fbfcbqjh0000gp/T/RtmpcRh9ZZ/temp_libpath4e4d53c96c8f</span></span>
<span><span class="co">#&gt;  [2] /Users/mayerdav/Library/R/arm64/4.5/library</span></span>
<span><span class="co">#&gt;  [3] /opt/homebrew/lib/R/4.5/site-library</span></span>
<span><span class="co">#&gt;  [4] /opt/homebrew/Cellar/r/4.5.1/lib/R/library</span></span>
<span><span class="co">#&gt;  * ── Packages attached to the search path.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.21/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2025) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2025) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.29. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-bose2025gxwasr"></a><a href="#cite-bose2025gxwasr">[2]</a><cite>
B. Bose, F. Blostein, J. Kim, et al. “GXwasR: A Toolkit for
Investigating Sex-Differentiated Genetic Effects on Complex Traits”. In:
<em>medRxiv 2025.06.10.25329327</em> (2025). DOI:
<a href="https://doi.org/10.1101/2025.06.10.25329327" class="external-link">10.1101/2025.06.10.25329327</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[3]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2025biocstyle"></a><a href="#cite-ole2025biocstyle">[4]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.36.0. 2025. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2025language"></a><a href="#cite-2025language">[5]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2025. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2025sessioninfo"></a><a href="#cite-wickham2025sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.3. 2025. DOI:
<a href="https://doi.org/10.32614/CRAN.package.sessioninfo" class="external-link">10.32614/CRAN.package.sessioninfo</a>.
URL:
<a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">https://CRAN.R-project.org/package=sessioninfo</a>.</cite>
</p>
<p>
<a id="bib-xie2025knitr"></a><a href="#cite-xie2025knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.50. 2025. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Banabithi Bose.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
